diff -ru abs/gtk2.ORIG/src/gtk+-2.24.16/gtk/gtkfilechooserdefault.c abs/gtk2/src/gtk+-2.24.16/gtk/gtkfilechooserdefault.c
--- abs/gtk2.ORIG/src/gtk+-2.24.16/gtk/gtkfilechooserdefault.c	2013-02-28 16:40:25.256962374 -0800
+++ abs/gtk2/src/gtk+-2.24.16/gtk/gtkfilechooserdefault.c	2013-02-28 16:40:31.503778425 -0800
@@ -6004,7 +6004,7 @@
   switch (impl->startup_mode)
     {
     case STARTUP_MODE_RECENT:
-      recent_shortcut_handler (impl);
+      home_folder_handler (impl);
       break;
 
     case STARTUP_MODE_CWD:
