diff -ru abs/gtk2.ORIG/src/gtk+-2.24.17/gtk/gtkfilechooserdefault.c abs/gtk2/src/gtk+-2.24.17/gtk/gtkfilechooserdefault.c
--- abs/gtk2.ORIG/src/gtk+-2.24.17/gtk/gtkfilechooserdefault.c	2013-03-18 10:44:08.441357310 -0700
+++ abs/gtk2/src/gtk+-2.24.17/gtk/gtkfilechooserdefault.c	2013-03-18 10:44:12.708138881 -0700
@@ -6004,7 +6004,7 @@
   switch (impl->startup_mode)
     {
     case STARTUP_MODE_RECENT:
-      recent_shortcut_handler (impl);
+      home_folder_handler (impl);
       break;
 
     case STARTUP_MODE_CWD:
