diff -ru abs/gtk2.ORIG/src/gtk+-2.24.23/gtk/gtkfilechooserdefault.c abs/gtk2/src/gtk+-2.24.23/gtk/gtkfilechooserdefault.c
--- abs/gtk2.ORIG/src/gtk+-2.24.23/gtk/gtkfilechooserdefault.c	2014-04-15 10:53:09.571045989 -0700
+++ abs/gtk2/src/gtk+-2.24.23/gtk/gtkfilechooserdefault.c	2014-04-15 10:53:55.978034652 -0700
@@ -6005,7 +6005,7 @@
   switch (impl->startup_mode)
     {
     case STARTUP_MODE_RECENT:
-      recent_shortcut_handler (impl);
+      home_folder_handler (impl);
       break;
 
     case STARTUP_MODE_CWD:
