diff -ru abs/gtk2.ORIG/src/gtk+-2.24.14/gtk/gtkitemfactory.c abs/gtk2/src/gtk+-2.24.14/gtk/gtkitemfactory.c
--- abs/gtk2.ORIG/src/gtk+-2.24.14/gtk/gtkitemfactory.c	2012-12-06 18:34:54.504461717 -0800
+++ abs/gtk2/src/gtk+-2.24.14/gtk/gtkitemfactory.c	2012-12-06 18:35:49.229105907 -0800
@@ -964,7 +964,7 @@
   else if (type_id == quark_type_toggle_item)
     type = GTK_TYPE_CHECK_MENU_ITEM;
   else if (type_id == quark_type_separator_item)
-    type = GTK_TYPE_MENU_ITEM;
+    type = GTK_TYPE_SEPARATOR_MENU_ITEM;
   else if (type_id == quark_type_branch)
     type = GTK_TYPE_MENU_ITEM;
   else if (type_id == quark_type_last_branch)
