diff -ru abs/gtk2.ORIG/src/gtk+-2.24.15/gtk/gtkitemfactory.c abs/gtk2/src/gtk+-2.24.15/gtk/gtkitemfactory.c
--- abs/gtk2.ORIG/src/gtk+-2.24.15/gtk/gtkitemfactory.c	2013-02-10 12:10:50.548639839 -0800
+++ abs/gtk2/src/gtk+-2.24.15/gtk/gtkitemfactory.c	2013-02-10 12:11:48.940333779 -0800
@@ -964,7 +964,7 @@
   else if (type_id == quark_type_toggle_item)
     type = GTK_TYPE_CHECK_MENU_ITEM;
   else if (type_id == quark_type_separator_item)
-    type = GTK_TYPE_MENU_ITEM;
+    type = GTK_TYPE_SEPARATOR_MENU_ITEM;
   else if (type_id == quark_type_branch)
     type = GTK_TYPE_MENU_ITEM;
   else if (type_id == quark_type_last_branch)
