diff -ru abs/gtk2.ORIG/src/gtk+-2.24.14/gtk/gtkfilechooserdefault.c abs/gtk2/src/gtk+-2.24.14/gtk/gtkfilechooserdefault.c
--- abs/gtk2.ORIG/src/gtk+-2.24.14/gtk/gtkfilechooserdefault.c	2012-12-06 18:34:54.497794890 -0800
+++ abs/gtk2/src/gtk+-2.24.14/gtk/gtkfilechooserdefault.c	2012-12-06 18:45:25.312909689 -0800
@@ -6061,7 +6061,7 @@
   switch (impl->startup_mode)
     {
     case STARTUP_MODE_RECENT:
-      recent_shortcut_handler (impl);
+      home_folder_handler (impl);
       break;
 
     case STARTUP_MODE_CWD:
