--- ../abs/gtk2/src/gtk+-2.24.6/gtk/gtkfilechooserdefault.c.ORIG	2011-09-05 13:22:14.866431660 -0700
+++ ../abs/gtk2/src/gtk+-2.24.6/gtk/gtkfilechooserdefault.c	2011-09-05 13:23:11.769767554 -0700
@@ -2162,7 +2162,7 @@
 
   shortcuts_append_search (impl);
 
-  if (impl->recent_manager)
+  if (0) /* gavin: no Recently Used */
     {
       shortcuts_append_recent (impl);
       shortcuts_insert_separator (impl, SHORTCUTS_RECENT_SEPARATOR);
