diff -ru orage.ORIG/src/orage-4.12.1/panel-plugin/xfce4-orageclock-plugin.c orage/src/orage-4.12.1/panel-plugin/xfce4-orageclock-plugin.c
--- orage.ORIG/src/orage-4.12.1/panel-plugin/xfce4-orageclock-plugin.c	2019-12-17 21:34:20.109479463 -0800
+++ orage/src/orage-4.12.1/panel-plugin/xfce4-orageclock-plugin.c	2019-12-17 21:34:21.752812728 -0800
@@ -100,6 +100,8 @@
 void oc_set_line(Clock *clock, ClockLine *clock_line, int pos)
 {
     clock_line->label = gtk_label_new("");
+    gtk_misc_set_alignment(GTK_MISC(clock_line->label), 1.0f, 1.0f);
+    gtk_misc_set_padding(GTK_MISC(clock_line->label), 5, 0);
     gtk_box_pack_start(GTK_BOX(clock->mbox), clock_line->label
             , FALSE, FALSE, 0);
     gtk_box_reorder_child(GTK_BOX(clock->mbox), clock_line->label, pos);
