diff -ru abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.1/plugins/tasklist/tasklist-widget.c abs/xfce4-panel/src/xfce4-panel-4.12.1/plugins/tasklist/tasklist-widget.c
--- abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.1/plugins/tasklist/tasklist-widget.c	2017-06-16 16:11:18.285112502 -0700
+++ abs/xfce4-panel/src/xfce4-panel-4.12.1/plugins/tasklist/tasklist-widget.c	2017-06-16 16:11:18.908431810 -0700
@@ -45,7 +45,7 @@
 
 
 #define DEFAULT_BUTTON_SIZE          (25)
-#define DEFAULT_MAX_BUTTON_LENGTH    (200)
+#define DEFAULT_MAX_BUTTON_LENGTH    (300)
 #define DEFAULT_MENU_ICON_SIZE       (16)
 #define DEFAULT_MIN_BUTTON_LENGTH    (DEFAULT_MAX_BUTTON_LENGTH / 4)
 #define DEFAULT_ICON_LUCENCY         (50)
@@ -1583,9 +1583,14 @@
       if (child->window == active_window)
         g_get_current_time (&child->last_focused);
 
-      /* set the toggle button state */
-      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button),
-                                    !!(child->window == active_window));
+      /* bold only active tasks */
+      if (child->window == active_window) {
+        gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), TRUE);
+        gtk_widget_modify_font (GTK_LABEL (child->label), pango_font_description_from_string ("semibold"));
+      } else {
+          gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), FALSE);
+          gtk_widget_modify_font (GTK_LABEL (child->label), pango_font_description_from_string (""));
+        }
     }
 
   /* release the lock */
@@ -2493,13 +2498,6 @@
   name = wnck_window_get_name (child->window);
   gtk_widget_set_tooltip_text (GTK_WIDGET (child->button), name);
 
-  /* create the button label */
-  if (!child->tasklist->only_minimized
-      && wnck_window_is_minimized (child->window))
-    name = label = g_strdup_printf ("[%s]", name);
-  else if (wnck_window_is_shaded (child->window))
-    name = label = g_strdup_printf ("=%s=", name);
-
   gtk_label_set_text (GTK_LABEL (child->label), name);
 
   g_free (label);
@@ -2563,33 +2561,6 @@
           xfce_tasklist_button_icon_changed (window, child);
         }
     }
-
-  /* update the blinking state */
-  if (PANEL_HAS_FLAG (changed_state, WNCK_WINDOW_STATE_DEMANDS_ATTENTION)
-      || PANEL_HAS_FLAG (changed_state, WNCK_WINDOW_STATE_URGENT))
-    {
-      /* only start blinking if the window requesting urgency
-       * notification is not the active window */
-      blink = wnck_window_or_transient_needs_attention (window);
-      if (!blink || (blink && !wnck_window_is_active (window)))
-        {
-          /* if we have all_blinking set make sure we toggle visibility of the button
-           * in case the window is not in the current workspace */
-          active_ws = wnck_screen_get_active_workspace (child->tasklist->screen);
-          if (child->tasklist->all_blinking && blink
-              && !xfce_tasklist_button_visible (child, active_ws))
-            {
-              gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), FALSE);
-              gtk_widget_show (child->button);
-            }
-
-          xfce_arrow_button_set_blinking (XFCE_ARROW_BUTTON (child->button), blink);
-
-          if (child->tasklist->all_blinking
-              && !xfce_tasklist_button_visible (child, active_ws))
-            gtk_widget_hide (child->button);
-        }
-    }
 }
 
 
@@ -2607,6 +2578,7 @@
 
   /* make sure we don't have two active windows (bug #6474) */
   gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), FALSE);
+  gtk_widget_modify_font (GTK_LABEL (child->label), pango_font_description_from_string (""));
 
   if (!tasklist->all_workspaces)
     xfce_tasklist_active_workspace_changed (tasklist->screen, NULL, tasklist);
