diff -ru abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.1/panel/panel-base-window.c abs/xfce4-panel/src/xfce4-panel-4.12.1/panel/panel-base-window.c
--- abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.1/panel/panel-base-window.c	2016-10-25 09:54:08.893642378 -0700
+++ abs/xfce4-panel/src/xfce4-panel-4.12.1/panel/panel-base-window.c	2016-10-25 09:54:09.323635965 -0700
@@ -840,6 +840,7 @@
   else if (window->background_style != PANEL_BG_STYLE_NONE)
     return PANEL_BORDER_NONE;
 
+  priv->borders = PANEL_BORDER_NONE;
   return priv->borders;
 }
 
