diff -ru abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.0/plugins/launcher/launcher.c abs/xfce4-panel/src/xfce4-panel-4.12.0/plugins/launcher/launcher.c
--- abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.0/plugins/launcher/launcher.c	2013-05-07 00:36:03.466822514 -0700
+++ abs/xfce4-panel/src/xfce4-panel-4.12.0/plugins/launcher/launcher.c	2013-05-07 00:38:25.839765229 -0700
@@ -2086,12 +2086,7 @@
 launcher_plugin_arrow_visibility (LauncherPlugin *plugin)
 {
   panel_return_if_fail (XFCE_IS_LAUNCHER_PLUGIN (plugin));
-
-  if (plugin->arrow_position != LAUNCHER_ARROW_INTERNAL
-       && LIST_HAS_TWO_OR_MORE_ENTRIES (plugin->items))
-    gtk_widget_show (plugin->arrow);
-  else
-    gtk_widget_hide (plugin->arrow);
+  gtk_widget_hide (plugin->arrow);
 }
 
 
