diff -ru abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.1/panel/panel-base-window.c abs/xfce4-panel/src/xfce4-panel-4.12.1/panel/panel-base-window.c
--- abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.1/panel/panel-base-window.c	2016-12-23 11:36:46.985391293 -0800
+++ abs/xfce4-panel/src/xfce4-panel-4.12.1/panel/panel-base-window.c	2016-12-23 11:36:47.148722560 -0800
@@ -840,6 +840,7 @@
   else if (window->background_style != PANEL_BG_STYLE_NONE)
     return PANEL_BORDER_NONE;
 
+  priv->borders = PANEL_BORDER_NONE;
   return priv->borders;
 }
 
