diff -ru abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.1/plugins/launcher/launcher.c abs/xfce4-panel/src/xfce4-panel-4.12.1/plugins/launcher/launcher.c
--- abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.1/plugins/launcher/launcher.c	2016-12-23 11:36:46.978724711 -0800
+++ abs/xfce4-panel/src/xfce4-panel-4.12.1/plugins/launcher/launcher.c	2016-12-23 11:36:47.115389648 -0800
@@ -2086,12 +2086,7 @@
 launcher_plugin_arrow_visibility (LauncherPlugin *plugin)
 {
   panel_return_if_fail (XFCE_IS_LAUNCHER_PLUGIN (plugin));
-
-  if (plugin->arrow_position != LAUNCHER_ARROW_INTERNAL
-       && LIST_HAS_TWO_OR_MORE_ENTRIES (plugin->items))
-    gtk_widget_show (plugin->arrow);
-  else
-    gtk_widget_hide (plugin->arrow);
+  gtk_widget_hide (plugin->arrow);
 }
 
 
