diff -ru abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.0/panel/panel-base-window.c abs/xfce4-panel/src/xfce4-panel-4.12.0/panel/panel-base-window.c
--- abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.0/panel/panel-base-window.c	2016-05-25 14:22:00.028270128 -0700
+++ abs/xfce4-panel/src/xfce4-panel-4.12.0/panel/panel-base-window.c	2016-05-25 14:22:00.328264106 -0700
@@ -840,6 +840,7 @@
   else if (window->background_style != PANEL_BG_STYLE_NONE)
     return PANEL_BORDER_NONE;
 
+  priv->borders = PANEL_BORDER_NONE;
   return priv->borders;
 }
 
