--- main.js.ORIG	2022-06-15 12:56:35.674626057 -0700
+++ main.js	2022-06-15 12:56:47.371823433 -0700
@@ -19915,7 +19915,7 @@
                 n = i.screen.getDisplayMatching(t);
             if (!g.isIntersect(n.bounds, t)) {
                 this.loggingService.logInfo("Window is being displayed in unavailable display. Updating the window bounds");
-                const i = g.getSafeBound(n, t);
+                const i = this._window.getBounds();
                 this.loggingService.logInfo("Updated bounds being set: " + JSON.stringify(i));
                 let o = !1,
                     r = !1;
