--- main.full.js.ORIG	2022-10-31 13:29:34.938042448 -0700
+++ main.full.js	2022-10-31 13:30:29.070689168 -0700
@@ -20510,7 +20510,7 @@
                 n = i.screen.getDisplayMatching(t);
             if (!g.isIntersect(n.bounds, t)) {
                 this.loggingService.logInfo("Window is being displayed in unavailable display. Updating the window bounds");
-                const i = g.getSafeBound(n, t);
+                const i = this._window.getBounds();
                 this.loggingService.logInfo("Updated bounds being set: " + JSON.stringify(i));
                 let o = !1,
                     r = !1;
