diff -ru xfce4-datetime-plugin.ORIG/src/xfce4-datetime-plugin-0.8.3/panel-plugin/datetime.c xfce4-datetime-plugin/src/xfce4-datetime-plugin-0.8.3/panel-plugin/datetime.c
--- xfce4-datetime-plugin.ORIG/src/xfce4-datetime-plugin-0.8.3/panel-plugin/datetime.c	2024-07-15 00:17:04.548050446 -0700
+++ xfce4-datetime-plugin/src/xfce4-datetime-plugin-0.8.3/panel-plugin/datetime.c	2024-07-15 00:17:04.551383894 -0700
@@ -243,6 +243,7 @@
   parent = g_object_get_data(G_OBJECT(widget), "calendar-parent");
   xfce_panel_plugin_position_widget(datetime->plugin, widget, parent, &x, &y);
   gtk_window_move(GTK_WINDOW(widget), x, y);
+  gtk_window_set_role(GTK_WINDOW(widget), "xfce4-datetime-plugin-calendar");
 }
 
 static gboolean close_calendar_window(t_datetime *datetime)
