diff --color -ru tigervnc.ORIG/src/tigervnc-1.13.0/common/rfb/CSecurityRSAAES.cxx tigervnc/src/tigervnc-1.13.0/common/rfb/CSecurityRSAAES.cxx
--- tigervnc.ORIG/src/tigervnc-1.13.0/common/rfb/CSecurityRSAAES.cxx	2023-02-22 12:21:46.101748073 -0800
+++ tigervnc/src/tigervnc-1.13.0/common/rfb/CSecurityRSAAES.cxx	2023-02-22 12:22:45.075884681 -0800
@@ -101,7 +101,6 @@
   switch (state) {
     case ReadPublicKey:
       if (readPublicKey()) {
-        verifyServer();
         writePublicKey();
         writeRandom();
         state = ReadRandom;
