--- pre/gedit/src/gedit-2.30.4/gedit/gedit-notebook.c.ORIG	2012-09-14 10:32:54.178883241 -0700
+++ pre/gedit/src/gedit-2.30.4/gedit/gedit-notebook.c	2012-09-14 10:33:26.349204488 -0700
@@ -915,24 +915,7 @@
 	jump_to = GPOINTER_TO_INT (g_object_get_data
 				   (G_OBJECT (tab), "jump_to"));
 
-	if (!jump_to || !nb->priv->focused_pages)
-	{
-		gtk_notebook_next_page (GTK_NOTEBOOK (nb));
-	}
-	else
-	{
-		GList *l;
-		GtkWidget *child;
-		int page_num;
-
-		/* activate the last focused tab */
-		l = g_list_last (nb->priv->focused_pages);
-		child = GTK_WIDGET (l->data);
-		page_num = gtk_notebook_page_num (GTK_NOTEBOOK (nb),
-						  child);
-		gtk_notebook_set_current_page (GTK_NOTEBOOK (nb), 
-					       page_num);
-	}
+	gtk_notebook_next_page (GTK_NOTEBOOK (nb));
 }
 
 static void
