--- pre/gedit/src/gedit-2.30.4/gedit/gedit-tab-label.c.ORIG	2012-02-12 14:35:05.583404120 -0800
+++ pre/gedit/src/gedit-2.30.4/gedit/gedit-tab-label.c	2012-02-12 14:47:13.533381632 -0800
@@ -166,14 +166,6 @@
 	    (state == GEDIT_TAB_STATE_SAVING)    ||
 	    (state == GEDIT_TAB_STATE_REVERTING))
 	{
-		gtk_widget_hide (tab_label->priv->icon);
-
-		gtk_widget_show (tab_label->priv->spinner);
-#ifdef BUILD_SPINNER
-		gedit_spinner_start (GEDIT_SPINNER (tab_label->priv->spinner));
-#else
-		gtk_spinner_start (GTK_SPINNER (tab_label->priv->spinner));
-#endif
 	}
 	else
 	{
@@ -187,12 +179,6 @@
 
 		gtk_widget_show (tab_label->priv->icon);
 
-		gtk_widget_hide (tab_label->priv->spinner);
-#ifdef BUILD_SPINNER
-		gedit_spinner_stop (GEDIT_SPINNER (tab_label->priv->spinner));
-#else
-		gtk_spinner_stop (GTK_SPINNER (tab_label->priv->spinner));
-#endif
 	}
 
 	/* sync tip since encoding is known only after load/save end */
@@ -283,14 +269,8 @@
 
 	close_button = gedit_close_button_new ();
 	gtk_widget_set_tooltip_text (close_button, _("Close document"));
-	gtk_box_pack_start (GTK_BOX (tab_label), close_button, FALSE, FALSE, 0);
 	tab_label->priv->close_button = close_button;
 
-	g_signal_connect (close_button,
-			  "clicked",
-			  G_CALLBACK (close_button_clicked_cb),
-			  tab_label);
-
 #ifdef BUILD_SPINNER
 	spinner = gedit_spinner_new ();
 	gedit_spinner_set_size (GEDIT_SPINNER (spinner), GTK_ICON_SIZE_MENU);
