--- ../pre/gedit/src/gedit-2.30.4/gedit/gedit-window.c.ORIG	2012-01-08 17:18:53.764912483 -0800
+++ ../pre/gedit/src/gedit-2.30.4/gedit/gedit-window.c	2012-01-08 17:20:37.138250504 -0800
@@ -3649,6 +3649,7 @@
 	gedit_debug (DEBUG_WINDOW);
 
 	window->priv->side_panel = gedit_panel_new (GTK_ORIENTATION_VERTICAL);
+	gtk_container_set_border_width (GTK_CONTAINER (window->priv->side_panel), 5); /* gavin: padding */
 
 	gtk_paned_pack1 (GTK_PANED (window->priv->hpaned), 
 			 window->priv->side_panel, 
@@ -3740,6 +3741,7 @@
 	gedit_debug (DEBUG_WINDOW);
 
 	window->priv->bottom_panel = gedit_panel_new (GTK_ORIENTATION_HORIZONTAL);
+	gtk_container_set_border_width (GTK_CONTAINER (window->priv->bottom_panel), 5); /* gavin: padding */
 
 	gtk_paned_pack2 (GTK_PANED (window->priv->vpaned), 
 			 window->priv->bottom_panel, 
