diff --color -ru solaar.ORIG/src/Solaar/lib/solaar/gtk.py solaar/src/Solaar/lib/solaar/gtk.py
--- solaar.ORIG/src/Solaar/lib/solaar/gtk.py	2021-11-12 10:33:12.169280022 -0800
+++ solaar/src/Solaar/lib/solaar/gtk.py	2021-11-12 10:35:03.259300529 -0800
@@ -165,7 +165,7 @@
             _upower.watch(lambda: listener.ping_all(True))
 
         # main UI event loop
-        ui.run_loop(listener.start_all, listener.stop_all, args.window != 'only', args.window != 'hide')
+        ui.run_loop(listener.start_all, listener.stop_all, False, args.window != 'hide')
     except Exception:
         import sys
         from traceback import format_exc
diff --color -ru solaar.ORIG/src/Solaar/lib/solaar/ui/__init__.py solaar/src/Solaar/lib/solaar/ui/__init__.py
--- solaar.ORIG/src/Solaar/lib/solaar/ui/__init__.py	2021-11-12 10:33:12.169280022 -0800
+++ solaar/src/Solaar/lib/solaar/ui/__init__.py	2021-11-12 10:34:43.759297664 -0800
@@ -151,7 +151,6 @@
 
 
 def run_loop(startup_hook, shutdown_hook, use_tray, show_window, args=None):
-    assert use_tray or show_window, 'need either tray or visible window'
     # from gi.repository.Gio import ApplicationFlags as _ApplicationFlags
     APP_ID = 'io.github.pwr.solaar'
     application = Gtk.Application.new(APP_ID, 0)  # _ApplicationFlags.HANDLES_COMMAND_LINE)
