diff --color -ru solaar.ORIG/src/Solaar/lib/solaar/gtk.py solaar/src/Solaar/lib/solaar/gtk.py
--- solaar.ORIG/src/Solaar/lib/solaar/gtk.py	2022-03-27 12:10:17.764304927 -0700
+++ solaar/src/Solaar/lib/solaar/gtk.py	2022-03-28 12:32:27.588792280 -0700
@@ -166,7 +166,7 @@
             _upower.watch(lambda: listener.ping_all(True))
 
         # main UI event loop
-        ui.run_loop(listener.start_all, listener.stop_all, args.window != 'only', args.window != 'hide')
+        ui.run_loop(listener.start_all, listener.stop_all, False, args.window != 'hide')
     except Exception:
         import sys
         from traceback import format_exc
diff --color -ru solaar.ORIG/src/Solaar/lib/solaar/ui/__init__.py solaar/src/Solaar/lib/solaar/ui/__init__.py
--- solaar.ORIG/src/Solaar/lib/solaar/ui/__init__.py	2022-03-27 12:10:17.764304927 -0700
+++ solaar/src/Solaar/lib/solaar/ui/__init__.py	2022-03-28 12:31:13.631931686 -0700
@@ -161,7 +161,6 @@
 
 
 def run_loop(startup_hook, shutdown_hook, use_tray, show_window):
-    assert use_tray or show_window, 'need either tray or visible window'
     # from gi.repository.Gio import ApplicationFlags as _ApplicationFlags
     APP_ID = 'io.github.pwr.solaar'
     application = Gtk.Application.new(APP_ID, Gio.ApplicationFlags.HANDLES_COMMAND_LINE)
