diff --color -ru solaar.ORIG/src/Solaar/lib/solaar/ui/notify.py solaar/src/Solaar/lib/solaar/ui/notify.py
--- solaar.ORIG/src/Solaar/lib/solaar/ui/notify.py	2022-03-27 12:10:17.764304927 -0700
+++ solaar/src/Solaar/lib/solaar/ui/notify.py	2022-03-28 12:33:58.962313540 -0700
@@ -93,13 +93,6 @@
             n.set_urgency(Notify.Urgency.NORMAL)
             n.set_hint('desktop-entry', GLib.Variant('s', NAME.lower()))
 
-            try:
-                # if _log.isEnabledFor(_DEBUG):
-                #     _log.debug("showing %s", n)
-                n.show()
-            except Exception:
-                _log.exception('showing %s', n)
-
     def show(dev, reason=None, icon=None, progress=None):
         """Show a notification with title and text.
         Optionally displays the `progress` integer value
@@ -132,13 +125,6 @@
             if progress:
                 n.set_hint('value', GLib.Variant('i', progress))
 
-            try:
-                # if _log.isEnabledFor(_DEBUG):
-                #     _log.debug("showing %s", n)
-                n.show()
-            except Exception:
-                _log.exception('showing %s', n)
-
 else:
     init = lambda: False
     uninit = lambda: None
