diff -ru abs/pacman.ORIG/src/pacman-4.1.0/src/pacman/util.c abs/pacman/src/pacman-4.1.0/src/pacman/util.c
--- abs/pacman.ORIG/src/pacman-4.1.0/src/pacman/util.c	2013-04-04 22:02:36.257818334 -0700
+++ abs/pacman/src/pacman-4.1.0/src/pacman/util.c	2013-04-04 22:23:34.536052303 -0700
@@ -1673,6 +1673,7 @@
 					config->colstr.nocolor);
 			break;
 		case ALPM_LOG_WARNING:
+			if (strstr(format, "ignoring package upgrade") || strstr(format, "is newer than")) { return 1; }
 			fprintf(stream, "%s%s%s", config->colstr.warn, _("warning: "),
 					config->colstr.nocolor);
 			break;
