diff -ru abs/pacman.ORIG/src/pacman-4.1.1/src/pacman/util.c abs/pacman/src/pacman-4.1.1/src/pacman/util.c
--- abs/pacman.ORIG/src/pacman-4.1.1/src/pacman/util.c	2013-05-09 17:36:34.035946447 -0700
+++ abs/pacman/src/pacman-4.1.1/src/pacman/util.c	2013-05-09 17:36:35.449289695 -0700
@@ -1674,6 +1674,7 @@
 					config->colstr.nocolor);
 			break;
 		case ALPM_LOG_WARNING:
+			if (strstr(format, "ignoring package upgrade") || strstr(format, "is newer than")) { return 1; }
 			fprintf(stream, "%s%s%s", config->colstr.warn, _("warning: "),
 					config->colstr.nocolor);
 			break;
