diff -ru abs/pacman/src/pacman-4.2.0/src/pacman/util.c abs/pacman.ORIG/src/pacman-4.2.0/src/pacman/util.c
--- abs/pacman/src/pacman-4.2.0/src/pacman/util.c	2015-01-02 10:31:59.252579709 -0800
+++ abs/pacman.ORIG/src/pacman-4.2.0/src/pacman/util.c	2015-01-02 10:30:45.854233491 -0800
@@ -1701,9 +1701,6 @@
 					config->colstr.nocolor);
 			break;
 		case ALPM_LOG_WARNING:
-			if (strstr(format, "ignoring package upgrade") ||
-					strstr(format, "ignoring package replacement") ||
-					strstr(format, "is newer than")) { return 1; }
 			fprintf(stream, "%s%s%s", config->colstr.warn, _("warning: "),
 					config->colstr.nocolor);
 			break;
