diff -ru abs/pacman.ORIG/src/pacman-4.1.2/src/pacman/util.c abs/pacman/src/pacman-4.1.2/src/pacman/util.c
--- abs/pacman.ORIG/src/pacman-4.1.2/src/pacman/util.c	2013-06-25 08:20:07.184479928 -0700
+++ abs/pacman/src/pacman-4.1.2/src/pacman/util.c	2013-06-25 08:26:49.802204953 -0700
@@ -1674,6 +1674,9 @@
 					config->colstr.nocolor);
 			break;
 		case ALPM_LOG_WARNING:
+			if (strstr(format, "ignoring package upgrade") ||
+					strstr(format, "ignoring package replacement") ||
+					strstr(format, "is newer than")) { return 1; }
 			fprintf(stream, "%s%s%s", config->colstr.warn, _("warning: "),
 					config->colstr.nocolor);
 			break;
