diff --color -ru pacman.ORIG/src/pacman/lib/libalpm/sync.c pacman/src/pacman/lib/libalpm/sync.c
--- pacman.ORIG/src/pacman/lib/libalpm/sync.c	2025-02-07 12:04:33.724647129 -0800
+++ pacman/src/pacman/lib/libalpm/sync.c	2025-02-07 12:05:53.109349910 -0800
@@ -93,8 +93,6 @@
 		/* check IgnorePkg/IgnoreGroup */
 		if(alpm_pkg_should_ignore(handle, spkg)
 				|| alpm_pkg_should_ignore(handle, lpkg)) {
-			_alpm_log(handle, ALPM_LOG_WARNING, _("%s: ignoring package upgrade (%s => %s)\n"),
-					lpkg->name, lpkg->version, spkg->version);
 		} else {
 			_alpm_log(handle, ALPM_LOG_DEBUG, "adding package %s-%s to the transaction targets\n",
 					spkg->name, spkg->version);
@@ -113,9 +111,6 @@
 				return 1;
 			}
 		} else {
-			alpm_db_t *sdb = alpm_pkg_get_db(spkg);
-			_alpm_log(handle, ALPM_LOG_WARNING, _("%s: local (%s) is newer than %s (%s)\n"),
-					lpkg->name, lpkg->version, sdb->treename, spkg->version);
 		}
 	}
 	return 0;
@@ -154,9 +149,6 @@
 			/* check IgnorePkg/IgnoreGroup */
 			if(alpm_pkg_should_ignore(handle, spkg)
 					|| alpm_pkg_should_ignore(handle, lpkg)) {
-				_alpm_log(handle, ALPM_LOG_WARNING,
-						_("ignoring package replacement (%s-%s => %s-%s)\n"),
-						lpkg->name, lpkg->version, spkg->name, spkg->version);
 				continue;
 			}
 
