diff -ru file-roller2-nn.ORIG/src/file-roller-2.32.2/src/fr-command-7z.c file-roller2-nn/src/file-roller-2.32.2/src/fr-command-7z.c
--- file-roller2-nn.ORIG/src/file-roller-2.32.2/src/fr-command-7z.c	2016-10-03 17:20:58.354835866 -0700
+++ file-roller2-nn/src/file-roller-2.32.2/src/fr-command-7z.c	2016-10-03 17:22:45.534442436 -0700
@@ -105,7 +105,7 @@
 			strncpy (version, ver_start, ver_len);
 			version[ver_len] = 0;
 
-			if (strcmp (version, "4.55") < 0)
+			if ((strcmp (version, "4.55") < 0) && (ver_len > 1) && (version[1] == '.'))
 				p7z_comm->old_style = TRUE;
 			else
 				p7z_comm->old_style = FALSE;
