diff -ru abs/gtk3.ORIG/src/gtk/gtk/gtkwindow.c abs/gtk3/src/gtk/gtk/gtkwindow.c
--- abs/gtk3.ORIG/src/gtk/gtk/gtkwindow.c	2018-06-13 00:59:46.822046705 -0700
+++ abs/gtk3/src/gtk/gtk/gtkwindow.c	2018-06-13 01:14:41.217707719 -0700
@@ -7438,6 +7438,19 @@
   gtk_widget_register_window (widget, gdk_window);
   gtk_widget_set_realized (widget, TRUE);
 
+  /* Set background color from theme so that the lag
+   * between resize and content redraw is less noticable.
+   */
+  if (!priv->client_decorated && !gtk_widget_get_app_paintable (widget))
+    {
+      GtkStyleContext *context;
+      const GdkRGBA *background_color;
+
+      context = gtk_widget_get_style_context (GTK_WIDGET (window));
+      background_color = _gtk_css_rgba_value_get_rgba (_gtk_style_context_peek_property (context, GTK_CSS_PROPERTY_BACKGROUND_COLOR));
+      gdk_window_set_background_rgba (gdk_window, background_color);
+    }
+
   attributes.x = allocation.x;
   attributes.y = allocation.y;
   attributes.width = allocation.width;
