diff -ru aur/roxterm-gtk2.ORIG/src/roxterm-2.9.3/src/profilegui.c aur/roxterm-gtk2/src/roxterm-2.9.3/src/profilegui.c
--- aur/roxterm-gtk2.ORIG/src/roxterm-2.9.3/src/profilegui.c	2014-11-19 08:58:31.560546738 -0800
+++ aur/roxterm-gtk2/src/roxterm-2.9.3/src/profilegui.c	2014-11-19 08:58:39.327322048 -0800
@@ -817,6 +817,7 @@
     pg->list_store = gtk_list_store_new(2, G_TYPE_STRING, G_TYPE_INT);
     for (n = 0; n < G_N_ELEMENTS(labels); ++n)
     {
+        if (n == 3 || n == 4) { continue; }
         gtk_list_store_append(pg->list_store, &iter);
         gtk_list_store_set(pg->list_store, &iter,
                 0, gettext(labels[n]), 1, n, -1);
diff -ru aur/roxterm-gtk2.ORIG/src/roxterm-2.9.3/src/roxterm.c aur/roxterm-gtk2/src/roxterm-2.9.3/src/roxterm.c
--- aur/roxterm-gtk2.ORIG/src/roxterm-2.9.3/src/roxterm.c	2014-11-19 08:58:31.560546738 -0800
+++ aur/roxterm-gtk2/src/roxterm-2.9.3/src/roxterm.c	2014-11-19 08:58:39.330655428 -0800
@@ -3303,7 +3303,6 @@
     roxterm_apply_title_template(roxterm);
     roxterm_apply_show_tab_number(roxterm);
     roxterm_apply_show_tab_status(roxterm);
-    roxterm_apply_match_files(roxterm, vte);
 #if GTK_CHECK_VERSION(3, 0, 0)
     roxterm_apply_show_resize_grip(roxterm);
 #endif
@@ -3503,8 +3502,6 @@
         gtk_widget_show(roxterm->widget);
     }
 
-    roxterm_add_matches(roxterm, vte);
-
     roxterm_apply_profile(roxterm, vte, FALSE);
     tab_name = global_options_lookup_string("tab-name");
     if (tab_name)
@@ -3769,10 +3766,6 @@
         {
             roxterm_apply_show_tab_status(roxterm);
         }
-        else if (!strcmp(key, "match_plain_files"))
-        {
-            roxterm_apply_match_files(roxterm, vte);
-        }
 #if GTK_CHECK_VERSION(3, 0, 0)
         else if (!strcmp(key, "show_resize_grip"))
         {
