diff -ru abs/systemd.ORIG/src/systemd-stable/src/core/job.c abs/systemd/src/systemd-stable/src/core/job.c
--- abs/systemd.ORIG/src/systemd-stable/src/core/job.c	2018-02-23 14:11:08.925384380 -0800
+++ abs/systemd/src/systemd-stable/src/core/job.c	2018-02-23 14:12:24.560893455 -0800
@@ -779,7 +779,7 @@
          * xsprintf() on purpose here: we are fine with truncation and don't
          * consider that an error. */
         DISABLE_WARNING_FORMAT_NONLITERAL;
-        (void) snprintf(buf, sizeof(buf), format, unit_description(u));
+        (void) snprintf(buf, sizeof(buf), format, strna(u->id));
         REENABLE_WARNING;
 
         switch (t) {
diff -ru abs/systemd.ORIG/src/systemd-stable/src/core/unit.c abs/systemd/src/systemd-stable/src/core/unit.c
--- abs/systemd.ORIG/src/systemd-stable/src/core/unit.c	2018-02-23 14:11:08.928384322 -0800
+++ abs/systemd/src/systemd-stable/src/core/unit.c	2018-02-23 14:12:40.132581901 -0800
@@ -1635,7 +1635,7 @@
 
 void unit_status_printf(Unit *u, const char *status, const char *unit_status_msg_format) {
         DISABLE_WARNING_FORMAT_NONLITERAL;
-        manager_status_printf(u->manager, STATUS_TYPE_NORMAL, status, unit_status_msg_format, unit_description(u));
+        manager_status_printf(u->manager, STATUS_TYPE_NORMAL, status, unit_status_msg_format, strna(u->id));
         REENABLE_WARNING;
 }
 
