diff -ru abs/systemd.ORIG/src/systemd-stable/src/core/job.c abs/systemd/src/systemd-stable/src/core/job.c
--- abs/systemd.ORIG/src/systemd-stable/src/core/job.c	2017-12-21 09:35:33.145266486 -0800
+++ abs/systemd/src/systemd-stable/src/core/job.c	2017-12-21 09:36:23.294320256 -0800
@@ -777,7 +777,7 @@
 
         /* The description might be longer than the buffer, but that's OK, we'll just truncate it here */
         DISABLE_WARNING_FORMAT_NONLITERAL;
-        xsprintf(buf, format, unit_description(u));
+        xsprintf(buf, format, strna(u->id));
         REENABLE_WARNING;
 
         switch (t) {
diff -ru abs/systemd.ORIG/src/systemd-stable/src/core/unit.c abs/systemd/src/systemd-stable/src/core/unit.c
--- abs/systemd.ORIG/src/systemd-stable/src/core/unit.c	2017-12-21 09:35:33.145266486 -0800
+++ abs/systemd/src/systemd-stable/src/core/unit.c	2017-12-21 09:36:43.897261203 -0800
@@ -1631,7 +1631,7 @@
 
 void unit_status_printf(Unit *u, const char *status, const char *unit_status_msg_format) {
         DISABLE_WARNING_FORMAT_NONLITERAL;
-        manager_status_printf(u->manager, STATUS_TYPE_NORMAL, status, unit_status_msg_format, unit_description(u));
+        manager_status_printf(u->manager, STATUS_TYPE_NORMAL, status, unit_status_msg_format, strna(u->id));
         REENABLE_WARNING;
 }
 
