diff -ru abs/systemd.ORIG/src/systemd-stable/src/core/job.c abs/systemd/src/systemd-stable/src/core/job.c
--- abs/systemd.ORIG/src/systemd-stable/src/core/job.c	2019-01-09 10:21:09.102538876 -0800
+++ abs/systemd/src/systemd-stable/src/core/job.c	2019-01-09 10:31:39.233331651 -0800
@@ -905,7 +905,7 @@
          * xsprintf() on purpose here: we are fine with truncation and don't
          * consider that an error. */
         DISABLE_WARNING_FORMAT_NONLITERAL;
-        (void) snprintf(buf, sizeof(buf), format, unit_description(u));
+        (void) snprintf(buf, sizeof(buf), format, strna(u->id));
         REENABLE_WARNING;
 
         switch (t) {
diff -ru abs/systemd.ORIG/src/systemd-stable/src/core/unit.c abs/systemd/src/systemd-stable/src/core/unit.c
--- abs/systemd.ORIG/src/systemd-stable/src/core/unit.c	2019-01-09 10:21:09.102538876 -0800
+++ abs/systemd/src/systemd-stable/src/core/unit.c	2019-01-09 10:32:55.497521182 -0800
@@ -1658,7 +1658,7 @@
 void unit_status_printf(Unit *u, const char *status, const char *unit_status_msg_format) {
         const char *d;
 
-        d = unit_description(u);
+        d = strna(u->id);
         if (log_get_show_color())
                 d = strjoina(ANSI_HIGHLIGHT, d, ANSI_NORMAL);
 
