diff -ru abs/xchat.ORIG/src/xchat-2.8.8/src/fe-gtk/maingui.c abs/xchat/src/xchat-2.8.8/src/fe-gtk/maingui.c
--- abs/xchat.ORIG/src/xchat-2.8.8/src/fe-gtk/maingui.c	2015-06-16 10:30:44.664947678 -0700
+++ abs/xchat/src/xchat-2.8.8/src/fe-gtk/maingui.c	2015-06-16 10:30:52.955060426 -0700
@@ -3430,6 +3430,8 @@
 		gui->is_tab = TRUE;
 	}
 
+	gtk_window_set_skip_taskbar_hint (GTK_WINDOW (sess->gui->window), TRUE);
+
 	if (user && user->hostname)
 		set_topic (sess, user->hostname, user->hostname);
 
