diff -ru gmrun.ORIG/src/gmrun-1.0w/src/main.c gmrun/src/gmrun-1.0w/src/main.c
--- gmrun.ORIG/src/gmrun-1.0w/src/main.c	2021-02-08 13:55:06.238496790 -0800
+++ gmrun/src/gmrun-1.0w/src/main.c	2021-02-12 00:26:08.085540101 -0800
@@ -521,12 +521,11 @@
 
    // this removes the title bar..
    GdkWindow *gwin = gtk_widget_get_window (GTK_WIDGET(dialog));
-   gdk_window_set_decorations (gwin, GDK_DECOR_BORDER);
 
    gtk_widget_set_name (GTK_WIDGET (dialog), "gmrun");
-   gtk_window_set_title (GTK_WINDOW(window), "A simple launcher with completion");
+   gtk_window_set_title (GTK_WINDOW(window), "gmrun");
 
-   gtk_container_set_border_width(GTK_CONTAINER(dialog), 4);
+   gtk_container_set_border_width(GTK_CONTAINER(dialog), 2);
    g_signal_connect(G_OBJECT(dialog), "destroy",
                   G_CALLBACK(gmrun_exit), NULL);
 
@@ -534,12 +533,9 @@
    gtk_box_pack_start (GTK_BOX (main_vbox), hhbox, FALSE, FALSE, 0);
 
    GtkWidget *label = gtk_label_new("Run program:");
-   gtk_box_pack_start (GTK_BOX(hhbox), label, FALSE, FALSE, 10);
-   gtkcompat_widget_set_halign_left (GTK_WIDGET (label));
    wlabel = label;
 
    label_search = gtk_label_new("");
-   gtk_box_pack_start (GTK_BOX (hhbox), label_search, FALSE, TRUE, 0);
    wlabel_search = label_search;
 
    compline = gtk_completion_line_new();
@@ -606,7 +602,7 @@
                                    window_geom.height);
    } else {
       /* default width = 450 */
-      gtk_window_set_default_size (GTK_WINDOW (dialog), 450, -1);
+      gtk_window_set_default_size (GTK_WINDOW (dialog), 350, -1);
    }
 
    // window icon
