--- abs/gmrun/src/gmrun-0.9.2/src/main.cc.ORIG	2011-11-23 15:57:11.345648232 -0800
+++ abs/gmrun/src/gmrun-0.9.2/src/main.cc	2011-11-23 16:04:56.885634021 -0800
@@ -537,12 +537,12 @@
 
   win = gtk_window_new(GTK_WINDOW_TOPLEVEL);
   gtk_widget_realize(win);
-  gdk_window_set_decorations(win->window, GDK_DECOR_BORDER);
+  /* gdk_window_set_decorations(win->window, GDK_DECOR_BORDER); */
   gtk_widget_set_name(win, "Msh_Run_Window");
-  gtk_window_set_title(GTK_WINDOW(win), "Execute program feat. completion");
+  gtk_window_set_title(GTK_WINDOW(win), "gmrun");
   gtk_window_set_policy(GTK_WINDOW(win), FALSE, FALSE, TRUE);
   // gtk_window_set_position(GTK_WINDOW(win), GTK_WIN_POS_CENTER);
-  gtk_container_set_border_width(GTK_CONTAINER(win), 4);
+  gtk_container_set_border_width(GTK_CONTAINER(win), 6); /* gavin: set padding from 4 to 6 */
   gtk_signal_connect(GTK_OBJECT(win), "destroy",
                      GTK_SIGNAL_FUNC(gtk_main_quit), NULL);
 
@@ -554,17 +554,20 @@
   gtk_widget_show(hhbox);
   gtk_box_pack_start(GTK_BOX(hbox), hhbox, FALSE, FALSE, 0);
 
-  GtkWidget *label = gtk_label_new("Run program:");
+	GtkWidget *label = gtk_label_new("Run program:");
+
+  /* gavin: hide label, search info, etc. */ /*
   gtk_widget_show(label);
   gtk_misc_set_alignment(GTK_MISC(label), 0.0, 1.0);
   gtk_misc_set_padding(GTK_MISC(label), 10, 0);
   gtk_box_pack_start(GTK_BOX(hhbox), label, FALSE, FALSE, 0);
-
+  
   label_search = gtk_label_new("");
   gtk_widget_show(label_search);
   gtk_misc_set_alignment(GTK_MISC(label_search), 1.0, 0.5);
   gtk_misc_set_padding(GTK_MISC(label_search), 10, 0);
   gtk_box_pack_start(GTK_BOX(hhbox), label_search, TRUE, TRUE, 0);
+	*/
 
   compline = gtk_completion_line_new();
   gtk_widget_set_name(compline, "Msh_Run_Compline");
