diff -ru abs/parcellite.ORIG/src/parcellite-1.1.5/src/main.c abs/parcellite/src/parcellite-1.1.5/src/main.c
--- abs/parcellite.ORIG/src/parcellite-1.1.5/src/main.c	2013-07-19 10:24:08.011499869 -0700
+++ abs/parcellite/src/parcellite-1.1.5/src/main.c	2013-07-19 10:24:48.394831426 -0700
@@ -1697,7 +1697,7 @@
       while (i < string->len)
       {	 /**fix 100% CPU utilization for odd data. - bug 2976890   */
 				gsize nline=0;
-				while(string->str[i+nline] == '\n' && nline+i<string->len)
+				while((string->str[i+nline] == '\r' || string->str[i+nline] == '\n') && nline+i<string->len)
 					nline++;
 				if(nline){
 					g_string_erase(string, i, nline);
