diff -ru abs/parcellite.ORIG/src/parcellite-1.1.6/src/main.c abs/parcellite/src/parcellite-1.1.6/src/main.c
--- abs/parcellite.ORIG/src/parcellite-1.1.6/src/main.c	2013-08-05 11:08:24.270440064 -0700
+++ abs/parcellite/src/parcellite-1.1.6/src/main.c	2013-08-05 11:09:26.886487606 -0700
@@ -1715,7 +1715,7 @@
       while (i < string->len)
       {	 /**fix 100% CPU utilization for odd data. - bug 2976890   */
 				gsize nline=0;
-				while(string->str[i+nline] == '\n' && nline+i<string->len)
+				while((string->str[i+nline] == '\r' || string->str[i+nline] == '\n') && nline+i<string->len)
 					nline++;
 				if(nline){
 					g_string_erase(string, i, nline);
