diff -ru abs/parcellite.ORIG/src/parcellite-1.1.3/src/main.c abs/parcellite/src/parcellite-1.1.3/src/main.c
--- abs/parcellite.ORIG/src/parcellite-1.1.3/src/main.c	2013-01-18 19:58:47.618313763 -0800
+++ abs/parcellite/src/parcellite-1.1.3/src/main.c	2013-01-18 21:02:45.094682495 -0800
@@ -1470,7 +1470,7 @@
       while (i < string->len)
       {	 /**fix 100% CPU utilization for odd data. - bug 2976890   */
 				gsize nline=0;
-				while(string->str[i+nline] == '\n' && nline+i<string->len)
+				while((string->str[i+nline] == '\r' || string->str[i+nline] == '\n') && nline+i<string->len)
 					nline++;
 				if(nline){
 					g_string_erase(string, i, nline);
