diff -ru abs/parcellite.ORIG/src/parcellite-1.1.9/src/main.c abs/parcellite/src/parcellite-1.1.9/src/main.c
--- abs/parcellite.ORIG/src/parcellite-1.1.9/src/main.c	2014-10-06 11:29:53.428399061 -0700
+++ abs/parcellite/src/parcellite-1.1.9/src/main.c	2014-10-06 11:30:51.759159264 -0700
@@ -1897,7 +1897,7 @@
       while (i < string->len)
       {	 /**fix 100% CPU utilization for odd data. - bug 2976890   */
 				gsize nline=0;
-				while(string->str[i+nline] == '\n' && nline+i<string->len)
+				while((string->str[i+nline] == '\r' || string->str[i+nline] == '\n') && nline+i<string->len)
 					nline++;
 				if(nline){
 					g_string_erase(string, i, nline);
