diff -ru parcellite.ORIG/src/parcellite-1.2.1/src/main.c parcellite/src/parcellite-1.2.1/src/main.c
--- parcellite.ORIG/src/parcellite-1.2.1/src/main.c	2017-01-30 09:50:49.399524780 -0800
+++ parcellite/src/parcellite-1.2.1/src/main.c	2017-01-30 09:50:49.509522063 -0800
@@ -1899,7 +1899,7 @@
       while (i < string->len)
       {	 /**fix 100% CPU utilization for odd data. - bug 2976890   */
 				gsize nline=0;
-				while(string->str[i+nline] == '\n' && nline+i<string->len)
+				while((string->str[i+nline] == '\r' || string->str[i+nline] == '\n') && nline+i<string->len)
 					nline++;
 				if(nline){
 					g_string_erase(string, i, nline);
