diff -ru abs/blueman.ORIG/src/blueman-1.23/blueman/plugins/applet/PowerManager.py abs/blueman/src/blueman-1.23/blueman/plugins/applet/PowerManager.py
--- abs/blueman.ORIG/src/blueman-1.23/blueman/plugins/applet/PowerManager.py	2013-04-09 07:42:29.946317618 -0700
+++ abs/blueman/src/blueman-1.23/blueman/plugins/applet/PowerManager.py	2013-04-09 07:42:42.659875449 -0700
@@ -93,7 +93,7 @@
 		if state:
 			def timeout():
 				self.adapter_state = self.get_adapter_state()
-				self.RequestPowerState(self.adapter_state)
+				self.RequestPowerState(False)
 			gobject.timeout_add(1000, timeout)
 		
 	def get_adapter_state(self):
