diff --color -ru xfce4-settings.ORIG/src/xfce4-settings/xfsettingsd/accessibility.c xfce4-settings/src/xfce4-settings/xfsettingsd/accessibility.c
--- xfce4-settings.ORIG/src/xfce4-settings/xfsettingsd/accessibility.c	2024-12-23 11:47:52.110591762 -0800
+++ xfce4-settings/src/xfce4-settings/xfsettingsd/accessibility.c	2024-12-23 13:23:44.767532246 -0800
@@ -138,9 +138,6 @@
 
         /* add event filter */
         XkbSelectEvents (GDK_DISPLAY_XDISPLAY (gdk_display_get_default ()), XkbUseCoreKbd, XkbControlsNotifyMask, XkbControlsNotifyMask);
-
-        /* monitor all window events */
-        gdk_window_add_filter (NULL, xfce_accessibility_helper_event_filter, helper);
 #endif /* !HAVE_LIBNOTIFY */
     }
     else
