--- ../abs/gthumb/src/gthumb-2.14.2/extensions/image_viewer/gth-image-viewer-page.c.ORIG	2011-08-28 17:28:56.407313870 -0700
+++ ../abs/gthumb/src/gthumb-2.14.2/extensions/image_viewer/gth-image-viewer-page.c	2011-08-28 17:29:35.990645998 -0700
@@ -197,7 +197,7 @@
 	  N_("Paste Image"), "<control>p",
 	  N_("Paste the image from the clipboard"),
 	  G_CALLBACK (image_viewer_activate_action_edit_paste_image) },
-
+/*
 	{ "ImageViewer_View_ZoomIn", GTK_STOCK_ZOOM_IN,
 	  N_("In"), "<control>plus",
 	  N_("Zoom in"),
@@ -222,6 +222,7 @@
 	  N_("Width"), "",
 	  N_("Zoom to fit width"),
 	  G_CALLBACK (image_viewer_activate_action_view_zoom_fit_width) },
+*/
 };
 
 
