diff --color -ru rofi.ORIG/src/rofi-1.7.4/source/view.c rofi/src/rofi-1.7.4/source/view.c
--- rofi.ORIG/src/rofi-1.7.4/source/view.c	2022-07-09 23:36:41.700690658 -0700
+++ rofi/src/rofi-1.7.4/source/view.c	2022-07-09 23:39:49.901585279 -0700
@@ -2075,6 +2075,8 @@
 }
 
 int rofi_view_error_dialog(const char *msg, int markup) {
+  return TRUE;
+
   RofiViewState *state = __rofi_view_state_create();
   state->retv = MENU_CANCEL;
   state->menu_flags = MENU_ERROR_DIALOG;
