diff --color -ru emerald-gtk3.ORIG/src/emerald-v0.8.18/src/main.c emerald-gtk3/src/emerald-v0.8.18/src/main.c
--- emerald-gtk3.ORIG/src/emerald-v0.8.18/src/main.c	2022-06-06 12:02:16.978054316 -0700
+++ emerald-gtk3/src/emerald-v0.8.18/src/main.c	2022-06-06 12:13:09.753152547 -0700
@@ -5492,6 +5492,7 @@
     PangoFontMetrics *metrics;
     PangoLanguage *lang;
 
+    pango_font_description_set_weight(ws->font_desc, PANGO_WEIGHT_SEMIBOLD);
     pango_context_set_font_description(ws->pango_context, ws->font_desc);
     lang = pango_context_get_language(ws->pango_context);
     metrics =
