diff -ru abs/xfce4-genmon-plugin.ORIG/src/xfce4-genmon-plugin-4.0.0/panel-plugin/main.c abs/xfce4-genmon-plugin/src/xfce4-genmon-plugin-4.0.0/panel-plugin/main.c
--- abs/xfce4-genmon-plugin.ORIG/src/xfce4-genmon-plugin-4.0.0/panel-plugin/main.c	2017-02-26 06:51:59.944843116 -0800
+++ abs/xfce4-genmon-plugin/src/xfce4-genmon-plugin-4.0.0/panel-plugin/main.c	2017-02-26 06:58:22.999255311 -0800
@@ -302,6 +302,13 @@
 }/* SetTimer() */
 
 /**************************************************************/
+static void copy_to_clipboard (GtkWidget *widget, gpointer user_data)
+{
+    struct genmon_t *poPlugin = (genmon_t *) user_data;
+
+    GtkClipboard *clipboard = gtk_clipboard_get (GDK_SELECTION_CLIPBOARD);
+    gtk_clipboard_set_text (clipboard, poPlugin->acValue, -1);
+}
 
 static genmon_t *genmon_create_control (XfcePanelPlugin *plugin)
 /* Plugin API */
@@ -313,6 +320,7 @@
     GtkOrientation orientation = xfce_panel_plugin_get_orientation (plugin);
     GtkSettings *settings;
     gchar *default_font;
+    GtkWidget *separator, *copy;
 
     #if GTK_CHECK_VERSION (3, 16, 0)
         GtkCssProvider *css_provider;
@@ -453,6 +461,15 @@
     g_free(css);
     g_free(default_font);
 
+    separator = gtk_separator_menu_item_new();
+    xfce_panel_plugin_menu_insert_item (plugin, GTK_MENU_ITEM (separator));
+    gtk_widget_show (separator);
+
+    copy = gtk_menu_item_new_with_mnemonic (_("_Copy to Clipboard"));
+    g_signal_connect(G_OBJECT (copy), "activate", G_CALLBACK (copy_to_clipboard), poPlugin);
+    xfce_panel_plugin_menu_insert_item (plugin, GTK_MENU_ITEM (copy));
+    gtk_widget_show (copy);
+
     return poPlugin;
 }/* genmon_create_control() */
 
