diff -ru abs/thunar.ORIG/src/Thunar-1.6.10/thunar/thunar-location-buttons.c abs/thunar/src/Thunar-1.6.10/thunar/thunar-location-buttons.c
--- abs/thunar.ORIG/src/Thunar-1.6.10/thunar/thunar-location-buttons.c	2015-10-29 16:39:24.932496261 -0700
+++ abs/thunar/src/Thunar-1.6.10/thunar/thunar-location-buttons.c	2015-10-29 16:41:03.755832348 -0700
@@ -127,6 +127,8 @@
                                                                          ThunarLocationButtons      *buttons);
 static void           thunar_location_buttons_action_open_in_new_window (GtkAction                  *action,
                                                                          ThunarLocationButtons      *buttons);
+static void           thunar_location_buttons_action_open_terminal      (GtkAction                  *action,
+                                                                         ThunarLocationButtons      *buttons);
 static void           thunar_location_buttons_action_paste_into_folder  (GtkAction                  *action,
                                                                          ThunarLocationButtons      *buttons);
 static void           thunar_location_buttons_action_properties         (GtkAction                  *action,
@@ -175,6 +177,7 @@
   { "location-buttons-open", GTK_STOCK_OPEN, N_("_Open"), "", NULL, G_CALLBACK (thunar_location_buttons_action_open), },
   { "location-buttons-open-in-new-tab", NULL, N_("Open in New Tab"), "", NULL, G_CALLBACK (thunar_location_buttons_action_open_in_new_tab), },
   { "location-buttons-open-in-new-window", NULL, N_("Open in New Window"), "", NULL, G_CALLBACK (thunar_location_buttons_action_open_in_new_window), },
+  { "location-buttons-open-terminal", NULL, N_("Open Terminal Here"), "", NULL, G_CALLBACK (thunar_location_buttons_action_open_terminal), },
   { "location-buttons-create-folder", "folder-new", N_("Create _Folder..."), "", NULL, G_CALLBACK (thunar_location_buttons_action_create_folder), },
   { "location-buttons-empty-trash", NULL, N_("_Empty Trash"), "", N_("Delete all files and folders in the Trash"), G_CALLBACK (thunar_location_buttons_action_empty_trash), },
   { "location-buttons-paste-into-folder", GTK_STOCK_PASTE, N_("Paste Into Folder"), "", NULL, G_CALLBACK (thunar_location_buttons_action_paste_into_folder), },
@@ -1255,6 +1258,11 @@
           thunar_gtk_action_set_tooltip (action, _("Open \"%s\" in a new tab"), display_name);
           g_object_set_qdata_full (G_OBJECT (action), thunar_file_quark, g_object_ref (G_OBJECT (file)), (GDestroyNotify) g_object_unref);
 
+          /* setup the "Open Terminal Here" action */
+          action = gtk_action_group_get_action (buttons->action_group, "location-buttons-open-terminal");
+          thunar_gtk_action_set_tooltip (action, _("Open \"%s\" in a terminal window"), display_name);
+          g_object_set_data_full (G_OBJECT (action), I_("thunar-file"), g_object_ref (G_OBJECT (file)), (GDestroyNotify) g_object_unref);
+
           /* setup the "Create Folder..." action */
           action = gtk_action_group_get_action (buttons->action_group, "location-buttons-create-folder");
           thunar_gtk_action_set_tooltip (action, _("Create a new folder in \"%s\""), display_name);
@@ -1439,6 +1447,39 @@
 }
 
 
+
+static void
+thunar_location_buttons_action_open_terminal (GtkAction             *action,
+                                              ThunarLocationButtons *buttons)
+{
+  ThunarFile *directory;
+  GString *command = g_string_new(NULL);
+  gchar *path;
+
+  _thunar_return_if_fail (THUNAR_IS_LOCATION_BUTTONS (buttons));
+  _thunar_return_if_fail (GTK_IS_ACTION (action));
+
+  directory = g_object_get_data (G_OBJECT (action), "thunar-file");
+
+  if (G_LIKELY (directory != NULL)) {
+    path = g_file_get_path (thunar_file_get_file (directory));
+
+    if (path != NULL) {
+      if (strlen(path) > 0) {
+        g_string_assign (command, "exo-open --launch TerminalEmulator --working-directory ");
+        g_string_append (command, g_shell_quote (path));
+        g_string_append (command, " ");
+        g_string_append (command, g_shell_quote (path));
+        system (command->str);
+      }
+      g_free (path);
+    }
+  }
+
+  g_string_free (command, TRUE);
+}
+
+
 
 static void
 thunar_location_buttons_action_paste_into_folder (GtkAction             *action,
diff -ru abs/thunar.ORIG/src/Thunar-1.6.10/thunar/thunar-location-buttons-ui.xml abs/thunar/src/Thunar-1.6.10/thunar/thunar-location-buttons-ui.xml
--- abs/thunar.ORIG/src/Thunar-1.6.10/thunar/thunar-location-buttons-ui.xml	2015-10-29 16:39:24.932496261 -0700
+++ abs/thunar/src/Thunar-1.6.10/thunar/thunar-location-buttons-ui.xml	2015-10-29 16:39:42.829163426 -0700
@@ -15,9 +15,11 @@
     <menuitem action="location-buttons-open-in-new-tab" />
     <menuitem action="location-buttons-open-in-new-window" />
     <separator />
-    <menuitem action="location-buttons-create-folder" />
+    <menuitem action="location-buttons-open-terminal" />
+    <separator />
     <menuitem action="location-buttons-empty-trash" />
     <separator />
+    <menuitem action="location-buttons-create-folder" />
     <menuitem action="location-buttons-paste-into-folder" />
     <separator />
     <menuitem action="location-buttons-properties" />
