diff -ru abs/thunar.ORIG/src/thunar/thunar/thunar-notify.c abs/thunar/src/thunar/thunar/thunar-notify.c
--- abs/thunar.ORIG/src/thunar/thunar/thunar-notify.c	2017-02-10 11:51:40.781771881 -0800
+++ abs/thunar/src/thunar/thunar/thunar-notify.c	2017-02-10 11:51:43.251648426 -0800
@@ -174,8 +174,7 @@
     {
       summary = _("Writing data to device");
       message = g_strdup_printf (_("There is data that needs to be written to the "
-                                   "device \"%s\" before it can be removed. Please "
-                                   "do not remove the media or disconnect the drive"),
+                                   "device \"%s\" before it can be removed."),
                                    name);
     }
 
@@ -213,8 +212,7 @@
     {
       summary = _("Writing data to device");
       message = g_strdup_printf (_("There is data that needs to be written to the "
-                                   "device \"%s\" before it can be removed. Please "
-                                   "do not remove the media or disconnect the drive"),
+                                   "device \"%s\" before it can be removed."),
                                    name);
     }
 
