diff -ru abs/thunar.ORIG/src/Thunar-1.6.14/thunar/thunar-window.c abs/thunar/src/Thunar-1.6.14/thunar/thunar-window.c
--- abs/thunar.ORIG/src/Thunar-1.6.14/thunar/thunar-window.c	2018-02-19 04:49:01.933310305 -0800
+++ abs/thunar/src/Thunar-1.6.14/thunar/thunar-window.c	2018-02-19 04:49:02.151304679 -0800
@@ -851,7 +851,7 @@
     }
 
   window->paned = gtk_hpaned_new ();
-  gtk_container_set_border_width (GTK_CONTAINER (window->paned), 0);
+  gtk_container_set_border_width (GTK_CONTAINER (window->paned), 5);
   gtk_table_attach (GTK_TABLE (window->table), window->paned, 0, 1, 4, 5, GTK_EXPAND | GTK_FILL, GTK_EXPAND | GTK_FILL, 0, 0);
   gtk_widget_show (window->paned);
 
