diff -ru abs/thunar.ORIG/src/Thunar-1.6.10/thunar/thunar-window.c abs/thunar/src/Thunar-1.6.10/thunar/thunar-window.c
--- abs/thunar.ORIG/src/Thunar-1.6.10/thunar/thunar-window.c	2016-05-25 14:20:02.490617060 -0700
+++ abs/thunar/src/Thunar-1.6.10/thunar/thunar-window.c	2016-05-25 14:20:02.627281014 -0700
@@ -851,7 +851,7 @@
     }
 
   window->paned = gtk_hpaned_new ();
-  gtk_container_set_border_width (GTK_CONTAINER (window->paned), 0);
+  gtk_container_set_border_width (GTK_CONTAINER (window->paned), 5);
   gtk_table_attach (GTK_TABLE (window->table), window->paned, 0, 1, 4, 5, GTK_EXPAND | GTK_FILL, GTK_EXPAND | GTK_FILL, 0, 0);
   gtk_widget_show (window->paned);
 
