diff -ru abs/thunar.ORIG/src/Thunar-1.8.0/thunar/thunar-file.c abs/thunar/src/Thunar-1.8.0/thunar/thunar-file.c
--- abs/thunar.ORIG/src/Thunar-1.8.0/thunar/thunar-file.c	2018-06-12 17:14:33.472510085 -0700
+++ abs/thunar/src/Thunar-1.8.0/thunar/thunar-file.c	2018-06-12 17:14:34.262491014 -0700
@@ -881,7 +881,7 @@
         }
 
       /* create a file or directory monitor */
-      file_watch->monitor = g_file_monitor (file->gfile, G_FILE_MONITOR_WATCH_MOUNTS | G_FILE_MONITOR_SEND_MOVED, NULL, NULL);
+      file_watch->monitor = g_file_monitor (file->gfile, G_FILE_MONITOR_WATCH_MOUNTS, NULL, NULL);
       if (G_LIKELY (file_watch->monitor != NULL))
         {
           /* watch monitor for file changes */
@@ -3938,7 +3938,7 @@
       file_watch->watch_count = 1;
 
       /* create a file or directory monitor */
-      file_watch->monitor = g_file_monitor (file->gfile, G_FILE_MONITOR_WATCH_MOUNTS | G_FILE_MONITOR_SEND_MOVED, NULL, &error);
+      file_watch->monitor = g_file_monitor (file->gfile, G_FILE_MONITOR_WATCH_MOUNTS, NULL, &error);
       if (G_UNLIKELY (file_watch->monitor == NULL))
         {
           g_debug ("Failed to create file monitor: %s", error->message);
diff -ru abs/thunar.ORIG/src/Thunar-1.8.0/thunar/thunar-folder.c abs/thunar/src/Thunar-1.8.0/thunar/thunar-folder.c
--- abs/thunar.ORIG/src/Thunar-1.8.0/thunar/thunar-folder.c	2018-06-12 17:14:33.472510085 -0700
+++ abs/thunar/src/Thunar-1.8.0/thunar/thunar-folder.c	2018-06-12 17:14:34.262491014 -0700
@@ -576,7 +576,7 @@
 
   /* add us to the file alteration monitor */
   folder->monitor = g_file_monitor_directory (thunar_file_get_file (folder->corresponding_file),
-                                              G_FILE_MONITOR_SEND_MOVED, NULL, NULL);
+                                              G_FILE_MONITOR_NONE, NULL, NULL);
   if (G_LIKELY (folder->monitor != NULL))
     g_signal_connect (folder->monitor, "changed", G_CALLBACK (thunar_folder_monitor), folder);
 
