diff -ru abs/thunar.ORIG/src/Thunar-1.8.3/thunar/thunar-history.c abs/thunar/src/Thunar-1.8.3/thunar/thunar-history.c
--- abs/thunar.ORIG/src/Thunar-1.8.3/thunar/thunar-history.c	2018-09-26 13:48:34.216751171 -0700
+++ abs/thunar/src/Thunar-1.8.3/thunar/thunar-history.c	2018-09-26 13:48:34.610089650 -0700
@@ -167,6 +167,7 @@
   g_signal_connect (G_OBJECT (history->action_back), "show-menu", G_CALLBACK (thunar_history_show_menu), history);
 G_GNUC_BEGIN_IGNORE_DEPRECATIONS
   gtk_action_set_sensitive (history->action_back, FALSE);
+  gtk_action_set_visible (history->action_back, FALSE);
 G_GNUC_END_IGNORE_DEPRECATIONS
 
   /* create the "forward" action */
@@ -175,6 +176,7 @@
   g_signal_connect (G_OBJECT (history->action_forward), "show-menu", G_CALLBACK (thunar_history_show_menu), history);
 G_GNUC_BEGIN_IGNORE_DEPRECATIONS
   gtk_action_set_sensitive (history->action_forward, FALSE);
+  gtk_action_set_visible (history->action_forward, FALSE);
 G_GNUC_END_IGNORE_DEPRECATIONS
 }
 
diff -ru abs/thunar.ORIG/src/Thunar-1.8.3/thunar/thunar-window.c abs/thunar/src/Thunar-1.8.3/thunar/thunar-window.c
--- abs/thunar.ORIG/src/Thunar-1.8.3/thunar/thunar-window.c	2018-09-26 13:48:34.220084548 -0700
+++ abs/thunar/src/Thunar-1.8.3/thunar/thunar-window.c	2018-09-26 13:48:34.610089650 -0700
@@ -376,7 +376,7 @@
   { "open-file-system", "drive-harddisk", N_ ("File System"), NULL, N_ ("Browse the file system"), G_CALLBACK (thunar_window_action_open_file_system), },
   { "open-network", "network-workgroup", N_("B_rowse Network"), NULL, N_ ("Browse local network connections"), G_CALLBACK (thunar_window_action_open_network), },
   { "open-templates", "text-x-generic-template", N_("T_emplates"), NULL, N_ ("Go to the templates folder"), G_CALLBACK (thunar_window_action_open_templates), },
-  { "open-location", NULL, N_ ("_Open Location..."), "<control>L", N_ ("Specify a location to open"), G_CALLBACK (thunar_window_action_open_location), },
+  { "open-location", NULL, N_ ("_Open Location..."), "<alt>D", N_ ("Specify a location to open"), G_CALLBACK (thunar_window_action_open_location), },
   { "help-menu", NULL, N_ ("_Help"), NULL, },
   { "contents", "help-browser", N_ ("_Contents"), "F1", N_ ("Display Thunar user manual"), G_CALLBACK (thunar_window_action_contents), },
   { "about", "help-about", N_ ("_About"), NULL, N_ ("Display information about Thunar"), G_CALLBACK (thunar_window_action_about), },
diff -ru abs/thunar.ORIG/src/Thunar-1.8.3/thunar/thunar-window-ui.xml abs/thunar/src/Thunar-1.8.3/thunar/thunar-window-ui.xml
--- abs/thunar.ORIG/src/Thunar-1.8.3/thunar/thunar-window-ui.xml	2018-09-26 13:48:34.220084548 -0700
+++ abs/thunar/src/Thunar-1.8.3/thunar/thunar-window-ui.xml	2018-09-26 13:48:34.610089650 -0700
@@ -126,9 +126,6 @@
   </popup>
 
   <toolbar name="location-toolbar">
-    <placeholder name="placeholder-history-actions" />
-    <toolitem action="open-parent" />
-    <toolitem action="open-home" />
   </toolbar>
 </ui>
 
