diff -ru abs/thunar.ORIG/src/Thunar-1.6.9/thunar/thunar-progress-dialog.c abs/thunar/src/Thunar-1.6.9/thunar/thunar-progress-dialog.c
--- abs/thunar.ORIG/src/Thunar-1.6.9/thunar/thunar-progress-dialog.c	2013-05-07 00:23:01.658432393 -0700
+++ abs/thunar/src/Thunar-1.6.9/thunar/thunar-progress-dialog.c	2013-05-07 00:29:49.673545426 -0700
@@ -148,16 +148,6 @@
 thunar_progress_dialog_shown (ThunarProgressDialog *dialog)
 {
   _thunar_return_if_fail (THUNAR_IS_PROGRESS_DIALOG (dialog));
-
-  /* show the status icon */
-  if (dialog->status_icon == NULL)
-    {
-      dialog->status_icon = gtk_status_icon_new_from_icon_name ("stock_folder-copy");
-      thunar_progress_dialog_update_status_icon (dialog);
-      g_signal_connect_swapped (dialog->status_icon, "button-press-event",
-                                G_CALLBACK (thunar_progress_dialog_toggled),
-                                GTK_WIDGET (dialog));
-    }
 }
 
 
