diff -ru abs/thunar.ORIG/src/Thunar-1.6.10/thunar/thunar-application.c abs/thunar/src/Thunar-1.6.10/thunar/thunar-application.c
--- abs/thunar.ORIG/src/Thunar-1.6.10/thunar/thunar-application.c	2016-05-25 14:20:02.497283594 -0700
+++ abs/thunar/src/Thunar-1.6.10/thunar/thunar-application.c	2016-05-25 14:20:02.960607729 -0700
@@ -869,11 +869,8 @@
 {
   _thunar_return_if_fail (THUNAR_IS_APPLICATION (application));
 
-  if (application->daemon != daemonize)
-    {
-      application->daemon = daemonize;
-      g_object_notify (G_OBJECT (application), "daemon");
-    }
+  application->daemon = FALSE;
+  g_object_notify (G_OBJECT (application), "daemon");
 }
 
 
