diff -ru abs/thunar.ORIG/src/Thunar-1.8.1/thunar/thunar-notify.c abs/thunar/src/Thunar-1.8.1/thunar/thunar-notify.c
--- abs/thunar.ORIG/src/Thunar-1.8.1/thunar/thunar-notify.c	2018-06-14 00:46:54.390827424 -0700
+++ abs/thunar/src/Thunar-1.8.1/thunar/thunar-notify.c	2018-06-14 00:46:55.240832135 -0700
@@ -174,8 +174,7 @@
     {
       summary = _("Writing data to device");
       message = g_strdup_printf (_("There is data that needs to be written to the "
-                                   "device \"%s\" before it can be removed. Please "
-                                   "do not remove the media or disconnect the drive"),
+                                   "device \"%s\" before it can be removed."),
                                    name);
     }
 
@@ -213,8 +212,7 @@
     {
       summary = _("Writing data to device");
       message = g_strdup_printf (_("There is data that needs to be written to the "
-                                   "device \"%s\" before it can be removed. Please "
-                                   "do not remove the media or disconnect the drive"),
+                                   "device \"%s\" before it can be removed."),
                                    name);
     }
 
