diff -ru abs/thunar.ORIG/src/Thunar-1.6.14/thunar/thunar-dialogs.c abs/thunar/src/Thunar-1.6.14/thunar/thunar-dialogs.c
--- abs/thunar.ORIG/src/Thunar-1.6.14/thunar/thunar-dialogs.c	2017-11-26 22:28:48.118946767 -0800
+++ abs/thunar/src/Thunar-1.6.14/thunar/thunar-dialogs.c	2017-11-26 22:28:48.245610727 -0800
@@ -66,7 +66,6 @@
   ThunarJob         *job = NULL;
   GtkWidget         *dialog;
   GtkWidget         *entry;
-  GtkWidget         *label;
   GtkWidget         *image;
   GtkWidget         *table;
   GtkWindow         *window;
@@ -106,26 +105,21 @@
   if (G_UNLIKELY (window == NULL && screen != NULL))
     gtk_window_set_screen (GTK_WINDOW (dialog), screen);
 
-  table = g_object_new (GTK_TYPE_TABLE, "border-width", 6, "column-spacing", 6, "row-spacing", 3, NULL);
+  table = g_object_new (GTK_TYPE_TABLE, "border-width", 6, "column-spacing", 6, "row-spacing", 0, NULL);
   gtk_box_pack_start (GTK_BOX (GTK_DIALOG (dialog)->vbox), table, TRUE, TRUE, 0);
   gtk_widget_show (table);
 
   icon_theme = gtk_icon_theme_get_for_screen (gtk_widget_get_screen (dialog));
   icon_factory = thunar_icon_factory_get_for_icon_theme (icon_theme);
-  icon = thunar_icon_factory_load_file_icon (icon_factory, file, THUNAR_FILE_ICON_STATE_DEFAULT, 48);
+  icon = thunar_icon_factory_load_file_icon (icon_factory, file, THUNAR_FILE_ICON_STATE_DEFAULT, 24);
   g_object_unref (G_OBJECT (icon_factory));
 
   image = gtk_image_new_from_pixbuf (icon);
-  gtk_misc_set_padding (GTK_MISC (image), 6, 6);
+  gtk_misc_set_padding (GTK_MISC (image), 2, 2);
   gtk_table_attach (GTK_TABLE (table), image, 0, 1, 0, 2, GTK_FILL, GTK_FILL, 0, 0);
   g_object_unref (G_OBJECT (icon));
   gtk_widget_show (image);
 
-  label = gtk_label_new (_("Enter the new name:"));
-  gtk_misc_set_alignment (GTK_MISC (label), 0.0f, 0.5f);
-  gtk_table_attach (GTK_TABLE (table), label, 1, 2, 0, 1, GTK_EXPAND | GTK_FILL, GTK_FILL, 0, 0);
-  gtk_widget_show (label);
-
   entry = gtk_entry_new ();
   gtk_entry_set_activates_default (GTK_ENTRY (entry), TRUE);
   gtk_table_attach (GTK_TABLE (table), entry, 1, 2, 1, 2, GTK_EXPAND | GTK_FILL, GTK_FILL, 0, 0);
@@ -168,7 +162,7 @@
     }
 
   /* resize the dialog to make long names fit as much as possible */
-  gtk_window_set_default_size (GTK_WINDOW (dialog), CLAMP (layout_width, 300, parent_width), -1);
+  gtk_window_set_default_size (GTK_WINDOW (dialog), CLAMP (layout_width, 400, parent_width), -1);
 
   /* run the dialog */
   response = gtk_dialog_run (GTK_DIALOG (dialog));
