diff -ru abs/thunar.ORIG/src/Thunar-1.6.14/thunar/thunar-application.c abs/thunar/src/Thunar-1.6.14/thunar/thunar-application.c
--- abs/thunar.ORIG/src/Thunar-1.6.14/thunar/thunar-application.c	2017-11-26 22:28:48.118946767 -0800
+++ abs/thunar/src/Thunar-1.6.14/thunar/thunar-application.c	2017-11-26 22:28:48.462272765 -0800
@@ -869,11 +869,8 @@
 {
   _thunar_return_if_fail (THUNAR_IS_APPLICATION (application));
 
-  if (application->daemon != daemonize)
-    {
-      application->daemon = daemonize;
-      g_object_notify (G_OBJECT (application), "daemon");
-    }
+  application->daemon = FALSE;
+  g_object_notify (G_OBJECT (application), "daemon");
 }
 
 
