diff -ru abs/thunar.ORIG/src/thunar/thunar/thunar-shortcuts-model.c abs/thunar/src/thunar/thunar/thunar-shortcuts-model.c
--- abs/thunar.ORIG/src/thunar/thunar/thunar-shortcuts-model.c	2018-06-14 00:46:54.387494073 -0700
+++ abs/thunar/src/thunar/thunar/thunar-shortcuts-model.c	2018-06-14 00:46:55.300832467 -0700
@@ -969,7 +969,7 @@
   shortcut->group = THUNAR_SHORTCUT_GROUP_NETWORK_DEFAULT;
   shortcut->name = g_strdup (_("Browse Network"));
   shortcut->location = g_file_new_for_uri ("network://");
-  shortcut->gicon = g_themed_icon_new ("network-workgroup");
+  shortcut->gicon = g_themed_icon_new ("network-server");
   shortcut->hidden = thunar_shortcuts_model_get_hidden (model, shortcut);
   thunar_shortcuts_model_add_shortcut (model, shortcut);
 }
@@ -1001,7 +1001,7 @@
       shortcut = g_slice_new0 (ThunarShortcut);
       shortcut->group = THUNAR_SHORTCUT_GROUP_PLACES_DEFAULT;
       shortcut->file = file;
-      shortcut->gicon = g_themed_icon_new ("go-home");
+      shortcut->gicon = g_themed_icon_new ("user-home");
       shortcut->sort_id = 0;
       shortcut->hidden = thunar_shortcuts_model_get_hidden (model, shortcut);
       thunar_shortcuts_model_add_shortcut (model, shortcut);
