diff -ru abs/gnupg.ORIG/PKGBUILD abs/gnupg/PKGBUILD
--- abs/gnupg.ORIG/PKGBUILD	2016-02-01 11:59:29.024110181 -0800
+++ abs/gnupg/PKGBUILD	2016-02-01 11:59:48.493395031 -0800
@@ -43,6 +43,7 @@
 		--libexecdir=/usr/lib/gnupg \
 		--enable-maintainer-mode \
 		--enable-symcryptrun \
+		--enable-large-secmem
 
 	make
 }
diff -ru abs/gnupg.ORIG/src/gnupg-2.1.12/g10/keygen.c abs/gnupg/src/gnupg-2.1.12/g10/keygen.c
--- abs/gnupg.ORIG/src/gnupg-2.1.12/g10/keygen.c	2016-02-01 11:59:29.027443392 -0800
+++ abs/gnupg/src/gnupg-2.1.12/g10/keygen.c	2016-02-01 11:59:34.787231408 -0800
@@ -2048,6 +2048,10 @@
       def=255;
       max=441;
       break;
+
+    case PUBKEY_ALGO_RSA:
+      if (opt.flags.large_rsa) { max = 8192; }
+      break;
     }
 
   tty_printf(_("%s keys may be between %u and %u bits long.\n"),
