diff -ru abs/gnupg.ORIG/PKGBUILD abs/gnupg/PKGBUILD
--- abs/gnupg.ORIG/PKGBUILD	2016-01-13 13:58:50.020326376 -0800
+++ abs/gnupg/PKGBUILD	2016-01-13 14:02:03.656468401 -0800
@@ -43,6 +43,7 @@
 		--libexecdir=/usr/lib/gnupg \
 		--enable-maintainer-mode \
 		--enable-symcryptrun \
+		--enable-large-secmem
 
 	make
 }
diff -ru abs/gnupg.ORIG/src/gnupg-2.1.10/g10/keygen.c abs/gnupg/src/gnupg-2.1.10/g10/keygen.c
--- abs/gnupg.ORIG/src/gnupg-2.1.10/g10/keygen.c	2016-01-13 13:58:50.026992911 -0800
+++ abs/gnupg/src/gnupg-2.1.10/g10/keygen.c	2016-01-13 13:59:57.122330092 -0800
@@ -2054,6 +2054,7 @@
 
     case PUBKEY_ALGO_RSA:
       min=1024;
+      if (opt.flags.large_rsa) { max = 8192; }
       break;
     }
 
