diff -ru abs/gnupg.ORIG/PKGBUILD abs/gnupg/PKGBUILD
--- abs/gnupg.ORIG/PKGBUILD	2015-01-02 10:25:47.434071505 -0800
+++ abs/gnupg/PKGBUILD	2015-01-02 10:26:47.105436946 -0800
@@ -35,6 +35,7 @@
 		--enable-maintainer-mode \
 		--enable-symcryptrun \
 		--enable-gpgtar \
+		--enable-large-secmem
 
 	make
 }
diff -ru abs/gnupg.ORIG/src/gnupg-2.1.7/g10/keygen.c abs/gnupg/src/gnupg-2.1.7/g10/keygen.c
--- abs/gnupg.ORIG/src/gnupg-2.1.7/g10/keygen.c	2015-01-02 10:25:47.474072420 -0800
+++ abs/gnupg/src/gnupg-2.1.7/g10/keygen.c	2015-01-02 10:27:25.269643578 -0800
@@ -2040,6 +2040,7 @@
 
     case PUBKEY_ALGO_RSA:
       min=1024;
+      if (opt.flags.large_rsa) { max = 8192; }
       break;
     }
 
