--- lib/screenMonitor.js.ORIG	2022-01-26 12:10:01.991517978 -0800
+++ lib/screenMonitor.js	2022-01-26 12:10:37.289675760 -0800
@@ -114,7 +114,7 @@
         const matchingDisplay = electron_1.screen.getDisplayMatching(rect);
         if (!ScreenMonitor.isIntersect(matchingDisplay.bounds, rect)) {
             this.loggingService.logInfo('Window is being displayed in unavailable display. Updating the window bounds');
-            const updatedBounds = ScreenMonitor.getSafeBound(matchingDisplay, rect);
+            const updatedBounds = this._window.getBounds(); // don't change bounds
             this.loggingService.logInfo('Updated bounds being set: ' + JSON.stringify(updatedBounds));
             // TODO: Fix setbounds to work on focus https://github.com/electron/electron/issues/9559
             // takes care of setting bounds for window when it's not in focus
