diff -ru abs/hexchat.ORIG/src/hexchat-2.14.1/src/fe-gtk/maingui.c abs/hexchat/src/hexchat-2.14.1/src/fe-gtk/maingui.c
--- abs/hexchat.ORIG/src/hexchat-2.14.1/src/fe-gtk/maingui.c	2017-03-24 14:46:09.860589023 -0700
+++ abs/hexchat/src/hexchat-2.14.1/src/fe-gtk/maingui.c	2017-03-24 14:55:34.147672270 -0700
@@ -3511,6 +3511,8 @@
 		gui->is_tab = TRUE;
 	}
 
+	gtk_window_set_skip_taskbar_hint (GTK_WINDOW (sess->gui->window), TRUE);
+
 	mg_add_chan (sess);
 
 	if (first_run || (prefs.hex_gui_tab_newtofront == FOCUS_NEW_ONLY_ASKED && focus)
