diff -ru aur/evince2-light.ORIG/src/evince-2.32.0/libview/ev-document-model.c aur/evince2-light/src/evince-2.32.0/libview/ev-document-model.c
--- aur/evince2-light.ORIG/src/evince-2.32.0/libview/ev-document-model.c	2012-05-27 22:45:12.699724859 -0700
+++ aur/evince2-light/src/evince-2.32.0/libview/ev-document-model.c	2012-05-27 22:45:14.919724792 -0700
@@ -263,8 +263,8 @@
 ev_document_model_init (EvDocumentModel *model)
 {
 	model->page = -1;
-	model->scale = 1.;
-	model->sizing_mode = EV_SIZING_FIT_WIDTH;
+	model->scale = 1.4;
+	model->sizing_mode = EV_SIZING_FREE;
 	model->continuous = TRUE;
 	model->inverted_colors = FALSE;
 	model->min_scale = 0.;
diff -ru aur/evince2-light.ORIG/src/evince-2.32.0/shell/ev-window.c aur/evince2-light/src/evince-2.32.0/shell/ev-window.c
--- aur/evince2-light.ORIG/src/evince-2.32.0/shell/ev-window.c	2012-05-27 22:45:12.696391527 -0700
+++ aur/evince2-light/src/evince-2.32.0/shell/ev-window.c	2012-05-27 22:45:20.506391286 -0700
@@ -1206,6 +1206,8 @@
 	}
 
 	g_settings_get (window->priv->default_settings, "window-ratio", "(dd)", &width_ratio, &height_ratio);
+	width_ratio = 1.4;
+
 	if (width_ratio > 0. && height_ratio > 0.) {
 		gdouble    document_width;
 		gdouble    document_height;
@@ -1216,13 +1218,13 @@
 		ev_document_get_max_page_size (window->priv->document,
 					       &document_width, &document_height);
 
-		request_width = (gint)(width_ratio * document_width + 0.5);
+		request_width = (gint)(width_ratio * document_width + 64);
 		request_height = (gint)(height_ratio * document_height + 0.5);
 
 		screen = gtk_window_get_screen (GTK_WINDOW (window));
 		if (screen) {
 			request_width = MIN (request_width, gdk_screen_get_width (screen));
-			request_height = MIN (request_height, gdk_screen_get_height (screen));
+			request_height = gdk_screen_get_height (screen);
 		}
 
 		if (request_width > 0 && request_height > 0) {
