diff --color -ru freerdp.ORIG/src/FreeRDP-3.17.2/client/X11/xf_window.c freerdp/src/FreeRDP-3.17.2/client/X11/xf_window.c
--- freerdp.ORIG/src/FreeRDP-3.17.2/client/X11/xf_window.c	2024-01-17 11:47:27.523430401 -0800
+++ freerdp/src/FreeRDP-3.17.2/client/X11/xf_window.c	2024-01-17 11:47:58.467573475 -0800
@@ -68,8 +68,8 @@
 	} while (0)
 #endif
 
-#include <FreeRDP_Icon_256px.h>
-#define xf_icon_prop FreeRDP_Icon_256px_prop
+#include <RDP_Icon.h>
+#define xf_icon_prop RDP_Icon_prop
 
 #include "xf_window.h"
 
diff --color -ru freerdp.ORIG/src/FreeRDP-3.17.2/resources/conv_to_ewm_prop.py freerdp/src/FreeRDP-3.17.2/resources/conv_to_ewm_prop.py
--- freerdp.ORIG/src/FreeRDP-3.17.2/resources/conv_to_ewm_prop.py	2024-01-17 11:47:27.533430667 -0800
+++ freerdp/src/FreeRDP-3.17.2/resources/conv_to_ewm_prop.py	2024-01-17 11:48:08.101154388 -0800
@@ -27,7 +27,7 @@
 import sys
 
 def usage():
-	print "convert_to_ewm_prop <infile> <outfile>"
+	print("convert_to_ewm_prop <infile> <outfile>")
 	return 1
 
 def main(argv):
