diff -ru freerdp.ORIG/src/FreeRDP-2.8.1/client/X11/xf_window.c freerdp/src/FreeRDP-2.8.1/client/X11/xf_window.c
--- freerdp.ORIG/src/FreeRDP-2.8.1/client/X11/xf_window.c	2020-05-29 09:55:34.968255271 -0700
+++ freerdp/src/FreeRDP-2.8.1/client/X11/xf_window.c	2020-05-29 09:55:35.018257192 -0700
@@ -66,8 +66,8 @@
 	} while (0)
 #endif
 
-#include "FreeRDP_Icon_256px.h"
-#define xf_icon_prop FreeRDP_Icon_256px_prop
+#include "RDP_Icon.h"
+#define xf_icon_prop RDP_Icon_prop
 
 #include "xf_window.h"
 
diff -ru freerdp.ORIG/src/FreeRDP-2.8.1/resources/conv_to_ewm_prop.py freerdp/src/FreeRDP-2.8.1/resources/conv_to_ewm_prop.py
--- freerdp.ORIG/src/FreeRDP-2.8.1/resources/conv_to_ewm_prop.py	2020-05-29 09:55:34.994922962 -0700
+++ freerdp/src/FreeRDP-2.8.1/resources/conv_to_ewm_prop.py	2020-05-29 09:55:35.018257192 -0700
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python2
 
 # Copyright 2011 Anthony Tong <atong@trustedcs.com>
 
