diff -ru mirage.ORIG/src/mirage-1.0_pre2/mirage.py mirage/src/mirage-1.0_pre2/mirage.py
--- mirage.ORIG/src/mirage-1.0_pre2/mirage.py	2020-12-16 15:35:10.182879296 -0800
+++ mirage/src/mirage-1.0_pre2/mirage.py	2020-12-16 15:35:30.679461191 -0800
@@ -492,7 +492,6 @@
 		self.refresh_recent_files_menu()
 		self.window.add_accel_group(self.UIManager.get_accel_group())
 		self.menubar = self.UIManager.get_widget('/MainMenu')
-		vbox.pack_start(self.menubar, False, False, 0)
 		self.toolbar = self.UIManager.get_widget('/MainToolbar')
 		vbox.pack_start(self.toolbar, False, False, 0)
 		self.layout = gtk.Layout()
@@ -1526,7 +1525,6 @@
 	def available_image_height(self):
 		height = self.window.get_size()[1]
 		if not self.fullscreen_mode:
-			height -= self.menubar.size_request()[1]
 			if self.usettings['toolbar_show']:
 				height -= self.toolbar.size_request()[1]
 			if self.usettings['statusbar_show']:
