diff -ru mirage.ORIG/src/mirage-1.0_pre2/mirage.py mirage/src/mirage-1.0_pre2/mirage.py
--- mirage.ORIG/src/mirage-1.0_pre2/mirage.py	2019-08-25 18:14:45.276140358 -0700
+++ mirage/src/mirage-1.0_pre2/mirage.py	2019-08-25 18:14:45.312808076 -0700
@@ -4503,13 +4503,7 @@
 		if len(self.image_list) == 0:
 			title = __appname__
 		else:
-			subfoldertitle = ''
-			firstimgindex_curr_subfolder = self.get_firstimgindex_curr_next_prev_subfolder(self.curr_img_in_list)[0]
-			if firstimgindex_curr_subfolder > -1:
-				currimg_subfolder = self.curr_img_in_list - firstimgindex_curr_subfolder + 1
-				numimg_curr_subfolder = self.get_numimg_subfolder(firstimgindex_curr_subfolder)
-				subfoldertitle = _("%(current)i of %(total)i") % {'current': currimg_subfolder, 'total': numimg_curr_subfolder} + ' '
-			title = __appname__ + " - " + subfoldertitle + _("[%(current)i of %(total)i]") % {'current': self.curr_img_in_list+1, 'total': len(self.image_list)} + ' ' + os.path.basename(self.currimg.name)
+			title = os.path.basename(self.currimg.name) + ' - ' + __appname__;
 
 			if self.slideshow_mode:
 				title = title + ' - ' + _('Slideshow Mode')
