diff --color -ru mirage.ORIG/src/mirage-0.11.1/mirage/__init__.py mirage/src/mirage-0.11.1/mirage/__init__.py
--- mirage.ORIG/src/mirage-0.11.1/mirage/__init__.py	2022-07-01 18:26:53.656234340 -0700
+++ mirage/src/mirage-0.11.1/mirage/__init__.py	2022-07-01 18:29:46.930867622 -0700
@@ -6405,14 +6405,8 @@
             title = "Mirage"
         else:
             title = (
-                "Mirage - "
-                + _("[%(current)i of %(total)i]")
-                % {"current": self.curr_img_in_list + 1, "total": len(self.image_list)}
-                + " "
-                + os.path.basename(self.currimg_name)
+                os.path.basename(self.currimg_name) + " - Mirage"
             )
-            if self.slideshow_mode:
-                title = title + " - " + _("Slideshow Mode")
         self.window.set_title(title)
 
     def slideshow_controls_show(self):
