diff -ru mirage.ORIG/src/mirage-1.0_pre2/mirage.py mirage/src/mirage-1.0_pre2/mirage.py
--- mirage.ORIG/src/mirage-1.0_pre2/mirage.py	2019-08-25 18:14:45.276140358 -0700
+++ mirage/src/mirage-1.0_pre2/mirage.py	2019-08-25 18:14:45.296140931 -0700
@@ -492,7 +492,6 @@
 		self.refresh_recent_files_menu()
 		self.window.add_accel_group(self.UIManager.get_accel_group())
 		self.menubar = self.UIManager.get_widget('/MainMenu')
-		vbox.pack_start(self.menubar, False, False, 0)
 		self.toolbar = self.UIManager.get_widget('/MainToolbar')
 		vbox.pack_start(self.toolbar, False, False, 0)
 		self.layout = gtk.Layout()
