diff --color -ru mirage.ORIG/src/mirage-0.11.1/mirage/__init__.py mirage/src/mirage-0.11.1/mirage/__init__.py
--- mirage.ORIG/src/mirage-0.11.1/mirage/__init__.py	2022-07-01 18:26:53.656234340 -0700
+++ mirage/src/mirage-0.11.1/mirage/__init__.py	2022-07-01 18:39:31.641007144 -0700
@@ -432,6 +432,8 @@
 
         # Create interface
         self.window = Gtk.ApplicationWindow.new(self)
+        self.window.set_show_menubar(False)
+
         self.window.insert_action_group("app", self.action_group)
         self.popup_menu.insert_action_group("app", self.action_group)
         self.fullscreen_controls.insert_action_group("app", self.action_group)
@@ -1920,7 +1922,6 @@
             self.statusbar.hide()
             self.statusbar2.hide()
             self.toolbar.hide()
-            self.window.set_show_menubar(False)
             self.thumbscroll.hide()
             self.thumbpane.hide()
             self.window.fullscreen()
@@ -1938,7 +1939,6 @@
             self.set_leave_fullscreen_visible(False)
             if self.toolbar_show:
                 self.toolbar.show()
-            self.window.set_show_menubar(True)
             if self.statusbar_show:
                 self.statusbar.show()
                 self.statusbar2.show()
