diff -ru mirage.ORIG/src/mirage-0.9.5.2/mirage.py mirage/src/mirage-0.9.5.2/mirage.py
--- mirage.ORIG/src/mirage-0.9.5.2/mirage.py	2018-02-09 15:05:10.176129994 -0800
+++ mirage/src/mirage-0.9.5.2/mirage.py	2018-02-09 15:05:10.178129942 -0800
@@ -683,7 +683,6 @@
 		self.layout.connect('drag_motion', self.motion_cb)
 		self.layout.connect('drag_data_received', self.drop_cb)
 		self.layout.add_events(gtk.gdk.KEY_PRESS_MASK | gtk.gdk.POINTER_MOTION_MASK | gtk.gdk.BUTTON_PRESS_MASK | gtk.gdk.BUTTON_MOTION_MASK | gtk.gdk.SCROLL_MASK)
-		self.layout.connect("scroll-event", self.mousewheel_scrolled)
 		self.layout.add_events(gtk.gdk.BUTTON_PRESS_MASK | gtk.gdk.KEY_PRESS_MASK)
 		self.layout.connect("button_press_event", self.button_pressed)
 		self.layout.add_events(gtk.gdk.POINTER_MOTION_MASK | gtk.gdk.POINTER_MOTION_HINT_MASK | gtk.gdk.BUTTON_RELEASE_MASK)
