--- ../abs/evince/src/evince-2.32.0/libview/ev-view.c.ORIG	2011-02-18 19:39:04.473448093 -0800
+++ ../abs/evince/src/evince-2.32.0/libview/ev-view.c	2011-02-18 19:49:12.118143716 -0800
@@ -4397,8 +4397,8 @@
 		page_surface = ev_pixbuf_cache_get_surface (view->pixbuf_cache, page);
 
 		if (!page_surface) {
-			if (page == current_page)
-				show_loading_window (view);
+			/* if (page == current_page)
+				show_loading_window (view); */ /* gavin: don't show "Loading..." while scrolling */
 
 			*page_ready = FALSE;
 
