diff -ru evince.ORIG/src/evince/shell/ev-metadata.c evince/src/evince/shell/ev-metadata.c
--- evince.ORIG/src/evince/shell/ev-metadata.c	2021-05-24 09:58:32.391212757 -0700
+++ evince/src/evince/shell/ev-metadata.c	2021-05-24 10:01:48.950806109 -0700
@@ -304,22 +304,5 @@
 gboolean
 ev_is_metadata_supported_for_file (GFile *file)
 {
-	GFileAttributeInfoList *namespaces;
-	gint i;
-	gboolean retval = FALSE;
-
-	namespaces = g_file_query_writable_namespaces (file, NULL, NULL);
-	if (!namespaces)
-		return retval;
-
-	for (i = 0; i < namespaces->n_infos; i++) {
-		if (strcmp (namespaces->infos[i].name, "metadata") == 0) {
-			retval = TRUE;
-			break;
-		}
-	}
-
-	g_file_attribute_info_list_unref (namespaces);
-
-	return retval;
+	return FALSE;
 }
