--- aur/physlock/src/physlock-0.4.3/main.c.ORIG	2012-09-08 22:56:35.721412015 -0700
+++ aur/physlock/src/physlock-0.4.3/main.c	2012-09-08 22:57:02.695467748 -0700
@@ -161,28 +161,15 @@
 	}
 
 	while (!auth) {
-		as = &root;
-		if (!only_root) {
-			tty_echo_on(&vt);
-			while (1) {
-				prompt(vt.ios, "\nUnlock as [%s/%s]: ", user.name, root.name);
-				if (!*buf || !strcmp(buf, user.name)) {
-					as = &user;
-					break;
-				} else if (!strcmp(buf, root.name)) {
-					break;
-				}
-			}
-			tty_echo_off(&vt);
-		} else {
-			prompt(vt.ios, "\nPress [Enter] to unlock.\n");
-		}
+  	prompt(vt.ios, "Press [Enter] to unlock\n");
+		as = only_root ? &root : &user;
 
-		prompt(vt.ios, "%s's password: ", as->name);
+		prompt(vt.ios, "[%s] Password: ", as->name);
 		auth = authenticate(as, buf);
 		if (!auth) {
-			fprintf(vt.ios, "\nAuthentication failed\n");
+			fprintf(vt.ios, "\n");
 			sleep(AUTH_FAIL_TIMEOUT);
+			fprintf(vt.ios, "\033[H\033[J");
 		}
 	}
 
