diff -ru xfce4-genmon-plugin-gtk2.ORIG/src/xfce4-genmon-plugin-3.4/panel-plugin/main.c xfce4-genmon-plugin-gtk2/src/xfce4-genmon-plugin-3.4/panel-plugin/main.c
--- xfce4-genmon-plugin-gtk2.ORIG/src/xfce4-genmon-plugin-3.4/panel-plugin/main.c	2017-02-27 10:28:57.288721142 -0800
+++ xfce4-genmon-plugin-gtk2/src/xfce4-genmon-plugin-3.4/panel-plugin/main.c	2017-02-27 10:28:57.298720960 -0800
@@ -261,6 +261,14 @@
 
 /**************************************************************/
 
+static void copy_to_clipboard (GtkWidget *widget, gpointer user_data)
+{
+    struct genmon_t *poPlugin = (genmon_t *) user_data;
+
+    GtkClipboard *clipboard = gtk_clipboard_get (GDK_SELECTION_CLIPBOARD);
+    gtk_clipboard_set_text (clipboard, poPlugin->acValue, -1);
+}
+
 static genmon_t *genmon_create_control (XfcePanelPlugin *plugin)
 /* Plugin API */
 /* Create the plugin */
@@ -270,6 +278,7 @@
     struct monitor_t *poMonitor;
     GtkOrientation orientation = xfce_panel_plugin_get_orientation (plugin);
     int size = xfce_panel_plugin_get_size (plugin);
+    GtkWidget *copy, *separator;
 
     poPlugin = g_new (genmon_t, 1);
     memset (poPlugin, 0, sizeof (genmon_t));
@@ -347,6 +356,15 @@
     else
         gtk_progress_bar_set_orientation(GTK_PROGRESS_BAR(poMonitor->wBar), GTK_PROGRESS_LEFT_TO_RIGHT);
 
+    separator = gtk_separator_menu_item_new();
+    xfce_panel_plugin_menu_insert_item (plugin, GTK_MENU_ITEM (separator));
+    gtk_widget_show (separator);
+
+    copy = gtk_menu_item_new_with_mnemonic (_("_Copy to Clipboard"));
+    g_signal_connect(G_OBJECT (copy), "activate", G_CALLBACK (copy_to_clipboard), poPlugin);
+    xfce_panel_plugin_menu_insert_item (plugin, GTK_MENU_ITEM (copy));
+    gtk_widget_show (copy);
+
     return poPlugin;
 }/* genmon_create_control() */
 
