diff -ru mkinitcpio.ORIG/src/mkinitcpio-41/functions mkinitcpio/src/mkinitcpio-41/functions
--- mkinitcpio.ORIG/src/mkinitcpio-41/functions	2026-06-23 13:41:37.489079877 -0700
+++ mkinitcpio/src/mkinitcpio-41/functions	2026-06-23 13:41:38.267116341 -0700
@@ -742,8 +742,7 @@
     fi
 
     if (( ${#firmware[*]} )); then
-        add_firmware "${firmware[@]}" ||
-            warning "Possibly missing firmware for module: '%s'" "$target"
+        add_firmware "${firmware[@]}"
     fi
 
     # handle module quirks
