--- ../abs/mkinitcpio/src/mkinitcpio-0.8.2/init_functions.ORIG	2012-01-21 14:04:34.140146135 -0800
+++ ../abs/mkinitcpio/src/mkinitcpio-0.8.2/init_functions	2012-01-21 14:04:48.140145709 -0800
@@ -15,7 +15,7 @@
     [ -b "$device" ] && return 0
 
     if [ "$udevd_running" -eq 1 ]; then
-        msg "Waiting $seconds seconds for device $device ..." >&2
+#        msg "Waiting $seconds seconds for device $device ..." >&2
         while [ ! -b "$device" -a "$seconds" -gt 0 ]; do
             sleep 1
             seconds=$(( $seconds - 1 ))
