diff -ru mkinitcpio.ORIG/src/mkinitcpio-39.1/hooks/encrypt mkinitcpio/src/mkinitcpio-39.1/hooks/encrypt
--- mkinitcpio.ORIG/src/mkinitcpio-39.1/hooks/encrypt	2024-03-04 10:45:44.457249286 -0800
+++ mkinitcpio/src/mkinitcpio-39.1/hooks/encrypt	2024-03-04 10:45:44.460582676 -0800
@@ -37,6 +37,29 @@
         [ ! -f "${ckeyfile}" ] && echo "Keyfile could not be opened. Reverting to passphrase."
     fi
 
+    # support detached headers
+    cheaderfile="/crypto_header"
+    if [ -n "$cryptheader" ]; then
+        IFS=: read chdev chfs chpath <<EOF
+$cryptheader
+EOF
+        if [ "$chdev" = "rootfs" ]; then
+            cheaderfile=$chpath
+        elif resolved=$(resolve_device "${chdev}" ${rootdelay}); then
+            # Use a file on the device
+            mkdir /cheader
+            mount -r -t "$chfs" "$resolved" /cheader
+            dd if="/cheader/$chpath" of="$cheaderfile" >/dev/null 2>&1
+            umount /cheader
+        fi
+
+        if [ -f "${cheaderfile}" ]; then
+            cryptargs="${cryptargs} --header ${cheaderfile}"
+        else
+            echo "Detached header could not be opened. Ignoring."
+        fi
+    fi
+
     if [ -n "${cryptdevice}" ]; then
         DEPRECATED_CRYPT=0
         IFS=: read -r cryptdev cryptname cryptoptions <<EOF
@@ -85,7 +108,7 @@
     unset OLDIFS
 
     if resolved=$(resolve_device "${cryptdev}" "${rootdelay}"); then
-        if cryptsetup isLuks "${resolved}" >/dev/null 2>&1; then
+        if (cryptsetup isLuks "${resolved}" >/dev/null 2>&1 || cryptsetup isLuks "${cheaderfile}" >/dev/null 2>&1); then
             [ "${DEPRECATED_CRYPT}" -eq 1 ] && warn_deprecated
             dopassphrase=1
             # If keyfile exists, try to use that
@@ -103,8 +126,7 @@
                         --prompt="A password is required to access the ${cryptname} volume" \
                         --command="cryptsetup open --type luks --key-file=- ${resolved} ${cryptname} ${cryptargs} ${CSQUIET}"
                 else
-                    echo ""
-                    echo "A password is required to access the ${cryptname} volume:"
+                    echo -e "\033[2J\033[1;1H"
 
                     #loop until we get a real password
                     while ! eval cryptsetup open --type luks "${resolved}" "${cryptname}" "${cryptargs}" "${CSQUIET}"; do
@@ -159,8 +181,10 @@
         else
             err "Failed to open encryption mapping: The device ${cryptdev} is not a LUKS volume and the crypto= paramater was not specified."
         fi
+    else
+        err "Could not resolve encryption device: ${cryptdev}"
     fi
-    rm -f "${ckeyfile}"
+    rm -f "${ckeyfile}" "${cheaderfile}"
 }
 
 # vim: set ft=sh ts=4 sw=4 et:
