diff -ru mkinitcpio.ORIG/src/mkinitcpio-41/functions mkinitcpio/src/mkinitcpio-41/functions
--- mkinitcpio.ORIG/src/mkinitcpio-41/functions	2026-06-23 13:41:37.489079877 -0700
+++ mkinitcpio/src/mkinitcpio-41/functions	2026-06-23 13:41:37.900099166 -0700
@@ -1329,7 +1329,7 @@
 
     (( ${#xz_comp[*]} )) && xz -d "${xz_comp[@]}"
     (( ${#gz_comp[*]} )) && gzip -d "${gz_comp[@]}"
-    (( ${#zst_comp[*]} )) && zstd -d --rm -q "${zst_comp[@]}"
+    (( ${#zst_comp[*]} )) && zstd -d --rm -qq "${zst_comp[@]}"
 
     msg "Generating module dependencies"
     for p in "$_d_kmoduledir"/modules.{builtin,builtin.modinfo,order}; do
