diff -ru mkinitcpio.ORIG/src/mkinitcpio-38/functions mkinitcpio/src/mkinitcpio-38/functions
--- mkinitcpio.ORIG/src/mkinitcpio-38/functions	2024-03-04 10:45:44.457249286 -0800
+++ mkinitcpio/src/mkinitcpio-38/functions	2024-03-04 10:45:44.467249456 -0800
@@ -1123,7 +1123,7 @@
 
     (( ${#xz_comp[*]} )) && xz -d "${xz_comp[@]}"
     (( ${#gz_comp[*]} )) && gzip -d "${gz_comp[@]}"
-    (( ${#zst_comp[*]} )) && zstd -d --rm -q "${zst_comp[@]}"
+    (( ${#zst_comp[*]} )) && zstd -d --rm -qq "${zst_comp[@]}"
 
     msg "Generating module dependencies"
     map add_file "$_d_kmoduledir"/modules.{builtin,builtin.modinfo,order}
