diff --color -ru xsecurelock.ORIG/src/xsecurelock-1.9.0/helpers/auth_x11.c xsecurelock/src/xsecurelock-1.9.0/helpers/auth_x11.c
--- xsecurelock.ORIG/src/xsecurelock-1.9.0/helpers/auth_x11.c	2022-10-16 13:45:40.120309550 -0700
+++ xsecurelock/src/xsecurelock-1.9.0/helpers/auth_x11.c	2022-10-16 17:17:00.520982443 -0700
@@ -85,7 +85,7 @@
 #define PARANOID_PASSWORD_MIN_CHANGE 4
 
 //! Border of the window around the text.
-#define WINDOW_BORDER 16
+#define WINDOW_BORDER 64
 
 //! Draw border rectangle (mainly for debugging).
 #undef DRAW_BORDER
@@ -786,19 +786,23 @@
 }
 
 void BuildTitle(char *output, size_t output_size, const char *input) {
-  if (show_username) {
-    size_t username_len = strlen(username);
-    StrAppend(&output, &output_size, username, username_len);
-  }
+  int show_prefix = strcmp(input, " ") != 0;
 
-  if (show_username && show_hostname) {
-    StrAppend(&output, &output_size, "@", 1);
-  }
+  if (show_prefix) {
+    if (show_username) {
+      size_t username_len = strlen(username);
+      StrAppend(&output, &output_size, username, username_len);
+    }
 
-  if (show_hostname) {
-    size_t hostname_len =
-        show_hostname > 1 ? strlen(hostname) : strcspn(hostname, ".");
-    StrAppend(&output, &output_size, hostname, hostname_len);
+    if (show_username && show_hostname) {
+      StrAppend(&output, &output_size, "@", 1);
+    }
+
+    if (show_hostname) {
+      size_t hostname_len =
+          show_hostname > 1 ? strlen(hostname) : strcspn(hostname, ".");
+      StrAppend(&output, &output_size, hostname, hostname_len);
+    }
   }
 
   if (*input == 0) {
@@ -806,9 +810,12 @@
     return;
   }
 
-  if (show_username || show_hostname) {
-    StrAppend(&output, &output_size, " - ", 3);
+  if (show_prefix) {
+    if (show_username || show_hostname) {
+      StrAppend(&output, &output_size, " - ", 3);
+    }
   }
+
   strncpy(output, input, output_size - 1);
   output[output_size - 1] = 0;
 }
@@ -941,24 +948,11 @@
                len_full_title);
     y += th * 2;
 
-    DrawString(i, cx - tw_str / 2, y, is_warning, str, len_str);
-    y += th;
-
-    DrawString(i, cx - tw_indicators / 2, y, indicators_warning, indicators,
-               len_indicators);
-    y += th;
-
-    if (have_multiple_layouts) {
-      DrawString(i, cx - tw_switch_layout / 2, y, 0, switch_layout,
-                 len_switch_layout);
-      y += th;
+    if (len_str > 0) {
+      DrawString(i, cx - tw_str / 2, y, is_warning, str, len_str);
     }
 
-    if (have_switch_user_command) {
-      DrawString(i, cx - tw_switch_user / 2, y, 0, switch_user,
-                 len_switch_user);
-      // y += th;
-    }
+    y += th;
   }
 
   // Make the things just drawn appear on the screen as soon as possible.
@@ -1469,10 +1463,10 @@
         }
         explicit_bzero(message, strlen(message));
         free(message);
-        DisplayMessage("Processing...", "", 0);
+        DisplayMessage(" ", "", 0);
         break;
       case PTYPE_PROMPT_LIKE_PASSWORD:
-        if (Prompt(message, &response, 0)) {
+        if (Prompt("", &response, 0)) {
           WritePacket(responsefd[1], PTYPE_RESPONSE_LIKE_PASSWORD, response);
           explicit_bzero(response, strlen(response));
           free(response);
@@ -1481,7 +1475,7 @@
         }
         explicit_bzero(message, strlen(message));
         free(message);
-        DisplayMessage("Processing...", "", 0);
+        DisplayMessage(" ", "", 0);
         break;
       case 0:
         goto done;
