diff -ru xsecurelock.ORIG/src/xsecurelock-1.7.0/main.c xsecurelock/src/xsecurelock-1.7.0/main.c
--- xsecurelock.ORIG/src/xsecurelock-1.7.0/main.c	2020-09-13 16:43:02.450907782 -0700
+++ xsecurelock/src/xsecurelock-1.7.0/main.c	2020-09-13 16:43:03.814236570 -0700
@@ -874,7 +874,7 @@
           display, root_window, incompatible_compositor_bits,
           incompatible_compositor_width, incompatible_compositor_height,
           BlackPixel(display, DefaultScreen(display)),
-          WhitePixel(display, DefaultScreen(display)),
+          BlackPixel(display, DefaultScreen(display)),
           DefaultDepth(display, DefaultScreen(display)));
       obscurer_window = XCreateWindow(
           display, root_window, 1, 1, w - 2, h - 2, 0, CopyFromParent,
