diff --color -ru xsecurelock.ORIG/src/xsecurelock-1.8.0/helpers/saver_multiplex.c xsecurelock/src/xsecurelock-1.8.0/helpers/saver_multiplex.c
--- xsecurelock.ORIG/src/xsecurelock-1.8.0/helpers/saver_multiplex.c	2022-09-26 22:43:10.678312310 -0700
+++ xsecurelock/src/xsecurelock-1.8.0/helpers/saver_multiplex.c	2022-09-26 23:29:55.579713586 -0700
@@ -60,7 +60,7 @@
     windows[i] =
         XCreateWindow(display, parent, monitors[i].x, monitors[i].y,
                       monitors[i].width, monitors[i].height, 0, CopyFromParent,
-                      InputOutput, CopyFromParent, 0, NULL);
+                      InputOutput, CopyFromParent, CWBackPixel, NULL);
     SetWMProperties(display, windows[i], "xsecurelock",
                     "saver_multiplex_screen", argc, argv);
     XMapRaised(display, windows[i]);
