diff -ru xsecurelock.ORIG/src/xsecurelock-1.9.0/main.c xsecurelock/src/xsecurelock-1.9.0/main.c
--- xsecurelock.ORIG/src/xsecurelock-1.9.0/main.c	2022-09-26 22:40:12.659706615 -0700
+++ xsecurelock/src/xsecurelock-1.9.0/main.c	2022-09-26 22:40:12.659706615 -0700
@@ -899,7 +899,7 @@
           display, root_window, incompatible_compositor_bits,
           incompatible_compositor_width, incompatible_compositor_height,
           BlackPixel(display, DefaultScreen(display)),
-          WhitePixel(display, DefaultScreen(display)),
+          BlackPixel(display, DefaultScreen(display)),
           DefaultDepth(display, DefaultScreen(display)));
       obscurer_window = XCreateWindow(
           display, root_window, 1, 1, w - 2, h - 2, 0, CopyFromParent,
