diff --color -ru xf86-input-synaptics.ORIG/src/xf86-input-synaptics-1.9.2/src/synaptics.c xf86-input-synaptics/src/xf86-input-synaptics-1.9.2/src/synaptics.c
--- xf86-input-synaptics.ORIG/src/xf86-input-synaptics-1.9.2/src/synaptics.c	2022-07-22 20:06:08.477251040 -0700
+++ xf86-input-synaptics/src/xf86-input-synaptics-1.9.2/src/synaptics.c	2022-07-22 20:08:16.233919682 -0700
@@ -2642,39 +2642,17 @@
     BUG_RETURN_VAL(hw->num_mt_mask > sizeof(close_point) * 8, 0);
 
     for (i = 0; i < hw->num_mt_mask - 1; i++) {
-        ValuatorMask *f1;
-
         if (hw->slot_state[i] == SLOTSTATE_EMPTY ||
             hw->slot_state[i] == SLOTSTATE_CLOSE)
             continue;
 
-        f1 = hw->mt_mask[i];
-
         for (j = i + 1; j < hw->num_mt_mask; j++) {
-            ValuatorMask *f2;
-            double x1, x2, y1, y2;
-
             if (hw->slot_state[j] == SLOTSTATE_EMPTY ||
                 hw->slot_state[j] == SLOTSTATE_CLOSE)
                 continue;
 
-            f2 = hw->mt_mask[j];
-
-            x1 = valuator_mask_get_double(f1, 0);
-            y1 = valuator_mask_get_double(f1, 1);
-
-            x2 = valuator_mask_get_double(f2, 0);
-            y2 = valuator_mask_get_double(f2, 1);
-
-            /* FIXME: fingers closer together than 30% of touchpad width, but
-             * really, this should be dependent on the touchpad size. Also,
-             * you'll need to find a touchpad that doesn't lie about it's
-             * size. Good luck. */
-            if (fabs(x1 - x2) < (priv->maxx - priv->minx) * .3 &&
-                fabs(y1 - y2) < (priv->maxy - priv->miny) * .3) {
-                close_point |= (1 << j);
-                close_point |= (1 << i);
-            }
+            close_point |= (1 << j);
+            close_point |= (1 << i);
         }
     }
 
