diff --color -ru xfce4-panel.ORIG/src/xfce4-panel/plugins/actions/actions.c xfce4-panel/src/xfce4-panel/plugins/actions/actions.c
--- xfce4-panel.ORIG/src/xfce4-panel/plugins/actions/actions.c	2024-12-23 11:47:52.087257979 -0800
+++ xfce4-panel/src/xfce4-panel/plugins/actions/actions.c	2024-12-23 12:24:40.573000071 -0800
@@ -1091,6 +1091,14 @@
 
 
 static void
+actions_plugin_end_session ()
+{
+  system("/usr/local/bin/xfce4-session-end");
+}
+
+
+
+static void
 actions_plugin_action_activate (GtkWidget *widget,
                                 ActionsPlugin *plugin)
 {
@@ -1112,21 +1120,25 @@
   switch (entry->type)
     {
     case ACTION_TYPE_LOGOUT:
+      actions_plugin_end_session ();
       succeed = actions_plugin_action_dbus_xfsm (plugin, "Logout", FALSE,
                                                  !unattended, &error);
       break;
 
     case ACTION_TYPE_LOGOUT_DIALOG:
+      actions_plugin_end_session ();
       succeed = actions_plugin_action_dbus_xfsm (plugin, "Logout", TRUE,
                                                  !unattended, &error);
       break;
 
     case ACTION_TYPE_RESTART:
+      actions_plugin_end_session ();
       succeed = actions_plugin_action_dbus_xfsm (plugin, "Restart", FALSE,
                                                  !unattended, &error);
       break;
 
     case ACTION_TYPE_SHUTDOWN:
+      actions_plugin_end_session ();
       succeed = actions_plugin_action_dbus_xfsm (plugin, "Shutdown", FALSE,
                                                  !unattended, &error);
       break;
