diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.18.1/plugins/pager/pager.c xfce4-panel/src/xfce4-panel-4.18.1/plugins/pager/pager.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.18.1/plugins/pager/pager.c	2022-12-15 20:02:45.618759582 -0800
+++ xfce4-panel/src/xfce4-panel-4.18.1/plugins/pager/pager.c	2022-12-15 20:02:45.742098441 -0800
@@ -890,14 +890,13 @@
     {
       n_workspaces = wnck_screen_get_workspace_count (plugin->wnck_screen);
       n_cols = MAX (1, (n_workspaces + plugin->rows - 1) / plugin->rows);
-      min_width = nat_width = (gint) (xfce_panel_plugin_get_size (XFCE_PANEL_PLUGIN (plugin)) / plugin->rows * plugin->ratio * n_cols);
     }
 
   if (minimum_width != NULL)
-    *minimum_width = min_width;
+    *minimum_width = 2 * min_width;
 
   if (natural_width != NULL)
-    *natural_width = nat_width;
+    *natural_width = 2 * nat_width;
 }
 
 static void
