diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.16.3/plugins/tasklist/tasklist-widget.c xfce4-panel/src/xfce4-panel-4.16.3/plugins/tasklist/tasklist-widget.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.16.3/plugins/tasklist/tasklist-widget.c	2020-12-30 15:10:14.511304419 -0800
+++ xfce4-panel/src/xfce4-panel-4.16.3/plugins/tasklist/tasklist-widget.c	2020-12-30 15:10:14.894631932 -0800
@@ -46,7 +46,7 @@
 
 
 #define DEFAULT_BUTTON_SIZE          (25)
-#define DEFAULT_MAX_BUTTON_LENGTH    (200)
+#define DEFAULT_MAX_BUTTON_LENGTH    (300)
 #define DEFAULT_MIN_BUTTON_LENGTH    (DEFAULT_MAX_BUTTON_LENGTH / 4)
 #define DEFAULT_ICON_LUCENCY         (50)
 #define DEFAULT_ELLIPSIZE_MODE       (PANGO_ELLIPSIZE_END)
@@ -1729,9 +1729,14 @@
             }
         }
 
-      /* set the toggle button state */
-      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button),
-                                    !!(child->window == active_window));
+      /* bold only active tasks */
+      if (child->window == active_window) {
+        gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), TRUE);
+        gtk_widget_modify_font (GTK_LABEL (child->label), pango_font_description_from_string ("medium"));
+      } else {
+          gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), FALSE);
+          gtk_widget_modify_font (GTK_LABEL (child->label), pango_font_description_from_string (""));
+      }
     }
   /* set the toggle button state for the group button */
   if (class_group)
@@ -2247,14 +2252,15 @@
                                                   | GDK_SMOOTH_SCROLL_MASK);
 
   child->box = gtk_box_new (!xfce_tasklist_vertical (tasklist) ?
-      GTK_ORIENTATION_HORIZONTAL : GTK_ORIENTATION_VERTICAL, 6);
+      GTK_ORIENTATION_HORIZONTAL : GTK_ORIENTATION_VERTICAL, 0);
   gtk_container_add (GTK_CONTAINER (child->button), child->box);
   gtk_widget_show (child->box);
 
   provider = gtk_css_provider_new ();
   /* silly workaround for gtkcss only accepting "." as decimal separator and floats returning
      with "," as decimal separator in some locales */
-  css_string = g_strdup_printf ("image { padding: 3px; } image.minimized { opacity: %d.%02d; }",
+  css_string = g_strdup_printf ("image { padding: %s; } image.minimized { opacity: %d.%02d; }",
+                                (tasklist->show_labels ? "3px 0 3px 8px" : "3px"),
                                 tasklist->minimized_icon_lucency / 100,
                                 tasklist->minimized_icon_lucency % 100);
   gtk_css_provider_load_from_data (provider, css_string, -1, NULL);
@@ -2273,7 +2279,7 @@
     gtk_widget_show (child->icon);
 
   child->label = gtk_label_new (NULL);
-  gtk_box_pack_start (GTK_BOX (child->box), child->label, TRUE, TRUE, 0);
+  gtk_box_pack_start (GTK_BOX (child->box), child->label, TRUE, TRUE, 6);
   if (!xfce_tasklist_vertical (tasklist))
     {
       /* gtk_box_reorder_child (GTK_BOX (child->box), child->icon, 0); */
@@ -2290,13 +2296,6 @@
       /* TODO can we already ellipsize here yet? */
     }
 
-  provider = gtk_css_provider_new ();
-  gtk_css_provider_load_from_data (provider, ".label-hidden { opacity: 0.75; }", -1, NULL);
-  gtk_style_context_add_provider (gtk_widget_get_style_context (child->label),
-                                  GTK_STYLE_PROVIDER (provider),
-                                  GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
-  g_object_unref (provider);
-
   /* don't show the label if we're in iconbox style */
   if (tasklist->show_labels)
     gtk_widget_show (child->label);
@@ -2691,16 +2690,7 @@
   ctx = gtk_widget_get_style_context (child->label);
   gtk_style_context_remove_class (ctx, "label-hidden");
 
-  if (child->tasklist->label_decorations)
-    {
-      /* create the button label */
-      if (!child->tasklist->only_minimized
-          && wnck_window_is_minimized (child->window))
-        name = label = g_strdup_printf ("[%s]", name);
-      else if (wnck_window_is_shaded (child->window))
-        name = label = g_strdup_printf ("=%s=", name);
-    }
-  else
+  if (!child->tasklist->label_decorations)
     {
       if ((!child->tasklist->only_minimized
           && wnck_window_is_minimized (child->window))
@@ -2785,45 +2775,6 @@
             }
         }
     }
-
-  /* update the blinking state */
-  if (PANEL_HAS_FLAG (changed_state, WNCK_WINDOW_STATE_DEMANDS_ATTENTION)
-      || PANEL_HAS_FLAG (changed_state, WNCK_WINDOW_STATE_URGENT))
-    {
-      /* only start blinking if the window requesting urgency
-       * notification is not the active window */
-      blink = wnck_window_or_transient_needs_attention (window);
-      if (!blink || (blink && !wnck_window_is_active (window)))
-        {
-          /* if we have all_blinking set make sure we toggle visibility of the button
-           * in case the window is not in the current workspace */
-          active_ws = wnck_screen_get_active_workspace (child->tasklist->screen);
-          if (child->tasklist->all_blinking && blink
-              && !xfce_tasklist_button_visible (child, active_ws))
-            {
-              gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), FALSE);
-              gtk_widget_show (child->button);
-            }
-
-
-          /* make sure the group button can blink too */
-          if (child->type == CHILD_TYPE_GROUP_MENU)
-            {
-              /* find the child for the group */
-              g_hash_table_lookup_extended (child->tasklist->class_groups,
-                                            child->class_group,
-                                            NULL, (gpointer *) &group_child);
-              xfce_arrow_button_set_blinking (XFCE_ARROW_BUTTON (group_child->button), blink);
-            }
-          /* otherwise just let the window button blink */
-          else
-            xfce_arrow_button_set_blinking (XFCE_ARROW_BUTTON (child->button), blink);
-
-          if (child->tasklist->all_blinking
-              && !xfce_tasklist_button_visible (child, active_ws))
-            gtk_widget_hide (child->button);
-        }
-    }
 }
 
 
@@ -2841,6 +2792,7 @@
 
   /* make sure we don't have two active windows (bug #6474) */
   gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), FALSE);
+  gtk_widget_modify_font (GTK_LABEL (child->label), pango_font_description_from_string (""));
 
   if (!tasklist->all_workspaces)
     xfce_tasklist_active_workspace_changed (tasklist->screen, NULL, tasklist);
