diff --color -ru xfce4-panel.ORIG/src/xfce4-panel-4.16.5/plugins/actions/actions.c xfce4-panel/src/xfce4-panel-4.16.5/plugins/actions/actions.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.16.5/plugins/actions/actions.c	2022-10-23 11:33:50.036253342 -0700
+++ xfce4-panel/src/xfce4-panel-4.16.5/plugins/actions/actions.c	2022-10-23 12:06:05.249480969 -0700
@@ -1024,6 +1024,14 @@
 
 
 static void
+actions_plugin_end_session ()
+{
+  system("/usr/local/bin/xfce4-session-end");
+}
+
+
+
+static void
 actions_plugin_action_activate (GtkWidget      *widget,
                                 ActionsPlugin  *plugin)
 {
@@ -1052,21 +1060,25 @@
   switch (entry->type)
     {
     case ACTION_TYPE_LOGOUT:
+      actions_plugin_end_session ();
       succeed = actions_plugin_action_dbus_xfsm ("Logout", FALSE,
                                                  allow_save, &error);
       break;
 
     case ACTION_TYPE_LOGOUT_DIALOG:
+      actions_plugin_end_session ();
       succeed = actions_plugin_action_dbus_xfsm ("Logout", TRUE,
                                                  allow_save, &error);
       break;
 
     case ACTION_TYPE_RESTART:
+      actions_plugin_end_session ();
       succeed = actions_plugin_action_dbus_xfsm ("Restart", FALSE,
                                                  allow_save, &error);
       break;
 
     case ACTION_TYPE_SHUTDOWN:
+      actions_plugin_end_session ();
       succeed = actions_plugin_action_dbus_xfsm ("Shutdown", FALSE,
                                                  allow_save, &error);
       break;
