diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.16.0/plugins/systray/sn-plugin.c xfce4-panel/src/xfce4-panel-4.16.0/plugins/systray/sn-plugin.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.16.0/plugins/systray/sn-plugin.c	2020-12-30 15:12:03.366288820 -0800
+++ xfce4-panel/src/xfce4-panel-4.16.0/plugins/systray/sn-plugin.c	2020-12-30 16:50:55.775928882 -0800
@@ -359,7 +359,6 @@
                             G_CALLBACK (sn_plugin_item_added), plugin);
   g_signal_connect_swapped (plugin->backend, "item-removed",
                             G_CALLBACK (sn_plugin_item_removed), plugin);
-  sn_backend_start (plugin->backend);
 #endif
 
   /* Systray arrow button */
diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.16.0/plugins/systray/systray-box.c xfce4-panel/src/xfce4-panel-4.16.0/plugins/systray/systray-box.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.16.0/plugins/systray/systray-box.c	2020-12-30 15:12:03.366288820 -0800
+++ xfce4-panel/src/xfce4-panel-4.16.0/plugins/systray/systray-box.c	2020-12-30 16:47:48.907288000 -0800
@@ -36,7 +36,7 @@
 #include "systray-socket.h"
 #include "sn-config.h"
 
-#define SPACING    (2)
+#define SPACING    (6)
 #define OFFSCREEN  (-9999)
 
 /* some icon implementations request a 1x1 size for invisible icons */
@@ -742,8 +742,8 @@
   if (hidden_a != hidden_b)
     return hidden_a ? 1 : -1;
 
-  name_a = systray_socket_get_name (XFCE_SYSTRAY_SOCKET (a));
-  name_b = systray_socket_get_name (XFCE_SYSTRAY_SOCKET (b));
+  name_a = systray_socket_get_sort (XFCE_SYSTRAY_SOCKET (a));
+  name_b = systray_socket_get_sort (XFCE_SYSTRAY_SOCKET (b));
 
   if (name_a != NULL && g_hash_table_lookup_extended (box->names_ordered, name_a, NULL, &value))
     index_a = GPOINTER_TO_INT (value);
diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.16.0/plugins/systray/systray-socket.c xfce4-panel/src/xfce4-panel-4.16.0/plugins/systray/systray-socket.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.16.0/plugins/systray/systray-socket.c	2020-12-30 15:12:03.366288820 -0800
+++ xfce4-panel/src/xfce4-panel-4.16.0/plugins/systray/systray-socket.c	2020-12-30 16:47:48.907288000 -0800
@@ -56,6 +56,7 @@
   Window           window;
 
   gchar           *name;
+  gchar           *sort;
 
   guint            is_composited : 1;
   guint            parent_relative_bg : 1;
@@ -102,6 +103,7 @@
 {
   socket->hidden = FALSE;
   socket->name = NULL;
+  socket->sort = NULL;
 }
 
 
@@ -112,6 +114,7 @@
   SystraySocket *socket = XFCE_SYSTRAY_SOCKET (object);
 
   g_free (socket->name);
+  g_free (socket->sort);
 
   G_OBJECT_CLASS (systray_socket_parent_class)->finalize (object);
 }
@@ -418,6 +421,29 @@
 }
 
 
+
+const gchar *
+systray_socket_get_sort (SystraySocket *socket)
+{
+  panel_return_val_if_fail (XFCE_IS_SYSTRAY_SOCKET (socket), NULL);
+
+  socket->name = systray_socket_get_name(socket);
+  socket->sort = g_strdup(socket->name);
+
+  if (!g_strcmp0(socket->sort, "hexchat")) { g_strlcpy(socket->sort, "zzzz1000", 8); }
+  else if (!g_strcmp0(socket->sort, "teams")) { g_strlcpy(socket->sort, "zzzz1100", 8); }
+  else if (!g_strcmp0(socket->sort, "pidgin")) { g_strlcpy(socket->sort, "zzzz1200", 8); }
+  else if (!g_strcmp0(socket->sort, "dropbox-tray")) { g_strlcpy(socket->sort, "zzzz1300", 8); }
+  else if (!g_strcmp0(socket->sort, "parcellite")) { g_strlcpy(socket->sort, "zzzz1400", 8); }
+  else if (!g_strcmp0(socket->sort, "blueman-applet")) { g_strlcpy(socket->sort, "zzzz1500", 8); }
+  else if (!g_strcmp0(socket->sort, "networkmanager applet")) { g_strlcpy(socket->sort, "zzzz1600", 8); }
+  else if (!g_strcmp0(socket->sort, "batterymon")) { g_strlcpy(socket->sort, "zzzz1700", 8); }
+  else if (!g_strcmp0(socket->sort, "volatile")) { g_strlcpy(socket->sort, "zzzz1800", 8); }
+
+  return socket->sort;
+}
+
+
 
 Window *
 systray_socket_get_window (SystraySocket *socket)
diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.16.0/plugins/systray/systray-socket.h xfce4-panel/src/xfce4-panel-4.16.0/plugins/systray/systray-socket.h
--- xfce4-panel.ORIG/src/xfce4-panel-4.16.0/plugins/systray/systray-socket.h	2020-12-30 15:12:03.366288820 -0800
+++ xfce4-panel/src/xfce4-panel-4.16.0/plugins/systray/systray-socket.h	2020-12-30 16:47:48.907288000 -0800
@@ -48,6 +48,8 @@
 
 const gchar     *systray_socket_get_name      (SystraySocket   *socket);
 
+const gchar     *systray_socket_get_sort      (SystraySocket   *socket);
+
 Window          *systray_socket_get_window    (SystraySocket   *socket);
 
 gboolean         systray_socket_get_hidden    (SystraySocket   *socket);
