diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.18.0/plugins/launcher/launcher.c xfce4-panel/src/xfce4-panel-4.18.0/plugins/launcher/launcher.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.18.0/plugins/launcher/launcher.c	2022-12-15 20:02:45.618759582 -0800
+++ xfce4-panel/src/xfce4-panel-4.18.0/plugins/launcher/launcher.c	2022-12-15 20:02:45.695429683 -0800
@@ -2240,12 +2240,7 @@
 launcher_plugin_arrow_visibility (LauncherPlugin *plugin)
 {
   panel_return_if_fail (XFCE_IS_LAUNCHER_PLUGIN (plugin));
-
-  if (plugin->arrow_position != LAUNCHER_ARROW_INTERNAL
-       && LIST_HAS_TWO_OR_MORE_ENTRIES (plugin->items))
-    gtk_widget_show (plugin->arrow);
-  else
-    gtk_widget_hide (plugin->arrow);
+  gtk_widget_hide (plugin->arrow);
 }
 
 
