--- ../abs/xfce4-panel/src/xfce4-panel-4.8.6/plugins/tasklist/tasklist-widget.c.ORIG	2012-01-27 14:21:43.163301714 -0800
+++ ../abs/xfce4-panel/src/xfce4-panel-4.8.6/plugins/tasklist/tasklist-widget.c	2012-01-27 14:22:14.376634093 -0800
@@ -45,7 +45,7 @@
 
 
 #define DEFAULT_BUTTON_SIZE          (25)
-#define DEFAULT_MAX_BUTTON_LENGTH    (200)
+#define DEFAULT_MAX_BUTTON_LENGTH    (300) /* gavin: increase tasklist width from 200 to 300 */
 #define DEFAULT_MENU_ICON_SIZE       (16)
 #define DEFAULT_MIN_BUTTON_LENGTH    (DEFAULT_MAX_BUTTON_LENGTH / 4)
 #define DEFAULT_ICON_LUCENCY         (50)
@@ -1501,8 +1501,13 @@
         g_get_current_time (&child->last_focused);
 
       /* set the toggle button state */
-      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button),
-                                    !!(child->window == active_window));
+      if (child->window == active_window) { /* gavin: bold active tasks, unbold others */
+        gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), TRUE);
+        gtk_widget_modify_font (GTK_LABEL (child->label), pango_font_description_from_string ("bold"));
+      } else {
+        gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), FALSE);
+        gtk_widget_modify_font (GTK_LABEL (child->label), pango_font_description_from_string (""));
+      } /* END gavin: bold active tasks, unbold others */
     }
 
   /* release the lock */
@@ -2391,11 +2396,12 @@
   gtk_widget_set_tooltip_text (GTK_WIDGET (child->button), name);
 
   /* create the button label */
-  if (!child->tasklist->only_minimized
+  /* gavin: no makers on minimized/shaded */
+  /* if (!child->tasklist->only_minimized
       && wnck_window_is_minimized (child->window))
     name = label = g_strdup_printf ("[%s]", name);
   else if (wnck_window_is_shaded (child->window))
-    name = label = g_strdup_printf ("=%s=", name);
+    name = label = g_strdup_printf ("=%s=", name); */
 
   gtk_label_set_text (GTK_LABEL (child->label), name);
 
@@ -2461,8 +2467,9 @@
     }
 
   /* update the blinking state */
-  if (PANEL_HAS_FLAG (changed_state, WNCK_WINDOW_STATE_DEMANDS_ATTENTION)
-      || PANEL_HAS_FLAG (changed_state, WNCK_WINDOW_STATE_URGENT))
+  /*if (PANEL_HAS_FLAG (changed_state, WNCK_WINDOW_STATE_DEMANDS_ATTENTION)
+      || PANEL_HAS_FLAG (changed_state, WNCK_WINDOW_STATE_URGENT))*/
+  if (FALSE) /* gavin: no blinking for urgent */
     {
       /* only start blinking if the window requesting urgentcy
        * notification is not the active window */
@@ -2487,6 +2494,7 @@
 
   /* make sure we don't have two active windows (bug #6474) */
   gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), FALSE);
+  gtk_widget_modify_font (GTK_LABEL (child->label), pango_font_description_from_string ("")); /* gavin: unbold if not active */
 
   if (!tasklist->all_workspaces)
     xfce_tasklist_active_workspace_changed (tasklist->screen, NULL, tasklist);
