diff -ru abs/xfce4-panel.ORIG/src/xfce4-panel-4.10.1/plugins/applicationsmenu/applicationsmenu.c abs/xfce4-panel/src/xfce4-panel-4.10.1/plugins/applicationsmenu/applicationsmenu.c
--- abs/xfce4-panel.ORIG/src/xfce4-panel-4.10.1/plugins/applicationsmenu/applicationsmenu.c	2013-05-07 00:36:03.466822514 -0700
+++ abs/xfce4-panel/src/xfce4-panel-4.10.1/plugins/applicationsmenu/applicationsmenu.c	2013-05-07 00:38:25.756434191 -0700
@@ -246,7 +246,8 @@
   gtk_widget_show (plugin->icon);
 
   plugin->label = gtk_label_new (DEFAULT_TITLE);
-  gtk_box_pack_start (GTK_BOX (plugin->box), plugin->label, FALSE, FALSE, 0);
+	gtk_widget_modify_font (GTK_LABEL (plugin->label), pango_font_description_from_string ("bold"));
+	gtk_box_pack_start (GTK_BOX (plugin->box), plugin->label, TRUE, TRUE, 0);
   gtk_widget_show (plugin->label);
 }
 
@@ -333,10 +334,13 @@
 
     case PROP_SHOW_BUTTON_TITLE:
       plugin->show_button_title = g_value_get_boolean (value);
-      if (plugin->show_button_title)
+      if (plugin->show_button_title) {
         gtk_widget_show (plugin->label);
-      else
+        gtk_widget_hide (plugin->icon);
+      } else {
         gtk_widget_hide (plugin->label);
+        gtk_widget_show(plugin->icon);
+      }
       applications_menu_plugin_size_changed (XFCE_PANEL_PLUGIN (plugin),
           xfce_panel_plugin_get_size (XFCE_PANEL_PLUGIN (plugin)));
       return;
