diff --color -ru xfce4-panel.ORIG/src/xfce4-panel-4.16.3/plugins/launcher/launcher.c xfce4-panel/src/xfce4-panel-4.16.3/plugins/launcher/launcher.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.16.3/plugins/launcher/launcher.c	2022-02-02 15:56:54.970940814 -0800
+++ xfce4-panel/src/xfce4-panel-4.16.3/plugins/launcher/launcher.c	2022-02-02 16:28:11.551885523 -0800
@@ -1648,6 +1648,7 @@
       gtk_label_set_xalign (GTK_LABEL (label), 0.0);
       box = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 4);
       gtk_box_pack_end (GTK_BOX (box), label, TRUE, TRUE, 0);
+      gtk_widget_set_margin_end (GTK_WIDGET (box), 25);
       gtk_container_add (GTK_CONTAINER (mi), box);
       g_object_set_qdata (G_OBJECT (mi), launcher_plugin_quark, plugin);
       gtk_widget_show_all (mi);
