--- ../abs/xfce4-panel/src/xfce4-panel-4.8.6/plugins/applicationsmenu/applicationsmenu.c.ORIG	2011-02-01 13:04:48.164195832 -0800
+++ ../abs/xfce4-panel/src/xfce4-panel-4.8.6/plugins/applicationsmenu/applicationsmenu.c	2011-02-01 14:24:03.546946196 -0800
@@ -235,7 +235,8 @@
   gtk_widget_show (plugin->icon);
 
   plugin->label = gtk_label_new (DEFAULT_TITLE);
-  gtk_box_pack_start (GTK_BOX (plugin->box), plugin->label, FALSE, FALSE, 0);
+  gtk_widget_modify_font (GTK_LABEL (plugin->label), pango_font_description_from_string ("bold")); /* gavin: Xfce menu bold label */
+  gtk_box_pack_start (GTK_BOX (plugin->box), plugin->label, TRUE, TRUE, 0); /* gavin: extra padding for Xfce menu label */
   gtk_widget_show (plugin->label);
 }
 
@@ -317,10 +318,13 @@
 
     case PROP_SHOW_BUTTON_TITLE:
       plugin->show_button_title = g_value_get_boolean (value);
-      if (plugin->show_button_title)
+      if (plugin->show_button_title) { /* gavin: title or label, not both */
         gtk_widget_show (plugin->label);
-      else
+        gtk_widget_hide (plugin->icon);
+      } else {
         gtk_widget_hide (plugin->label);
+        gtk_widget_show (plugin->icon);
+      }
       applications_menu_plugin_size_changed (XFCE_PANEL_PLUGIN (plugin),
           xfce_panel_plugin_get_size (XFCE_PANEL_PLUGIN (plugin)));
       return;
