diff -ru abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.0/plugins/actions/actions.c abs/xfce4-panel/src/xfce4-panel-4.12.0/plugins/actions/actions.c
--- abs/xfce4-panel.ORIG/src/xfce4-panel-4.12.0/plugins/actions/actions.c	2013-07-01 18:06:38.974973430 -0700
+++ abs/xfce4-panel/src/xfce4-panel-4.12.0/plugins/actions/actions.c	2013-07-01 18:19:24.308348262 -0700
@@ -1071,7 +1071,7 @@
   GtkWidget           *label;
   GtkWidget           *button;
   GtkWidget           *widget;
-  const gchar         *username;
+  gchar               hostname[128];
   GtkWidget           *child;
   GtkWidget           *box;
   guint                i;
@@ -1131,16 +1131,8 @@
     }
   else
     {
-      /* get a decent username, not the glib defaults */
-      username = g_get_real_name ();
-      if (exo_str_is_empty (username)
-          || strcmp (username, "Unknown") == 0)
-        {
-          username = g_get_user_name ();
-          if (exo_str_is_empty (username)
-              || strcmp (username, "somebody") == 0)
-            username = _("John Doe");
-        }
+      gethostname(hostname, sizeof hostname);
+      hostname[0] = toupper(hostname[0]);
 
       button = xfce_arrow_button_new (GTK_ARROW_NONE);
       gtk_widget_set_name (button, "actions-button");
@@ -1151,7 +1143,8 @@
           G_CALLBACK (actions_plugin_menu), plugin);
       gtk_widget_show (button);
 
-      label = gtk_label_new (username);
+      label = gtk_label_new (hostname);
+      gtk_widget_modify_font (GTK_LABEL (label), pango_font_description_from_string ("bold"));
       gtk_container_add (GTK_CONTAINER (button), label);
       mode = xfce_panel_plugin_get_mode (XFCE_PANEL_PLUGIN (plugin));
       gtk_label_set_angle (GTK_LABEL (label),
