diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.16.1/plugins/launcher/launcher.c xfce4-panel/src/xfce4-panel-4.16.1/plugins/launcher/launcher.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.16.1/plugins/launcher/launcher.c	2021-01-14 10:16:48.404391772 -0800
+++ xfce4-panel/src/xfce4-panel-4.16.1/plugins/launcher/launcher.c	2021-01-14 10:16:50.917674731 -0800
@@ -1614,8 +1614,8 @@
   GtkArrowType    arrow_type;
   guint           n;
   GarconMenuItem *item;
-  GtkWidget      *mi, *box, *label, *image;
-  const gchar    *name, *icon_name;
+  GtkWidget      *mi, *box, *label;
+  const gchar    *name;
   GSList         *li;
 
   panel_return_if_fail (XFCE_IS_LAUNCHER_PLUGIN (plugin));
@@ -1673,32 +1673,6 @@
         gtk_menu_shell_prepend (GTK_MENU_SHELL (plugin->menu), mi);
       else
         gtk_menu_shell_append (GTK_MENU_SHELL (plugin->menu), mi);
-
-      /* set the icon if one is set */
-      icon_name = garcon_menu_item_get_icon_name (item);
-
-      if (panel_str_is_empty (icon_name))
-        {
-          /* use an empty placeholder icon */
-          image = gtk_image_new_from_icon_name ("", GTK_ICON_SIZE_MENU);
-        }
-      else if (g_path_is_absolute (icon_name))
-        {
-          /* remember the icon name for recreating the pixbuf when panel
-              size changes */
-          plugin->icon_name = g_strdup (icon_name);
-          plugin->pixbuf = gdk_pixbuf_new_from_file_at_size (icon_name, 16, 16, NULL);
-          image = gtk_image_new_from_pixbuf (plugin->pixbuf);
-        }
-      else
-        {
-          image = gtk_image_new_from_icon_name (icon_name, GTK_ICON_SIZE_MENU);
-          gtk_image_set_pixel_size (GTK_IMAGE (image), 16);
-          plugin->icon_name = NULL;
-        }
-
-      gtk_box_pack_start (GTK_BOX (box), image, FALSE, TRUE, 3);
-      gtk_widget_show (image);
     }
 }
 
