diff -ru xfce4-panel.ORIG/src/xfce4-panel/plugins/actions/actions.c xfce4-panel/src/xfce4-panel/plugins/actions/actions.c
--- xfce4-panel.ORIG/src/xfce4-panel/plugins/actions/actions.c	2023-03-23 22:46:26.410500168 -0700
+++ xfce4-panel/src/xfce4-panel/plugins/actions/actions.c	2023-03-23 22:46:26.417166935 -0700
@@ -1359,6 +1359,7 @@
       gtk_widget_show (button);
 
       label = gtk_label_new (button_title);
+      gtk_widget_override_font (GTK_WIDGET (label), pango_font_description_from_string ("semi-bold"));
       gtk_container_add (GTK_CONTAINER (button), label);
       mode = xfce_panel_plugin_get_mode (XFCE_PANEL_PLUGIN (plugin));
       gtk_label_set_angle (GTK_LABEL (label),
