diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.16.1/plugins/pager/pager.c xfce4-panel/src/xfce4-panel-4.16.1/plugins/pager/pager.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.16.1/plugins/pager/pager.c	2020-12-30 15:10:14.507971136 -0800
+++ xfce4-panel/src/xfce4-panel-4.16.1/plugins/pager/pager.c	2020-12-30 15:10:14.774633754 -0800
@@ -777,14 +777,13 @@
     {
       n_workspaces = wnck_screen_get_workspace_count (plugin->wnck_screen);
       n_cols = MAX (1, (n_workspaces + plugin->rows - 1) / plugin->rows);
-      min_width = nat_width = (gint) (xfce_panel_plugin_get_size (XFCE_PANEL_PLUGIN (plugin)) / plugin->rows * plugin->ratio * n_cols);
     }
 
   if (minimum_width != NULL)
-    *minimum_width = min_width;
+    *minimum_width = min_width - n_cols - 2;
 
   if (natural_width != NULL)
-    *natural_width = nat_width;
+    *natural_width = nat_width - n_cols - 2;
 }
 
 static void
