diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.18.1/plugins/systray/sn-plugin.c xfce4-panel/src/xfce4-panel-4.18.1/plugins/systray/sn-plugin.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.18.1/plugins/systray/sn-plugin.c	2022-12-15 20:02:45.618759582 -0800
+++ xfce4-panel/src/xfce4-panel-4.18.1/plugins/systray/sn-plugin.c	2022-12-15 20:02:45.765432820 -0800
@@ -252,8 +252,7 @@
 static void
 update_button_visibility (SnPlugin *plugin)
 {
-  gboolean visible = plugin->has_hidden_systray_items || plugin->has_hidden_sn_items;
-  gtk_widget_set_visible(GTK_WIDGET(plugin->button), visible);
+  gtk_widget_set_visible(GTK_WIDGET(plugin->button), FALSE);
 }
 
 
@@ -372,7 +371,6 @@
                             G_CALLBACK (sn_plugin_item_added), plugin);
   g_signal_connect_swapped (plugin->backend, "item-removed",
                             G_CALLBACK (sn_plugin_item_removed), plugin);
-  sn_backend_start (plugin->backend);
 #endif
 
   /* Systray arrow button */
diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.18.1/plugins/systray/systray-box.c xfce4-panel/src/xfce4-panel-4.18.1/plugins/systray/systray-box.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.18.1/plugins/systray/systray-box.c	2022-12-15 20:02:45.618759582 -0800
+++ xfce4-panel/src/xfce4-panel-4.18.1/plugins/systray/systray-box.c	2022-12-15 20:02:45.765432820 -0800
@@ -36,7 +36,7 @@
 #include "systray-socket.h"
 #include "sn-config.h"
 
-#define SPACING    (2)
+#define SPACING    (10)
 #define OFFSCREEN  (-9999)
 
 /* some icon implementations request a 1x1 size for invisible icons */
