diff -ru xfce4-panel.ORIG/src/xfce4-panel/panel/panel-item-dialog.c xfce4-panel/src/xfce4-panel/panel/panel-item-dialog.c
--- xfce4-panel.ORIG/src/xfce4-panel/panel/panel-item-dialog.c	2023-03-23 22:46:26.407166785 -0700
+++ xfce4-panel/src/xfce4-panel/panel/panel-item-dialog.c	2023-03-23 22:46:26.550502265 -0700
@@ -230,7 +230,6 @@
   entry = gtk_entry_new ();
   gtk_box_pack_start (GTK_BOX (hbox), entry, FALSE, FALSE, 0);
   gtk_label_set_mnemonic_widget (GTK_LABEL (label), entry);
-  gtk_widget_set_tooltip_text (entry, _("Enter search phrase here"));
   gtk_entry_set_icon_from_icon_name (GTK_ENTRY (entry), GTK_ENTRY_ICON_PRIMARY, "edit-find");
   gtk_widget_show (entry);
 
diff -ru xfce4-panel.ORIG/src/xfce4-panel/plugins/actions/actions.c xfce4-panel/src/xfce4-panel/plugins/actions/actions.c
--- xfce4-panel.ORIG/src/xfce4-panel/plugins/actions/actions.c	2023-03-23 22:46:26.410500168 -0700
+++ xfce4-panel/src/xfce4-panel/plugins/actions/actions.c	2023-03-23 22:46:26.550502265 -0700
@@ -1153,7 +1153,6 @@
       widget = xfce_panel_create_button ();
       gtk_button_set_relief (GTK_BUTTON (widget), GTK_RELIEF_NONE);
       g_object_set_qdata (G_OBJECT (widget), action_quark, entry);
-      gtk_widget_set_tooltip_text (widget, _(entry->display_name));
       g_signal_connect (G_OBJECT (widget), "clicked",
           G_CALLBACK (actions_plugin_action_activate), plugin);
 
diff -ru xfce4-panel.ORIG/src/xfce4-panel/plugins/applicationsmenu/applicationsmenu.c xfce4-panel/src/xfce4-panel/plugins/applicationsmenu/applicationsmenu.c
--- xfce4-panel.ORIG/src/xfce4-panel/plugins/applicationsmenu/applicationsmenu.c	2023-03-23 22:46:26.410500168 -0700
+++ xfce4-panel/src/xfce4-panel/plugins/applicationsmenu/applicationsmenu.c	2023-03-23 22:46:26.550502265 -0700
@@ -228,7 +228,6 @@
   gtk_container_add (GTK_CONTAINER (plugin), plugin->button);
   gtk_widget_set_name (plugin->button, "applicationmenu-button");
   gtk_button_set_relief (GTK_BUTTON (plugin->button), GTK_RELIEF_NONE);
-  gtk_widget_set_tooltip_text (plugin->button, DEFAULT_TITLE);
   g_signal_connect (G_OBJECT (plugin->button), "button-press-event",
       G_CALLBACK (applications_menu_plugin_menu), plugin);
 
@@ -373,8 +372,6 @@
       plugin->button_title = g_value_dup_string (value);
       gtk_label_set_text (GTK_LABEL (plugin->label),
           plugin->button_title != NULL ? plugin->button_title : "");
-      gtk_widget_set_tooltip_text (plugin->button,
-          panel_str_is_empty (plugin->button_title) ? NULL : plugin->button_title);
 
       /* check if the label still fits */
       if (xfce_panel_plugin_get_mode (XFCE_PANEL_PLUGIN (plugin)) == XFCE_PANEL_PLUGIN_MODE_DESKBAR
diff -ru xfce4-panel.ORIG/src/xfce4-panel/plugins/directorymenu/directorymenu.c xfce4-panel/src/xfce4-panel/plugins/directorymenu/directorymenu.c
--- xfce4-panel.ORIG/src/xfce4-panel/plugins/directorymenu/directorymenu.c	2023-03-23 22:46:26.410500168 -0700
+++ xfce4-panel/src/xfce4-panel/plugins/directorymenu/directorymenu.c	2023-03-23 22:46:26.550502265 -0700
@@ -294,7 +294,6 @@
       plugin->base_directory = g_file_new_for_commandline_arg (path);
 
       display_name = g_file_get_parse_name (plugin->base_directory);
-      gtk_widget_set_tooltip_text (plugin->button, display_name);
 
       panel_utils_set_atk_info (plugin->button, _("Directory Menu"), display_name);
 
@@ -1060,7 +1059,6 @@
   GFileType        file_type;
 #ifdef HAVE_GIO_UNIX
   GDesktopAppInfo *desktopinfo;
-  const gchar     *description;
 #endif
 
   panel_return_if_fail (XFCE_IS_DIRECTORY_MENU_PLUGIN (plugin));
@@ -1251,10 +1249,6 @@
 #ifdef HAVE_GIO_UNIX
           else if (G_UNLIKELY (desktopinfo != NULL))
             {
-              description = g_app_info_get_description (G_APP_INFO (desktopinfo));
-              if (!panel_str_is_empty (description))
-                gtk_widget_set_tooltip_text (mi, description);
-
               g_signal_connect_data (G_OBJECT (mi), "activate",
                   G_CALLBACK (directory_menu_plugin_menu_launch_desktop_file),
                   desktopinfo, (GClosureNotify) (void (*)(void)) g_object_unref, 0);
diff -ru xfce4-panel.ORIG/src/xfce4-panel/plugins/showdesktop/showdesktop.c xfce4-panel/src/xfce4-panel/plugins/showdesktop/showdesktop.c
--- xfce4-panel.ORIG/src/xfce4-panel/plugins/showdesktop/showdesktop.c	2023-03-23 22:46:26.410500168 -0700
+++ xfce4-panel/src/xfce4-panel/plugins/showdesktop/showdesktop.c	2023-03-23 22:46:26.550502265 -0700
@@ -242,7 +242,6 @@
   else
     text = _("Minimize all open windows and show the desktop");
 
-  gtk_widget_set_tooltip_text (GTK_WIDGET (button), text);
   panel_utils_set_atk_info (GTK_WIDGET (button), _("Show Desktop"), text);
 }
 
diff -ru xfce4-panel.ORIG/src/xfce4-panel/plugins/tasklist/tasklist-widget.c xfce4-panel/src/xfce4-panel/plugins/tasklist/tasklist-widget.c
--- xfce4-panel.ORIG/src/xfce4-panel/plugins/tasklist/tasklist-widget.c	2023-03-23 22:46:26.410500168 -0700
+++ xfce4-panel/src/xfce4-panel/plugins/tasklist/tasklist-widget.c	2023-03-23 22:46:26.550502265 -0700
@@ -2817,7 +2817,6 @@
   panel_return_if_fail (XFCE_IS_TASKLIST (child->tasklist));
 
   name = wnck_window_get_name (child->window);
-  gtk_widget_set_tooltip_text (GTK_WIDGET (child->button), name);
   gtk_widget_set_has_tooltip (GTK_WIDGET (child->button), child->tasklist->show_tooltips);
 
   ctx = gtk_widget_get_style_context (child->label);
diff -ru xfce4-panel.ORIG/src/xfce4-panel/plugins/windowmenu/windowmenu.c xfce4-panel/src/xfce4-panel/plugins/windowmenu/windowmenu.c
--- xfce4-panel.ORIG/src/xfce4-panel/plugins/windowmenu/windowmenu.c	2023-03-23 22:46:26.410500168 -0700
+++ xfce4-panel/src/xfce4-panel/plugins/windowmenu/windowmenu.c	2023-03-23 22:46:26.550502265 -0700
@@ -647,8 +647,6 @@
   if (! wnck_window_is_active (window))
     return;
 
-  gtk_widget_set_tooltip_text (plugin->icon, wnck_window_get_name (window));
-
   icon_size = xfce_panel_plugin_get_icon_size (XFCE_PANEL_PLUGIN (plugin));
   scale_factor = gtk_widget_get_scale_factor (GTK_WIDGET (plugin));
   if (icon_size < WNCK_DEFAULT_ICON_SIZE)
@@ -707,7 +705,6 @@
           icon_size = xfce_panel_plugin_get_icon_size (XFCE_PANEL_PLUGIN (plugin));
           gtk_image_set_from_icon_name (GTK_IMAGE (icon), "user-desktop", icon_size);
           gtk_image_set_pixel_size (GTK_IMAGE (icon), icon_size);
-          gtk_widget_set_tooltip_text (GTK_WIDGET (icon), _("Desktop"));
         }
     }
 }
@@ -1047,7 +1044,6 @@
 
   /* create the menu item */
   mi = panel_image_menu_item_new_with_label (name);
-  gtk_widget_set_tooltip_text (mi, tooltip);
   g_object_set_qdata (G_OBJECT (mi), window_quark, window);
   g_signal_connect (G_OBJECT (mi), "button-release-event",
       G_CALLBACK (window_menu_plugin_menu_window_item_activate), plugin);
