diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.14.0/plugins/tasklist/tasklist-widget.c xfce4-panel/src/xfce4-panel-4.14.0/plugins/tasklist/tasklist-widget.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.14.0/plugins/tasklist/tasklist-widget.c	2019-08-12 12:54:49.048828575 -0700
+++ xfce4-panel/src/xfce4-panel-4.14.0/plugins/tasklist/tasklist-widget.c	2019-08-12 12:54:49.395505737 -0700
@@ -46,7 +46,7 @@
 
 
 #define DEFAULT_BUTTON_SIZE          (25)
-#define DEFAULT_MAX_BUTTON_LENGTH    (200)
+#define DEFAULT_MAX_BUTTON_LENGTH    (300)
 #define DEFAULT_MIN_BUTTON_LENGTH    (DEFAULT_MAX_BUTTON_LENGTH / 4)
 #define DEFAULT_ICON_LUCENCY         (50)
 #define DEFAULT_ELLIPSIZE_MODE       (PANGO_ELLIPSIZE_END)
@@ -1730,9 +1730,14 @@
             }
         }
 
-      /* set the toggle button state */
-      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button),
-                                    !!(child->window == active_window));
+      /* bold only active tasks */
+      if (child->window == active_window) {
+        gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), TRUE);
+        gtk_widget_modify_font (GTK_LABEL (child->label), pango_font_description_from_string ("semibold"));
+      } else {
+          gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), FALSE);
+          gtk_widget_modify_font (GTK_LABEL (child->label), pango_font_description_from_string (""));
+      }
     }
   /* set the toggle button state for the group button */
   if (class_group)
@@ -2246,7 +2251,7 @@
                                                   | GDK_SMOOTH_SCROLL_MASK);
 
   child->box = gtk_box_new (!xfce_tasklist_vertical (tasklist) ?
-      GTK_ORIENTATION_HORIZONTAL : GTK_ORIENTATION_VERTICAL, 6);
+      GTK_ORIENTATION_HORIZONTAL : GTK_ORIENTATION_VERTICAL, 2);
   gtk_container_add (GTK_CONTAINER (child->button), child->box);
   gtk_widget_show (child->box);
 
@@ -2265,9 +2270,9 @@
   g_free (css_string);
 
   if (tasklist->show_labels)
-    gtk_box_pack_start (GTK_BOX (child->box), child->icon, FALSE, TRUE, 0);
+    gtk_box_pack_start (GTK_BOX (child->box), child->icon, FALSE, TRUE, 4);
   else
-    gtk_box_pack_start (GTK_BOX (child->box), child->icon, TRUE, TRUE, 0);
+    gtk_box_pack_start (GTK_BOX (child->box), child->icon, TRUE, TRUE, 4);
   if (tasklist->minimized_icon_lucency > 0)
     gtk_widget_show (child->icon);
 
@@ -2678,16 +2683,7 @@
   ctx = gtk_widget_get_style_context (child->label);
   gtk_style_context_remove_class (ctx, "label-hidden");
 
-  if (child->tasklist->label_decorations)
-    {
-      /* create the button label */
-      if (!child->tasklist->only_minimized
-          && wnck_window_is_minimized (child->window))
-        name = label = g_strdup_printf ("[%s]", name);
-      else if (wnck_window_is_shaded (child->window))
-        name = label = g_strdup_printf ("=%s=", name);
-    }
-  else
+  if (!child->tasklist->label_decorations)
     {
       if ((!child->tasklist->only_minimized
           && wnck_window_is_minimized (child->window))
@@ -2772,45 +2768,6 @@
             }
         }
     }
-
-  /* update the blinking state */
-  if (PANEL_HAS_FLAG (changed_state, WNCK_WINDOW_STATE_DEMANDS_ATTENTION)
-      || PANEL_HAS_FLAG (changed_state, WNCK_WINDOW_STATE_URGENT))
-    {
-      /* only start blinking if the window requesting urgency
-       * notification is not the active window */
-      blink = wnck_window_or_transient_needs_attention (window);
-      if (!blink || (blink && !wnck_window_is_active (window)))
-        {
-          /* if we have all_blinking set make sure we toggle visibility of the button
-           * in case the window is not in the current workspace */
-          active_ws = wnck_screen_get_active_workspace (child->tasklist->screen);
-          if (child->tasklist->all_blinking && blink
-              && !xfce_tasklist_button_visible (child, active_ws))
-            {
-              gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), FALSE);
-              gtk_widget_show (child->button);
-            }
-
-
-          /* make sure the group button can blink too */
-          if (child->type == CHILD_TYPE_GROUP_MENU)
-            {
-              /* find the child for the group */
-              g_hash_table_lookup_extended (child->tasklist->class_groups,
-                                            child->class_group,
-                                            NULL, (gpointer *) &group_child);
-              xfce_arrow_button_set_blinking (XFCE_ARROW_BUTTON (group_child->button), blink);
-            }
-          /* otherwise just let the window button blink */
-          else
-            xfce_arrow_button_set_blinking (XFCE_ARROW_BUTTON (child->button), blink);
-
-          if (child->tasklist->all_blinking
-              && !xfce_tasklist_button_visible (child, active_ws))
-            gtk_widget_hide (child->button);
-        }
-    }
 }
 
 
@@ -2828,6 +2785,7 @@
 
   /* make sure we don't have two active windows (bug #6474) */
   gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (child->button), FALSE);
+  gtk_widget_modify_font (GTK_LABEL (child->label), pango_font_description_from_string (""));
 
   if (!tasklist->all_workspaces)
     xfce_tasklist_active_workspace_changed (tasklist->screen, NULL, tasklist);
