--- ../abs/xfce4-panel/src/xfce4-panel-4.8.6/panel/panel-base-window.c.ORIG	2011-04-26 22:01:05.338933036 -0700
+++ ../abs/xfce4-panel/src/xfce4-panel-4.8.6/panel/panel-base-window.c	2011-04-26 22:01:08.725610803 -0700
@@ -576,8 +576,8 @@
       cairo_rectangle (cr, 0.5, 0.5, width - 1, height - 1);
       cairo_stroke (cr);
     }
-  else if (window->background_style == PANEL_BG_STYLE_NONE)
-    {
+  else /* gavin: always draw borders */
+    { 
       if (PANEL_HAS_FLAG (priv->borders, PANEL_BORDER_BOTTOM | PANEL_BORDER_RIGHT))
         {
           /* use dark color for buttom and right line */
@@ -602,7 +602,7 @@
       if (PANEL_HAS_FLAG (priv->borders, PANEL_BORDER_TOP | PANEL_BORDER_LEFT))
         {
           /* use light color for top and left line */
-          color = &(widget->style->light[GTK_STATE_NORMAL]);
+          color = &(widget->style->dark[GTK_STATE_NORMAL]); /* gavin: all borders should be dark */
           panel_util_set_source_rgba (cr, color, alpha);
 
           if (PANEL_HAS_FLAG (priv->borders, PANEL_BORDER_LEFT))
@@ -832,8 +832,7 @@
   if (priv->active_timeout_id != 0)
     return PANEL_BORDER_TOP | PANEL_BORDER_BOTTOM
            | PANEL_BORDER_LEFT | PANEL_BORDER_RIGHT;
-  else if (window->background_style != PANEL_BG_STYLE_NONE)
-    return PANEL_BORDER_NONE;
+  /* gavin: always use borders */
 
   return priv->borders;
 }
