diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.16.3/plugins/launcher/launcher.c xfce4-panel/src/xfce4-panel-4.16.3/plugins/launcher/launcher.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.16.3/plugins/launcher/launcher.c	2020-12-30 15:10:14.511304419 -0800
+++ xfce4-panel/src/xfce4-panel-4.16.3/plugins/launcher/launcher.c	2020-12-30 15:10:14.651302293 -0800
@@ -2274,12 +2274,7 @@
 launcher_plugin_arrow_visibility (LauncherPlugin *plugin)
 {
   panel_return_if_fail (XFCE_IS_LAUNCHER_PLUGIN (plugin));
-
-  if (plugin->arrow_position != LAUNCHER_ARROW_INTERNAL
-       && LIST_HAS_TWO_OR_MORE_ENTRIES (plugin->items))
-    gtk_widget_show (plugin->arrow);
-  else
-    gtk_widget_hide (plugin->arrow);
+  gtk_widget_hide (plugin->arrow);
 }
 
 
