diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.16.1/panel/panel-base-window.c xfce4-panel/src/xfce4-panel-4.16.1/panel/panel-base-window.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.16.1/panel/panel-base-window.c	2020-12-30 15:10:14.507971136 -0800
+++ xfce4-panel/src/xfce4-panel-4.16.1/panel/panel-base-window.c	2020-12-30 15:10:14.594636486 -0800
@@ -825,15 +825,6 @@
   PanelBaseWindowPrivate *priv = window->priv;
 
   panel_return_if_fail (PANEL_IS_BASE_WINDOW (window));
-
-  if (priv->borders != borders)
-    {
-      priv->borders = borders;
-      gtk_widget_queue_resize (GTK_WIDGET (window));
-      /* Re-draw the borders if system colors are being used */
-      if (window->background_style == PANEL_BG_STYLE_NONE)
-        panel_base_window_reset_background_css (window);
-    }
 }
 
 
@@ -850,5 +841,6 @@
     return PANEL_BORDER_TOP | PANEL_BORDER_BOTTOM
            | PANEL_BORDER_LEFT | PANEL_BORDER_RIGHT;
 
+  priv->borders = PANEL_BORDER_NONE;
   return priv->borders;
 }
