diff -ru xfce4-panel.ORIG/src/xfce4-panel/plugins/actions/actions.c xfce4-panel/src/xfce4-panel/plugins/actions/actions.c
--- xfce4-panel.ORIG/src/xfce4-panel/plugins/actions/actions.c	2023-03-23 22:46:26.410500168 -0700
+++ xfce4-panel/src/xfce4-panel/plugins/actions/actions.c	2023-03-23 22:46:26.653837145 -0700
@@ -1036,6 +1036,14 @@
 
 
 static void
+actions_plugin_end_session ()
+{
+  system("/usr/local/bin/xfce4-session-end");
+}
+
+
+
+static void
 actions_plugin_action_activate (GtkWidget      *widget,
                                 ActionsPlugin  *plugin)
 {
@@ -1064,21 +1072,25 @@
   switch (entry->type)
     {
     case ACTION_TYPE_LOGOUT:
+      actions_plugin_end_session ();
       succeed = actions_plugin_action_dbus_xfsm ("Logout", FALSE,
                                                  allow_save, &error);
       break;
 
     case ACTION_TYPE_LOGOUT_DIALOG:
+      actions_plugin_end_session ();
       succeed = actions_plugin_action_dbus_xfsm ("Logout", TRUE,
                                                  allow_save, &error);
       break;
 
     case ACTION_TYPE_RESTART:
+      actions_plugin_end_session ();
       succeed = actions_plugin_action_dbus_xfsm ("Restart", FALSE,
                                                  allow_save, &error);
       break;
 
     case ACTION_TYPE_SHUTDOWN:
+      actions_plugin_end_session ();
       succeed = actions_plugin_action_dbus_xfsm ("Shutdown", FALSE,
                                                  allow_save, &error);
       break;
