diff -ru abs/xfce4-panel.ORIG/src/xfce4-panel-4.10.0/panel/panel-base-window.c abs/xfce4-panel/src/xfce4-panel-4.10.0/panel/panel-base-window.c
--- abs/xfce4-panel.ORIG/src/xfce4-panel-4.10.0/panel/panel-base-window.c	2012-11-19 20:31:05.014843181 -0800
+++ abs/xfce4-panel/src/xfce4-panel-4.10.0/panel/panel-base-window.c	2012-11-19 20:31:36.579052093 -0800
@@ -579,33 +579,12 @@
       cairo_rectangle (cr, 0.5, 0.5, width - 1, height - 1);
       cairo_stroke (cr);
     }
-  else if (window->background_style == PANEL_BG_STYLE_NONE)
+  else
     {
-      if (PANEL_HAS_FLAG (priv->borders, PANEL_BORDER_BOTTOM | PANEL_BORDER_RIGHT))
-        {
-          /* use dark color for buttom and right line */
-          color = &(widget->style->dark[GTK_STATE_NORMAL]);
-          panel_util_set_source_rgba (cr, color, alpha);
-
-          if (PANEL_HAS_FLAG (priv->borders, PANEL_BORDER_BOTTOM))
-            {
-              cairo_move_to (cr, 0.50, height - 1);
-              cairo_rel_line_to (cr, width, 0.50);
-            }
-
-          if (PANEL_HAS_FLAG (priv->borders, PANEL_BORDER_RIGHT))
-            {
-              cairo_move_to (cr, width - 1, 0.50);
-              cairo_rel_line_to (cr, 0.50, height);
-            }
-
-          cairo_stroke (cr);
-        }
-
       if (PANEL_HAS_FLAG (priv->borders, PANEL_BORDER_TOP | PANEL_BORDER_LEFT))
         {
           /* use light color for top and left line */
-          color = &(widget->style->light[GTK_STATE_NORMAL]);
+          color = &(widget->style->light[GTK_STATE_INSENSITIVE]);
           panel_util_set_source_rgba (cr, color, alpha);
 
           if (PANEL_HAS_FLAG (priv->borders, PANEL_BORDER_LEFT))
@@ -840,6 +819,7 @@
   else if (window->background_style != PANEL_BG_STYLE_NONE)
     return PANEL_BORDER_NONE;
 
+  priv->borders = priv->borders & !PANEL_BORDER_BOTTOM & !PANEL_BORDER_RIGHT;
   return priv->borders;
 }
 
