diff --color -ru xfce4-panel.ORIG/src/xfce4-panel/plugins/systray/sn-plugin.c xfce4-panel/src/xfce4-panel/plugins/systray/sn-plugin.c
--- xfce4-panel.ORIG/src/xfce4-panel/plugins/systray/sn-plugin.c	2024-12-23 11:47:52.087257979 -0800
+++ xfce4-panel/src/xfce4-panel/plugins/systray/sn-plugin.c	2024-12-23 12:19:17.881677296 -0800
@@ -260,8 +260,7 @@
 static void
 update_button_visibility (SnPlugin *plugin)
 {
-  gboolean visible = plugin->has_hidden_systray_items || plugin->has_hidden_sn_items;
-  gtk_widget_set_visible (GTK_WIDGET (plugin->button), visible);
+  gtk_widget_set_visible (GTK_WIDGET(plugin->button), FALSE);
 }
 
 
@@ -384,7 +383,6 @@
                             G_CALLBACK (sn_plugin_item_added), plugin);
   g_signal_connect_swapped (plugin->backend, "item-removed",
                             G_CALLBACK (sn_plugin_item_removed), plugin);
-  sn_backend_start (plugin->backend);
 #endif
 
   /* Systray arrow button */
diff --color -ru xfce4-panel.ORIG/src/xfce4-panel/plugins/systray/systray-box.c xfce4-panel/src/xfce4-panel/plugins/systray/systray-box.c
--- xfce4-panel.ORIG/src/xfce4-panel/plugins/systray/systray-box.c	2024-12-23 11:47:52.087257979 -0800
+++ xfce4-panel/src/xfce4-panel/plugins/systray/systray-box.c	2024-12-23 12:19:25.375043812 -0800
@@ -31,7 +31,7 @@
 #include <math.h>
 #endif
 
-#define SPACING (2)
+#define SPACING (12)
 
 /* some icon implementations request a 1x1 size for invisible icons */
 #define REQUISITION_IS_INVISIBLE(child_req) ((child_req).width <= 1 && (child_req).height <= 1)
