--- ../abs/xfce4-panel/src/xfce4-panel-4.8.6/plugins/systray/systray-box.c.ORIG	2011-04-26 19:04:04.767654125 -0700
+++ ../abs/xfce4-panel/src/xfce4-panel-4.8.6/plugins/systray/systray-box.c	2011-04-26 21:48:59.394030393 -0700
@@ -36,7 +36,8 @@
 #include "systray-box.h"
 #include "systray-socket.h"
 
-#define SPACING    (2)
+/* gavin: increase spacing */
+#define SPACING    (4)
 #define OFFSCREEN  (-9999)
 
 /* some icon implementations request a 1x1 size for invisible icons */
@@ -649,9 +650,9 @@
   if (hidden_a != hidden_b)
     return hidden_a ? 1 : -1;
 
-  /* sort icons by name */
-  name_a = systray_socket_get_name (XFCE_SYSTRAY_SOCKET (a));
-  name_b = systray_socket_get_name (XFCE_SYSTRAY_SOCKET (b));
+  /* sort icons by name */ /* gavin: use sort functions here */
+  name_a = systray_socket_get_sort (XFCE_SYSTRAY_SOCKET (a));
+  name_b = systray_socket_get_sort (XFCE_SYSTRAY_SOCKET (b));
 
 #if GLIB_CHECK_VERSION (2, 16, 0)
   return g_strcmp0 (name_a, name_b);
