diff -ru xfce4-panel.ORIG/src/xfce4-panel-4.18.5/plugins/launcher/launcher.c xfce4-panel/src/xfce4-panel-4.18.5/plugins/launcher/launcher.c
--- xfce4-panel.ORIG/src/xfce4-panel-4.18.5/plugins/launcher/launcher.c	2023-03-23 22:46:26.410500168 -0700
+++ xfce4-panel/src/xfce4-panel-4.18.5/plugins/launcher/launcher.c	2023-03-23 22:46:26.493834750 -0700
@@ -2226,12 +2226,7 @@
 launcher_plugin_arrow_visibility (LauncherPlugin *plugin)
 {
   panel_return_if_fail (XFCE_IS_LAUNCHER_PLUGIN (plugin));
-
-  if (plugin->arrow_position != LAUNCHER_ARROW_INTERNAL
-       && LIST_HAS_TWO_OR_MORE_ENTRIES (plugin->items))
-    gtk_widget_show (plugin->arrow);
-  else
-    gtk_widget_hide (plugin->arrow);
+  gtk_widget_hide (plugin->arrow);
 }
 
 
