diff -ru pakku.ORIG/src/pakku/Makefile pakku/src/pakku/Makefile
--- pakku.ORIG/src/pakku/Makefile	2021-06-01 12:19:14.794764482 -0700
+++ pakku/src/pakku/Makefile	2021-06-01 12:19:46.481420768 -0700
@@ -8,7 +8,6 @@
 
 TARGETS =  \
 	completion/bash \
-	completion/zsh \
 	lib/tools \
 	src/pakku \
 	${MAN_PAGES}
@@ -141,7 +140,6 @@
 
 install:
 	$(call install,644,'completion/bash','${BASHCOMPLETIONSDIR}/pakku')
-	$(call install,644,'completion/zsh','${ZSHCOMPLETIONSDIR}/_pakku')
 	$(call install,644,'doc/pakku.8','${MANDIR}/man8/pakku.8')
 	$(call install,644,'doc/pakku.conf.5','${MANDIR}/man5/pakku.conf.5')
 	$(call install,755,'lib/tools','${PKGLIBDIR}/tools')
@@ -154,7 +152,6 @@
 
 uninstall:
 	$(call uninstall,'${BASHCOMPLETIONSDIR}','pakku')
-	$(call uninstall,'${ZSHCOMPLETIONSDIR}','_pakku')
 	$(call uninstall,'${MANDIR}/man8','pakku.8')
 	$(call uninstall,'${MANDIR}/man5','pakku.conf.5')
 	$(call uninstall,'${PKGLIBDIR}','tools')
