diff --color -ru nct6687d-dkms-git.ORIG/PKGBUILD nct6687d-dkms-git/PKGBUILD
--- nct6687d-dkms-git.ORIG/PKGBUILD	2023-05-22 12:59:51.626561534 -0700
+++ nct6687d-dkms-git/PKGBUILD	2023-05-22 13:00:38.517125455 -0700
@@ -24,11 +24,8 @@
 	cd "${pkgname}"
 
 	install -dm755 "${pkgdir}/usr/src/${pkgname}-${pkgver}"
-	cp -a LICENSE README* *.c "${pkgdir}/usr/src/${pkgname}-${pkgver}"
+	cp -a LICENSE README* *.c dkms.conf "${pkgdir}/usr/src/${pkgname}-${pkgver}"
 
 	sed -e "s/(shell uname -r)/{TARGET}/" Makefile > \
 		"${pkgdir}/usr/src/${pkgname}-${pkgver}/Makefile"
-
-	sed -e "s/@PACKAGE_VERSION@/${pkgver}/" "${srcdir}/dkms.conf.in" > \
-		"${pkgdir}/usr/src/${pkgname}-${pkgver}/dkms.conf"
 }
diff --color -ru nct6687d-dkms-git.ORIG/src/nct6687d-dkms-git/Makefile nct6687d-dkms-git/src/nct6687d-dkms-git/Makefile
--- nct6687d-dkms-git.ORIG/src/nct6687d-dkms-git/Makefile	2023-05-22 12:59:51.626561534 -0700
+++ nct6687d-dkms-git/src/nct6687d-dkms-git/Makefile	2023-05-22 13:00:08.770101148 -0700
@@ -2,9 +2,6 @@
 
 curpwd      := $(shell pwd)
 kver        := $(shell uname -r)
-commitcount := $(shell git rev-list --all --count)
-commithash  := $(shell git rev-parse --short HEAD)
-
 
 build:
 	rm -rf ${curpwd}/${kver}
