diff -ru pidgin.ORIG/src/pidgin-2.14.6/pidgin/gtkblist.c pidgin/src/pidgin-2.14.6/pidgin/gtkblist.c
--- pidgin.ORIG/src/pidgin-2.14.6/pidgin/gtkblist.c	2016-12-29 10:18:15.920693074 -0800
+++ pidgin/src/pidgin-2.14.6/pidgin/gtkblist.c	2016-12-29 10:18:15.954025765 -0800
@@ -5639,7 +5639,7 @@
 			g_signal_connect(G_OBJECT(rend), "editing-started", G_CALLBACK(gtk_blist_renderer_editing_started_cb), NULL);
 			g_signal_connect(G_OBJECT(rend), "editing-canceled", G_CALLBACK(gtk_blist_renderer_editing_cancelled_cb), list);
 			g_signal_connect(G_OBJECT(rend), "edited", G_CALLBACK(gtk_blist_renderer_edited_cb), list);
-			g_object_set(rend, "ypad", 0, "yalign", 0.5, NULL);
+			g_object_set(rend, "ypad", 3, "yalign", 0.5, NULL);
 			g_object_set(rend, "ellipsize", PANGO_ELLIPSIZE_END, NULL);
 
 			/* idle */
