diff -ru abs/pidgin.ORIG/src/pidgin-2.10.9/libpurple/certificate.c abs/pidgin/src/pidgin-2.10.9/libpurple/certificate.c
--- abs/pidgin.ORIG/src/pidgin-2.10.9/libpurple/certificate.c	2014-09-23 10:34:35.182215569 -0700
+++ abs/pidgin/src/pidgin-2.10.9/libpurple/certificate.c	2014-09-23 10:34:36.798902122 -0700
@@ -1701,7 +1701,6 @@
 	g_free(ca_id);
 	g_free(ca2_id);
 	if ( NULL == ca_crts ) {
-		flags |= PURPLE_CERTIFICATE_CA_UNKNOWN;
 
 		purple_debug_warning("certificate/x509/tls_cached",
 				  "No Certificate Authorities with either DN found "
