diff -ru pidgin.ORIG/src/pidgin-2.13.0/pidgin/gtkblist.c pidgin/src/pidgin-2.13.0/pidgin/gtkblist.c
--- pidgin.ORIG/src/pidgin-2.13.0/pidgin/gtkblist.c	2016-12-29 10:18:15.920693074 -0800
+++ pidgin/src/pidgin-2.13.0/pidgin/gtkblist.c	2016-12-29 10:18:16.937340142 -0800
@@ -5780,6 +5780,8 @@
 			 G_CALLBACK(blist_focus_cb), gtkblist);
 	GTK_WINDOW(gtkblist->window)->allow_shrink = TRUE;
 
+	gtk_window_set_skip_taskbar_hint(gtkblist->window, TRUE);
+
 	gtkblist->main_vbox = gtk_vbox_new(FALSE, 0);
 	gtk_widget_show(gtkblist->main_vbox);
 	gtk_container_add(GTK_CONTAINER(gtkblist->window), gtkblist->main_vbox);
