diff --color -ru neofetch.ORIG/src/neofetch/neofetch neofetch/src/neofetch/neofetch
--- neofetch.ORIG/src/neofetch/neofetch	2023-05-24 01:29:26.357246710 -0700
+++ neofetch/src/neofetch/neofetch	2023-05-24 01:29:54.074359304 -0700
@@ -1491,7 +1491,6 @@
             has kiss       && tot kiss l
             has pacman-key && tot pacman -Qq --color never
             has dpkg       && tot dpkg-query -f '.\n' -W
-            has rpm        && tot rpm -qa
             has xbps-query && tot xbps-query -l
             has apk        && tot apk info
             has opkg       && tot opkg list-installed
