diff --color -ru clipit.ORIG/src/ClipIt-1.4.5/src/main.c clipit/src/ClipIt-1.4.5/src/main.c
--- clipit.ORIG/src/ClipIt-1.4.5/src/main.c	2023-11-29 05:01:20.249405437 -0800
+++ clipit/src/ClipIt-1.4.5/src/main.c	2023-11-29 05:51:09.659665580 -0800
@@ -899,7 +899,6 @@
   /* Popup the menu... */
   gtk_widget_show_all(menu);
   gtk_menu_popup((GtkMenu*)menu, NULL, NULL, NULL, NULL, 1, gtk_get_current_event_time());
-  gtk_menu_shell_select_first((GtkMenuShell*)menu, TRUE);
   /* Return FALSE so the g_timeout_add() function is called only once */
   return FALSE;
 }
@@ -979,7 +978,6 @@
 	/* GENERATE THE MENU*/
 	gtk_widget_set_visible(statusicon_menu, TRUE);
 	gtk_menu_popup((GtkMenu*)statusicon_menu, NULL, NULL, gtk_status_icon_position_menu, status_icon, 1, gtk_get_current_event_time());
-	gtk_menu_shell_select_first((GtkMenuShell*)statusicon_menu, TRUE);
 
 	status_menu_lock = FALSE;
 }
