diff --color -ru clipit.ORIG/src/ClipIt-1.4.5/src/main.c clipit/src/ClipIt-1.4.5/src/main.c
--- clipit.ORIG/src/ClipIt-1.4.5/src/main.c	2023-11-29 05:01:20.249405437 -0800
+++ clipit/src/ClipIt-1.4.5/src/main.c	2023-11-29 05:24:39.033488995 -0800
@@ -47,8 +47,8 @@
 #include <stdbool.h>
 #include <string.h>
 
-#define ICON "clipit-trayicon"
-#define ICON_OFFLINE "clipit-trayicon-offline"
+#define ICON "clipboard"
+#define ICON_OFFLINE "clipboard-disabled"
 
 static gchar* primary_text;
 static gchar* clipboard_text;
