diff --color -ru clipit.ORIG/src/ClipIt-1.4.5/src/utils.c clipit/src/ClipIt-1.4.5/src/utils.c
--- clipit.ORIG/src/ClipIt-1.4.5/src/utils.c	2023-11-29 05:01:20.249405437 -0800
+++ clipit/src/ClipIt-1.4.5/src/utils.c	2023-11-29 05:11:05.941491785 -0800
@@ -105,7 +105,7 @@
 	int i = 0;
 	while (i < string->len)
 	{
-		if (string->str[i] == '\n')
+		if (string->str[i] == '\n' || string->str[i] == '\r')
 		g_string_overwrite(string, i, " ");
 		i++;
 	}
