diff -ru pre/networkmanager-openconnect.ORIG/src/NetworkManager-openconnect-0.8.6.0/src/nm-openconnect-service.h pre/networkmanager-openconnect/src/NetworkManager-openconnect-0.8.6.0/src/nm-openconnect-service.h
--- pre/networkmanager-openconnect.ORIG/src/NetworkManager-openconnect-0.8.6.0/src/nm-openconnect-service.h	2012-05-31 20:57:52.658168751 -0700
+++ pre/networkmanager-openconnect/src/NetworkManager-openconnect-0.8.6.0/src/nm-openconnect-service.h	2012-05-31 21:00:22.881497443 -0700
@@ -25,7 +25,7 @@
 #ifndef NM_OPENCONNECT_PLUGIN_H
 #define NM_OPENCONNECT_PLUGIN_H
 
-#include <glib/gtypes.h>
+#include <glib.h>
 #include <glib-object.h>
 #include <nm-vpn-plugin.h>
 
