diff -ru pre/networkmanager-openconnect.ORIG/src/NetworkManager-openconnect-0.8.6.0/configure pre/networkmanager-openconnect/src/NetworkManager-openconnect-0.8.6.0/configure
--- pre/networkmanager-openconnect.ORIG/src/NetworkManager-openconnect-0.8.6.0/configure	2012-05-31 20:57:52.661502085 -0700
+++ pre/networkmanager-openconnect/src/NetworkManager-openconnect-0.8.6.0/configure	2012-05-31 20:59:42.561498691 -0700
@@ -13289,7 +13289,7 @@
 if test "$GCC" = "yes" -a "$set_more_warnings" != "no"; then
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-	CFLAGS="-Wall -Werror -std=gnu89 $CFLAGS"
+	CFLAGS="-Wall -lm -std=gnu89 $CFLAGS"
 
 	for option in -Wshadow -Wmissing-declarations -Wmissing-prototypes \
 		      -Wdeclaration-after-statement -Wstrict-prototypes \
