diff -ru xfdesktop.ORIG/src/xfdesktop/src/xfdesktop-icon-view.c xfdesktop/src/xfdesktop/src/xfdesktop-icon-view.c
--- xfdesktop.ORIG/src/xfdesktop/src/xfdesktop-icon-view.c	2024-11-03 11:18:30.626998085 -0800
+++ xfdesktop/src/xfdesktop/src/xfdesktop-icon-view.c	2024-11-03 11:18:30.636998023 -0800
@@ -2222,7 +2222,7 @@
     GdkScreen *gscreen;
     GdkWindow *groot;
 
-    gtk_window_set_accept_focus(GTK_WINDOW(icon_view->priv->parent_window), FALSE);
+    gtk_window_set_accept_focus(GTK_WINDOW(icon_view->priv->parent_window), TRUE);
 
     gscreen = gtk_widget_get_screen(widget);
     groot = gdk_screen_get_root_window(gscreen);
