--- ../abs/xfdesktop/src/xfdesktop-4.8.3/src/xfce-desktop.c.ORIG	2011-12-18 14:48:25.246478118 -0800
+++ ../abs/xfdesktop/src/xfdesktop-4.8.3/src/xfce-desktop.c	2011-12-18 14:51:43.756472059 -0800
@@ -616,7 +616,9 @@
     GTK_WINDOW(desktop)->type = GTK_WINDOW_TOPLEVEL;
     
     gtk_window_set_type_hint(GTK_WINDOW(desktop), GDK_WINDOW_TYPE_HINT_DESKTOP);
-    gtk_window_set_accept_focus(GTK_WINDOW(desktop), FALSE);
+
+    /* gavin: always accept focus, workround for restoring windows with compiz */
+    gtk_window_set_accept_focus(GTK_WINDOW(desktop), TRUE);
 }
 
 static void
