diff -ru xfdesktop.ORIG/src/xfdesktop-4.14.3/src/xfdesktop-icon-view.c xfdesktop/src/xfdesktop-4.14.3/src/xfdesktop-icon-view.c
--- xfdesktop.ORIG/src/xfdesktop-4.14.3/src/xfdesktop-icon-view.c	2019-03-09 04:14:50.905535633 -0800
+++ xfdesktop/src/xfdesktop-4.14.3/src/xfdesktop-icon-view.c	2019-03-09 04:15:05.232615140 -0800
@@ -2025,7 +2025,7 @@
     GdkScreen *gscreen;
     GdkWindow *groot;
 
-    gtk_window_set_accept_focus(GTK_WINDOW(icon_view->priv->parent_window), FALSE);
+    gtk_window_set_accept_focus(GTK_WINDOW(icon_view->priv->parent_window), TRUE);
 
     gscreen = gtk_widget_get_screen(widget);
     groot = gdk_screen_get_root_window(gscreen);
