--- ../abs/xfdesktop/src/xfdesktop-4.8.3/src/xfdesktop-icon-view.c.ORIG	2011-12-18 15:20:31.936419320 -0800
+++ ../abs/xfdesktop/src/xfdesktop-4.8.3/src/xfdesktop-icon-view.c	2011-12-18 15:21:15.563084655 -0800
@@ -1646,8 +1646,6 @@
     GdkWindow *groot;
     GList *l;
     
-    gtk_window_set_accept_focus(GTK_WINDOW(icon_view->priv->parent_window), FALSE);
-    
     gscreen = gtk_widget_get_screen(widget);
     groot = gdk_screen_get_root_window(gscreen);
     gdk_window_remove_filter(groot, xfdesktop_rootwin_watch_workarea, icon_view);
