diff --color -ru xfdesktop.ORIG/src/xfdesktop/src/xfdesktop-icon-view.c xfdesktop/src/xfdesktop/src/xfdesktop-icon-view.c
--- xfdesktop.ORIG/src/xfdesktop/src/xfdesktop-icon-view.c	2024-12-23 11:47:52.127258749 -0800
+++ xfdesktop/src/xfdesktop/src/xfdesktop-icon-view.c	2024-12-23 13:21:44.275351267 -0800
@@ -2844,7 +2844,7 @@
     GdkScreen *gscreen;
 
     GtkWidget *toplevel = gtk_widget_get_toplevel(widget);
-    gtk_window_set_accept_focus(GTK_WINDOW(toplevel), FALSE);
+    gtk_window_set_accept_focus(GTK_WINDOW(toplevel), TRUE);
 
     gscreen = gtk_widget_get_screen(widget);
 
