diff -ru abs/xfdesktop.ORIG/src/xfdesktop-4.10.2/src/xfce-desktop.c abs/xfdesktop/src/xfdesktop-4.10.2/src/xfce-desktop.c
--- abs/xfdesktop.ORIG/src/xfdesktop-4.10.2/src/xfce-desktop.c	2013-03-10 14:00:41.918297273 -0700
+++ abs/xfdesktop/src/xfdesktop-4.10.2/src/xfce-desktop.c	2013-03-10 14:00:51.345161368 -0700
@@ -672,7 +672,6 @@
     GTK_WINDOW(desktop)->type = GTK_WINDOW_TOPLEVEL;
     
     gtk_window_set_type_hint(GTK_WINDOW(desktop), GDK_WINDOW_TYPE_HINT_DESKTOP);
-    gtk_window_set_accept_focus(GTK_WINDOW(desktop), FALSE);
     gtk_window_set_resizable(GTK_WINDOW(desktop), FALSE);
 }
 
diff -ru abs/xfdesktop.ORIG/src/xfdesktop-4.10.2/src/xfdesktop-icon-view.c abs/xfdesktop/src/xfdesktop-4.10.2/src/xfdesktop-icon-view.c
--- abs/xfdesktop.ORIG/src/xfdesktop-4.10.2/src/xfdesktop-icon-view.c	2013-03-10 14:00:41.918297273 -0700
+++ abs/xfdesktop/src/xfdesktop-4.10.2/src/xfdesktop-icon-view.c	2013-03-10 14:00:51.345161368 -0700
@@ -1927,8 +1927,6 @@
     GdkWindow *groot;
     GList *l;
     
-    gtk_window_set_accept_focus(GTK_WINDOW(icon_view->priv->parent_window), FALSE);
-    
     gscreen = gtk_widget_get_screen(widget);
     groot = gdk_screen_get_root_window(gscreen);
     gdk_window_remove_filter(groot, xfdesktop_rootwin_watch_workarea, icon_view);
