diff --color -ru xfdesktop.ORIG/src/xfdesktop/src/xfdesktop-file-utils.c xfdesktop/src/xfdesktop/src/xfdesktop-file-utils.c
--- xfdesktop.ORIG/src/xfdesktop/src/xfdesktop-file-utils.c	2025-05-20 09:19:47.140198955 -0700
+++ xfdesktop/src/xfdesktop/src/xfdesktop-file-utils.c	2025-05-20 09:22:27.219999642 -0700
@@ -1275,10 +1275,10 @@
         g_object_unref(new_file);
     }
 
-    gchar *title = g_strdup_printf(_("Create Document from template \"%s\""), name);
+    gchar *title = g_strdup_printf(_("%s..."), name);
     gchar *filename = show_editable_file_create_dialog(title,
                                                        icon,
-                                                       _("Enter the name:"),
+                                                       NULL,
                                                        name,
                                                        parent,
                                                        _("Could not create a new file"));
@@ -1311,7 +1311,7 @@
 
     gchar *folder_name = show_editable_file_create_dialog(_("Create New Folder"),
                                                           icon,
-                                                          _("Enter the name:"),
+                                                          NULL,
                                                           prefill,
                                                           parent,
                                                           _("Could not create a new folder"));
