diff -ru abs/systemd.ORIG/src/systemd-200/src/core/job.c abs/systemd/src/systemd-200/src/core/job.c
--- abs/systemd.ORIG/src/systemd-200/src/core/job.c	2013-04-06 21:14:26.972678284 -0700
+++ abs/systemd/src/systemd-200/src/core/job.c	2013-04-06 21:17:30.093301561 -0700
@@ -715,7 +715,7 @@
         if (!format)
                 return;
 
-        snprintf(buf, sizeof(buf), format, unit_description(u));
+        snprintf(buf, sizeof(buf), format, strna(u->id));
         char_array_0(buf);
 
         if (t == JOB_START) {
diff -ru abs/systemd.ORIG/src/systemd-200/src/core/unit.c abs/systemd/src/systemd-200/src/core/unit.c
--- abs/systemd.ORIG/src/systemd-200/src/core/unit.c	2013-04-06 21:14:26.969344818 -0700
+++ abs/systemd/src/systemd-200/src/core/unit.c	2013-04-06 21:17:30.096635028 -0700
@@ -2549,7 +2549,7 @@
 }
 
 void unit_status_printf(Unit *u, const char *status, const char *unit_status_msg_format) {
-        manager_status_printf(u->manager, false, status, unit_status_msg_format, unit_description(u));
+        manager_status_printf(u->manager, false, status, unit_status_msg_format, strna(u->id));
 }
 
 bool unit_need_daemon_reload(Unit *u) {
Only in abs/systemd/src/systemd-200/src/core: unit.c.orig
