diff -ru systemd.ORIG/src/systemd-stable/src/core/job.c systemd/src/systemd-stable/src/core/job.c
--- systemd.ORIG/src/systemd-stable/src/core/job.c	2021-07-14 00:21:38.994352080 -0700
+++ systemd/src/systemd-stable/src/core/job.c	2021-07-14 00:25:13.487462743 -0700
@@ -566,7 +566,7 @@
         /* Log to the console second. */
         if (do_console) {
                 DISABLE_WARNING_FORMAT_NONLITERAL;
-                unit_status_printf(u, STATUS_TYPE_NORMAL, "", format, ident);
+                unit_status_printf(u, STATUS_TYPE_NORMAL, "", format, strna(u->id));
                 REENABLE_WARNING;
         }
 }
@@ -729,7 +729,7 @@
                 DISABLE_WARNING_FORMAT_NONLITERAL;
                 unit_status_printf(u,
                                    result == JOB_DONE ? STATUS_TYPE_NORMAL : STATUS_TYPE_NOTICE,
-                                   status, format, ident);
+                                   status, format, strna(u->id));
                 REENABLE_WARNING;
 
                 if (t == JOB_START && result == JOB_FAILED) {
