diff -ru abs/systemd.ORIG/src/systemd-216/src/core/job.c abs/systemd/src/systemd-216/src/core/job.c
--- abs/systemd.ORIG/src/systemd-216/src/core/job.c	2014-08-26 13:17:27.771897160 -0700
+++ abs/systemd/src/systemd-216/src/core/job.c	2014-08-26 13:17:30.991934835 -0700
@@ -705,7 +705,7 @@
                 return;
 
         DISABLE_WARNING_FORMAT_NONLITERAL;
-        snprintf(buf, sizeof(buf), format, unit_description(u));
+        snprintf(buf, sizeof(buf), format, strna(u->id));
         char_array_0(buf);
         REENABLE_WARNING;
 
diff -ru abs/systemd.ORIG/src/systemd-216/src/core/unit.c abs/systemd/src/systemd-216/src/core/unit.c
--- abs/systemd.ORIG/src/systemd-216/src/core/unit.c	2014-08-26 13:17:27.771897160 -0700
+++ abs/systemd/src/systemd-216/src/core/unit.c	2014-08-26 13:17:30.991934835 -0700
@@ -2716,7 +2716,7 @@
 
 void unit_status_printf(Unit *u, const char *status, const char *unit_status_msg_format) {
         DISABLE_WARNING_FORMAT_NONLITERAL;
-        manager_status_printf(u->manager, false, status, unit_status_msg_format, unit_description(u));
+        manager_status_printf(u->manager, false, status, unit_status_msg_format, strna(u->id));
         REENABLE_WARNING;
 }
 
Only in abs/systemd/src/systemd-216/src/core: unit.c.orig
