diff -ru abs/systemd.ORIG/src/systemd-198/src/core/job.c abs/systemd/src/systemd-198/src/core/job.c
--- abs/systemd.ORIG/src/systemd-198/src/core/job.c	2013-03-08 09:59:09.652481370 -0800
+++ abs/systemd/src/systemd-198/src/core/job.c	2013-03-08 21:27:22.228243610 -0800
@@ -715,7 +715,7 @@
         if (!format)
                 return;
 
-        snprintf(buf, sizeof(buf), format, unit_description(u));
+        snprintf(buf, sizeof(buf), format, strna(u->id));
         char_array_0(buf);
 
         if (t == JOB_START) {
diff -ru abs/systemd.ORIG/src/systemd-198/src/core/unit.c abs/systemd/src/systemd-198/src/core/unit.c
--- abs/systemd.ORIG/src/systemd-198/src/core/unit.c	2013-03-08 09:59:09.652481370 -0800
+++ abs/systemd/src/systemd-198/src/core/unit.c	2013-03-08 21:27:43.055399271 -0800
@@ -2553,7 +2553,7 @@
 }
 
 void unit_status_printf(Unit *u, const char *status, const char *unit_status_msg_format) {
-        manager_status_printf(u->manager, false, status, unit_status_msg_format, unit_description(u));
+        manager_status_printf(u->manager, false, status, unit_status_msg_format, strna(u->id));
 }
 
 bool unit_need_daemon_reload(Unit *u) {
