diff -ru slimlock.ORIG/src/slim-1.3.6/panel.cpp slimlock/src/slim-1.3.6/panel.cpp
--- slimlock.ORIG/src/slim-1.3.6/panel.cpp	2020-05-20 18:10:58.286433788 -0700
+++ slimlock/src/slim-1.3.6/panel.cpp	2020-05-20 18:13:58.530293609 -0700
@@ -271,7 +271,7 @@
 	Reset();
 	XClearWindow(Dpy, Root);
 	XClearWindow(Dpy, Win);
-	Cursor(SHOW);
+	Cursor(HIDE);
 	ShowText();
 	XFlush(Dpy);
 }
@@ -506,7 +506,7 @@
 	}
 
 	XftDrawDestroy (draw);
-	Cursor(SHOW);
+	Cursor(HIDE);
 	ShowText();
 }
 
@@ -674,7 +674,7 @@
 	}
 
 	XftDrawDestroy (draw);
-	Cursor(SHOW);
+	Cursor(HIDE);
 	return true;
 }
 
