Only in aur/slimlock: slimlock-1.3.6-1-x86_64.pkg.tar
diff -ru aur/slimlock.ORIG/src/slim-1.3.6/cfg.cpp aur/slimlock/src/slim-1.3.6/cfg.cpp
--- aur/slimlock.ORIG/src/slim-1.3.6/cfg.cpp	2014-08-14 20:12:29.540597719 -0700
+++ aur/slimlock/src/slim-1.3.6/cfg.cpp	2014-08-14 21:19:03.819951533 -0700
@@ -56,8 +56,10 @@
 	options.insert(option("reboot_msg","The system is rebooting..."));
 	options.insert(option("sessiondir",""));
 	options.insert(option("hidecursor","false"));
+	options.insert(option("single_display_width","0"));
 
 	/* Theme stuff */
+	options.insert(option("input_panel_x_single","50%"));
 	options.insert(option("input_panel_x","50%"));
 	options.insert(option("input_panel_y","40%"));
 	options.insert(option("input_name_x","200"));
diff -ru aur/slimlock.ORIG/src/slim-1.3.6/panel.cpp aur/slimlock/src/slim-1.3.6/panel.cpp
--- aur/slimlock.ORIG/src/slim-1.3.6/panel.cpp	2014-08-14 20:12:29.543931052 -0700
+++ aur/slimlock/src/slim-1.3.6/panel.cpp	2014-08-14 21:20:55.153463751 -0700
@@ -168,11 +168,15 @@
 		}
 	}
 
+  bool single = true;
+  int singleDisplayWidth = cfg->getIntOption("single_display_width");
+  string cfgSingleX = cfg->getOption("input_panel_x_single");
 	string cfgX = cfg->getOption("input_panel_x");
 	string cfgY = cfg->getOption("input_panel_y");
 
 	if (mode == Mode_Lock) {
-		X = Cfg::absolutepos(cfgX, viewport.width, image->Width());
+		single = singleDisplayWidth == 0 || viewport.width <= singleDisplayWidth;
+		X = Cfg::absolutepos(single ? cfgSingleX : cfgX, viewport.width, image->Width());
 		Y = Cfg::absolutepos(cfgY, viewport.height, image->Height());
 
 		input_name_x += X;
