diff -ru aur/slimlock.ORIG/src/slim-1.3.6/panel.cpp aur/slimlock/src/slim-1.3.6/panel.cpp
--- aur/slimlock.ORIG/src/slim-1.3.6/panel.cpp	2018-06-13 01:18:07.183138784 -0700
+++ aur/slimlock/src/slim-1.3.6/panel.cpp	2018-06-13 01:18:07.303140009 -0700
@@ -409,7 +409,7 @@
 
 	XGlyphInfo extents;
 	XftTextExtents8(Dpy, font, (XftChar8*)txth, strlen(txth), &extents);
-	cheight = extents.height;
+	cheight = extents.height - 2;
 	y2 = yy - extents.y + extents.height;
 	XftTextExtents8(Dpy, font, (XftChar8*)text, strlen(text), &extents);
 	xx += extents.width;
