diff --color -ru memtest86-efi.ORIG/memtest86-efi memtest86-efi/memtest86-efi
--- memtest86-efi.ORIG/memtest86-efi	2022-02-08 09:43:09.736738748 -0800
+++ memtest86-efi/memtest86-efi	2022-02-08 09:49:44.563518188 -0800
@@ -154,7 +154,8 @@
 	echo -e "${CB}2${CR}: Add a new EFI boot entry (more safe)"
 	echo -e "${CB}3${CR}: Add a boot entry for GRUB2 menu"
 	echo -e "${CB}4${CR}: Add a boot entry for systemd-boot menu"
-	echo -e "${CB}5${CR}: Cancel"
+	echo -e "${CB}5${CR}: Copy shellx64.efi file on ESP to user specified directory (bit safe)"
+	echo -e "${CB}6${CR}: Cancel"
 	choice=0
 	while [[ $choice -lt 1 ]] || [[ $choice -gt 5 ]]; do
 		read -r choice
@@ -211,6 +212,11 @@
 			bootctl --path="$esp_mount_path" update
 		;;
 
+		5) # Install MemTest86 in ${esp_mount_path}${memtest86_esp_path}/
+			memtest86_esp_bin="shell$ARCH.efi"
+			_common_install "${esp_mount_path}${memtest86_esp_path}" "$memtest86_esp_bin"
+		;;
+
 		*) # Do nothing and quit
 			echo -e "Canceled. MemTest86 will not be installed."
 			exit $CODE_OK
@@ -242,7 +248,7 @@
 			_common_install "$esp_mount_path" "$memtest86_esp_bin"
 		;;
 
-		2|3|4) # Update files in ${esp_mount_path}${memtest86_esp_path}/
+		2|3|4|5) # Update files in ${esp_mount_path}${memtest86_esp_path}/
 			_common_install "${esp_mount_path}${memtest86_esp_path}" "$memtest86_esp_bin"
 		;;
 	esac
@@ -298,6 +304,11 @@
 			rm -v "$esp_mount_path/loader/entries/memtest86-efi.conf"
 			bootctl --path="$esp_mount_path" update
 		;;
+
+		5) # Remove files in ${esp_mount_path}${memtest86_esp_path}
+			echo -e "MemTest86 will be removed from ${CB}${esp_mount_path}${memtest86_esp_path}/${CR}."
+			rm -rfv "${esp_mount_path:?}${memtest86_esp_path:?}"
+		;;
 	esac
 
 	echo "Writting configuration..."
