--- pre/brasero/src/brasero-2.32.1/src/brasero-project.c.ORIG	2011-02-12 18:15:01.635266065 -0800
+++ pre/brasero/src/brasero-2.32.1/src/brasero-project.c	2011-02-12 18:16:22.955604164 -0800
@@ -684,9 +684,10 @@
 	                            NULL);
 
 	gtk_text_buffer_get_start_iter (buffer, &iter);
-	gtk_text_buffer_insert_with_tags_by_name (buffer, &iter, _("To add files to this project click the \"Add\" button or drag files to this area"), -1, "Title", NULL);
+	/* gavin: this help is not needed :) */
+	/*gtk_text_buffer_insert_with_tags_by_name (buffer, &iter, _("To add files to this project click the \"Add\" button or drag files to this area"), -1, "Title", NULL);
 	gtk_text_buffer_insert (buffer, &iter, "\n\n\n", -1);
-	gtk_text_buffer_insert_with_tags_by_name (buffer, &iter, _("To remove files select them then click on the \"Remove\" button or press \"Delete\" key"), -1, "Title", NULL);
+	gtk_text_buffer_insert_with_tags_by_name (buffer, &iter, _("To remove files select them then click on the \"Remove\" button or press \"Delete\" key"), -1, "Title", NULL);*/
 
 	textview = gtk_text_view_new_with_buffer (buffer);
 	gtk_text_view_set_editable (GTK_TEXT_VIEW (textview), FALSE);
