diff -ru libgksu.ORIG/src/libgksu-2.0.12/libgksu/libgksu.c libgksu/src/libgksu-2.0.12/libgksu/libgksu.c
--- libgksu.ORIG/src/libgksu-2.0.12/libgksu/libgksu.c	2012-07-26 19:37:53.782660511 -0700
+++ libgksu/src/libgksu-2.0.12/libgksu/libgksu.c	2012-07-26 19:38:39.480631696 -0700
@@ -180,7 +180,7 @@
 FadeoutData *fade_data = NULL;
 static GList *fadeout_windows = NULL;
 
-#define FADE_DURATION 500.0
+#define FADE_DURATION 150.0
 
 int
 gsm_screen_get_width (GdkScreen *screen,
@@ -419,7 +419,7 @@
   fadeout->area.height = gsm_screen_get_height (screen, monitor);
 
   fadeout->root_window = gdk_screen_get_root_window (screen);
-  attr.window_type = GDK_WINDOW_CHILD;
+  attr.window_type = GDK_WINDOW_TOPLEVEL;
   attr.x = fadeout->area.x;
   attr.y = fadeout->area.y;
   attr.width = fadeout->area.width;
@@ -457,7 +457,6 @@
   fadeout->gc = gdk_gc_new_with_values (fadeout->root_window, &values, GDK_GC_SUBWINDOW);
 
   gdk_window_set_back_pixmap (fadeout->draw_window, NULL, FALSE);
-  gdk_window_show (fadeout->draw_window);
   gdk_draw_pixbuf (fadeout->draw_window,
 		   fadeout->gc,
 		   fadeout->frame,
@@ -469,7 +468,6 @@
 		   0, 0);
 
   g_get_current_time (&fadeout->start_time);
-  g_idle_add ((GSourceFunc) fadeout_callback, fadeout);
 
   fade_data = fadeout;
 }
