diff -ru aur/physlock-git.ORIG/src/physlock-git/config.h aur/physlock-git/src/physlock-git/config.h
--- aur/physlock-git.ORIG/src/physlock-git/config.h	2013-04-05 00:07:29.790142637 -0700
+++ aur/physlock-git/src/physlock-git/config.h	2013-04-05 00:07:40.727154071 -0700
@@ -9,5 +9,5 @@
 static const char * const SYSRQ_PATH = "/proc/sys/kernel/sysrq";
 
 /* timeout (seconds) after failed authentication try:  */
-enum { AUTH_FAIL_TIMEOUT = 2 };
+enum { AUTH_FAIL_TIMEOUT = 1 };
 
