diff -ru clipit-git.ORIG/src/clipit/src/main.c clipit-git/src/clipit/src/main.c
--- clipit-git.ORIG/src/clipit/src/main.c	2025-01-11 01:24:05.286958754 -0800
+++ clipit-git/src/clipit/src/main.c	2025-01-11 01:24:05.433627880 -0800
@@ -1070,7 +1070,7 @@
 	create_app_indicator(1);
 #else
 	status_icon = gtk_status_icon_new_from_icon_name(ICON);
-	gtk_status_icon_set_tooltip_text((GtkStatusIcon*)status_icon, _("Clipboard Manager"));
+	gtk_status_icon_set_tooltip_text((GtkStatusIcon*)status_icon, _("ClipIt"));
 	g_signal_connect((GObject*)status_icon, "button_press_event", (GCallback)status_icon_clicked, NULL);
 #endif
 	}
