diff -ru clipit-git.ORIG/src/clipit/src/main.c clipit-git/src/clipit/src/main.c
--- clipit-git.ORIG/src/clipit/src/main.c	2025-01-11 01:24:05.286958754 -0800
+++ clipit-git/src/clipit/src/main.c	2025-01-11 01:24:05.376960264 -0800
@@ -866,17 +866,6 @@
 			elem = elem->next;
 		}
 	}
-	/* Show a notice in offline mode */
-	if (prefs.offline_mode) {
-		gtk_menu_shell_append((GtkMenuShell*)history_menu, gtk_separator_menu_item_new());
-
-		menu_item = gtk_menu_item_new_with_label("");
-		item_label = gtk_bin_get_child((GtkBin*)menu_item);
-		gtk_label_set_markup((GtkLabel*)item_label, "<b>Offline mode is ON</b>");
-		gtk_label_set_single_line_mode((GtkLabel*)item_label, TRUE);
-		gtk_widget_set_sensitive(item_label, FALSE);
-		gtk_menu_shell_append((GtkMenuShell*)history_menu, menu_item);
-	}
     /* Append "Remove all" item */
     menu_item = gtk_menu_item_new_with_label(_("Remove all"));
     gtk_menu_shell_append((GtkMenuShell*)history_menu, menu_item);
