diff -ru clipit-git.ORIG/src/clipit/src/main.c clipit-git/src/clipit/src/main.c
--- clipit-git.ORIG/src/clipit/src/main.c	2025-01-11 01:24:05.286958754 -0800
+++ clipit-git/src/clipit/src/main.c	2025-01-11 01:24:05.290292144 -0800
@@ -562,7 +562,7 @@
   gtk_menu_shell_append((GtkMenuShell*)menu, menu_item);
   /* Popup the menu... */
   gtk_widget_show_all(menu);
-  gtk_menu_popup((GtkMenu*)menu, NULL, NULL, NULL, NULL, 1, gtk_get_current_event_time());
+  gtk_menu_popup((GtkMenu*)menu, NULL, NULL, gtk_status_icon_position_menu, status_icon, 1, gtk_get_current_event_time());
   /* Return false so the g_timeout_add() function is called only once */
   return FALSE;
 }
@@ -894,8 +894,7 @@
   g_signal_connect((GObject*)menu, "selection-done", (GCallback)gtk_widget_destroy, NULL);
   /* Popup the menu... */
   gtk_widget_show_all(menu);
-  gtk_menu_popup((GtkMenu*)menu, NULL, NULL, NULL, NULL, 1, gtk_get_current_event_time());
-  gtk_menu_shell_select_first((GtkMenuShell*)menu, TRUE);
+  gtk_menu_popup((GtkMenu*)menu, NULL, NULL, gtk_status_icon_position_menu, status_icon, 1, gtk_get_current_event_time());
   /* Return FALSE so the g_timeout_add() function is called only once */
   return FALSE;
 }
@@ -975,7 +974,6 @@
 	/* GENERATE THE MENU*/
 	gtk_widget_set_visible(statusicon_menu, TRUE);
 	gtk_menu_popup((GtkMenu*)statusicon_menu, NULL, NULL, gtk_status_icon_position_menu, status_icon, 1, gtk_get_current_event_time());
-	gtk_menu_shell_select_first((GtkMenuShell*)statusicon_menu, TRUE);
 
 	status_menu_lock = FALSE;
 }
