diff -ru clipit-git.ORIG/src/clipit/src/main.c clipit-git/src/clipit/src/main.c
--- clipit-git.ORIG/src/clipit/src/main.c	2025-01-11 01:24:05.286958754 -0800
+++ clipit-git/src/clipit/src/main.c	2025-01-11 01:24:05.390293820 -0800
@@ -591,7 +591,7 @@
 
 /* Handle user input while the menu is open */
 static gboolean menu_key_pressed(GtkWidget *history_menu, GdkEventKey *event, gpointer user_data) {
-  gboolean handled = selected_by_digit(history_menu, event);
+  gboolean handled = prefs.show_indexes && selected_by_digit(history_menu, event);
   if (!handled) {
     handled = selected_by_input(history_menu, event);
   }
