diff -ru clipit-git.ORIG/src/clipit/src/main.c clipit-git/src/clipit/src/main.c
--- clipit-git.ORIG/src/clipit/src/main.c	2025-01-11 01:24:05.286958754 -0800
+++ clipit-git/src/clipit/src/main.c	2025-01-11 01:24:05.303625700 -0800
@@ -623,6 +623,8 @@
  * underlined and non matching entries greyed out.
 */
 gboolean selected_by_input(const GtkWidget *history_menu, const GdkEventKey *event) {
+  int prev_input_index = input_index;
+
   if (event->keyval == GDK_KEY_Delete || event->keyval == GDK_KEY_BackSpace) {
     remove_from_input_buffer();
   } else if (event->keyval == GDK_KEY_KP_Enter || event->keyval == GDK_KEY_Return || event->keyval == GDK_KEY_Escape) {
@@ -637,6 +639,10 @@
   if (isprint(*event->string))
     append_to_input_buffer(event->string);
 
+  if (prev_input_index == input_index) {
+    return FALSE;
+  }
+
   GtkMenuShell* menu = (GtkMenuShell *) history_menu;
   GList* element = gtk_container_get_children(GTK_CONTAINER (menu));
   GtkMenuItem *menu_item, *first_match = 0;
