diff -ru clipit-git.ORIG/src/clipit/src/main.c clipit-git/src/clipit/src/main.c
--- clipit-git.ORIG/src/clipit/src/main.c	2025-01-11 01:24:05.286958754 -0800
+++ clipit-git/src/clipit/src/main.c	2025-01-11 01:24:05.346959760 -0800
@@ -815,6 +815,7 @@
 				element_number++;
 			element_number_small++;
 		}
+		gtk_menu_shell_append((GtkMenuShell*)history_menu, gtk_separator_menu_item_new());
 		/* Cleanup */
 		g_free(primary_temp);
 		g_free(clipboard_temp);
@@ -822,13 +823,6 @@
 		if (prefs.reverse_history)
 			history = g_list_reverse(history);
 	}
-	else
-	{
-		/* Nothing in history so adding empty */
-		menu_item = gtk_menu_item_new_with_label(_("Empty"));
-		gtk_widget_set_sensitive(menu_item, FALSE);
-		gtk_menu_shell_append((GtkMenuShell*)history_menu, menu_item);
-	}
 	if (prefs.statics_show) {
 		/* -------------------- */
 		gtk_menu_shell_append((GtkMenuShell*)history_menu, gtk_separator_menu_item_new());
