diff -ru clipit-git.ORIG/src/clipit/src/main.c clipit-git/src/clipit/src/main.c
--- clipit-git.ORIG/src/clipit/src/main.c	2025-01-11 01:24:05.286958754 -0800
+++ clipit-git/src/clipit/src/main.c	2025-01-11 01:24:05.320292646 -0800
@@ -983,19 +983,8 @@
 /* Called when status icon is clicked */
 /* (checks type of click and calls correct function) */
 static void status_icon_clicked(GtkStatusIcon *status_icon, GdkEventButton *event ) {
-	/* Check what type of click was recieved */
-	GdkModifierType state;
-	gtk_get_current_event_state(&state);
-	/* Control click */
-	if (state == GDK_MOD2_MASK+GDK_CONTROL_MASK || state == GDK_CONTROL_MASK)
-	{
-		if (actions_lock == FALSE)
-		{
-			g_timeout_add(POPUP_DELAY, show_actions_menu, NULL);
-		}
-	}
 	/* Left click */
-	else if (event->button == 1)
+	if (event->button == 1)
 	{
 		show_clipit_menu(1);
 	}
