diff --color -ru pnpm.ORIG/PKGBUILD pnpm/PKGBUILD
--- pnpm.ORIG/PKGBUILD	2023-07-17 12:47:35.068803268 -0700
+++ pnpm/PKGBUILD	2023-07-17 12:52:17.613255501 -0700
@@ -15,6 +15,8 @@
 b2sums=('b4d2004b32d542ded333118296d6a455f2119013c54d9138a4c1320a6fff7d4cbe039414100e5c20a9444a22426bda516f0a51c600a9b534ff505b85fe5094a6')
 
 build() {
+  rm -fr $srcdir/../.git
+
   cd $pkgname-$pkgver/$pkgname
   pnpm install --frozen-lockfile
   pnpm run compile
