--- abs/exo/src/exo-0.6.2/exo-helper/main.c.ORIG	2011-06-14 13:28:03.415392740 -0700
+++ abs/exo/src/exo-0.6.2/exo-helper/main.c	2011-06-14 13:28:53.168577629 -0700
@@ -180,7 +180,7 @@
       if (G_LIKELY (helper != NULL))
         {
           /* try to execute the helper with the given parameter */
-          if (!exo_helper_execute (helper, NULL, (argc > 1) ? argv[1] : NULL, &error))
+          if (!exo_helper_execute (helper, NULL, (argc > 1) ? argv[1] : NULL, &error) && 0) /* gavin: hide these errors */
             {
               dialog = gtk_message_dialog_new (NULL, 0, GTK_MESSAGE_ERROR, GTK_BUTTONS_CLOSE,
                                                "%s.", _(CATEGORY_EXEC_ERRORS[category]));
