diff -ru gimp.ORIG/src/gimp-2.10.22/app/display/gimpimagewindow.c gimp/src/gimp-2.10.22/app/display/gimpimagewindow.c
--- gimp.ORIG/src/gimp-2.10.22/app/display/gimpimagewindow.c	2020-02-24 09:11:23.259369588 -0800
+++ gimp/src/gimp-2.10.22/app/display/gimpimagewindow.c	2020-02-24 09:11:24.406068499 -0800
@@ -2354,8 +2354,6 @@
   GtkWidget *hbox;
   GtkWidget *view;
   GimpImage *image;
-  GtkWidget *button;
-  GtkWidget *gtk_image;
 
   hbox = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 2);
   gtk_widget_show (hbox);
@@ -2373,23 +2371,6 @@
   if (image)
     gimp_view_set_viewable (GIMP_VIEW (view), GIMP_VIEWABLE (image));
 
-  button = gtk_button_new ();
-  gtk_widget_set_can_focus (button, FALSE);
-  gtk_button_set_relief (GTK_BUTTON (button), GTK_RELIEF_NONE);
-  gtk_box_pack_start (GTK_BOX (hbox), button, FALSE, FALSE, 0);
-  gtk_widget_show (button);
-
-  gtk_image = gtk_image_new_from_icon_name (GIMP_ICON_CLOSE,
-                                            GTK_ICON_SIZE_MENU);
-  gtk_container_add (GTK_CONTAINER (button), gtk_image);
-  gtk_widget_show (gtk_image);
-
-  g_signal_connect_swapped (button, "clicked",
-                            G_CALLBACK (gimp_image_window_shell_close_button_callback),
-                            shell);
-
-  g_object_set_data (G_OBJECT (hbox), "close-button", button);
-
   return hbox;
 }
 
