diff -ru gimp.ORIG/src/gimp-2.10.24/app/widgets/gimpthumbbox.c gimp/src/gimp-2.10.24/app/widgets/gimpthumbbox.c
--- gimp.ORIG/src/gimp-2.10.24/app/widgets/gimpthumbbox.c	2020-02-24 09:11:23.276036722 -0800
+++ gimp/src/gimp-2.10.24/app/widgets/gimpthumbbox.c	2020-02-24 09:11:23.572711732 -0800
@@ -336,7 +336,6 @@
 
   button = gtk_button_new ();
   gtk_box_pack_start (GTK_BOX (vbox), button, FALSE, FALSE, 0);
-  gtk_widget_show (button);
 
   label = gtk_label_new_with_mnemonic (_("Pr_eview"));
   gtk_label_set_xalign (GTK_LABEL (label), 0.0);
diff -ru gimp.ORIG/src/gimp-2.10.24/app/widgets/gimpviewrendererimagefile.c gimp/src/gimp-2.10.24/app/widgets/gimpviewrendererimagefile.c
--- gimp.ORIG/src/gimp-2.10.24/app/widgets/gimpviewrendererimagefile.c	2020-02-24 09:11:23.269369869 -0800
+++ gimp/src/gimp-2.10.24/app/widgets/gimpviewrendererimagefile.c	2020-02-24 09:11:23.576045159 -0800
@@ -193,8 +193,7 @@
 
       pixbuf = gtk_icon_theme_load_icon (icon_theme,
                                          icon_name, size,
-                                         GTK_ICON_LOOKUP_USE_BUILTIN,
-                                         NULL);
+                                         0, NULL);
     }
 
   return pixbuf;
diff -ru gimp.ORIG/src/gimp-2.10.24/app/widgets/gimpwidgets-utils.c gimp/src/gimp-2.10.24/app/widgets/gimpwidgets-utils.c
--- gimp.ORIG/src/gimp-2.10.24/app/widgets/gimpwidgets-utils.c	2020-02-24 09:11:23.269369869 -0800
+++ gimp/src/gimp-2.10.24/app/widgets/gimpwidgets-utils.c	2020-02-24 09:11:23.576045159 -0800
@@ -433,8 +433,7 @@
   if (closest_size != -1)
     size = closest_size;
 
-  pixbuf = gtk_icon_theme_load_icon (icon_theme, icon_name, size,
-                                     GTK_ICON_LOOKUP_USE_BUILTIN, NULL);
+  pixbuf = gtk_icon_theme_load_icon (icon_theme, icon_name, size, 0, NULL);
 
   if (! pixbuf)
     {
