diff -ru gimp.ORIG/src/gimp/app/display/gimpdisplayshell-callbacks.c gimp/src/gimp/app/display/gimpdisplayshell-callbacks.c
--- gimp.ORIG/src/gimp/app/display/gimpdisplayshell-callbacks.c	2024-11-06 09:26:10.851102558 -0800
+++ gimp/src/gimp/app/display/gimpdisplayshell-callbacks.c	2024-11-06 09:26:11.561113612 -0800
@@ -637,8 +637,6 @@
 
   gimp_display_shell_draw_background (shell, cr);
 
-  gimp_cairo_draw_drop_wilber (shell->canvas, cr, shell->blink);
-
   cairo_restore (cr);
 
 #ifdef GIMP_UNSTABLE
