diff -ru gimp.ORIG/src/gimp-2.10.12/app/widgets/gimppanedbox.c gimp/src/gimp-2.10.12/app/widgets/gimppanedbox.c
--- gimp.ORIG/src/gimp-2.10.12/app/widgets/gimppanedbox.c	2018-05-17 13:27:52.278208702 -0700
+++ gimp/src/gimp-2.10.12/app/widgets/gimppanedbox.c	2018-05-17 13:45:42.982893550 -0700
@@ -56,7 +56,7 @@
 #define INSERT_INDEX_UNUSED       G_MININT
 
 #define INSTRUCTIONS_TEXT_PADDING 6
-#define INSTRUCTIONS_TEXT         _("You can drop dockable dialogs here")
+#define INSTRUCTIONS_TEXT         _("")
 
 
 struct _GimpPanedBoxPrivate
