diff -ru abs/gnupg.ORIG/PKGBUILD abs/gnupg/PKGBUILD
--- abs/gnupg.ORIG/PKGBUILD	2014-11-25 08:32:07.034778852 -0800
+++ abs/gnupg/PKGBUILD	2014-11-25 08:32:21.311636177 -0800
@@ -43,6 +43,7 @@
 		--enable-maintainer-mode \
 		--enable-symcryptrun \
 		--enable-gpgtar \
+		--enable-large-secmem
 
 	make
 }
diff -ru abs/gnupg.ORIG/src/gnupg-2.1.0/g10/keygen.c abs/gnupg/src/gnupg-2.1.0/g10/keygen.c
--- abs/gnupg.ORIG/src/gnupg-2.1.0/g10/keygen.c	2014-11-25 08:32:07.061445874 -0800
+++ abs/gnupg/src/gnupg-2.1.0/g10/keygen.c	2014-11-25 08:33:14.725670299 -0800
@@ -2040,6 +2040,7 @@
 
     case PUBKEY_ALGO_RSA:
       min=1024;
+      if (opt.flags.large_rsa) { max = 8192; }
       break;
     }
 
