diff -ru evince2.ORIG/src/evince-2.32.0/libview/ev-view.c evince2/src/evince-2.32.0/libview/ev-view.c
--- evince2.ORIG/src/evince-2.32.0/libview/ev-view.c	2018-05-23 14:41:40.578203677 -0700
+++ evince2/src/evince-2.32.0/libview/ev-view.c	2018-05-23 14:41:40.928195080 -0700
@@ -3315,13 +3315,6 @@
 	cairo_t   *cr;
 	gint       i;
 
-	if (view->loading) {
-		show_loading_window (view);
-	} else if (view->loading_window &&
-		   gtk_widget_get_visible (view->loading_window)) {
-		ev_view_loading_window_move (view);
-	}
-
 	if (view->document == NULL)
 		return FALSE;
 
@@ -4397,9 +4390,6 @@
 		page_surface = ev_pixbuf_cache_get_surface (view->pixbuf_cache, page);
 
 		if (!page_surface) {
-			if (page == current_page)
-				show_loading_window (view);
-
 			*page_ready = FALSE;
 
 			return;
