--- ../abs/xfwm4/src/xfwm4-4.8.2/src/compositor.c.ORIG	2011-04-20 15:38:48.993268216 -0700
+++ ../abs/xfwm4/src/xfwm4-4.8.2/src/compositor.c	2011-04-20 15:39:42.766590262 -0700
@@ -50,7 +50,7 @@
 #include <X11/extensions/Xrender.h>
 
 #ifndef SHADOW_RADIUS
-#define SHADOW_RADIUS   12
+#define SHADOW_RADIUS   4 /* gavin: change radius from 12 to 4 */
 #endif /* SHADOW_RADIUS */
 
 #ifndef SHADOW_OFFSET_X
@@ -930,6 +930,7 @@
     DisplayInfo *display_info;
     ScreenInfo *screen_info;
     XRectangle r;
+    double shadow_scalar = 1.00; /* gavin */
 
     g_return_val_if_fail (cw != NULL, None);
     TRACE ("entering win_extents: 0x%lx", cw->id);
@@ -952,7 +953,8 @@
     if ((screen_info->params->show_popup_shadow &&
               WIN_IS_OVERRIDE(cw) &&
               !(WIN_IS_ARGB(cw) || WIN_IS_SHAPED(cw)) &&
-              !WIN_IS_FULLSCREEN(cw)) ||
+              !WIN_IS_FULLSCREEN(cw) &&
+              (shadow_scalar = 0.35)) || /* gavin: popups get a lower shadow opacity than the rest */
           (screen_info->params->show_frame_shadow &&
               !WIN_IS_OVERRIDE(cw) &&
               !WIN_NO_SHADOW(cw) &&
@@ -961,8 +963,8 @@
           (screen_info->params->show_dock_shadow &&
               WIN_IS_DOCK(cw) &&
               !WIN_NO_SHADOW(cw) &&
-              !WIN_IS_OVERRIDE(cw) &&
-              (!(WIN_IS_ARGB(cw) || WIN_IS_SHAPED(cw)))))
+              !WIN_IS_OVERRIDE(cw)))
+              /* (!(WIN_IS_ARGB(cw) || WIN_IS_SHAPED(cw))))) */ /* gavin: how the dock got its shadow back */
     {
         XRectangle sr;
 
@@ -976,6 +978,7 @@
             shadow_opacity = (double) screen_info->params->frame_opacity
                            * (screen_info->params->shadow_opacity / 100.0)
                            * cw->opacity
+                           * shadow_scalar /* gavin */
                            / (NET_WM_OPAQUE * 100.0);
 
             cw->shadow = shadow_picture (screen_info, shadow_opacity,
