diff -ru abs/xfwm4.ORIG/src/xfwm4-4.10.1/src/tabwin.c abs/xfwm4/src/xfwm4-4.10.1/src/tabwin.c
--- abs/xfwm4.ORIG/src/xfwm4-4.10.1/src/tabwin.c	2013-06-27 17:26:30.997845911 -0700
+++ abs/xfwm4/src/xfwm4-4.10.1/src/tabwin.c	2013-06-27 17:41:00.851146611 -0700
@@ -25,7 +25,7 @@
 #endif
 
 #ifndef WIN_ICON_SIZE
-#define WIN_ICON_SIZE 48
+#define WIN_ICON_SIZE 16
 #endif
 
 #ifndef WIN_ICON_BORDER
@@ -276,7 +276,7 @@
     g_return_val_if_fail (screen_info->client_count > 0, NULL);
 
     gtk_widget_style_get (GTK_WIDGET (tbw), "icon-size", &icon_size, NULL);
-    tbw->grid_cols = (monitor_width / (icon_size + 2 * WIN_ICON_BORDER)) * 0.75;
+    tbw->grid_cols = (monitor_width / (icon_size + 2 * WIN_ICON_BORDER)) * 0.25 - 1;
     tbw->grid_rows = screen_info->client_count / tbw->grid_cols + 1;
     tbw->widgets = NULL;
     windowlist = gtk_table_new (tbw->grid_rows, tbw->grid_cols, FALSE);
@@ -420,7 +420,7 @@
     gtk_label_set_justify (GTK_LABEL (tbw->label), GTK_JUSTIFY_CENTER);
     gtk_label_set_use_markup (GTK_LABEL (tbw->class), TRUE);
     gtk_box_pack_start (GTK_BOX (vbox), tbw->label, TRUE, TRUE, 0);
-    gtk_widget_set_size_request (GTK_WIDGET (tbw->label), 240, -1);
+    gtk_widget_set_size_request (GTK_WIDGET (tbw->label), 360, -1);
 
     windowlist = createWindowlist (screen_info, tbw);
     tbw->container = windowlist;
