diff -ru abs/xfwm4.ORIG/src/xfwm4-4.10.1/src/compositor.c abs/xfwm4/src/xfwm4-4.10.1/src/compositor.c
--- abs/xfwm4.ORIG/src/xfwm4-4.10.1/src/compositor.c	2013-06-27 17:26:31.001179244 -0700
+++ abs/xfwm4/src/xfwm4-4.10.1/src/compositor.c	2013-06-27 17:38:41.401151841 -0700
@@ -934,6 +934,7 @@
     DisplayInfo *display_info;
     ScreenInfo *screen_info;
     XRectangle r;
+    double shadow_scalar = 1.00;
 
     g_return_val_if_fail (cw != NULL, None);
     TRACE ("entering win_extents: 0x%lx", cw->id);
@@ -956,7 +957,8 @@
     if ((screen_info->params->show_popup_shadow &&
               WIN_IS_OVERRIDE(cw) &&
               !(WIN_IS_ARGB(cw) || WIN_IS_SHAPED(cw)) &&
-              !WIN_IS_FULLSCREEN(cw)) ||
+              !WIN_IS_FULLSCREEN(cw) &&
+              (shadow_scalar = 0.75)) ||
           (screen_info->params->show_frame_shadow &&
               !WIN_IS_OVERRIDE(cw) &&
               !WIN_NO_SHADOW(cw) &&
@@ -965,8 +967,7 @@
           (screen_info->params->show_dock_shadow &&
               WIN_IS_DOCK(cw) &&
               !WIN_NO_SHADOW(cw) &&
-              !WIN_IS_OVERRIDE(cw) &&
-              (!(WIN_IS_ARGB(cw) || WIN_IS_SHAPED(cw)))))
+              !WIN_IS_OVERRIDE(cw)))
     {
         XRectangle sr;
 
@@ -980,6 +981,7 @@
             shadow_opacity = (double) screen_info->params->frame_opacity
                            * (screen_info->params->shadow_opacity / 100.0)
                            * cw->opacity
+                           * shadow_scalar
                            / (NET_WM_OPAQUE * 100.0);
 
             cw->shadow = shadow_picture (screen_info, shadow_opacity,
