diff -ru abs/scrot.ORIG/src/scrot-0.8/src/main.c abs/scrot/src/scrot-0.8/src/main.c
--- abs/scrot.ORIG/src/scrot-0.8/src/main.c	2018-07-24 02:19:08.065139702 -0700
+++ abs/scrot/src/scrot-0.8/src/main.c	2018-07-24 02:36:33.177427382 -0700
@@ -240,16 +240,9 @@
       switch (ev.type) {
         case MotionNotify:
           if (btn_pressed) {
-            if (rect_w) {
-              /* re-draw the last rect to clear it */
-              XDrawRectangle(disp, root, gc, rect_x, rect_y, rect_w, rect_h);
-            } else {
-              /* Change the cursor to show we're selecting a region */
-              XChangeActivePointerGrab(disp,
-                                       ButtonMotionMask | ButtonReleaseMask,
-                                       cursor2, CurrentTime);
-            }
-
+            XChangeActivePointerGrab(disp,
+                                     ButtonMotionMask | ButtonReleaseMask,
+                                     cursor2, CurrentTime);
             rect_x = rx;
             rect_y = ry;
             rect_w = ev.xmotion.x - rect_x;
@@ -263,9 +256,6 @@
               rect_y += rect_h;
               rect_h = 0 - rect_h;
             }
-            /* draw rectangle */
-            XDrawRectangle(disp, root, gc, rect_x, rect_y, rect_w, rect_h);
-            XFlush(disp);
           }
           break;
         case ButtonPress:
@@ -304,10 +294,7 @@
         && ((errno == ENOMEM) || (errno == EINVAL) || (errno == EBADF)))
       gib_eprintf("Connection to X display lost");
   }
-  if (rect_w) {
-    XDrawRectangle(disp, root, gc, rect_x, rect_y, rect_w, rect_h);
-    XFlush(disp);
-  }
+
   XUngrabPointer(disp, CurrentTime);
   XUngrabKeyboard(disp, CurrentTime);
   XFreeCursor(disp, cursor);
