diff -ru ../pre/network-manager-applet.ORIG/src/network-manager-applet-0.8.6.0/src/applet.c ../pre/network-manager-applet/src/network-manager-applet-0.8.6.0/src/applet.c
--- ../pre/network-manager-applet.ORIG/src/network-manager-applet-0.8.6.0/src/applet.c	2012-05-31 20:38:00.481538911 -0700
+++ ../pre/network-manager-applet/src/network-manager-applet-0.8.6.0/src/applet.c	2012-05-31 20:41:16.011532870 -0700
@@ -879,9 +879,9 @@
 	case NM_VPN_CONNECTION_STATE_ACTIVATED:
 		banner = nm_vpn_connection_get_banner (vpn);
 		if (banner && strlen (banner))
-			msg = g_strdup_printf ("VPN connection has been successfully established.\n\n%s\n", banner);
+			msg = g_strdup_printf ("VPN connection has been successfully established.\n\n%s", banner);
 		else
-			msg = g_strdup ("VPN connection has been successfully established.\n");
+			msg = g_strdup ("VPN connection has been successfully established.");
 
 		title = _("VPN Login Message");
 		applet_do_notify_with_pref (applet, title, msg, "gnome-lockscreen",
