diff --color -ru network-manager-applet.ORIG/src/network-manager-applet/meson.build network-manager-applet/src/network-manager-applet/meson.build
--- network-manager-applet.ORIG/src/network-manager-applet/meson.build	2022-01-26 10:20:44.564781723 -0800
+++ network-manager-applet/src/network-manager-applet/meson.build	2022-01-26 10:20:54.625238776 -0800
@@ -253,7 +253,6 @@
 
 foreach desktop: desktop_files
   i18n.merge_file(
-    desktop + '-desktop',
     input: desktop + '.desktop.in',
     output: desktop + '.desktop',
     install: true,
@@ -274,7 +273,6 @@
 appdata = 'nm-connection-editor.appdata.xml'
 
 i18n.merge_file(
-  'desktop',
   input: appdata + '.in',
   output: appdata,
   install: true,
