diff --color -ru network-manager-applet.ORIG/PKGBUILD network-manager-applet/PKGBUILD
--- network-manager-applet.ORIG/PKGBUILD	2022-01-26 10:20:44.561448238 -0800
+++ network-manager-applet/PKGBUILD	2022-01-26 10:23:19.805152900 -0800
@@ -29,7 +29,7 @@
 }
 
 build() {
-  arch-meson $pkgbase build -D selinux=false
+  arch-meson $pkgbase build -D selinux=false -D appindicator=no
   meson compile -C build
 }
 
