diff -ru abs/virtualbox.ORIG/src/VirtualBox-5.1.28/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineWindow.cpp abs/virtualbox/src/VirtualBox-5.1.28/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineWindow.cpp
--- abs/virtualbox.ORIG/src/VirtualBox-5.1.28/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineWindow.cpp	2017-09-12 15:07:51.365291578 -0700
+++ abs/virtualbox/src/VirtualBox-5.1.28/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineWindow.cpp	2017-09-12 15:14:34.586004463 -0700
@@ -554,8 +554,6 @@
             strSnapshotName = " (" + snapshot.GetName() + ")";
         }
         QString strMachineName = machineName() + strSnapshotName;
-        if (state != KMachineState_Null)
-            strMachineName += " [" + gpConverter->toString(state) + "]";
         /* Unusual on the Mac. */
 #ifndef VBOX_WS_MAC
         const QString strUserProductName = uisession()->machineWindowNamePostfix();
