--- aur/evince2-light.ORIG/src/evince-2.32.0/libdocument/ev-document-misc.c	2012-05-27 22:51:18.973046880 -0700
+++ aur/evince2-light/src/evince-2.32.0/libdocument/ev-document-misc.c	2012-05-27 22:52:43.349710938 -0700
@@ -63,11 +63,11 @@
 				 width_r + 4,
 				 height_r + 4);
 
-	/* make it black and fill in the middle */
+	/* make it gray and fill in the middle */
 	data = gdk_pixbuf_get_pixels (retval);
 	rowstride = gdk_pixbuf_get_rowstride (retval);
 
-	gdk_pixbuf_fill (retval, 0x000000ff);
+	gdk_pixbuf_fill (retval, 0x666666ff);
 	if (fill_bg) {
 		for (i = 1; i < height_r + 1; i++)
 			memset (data + (rowstride * i) + 4, 0xffffffff, width_r * 4);
@@ -119,16 +119,8 @@
 
 	border->left = 1;
 	border->top = 1;
-	if (page_width < 100) {
-		border->right = 2;
-		border->bottom = 2;
-	} else if (page_width < 500) {
-		border->right = 3;
-		border->bottom = 3;
-	} else {
-		border->right = 4;
-		border->bottom = 4;
-	}
+	border->right = 2;
+	border->bottom = 2;
 }
 
 
@@ -143,7 +135,7 @@
 	GtkStyle    *style = gtk_widget_get_style (widget);
 	GtkStateType state = gtk_widget_get_state (widget);
 
-	gdk_cairo_set_source_color (cr, highlight ? &style->text[state] : &style->dark[state]);
+	gdk_cairo_set_source_color (cr, &style->dark[state]);
 	gdk_cairo_rectangle (cr, area);
 	cairo_fill (cr);
 
