--- ../pre/notification-daemon/src/notification-daemon-0.4.0/src/capplet/notification-properties.c.ORIG	2012-01-06 09:13:07.124463458 -0800
+++ ../pre/notification-daemon/src/notification-daemon-0.4.0/src/capplet/notification-properties.c	2012-01-06 09:13:37.944462504 -0800
@@ -434,8 +434,7 @@
 
 	dialog->preview = notify_notification_new(_("Notification Test"),
 											  _("Just a test"),
-											  "gnome-util",
-											  NULL);
+											  "gnome-util");
 
 	if (!notify_notification_show(dialog->preview, &error))
 	{
