diff -ru abs/xfce4-power-manager.ORIG/src/xfce4-power-manager/src/xfpm-backlight.c abs/xfce4-power-manager/src/xfce4-power-manager/src/xfpm-backlight.c
--- abs/xfce4-power-manager.ORIG/src/xfce4-power-manager/src/xfpm-backlight.c	2014-05-08 22:49:26.362063138 -0700
+++ abs/xfce4-power-manager/src/xfce4-power-manager/src/xfpm-backlight.c	2014-05-08 22:51:26.762068559 -0700
@@ -320,9 +320,6 @@
         g_signal_connect (backlight->priv->idle, "reset",
                           G_CALLBACK(xfpm_backlight_reset_cb), backlight);
 			  
-	g_signal_connect (backlight->priv->button, "button-pressed",
-		          G_CALLBACK (xfpm_backlight_button_pressed_cb), backlight);
-			  
 	g_signal_connect_swapped (backlight->priv->conf, "notify::" BRIGHTNESS_ON_AC,
 				  G_CALLBACK (xfpm_backlight_brightness_on_ac_settings_changed), backlight);
 	
Only in abs/xfce4-power-manager/src/xfce4-power-manager/src: xfpm-backlight.c.orig
