diff -ru abs/xfce4-power-manager.ORIG/src/xfce4-power-manager/src/xfpm-power.c abs/xfce4-power-manager/src/xfce4-power-manager/src/xfpm-power.c
--- abs/xfce4-power-manager.ORIG/src/xfce4-power-manager/src/xfpm-power.c	2014-05-08 22:53:24.605407201 -0700
+++ abs/xfce4-power-manager/src/xfce4-power-manager/src/xfpm-power.c	2014-05-08 22:56:14.155414836 -0700
@@ -508,56 +508,6 @@
 
     menu = gtk_menu_new();
 
-    // Hibernate menu option
-    mi = gtk_image_menu_item_new_with_label (_("Hibernate"));
-    img = gtk_image_new_from_icon_name (XFPM_HIBERNATE_ICON, GTK_ICON_SIZE_MENU);
-    gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (mi), img);
-    gtk_widget_set_sensitive (mi, FALSE);
-
-    if ( power->priv->can_hibernate && power->priv->auth_hibernate)
-    {
-	gtk_widget_set_sensitive (mi, TRUE);
-	g_signal_connect_swapped (G_OBJECT (mi), "activate",
-				  G_CALLBACK (xfpm_power_hibernate_cb), power);
-    }
-    gtk_widget_show (mi);
-    gtk_menu_shell_append (GTK_MENU_SHELL (menu), mi);
-
-    // Suspend menu option
-    mi = gtk_image_menu_item_new_with_label (_("Suspend"));
-    img = gtk_image_new_from_icon_name (XFPM_SUSPEND_ICON, GTK_ICON_SIZE_MENU);
-    gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (mi), img);
-
-    gtk_widget_set_sensitive (mi, FALSE);
-
-    if ( power->priv->can_suspend && power->priv->auth_suspend)
-    {
-	gtk_widget_set_sensitive (mi, TRUE);
-	g_signal_connect_swapped (mi, "activate",
-				  G_CALLBACK (xfpm_power_suspend_cb), power);
-    }
-
-    gtk_widget_show (mi);
-    gtk_menu_shell_append (GTK_MENU_SHELL (menu), mi);
-/*
-    saver_inhibited = xfpm_screen_saver_get_inhibit (tray->priv->srv);
-    mi = gtk_check_menu_item_new_with_label (_("Monitor power control"));
-    gtk_check_menu_item_set_active (GTK_CHECK_MENU_ITEM (mi), !saver_inhibited);
-    gtk_widget_set_tooltip_text (mi, _("Disable or enable monitor power control, "\
-                                       "for example you could disable the screen power "\
-				       "control to avoid screen blanking when watching a movie."));
-
-    g_signal_connect (G_OBJECT (mi), "activate",
-		      G_CALLBACK (xfpm_tray_icon_inhibit_active_cb), tray);
-    gtk_widget_set_sensitive (mi, TRUE);
-    gtk_widget_show(mi);
-    gtk_menu_shell_append(GTK_MENU_SHELL(menu),mi);
-*/
-
-    mi = gtk_separator_menu_item_new ();
-    gtk_widget_show (mi);
-    gtk_menu_shell_append (GTK_MENU_SHELL (menu), mi);
-
     // Power information
     mi = gtk_image_menu_item_new_with_label (_("Power Information"));
     img = gtk_image_new_from_stock (GTK_STOCK_INFO, GTK_ICON_SIZE_MENU);
@@ -571,41 +521,6 @@
     gtk_widget_show (mi);
     gtk_menu_shell_append (GTK_MENU_SHELL (menu), mi);
 
-    /**
-     * Power Mode
-     **/
-    /* TRANSLATOR: Mode here is the power profile (presentation, power save, normal) */
-    mi = gtk_image_menu_item_new_with_label (_("Mode"));
-    img = gtk_image_new_from_icon_name (XFPM_AC_ADAPTER_ICON, GTK_ICON_SIZE_MENU);
-    gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (mi), img);
-    gtk_widget_set_sensitive (mi,TRUE);
-    gtk_widget_show (mi);
-
-    subm = gtk_menu_new ();
-    gtk_menu_item_set_submenu (GTK_MENU_ITEM (mi), subm);
-    gtk_menu_shell_append (GTK_MENU_SHELL (menu), mi);
-
-    /* Normal*/
-    mi = gtk_check_menu_item_new_with_label (_("Normal"));
-    gtk_check_menu_item_set_active (GTK_CHECK_MENU_ITEM (mi), power->priv->power_mode == XFPM_POWER_MODE_NORMAL);
-    gtk_widget_set_sensitive (mi,TRUE);
-
-    g_signal_connect_swapped (mi, "activate",
-			      G_CALLBACK (xfpm_power_normal_mode_cb), power);
-    gtk_widget_show (mi);
-    gtk_menu_shell_append (GTK_MENU_SHELL (subm), mi);
-
-    /* Normal*/
-    mi = gtk_check_menu_item_new_with_label (_("Presentation"));
-    gtk_check_menu_item_set_active (GTK_CHECK_MENU_ITEM (mi), power->priv->power_mode == XFPM_POWER_MODE_PRESENTATION);
-    gtk_widget_set_sensitive (mi, TRUE);
-
-    g_signal_connect_swapped (mi, "activate",
-			      G_CALLBACK (xfpm_power_presentation_mode_cb), power);
-    gtk_widget_show (mi);
-    gtk_menu_shell_append (GTK_MENU_SHELL (subm), mi);
-
-
     mi = gtk_separator_menu_item_new ();
     gtk_widget_show (mi);
     gtk_menu_shell_append (GTK_MENU_SHELL (menu), mi);
