--- ../abs/xfce4-power-manager/src/xfce4-power-manager-1.0.11/src/xfpm-backlight.c.ORIG	2012-03-30 11:38:12.903085578 -0700
+++ ../abs/xfce4-power-manager/src/xfce4-power-manager-1.0.11/src/xfpm-backlight.c	2012-03-30 11:39:51.689749191 -0700
@@ -284,6 +284,7 @@
 static void
 xfpm_backlight_button_pressed_cb (XfpmButton *button, XfpmButtonKey type, XfpmBacklight *backlight)
 {
+    return;
     glong level;
     gboolean ret = TRUE;
     
