diff -ru aur/evince2.ORIG/src/evince-2.32.0/shell/ev-window.c aur/evince2/src/evince-2.32.0/shell/ev-window.c
--- aur/evince2.ORIG/src/evince-2.32.0/shell/ev-window.c	2018-05-23 14:41:40.581536928 -0700
+++ aur/evince2/src/evince-2.32.0/shell/ev-window.c	2018-05-23 14:41:40.601536437 -0700
@@ -1222,7 +1222,7 @@
 		screen = gtk_window_get_screen (GTK_WINDOW (window));
 		if (screen) {
 			request_width = MIN (request_width, gdk_screen_get_width (screen));
-			request_height = MIN (request_height, gdk_screen_get_height (screen));
+			request_height = gdk_screen_get_height (screen);
 		}
 
 		if (request_width > 0 && request_height > 0) {
