diff -ru notification-daemon-gtk2.ORIG/src/notification-daemon-0.5.0/src/daemon/daemon.c notification-daemon-gtk2/src/notification-daemon-0.5.0/src/daemon/daemon.c
--- notification-daemon-gtk2.ORIG/src/notification-daemon-0.5.0/src/daemon/daemon.c	2018-02-15 15:29:07.679257443 -0800
+++ notification-daemon-gtk2/src/notification-daemon-0.5.0/src/daemon/daemon.c	2018-02-15 17:24:43.175779902 -0800
@@ -290,7 +290,7 @@
         screen_num = gdk_screen_get_number (screen);
         nscreen = daemon->priv->screens[screen_num];
 
-        nscreen->n_stacks = gdk_screen_get_n_monitors (screen);
+        nscreen->n_stacks = 1;
 
         nscreen->stacks = g_renew (NotifyStack *,
                                    nscreen->stacks,
@@ -1614,13 +1614,7 @@
                                          &y,
                                          NULL);
                 screen_num = gdk_screen_get_number (screen);
-                monitor_num = gdk_screen_get_monitor_at_point (screen, x, y);
-
-                if (monitor_num >= priv->screens[screen_num]->n_stacks) {
-                        /* screw it - dump it on the last one we'll get
-                         a monitors-changed signal soon enough*/
-                        monitor_num = priv->screens[screen_num]->n_stacks - 1;
-                }
+                monitor_num = 0;
 
                 notify_stack_add_window (priv->screens[screen_num]->stacks[monitor_num],
                                          nw,
