diff -ru orage-4.10.ORIG/src/orage-4.10.0/panel-plugin/xfce4-orageclock-plugin.c orage-4.10/src/orage-4.10.0/panel-plugin/xfce4-orageclock-plugin.c
--- orage-4.10.ORIG/src/orage-4.10.0/panel-plugin/xfce4-orageclock-plugin.c	2017-12-27 18:55:04.726904375 -0800
+++ orage-4.10/src/orage-4.10.0/panel-plugin/xfce4-orageclock-plugin.c	2017-12-27 19:05:19.281974905 -0800
@@ -94,6 +94,8 @@
     clock_line->clock = clock;
 
     clock_line->label = gtk_label_new("");
+    gtk_misc_set_alignment(GTK_MISC(clock_line->label), 1.0f, 1.0f);
+    gtk_misc_set_padding(GTK_MISC(clock_line->label), 5, 0);
     gtk_box_pack_start(GTK_BOX(clock->vbox), clock_line->label
             , FALSE, FALSE, 0);
     gtk_box_reorder_child(GTK_BOX(clock->vbox), clock_line->label, pos);
