diff --color -ru gimp-devel.ORIG/src/gimp-2.99.14/app/display/gimpdisplayshell-callbacks.c gimp-devel/src/gimp-2.99.14/app/display/gimpdisplayshell-callbacks.c
--- gimp-devel.ORIG/src/gimp-2.99.14/app/display/gimpdisplayshell-callbacks.c	2022-09-01 15:02:47.087962435 -0700
+++ gimp-devel/src/gimp-2.99.14/app/display/gimpdisplayshell-callbacks.c	2022-09-01 15:04:26.583121265 -0700
@@ -626,8 +626,6 @@
 
   gimp_display_shell_draw_background (shell, cr);
 
-  gimp_cairo_draw_drop_wilber (shell->canvas, cr, shell->blink);
-
   cairo_restore (cr);
 
 #ifdef GIMP_UNSTABLE
