diff -ru aur/xfce4-power-manager-1.2.ORIG/src/xfce4-power-manager-1.2.0/settings/xfpm-settings.c aur/xfce4-power-manager-1.2/src/xfce4-power-manager-1.2.0/settings/xfpm-settings.c
--- aur/xfce4-power-manager-1.2.ORIG/src/xfce4-power-manager-1.2.0/settings/xfpm-settings.c	2014-05-08 22:49:26.365396471 -0700
+++ aur/xfce4-power-manager-1.2/src/xfce4-power-manager-1.2.0/settings/xfpm-settings.c	2014-05-08 22:49:42.098730512 -0700
@@ -1560,7 +1560,7 @@
     gtk_tree_view_append_column(GTK_TREE_VIEW(view),col);
     
     /*General settings */
-    pix = xfpm_icon_load ("preferences-system", 48); 
+    pix = xfpm_icon_load ("preferences-system", 16); 
     
     gtk_list_store_append(list_store, &iter);
     
@@ -1576,7 +1576,7 @@
     i++;
     
     /* ON ac power */
-    pix = xfpm_icon_load (XFPM_AC_ADAPTER_ICON, 48); 
+    pix = xfpm_icon_load (XFPM_AC_ADAPTER_ICON, 16); 
     gtk_list_store_append(list_store, &iter);
     if ( pix )
     {
@@ -1591,7 +1591,7 @@
     
     if ( system_laptop )
     {
-	pix = xfpm_icon_load (XFPM_BATTERY_ICON, 48); 
+	pix = xfpm_icon_load (XFPM_BATTERY_ICON, 16); 
 	gtk_list_store_append(list_store, &iter);
 	if ( pix )
 	{
@@ -1605,7 +1605,7 @@
     }
     i++;
     
-    pix = xfpm_icon_load ("applications-other", 48); 
+    pix = xfpm_icon_load ("applications-other", 16); 
     gtk_list_store_append(list_store, &iter);
     if ( pix )
     {
Only in aur/xfce4-power-manager-1.2/src/xfce4-power-manager-1.2.0/settings: xfpm-settings.c.orig
diff -ru aur/xfce4-power-manager-1.2.ORIG/src/xfce4-power-manager-1.2.0/src/xfpm-notify.c aur/xfce4-power-manager-1.2/src/xfce4-power-manager-1.2.0/src/xfpm-notify.c
--- aur/xfce4-power-manager-1.2.ORIG/src/xfce4-power-manager-1.2.0/src/xfpm-notify.c	2014-05-08 22:49:26.358729804 -0700
+++ aur/xfce4-power-manager-1.2/src/xfce4-power-manager-1.2.0/src/xfpm-notify.c	2014-05-08 22:49:42.098730512 -0700
@@ -197,7 +197,7 @@
 static void
 xfpm_notify_set_notification_icon (NotifyNotification *n, const gchar *icon_name )
 {
-    GdkPixbuf *pix = xfpm_icon_load (icon_name, 48);
+    GdkPixbuf *pix = xfpm_icon_load (icon_name, 16);
     
     if ( pix )
     {
diff -ru aur/xfce4-power-manager-1.2.ORIG/src/xfce4-power-manager-1.2.0/src/xfpm-power-info.c aur/xfce4-power-manager-1.2/src/xfce4-power-manager-1.2.0/src/xfpm-power-info.c
--- aur/xfce4-power-manager-1.2.ORIG/src/xfce4-power-manager-1.2.0/src/xfpm-power-info.c	2014-05-08 22:49:26.362063138 -0700
+++ aur/xfce4-power-manager-1.2/src/xfce4-power-manager-1.2.0/src/xfpm-power-info.c	2014-05-08 22:49:42.098730512 -0700
@@ -279,7 +279,7 @@
     
     pix = gtk_icon_theme_load_icon (gtk_icon_theme_get_default (),
 				    icon_name,
-				    48,
+				    16,
 				    GTK_ICON_LOOKUP_USE_BUILTIN,
 				    NULL);
     
Only in aur/xfce4-power-manager-1.2/src/xfce4-power-manager-1.2.0/src: xfpm-power-info.c.orig
