diff -ru thunar.ORIG/src/thunar-4.16.6/thunar/thunar-launcher.c thunar/src/thunar-4.16.6/thunar/thunar-launcher.c
--- thunar.ORIG/src/thunar-4.16.6/thunar/thunar-launcher.c	2020-12-26 22:33:39.644938041 -0800
+++ thunar/src/thunar-4.16.6/thunar/thunar-launcher.c	2020-12-26 22:33:40.321593594 -0800
@@ -239,7 +239,7 @@
     { THUNAR_LAUNCHER_ACTION_EXECUTE,          "<Actions>/ThunarLauncher/execute",                 "",                  XFCE_GTK_IMAGE_MENU_ITEM, NULL,                                   NULL,                                                                                            "system-run",           G_CALLBACK (thunar_launcher_action_open),                },
     { THUNAR_LAUNCHER_ACTION_OPEN_IN_TAB,      "<Actions>/ThunarLauncher/open-in-new-tab",         "<Primary><shift>P", XFCE_GTK_MENU_ITEM,       NULL,                                   NULL,                                                                                            NULL,                   G_CALLBACK (thunar_launcher_action_open_in_new_tabs),    },
     { THUNAR_LAUNCHER_ACTION_OPEN_IN_WINDOW,   "<Actions>/ThunarLauncher/open-in-new-window",      "<Primary><shift>O", XFCE_GTK_MENU_ITEM,       NULL,                                   NULL,                                                                                            NULL,                   G_CALLBACK (thunar_launcher_action_open_in_new_windows), },
-    { THUNAR_LAUNCHER_ACTION_OPEN_WITH_OTHER,  "<Actions>/ThunarLauncher/open-with-other",         "",                  XFCE_GTK_MENU_ITEM,       N_ ("Open With Other _Application..."), N_ ("Choose another application with which to open the selected file"),                          NULL,                   G_CALLBACK (thunar_launcher_action_open_with_other),     },
+    { THUNAR_LAUNCHER_ACTION_OPEN_WITH_OTHER,  "<Actions>/ThunarLauncher/open-with-other",         "",                  XFCE_GTK_MENU_ITEM,       N_ ("Open _With..."),                   N_ ("Choose another application with which to open the selected file"),                          NULL,                   G_CALLBACK (thunar_launcher_action_open_with_other),     },
 
     /* For backward compatibility the old accel paths are re-used. Currently not possible to automatically migrate to new accel paths. */
     /* Waiting for https://gitlab.gnome.org/GNOME/gtk/issues/2375 to be able to fix that */
@@ -2755,7 +2755,7 @@
   /* add open with subitem per application */
   for (lp = applications; lp != NULL; lp = lp->next)
     {
-      label_text = g_strdup_printf (_("Open With \"%s\""), g_app_info_get_name (lp->data));
+      label_text = g_strdup_printf (_("Open with %s"), g_app_info_get_name (lp->data));
       tooltip_text = g_strdup_printf (ngettext ("Use \"%s\" to open the selected file",
                                            "Use \"%s\" to open the selected files",
                                            launcher->n_files_to_process), g_app_info_get_name (lp->data));
@@ -2822,7 +2822,7 @@
     }
   else if (G_LIKELY (applications != NULL))
     {
-      label_text = g_strdup_printf (_("_Open With \"%s\""), g_app_info_get_name (applications->data));
+      label_text = g_strdup_printf (_("_Open with %s"), g_app_info_get_name (applications->data));
       tooltip_text = g_strdup_printf (ngettext ("Use \"%s\" to open the selected file",
                                                 "Use \"%s\" to open the selected files",
                                                 launcher->n_files_to_process), g_app_info_get_name (applications->data));
