diff --color -ru thunar.ORIG/src/thunar-4.18.3/thunar/thunar-window.c thunar/src/thunar-4.18.3/thunar/thunar-window.c
--- thunar.ORIG/src/thunar-4.18.3/thunar/thunar-window.c	2023-02-16 00:05:16.450417064 -0800
+++ thunar/src/thunar-4.18.3/thunar/thunar-window.c	2023-02-16 00:07:32.251563261 -0800
@@ -4778,7 +4778,6 @@
                                          ThunarWindow *window)
 {
   gboolean      show_full_path;
-  gchar        *title;
   gchar        *parse_name = NULL;
   const gchar  *name;
 
@@ -4794,9 +4793,7 @@
     name = thunar_file_get_display_name (current_directory);
 
   /* set window title */
-  title = g_strdup_printf ("%s - %s", name, "Thunar");
-  gtk_window_set_title (GTK_WINDOW (window), title);
-  g_free (title);
+  gtk_window_set_title (GTK_WINDOW (window), name);
   g_free (parse_name);
 
   /* set window icon */
