diff -ru thunar.ORIG/src/thunar-4.18.10/thunar/thunar-window.c thunar/src/thunar-4.18.10/thunar/thunar-window.c
--- thunar.ORIG/src/thunar-4.18.10/thunar/thunar-window.c	2023-10-24 15:53:44.921473063 -0700
+++ thunar/src/thunar-4.18.10/thunar/thunar-window.c	2023-10-24 15:53:45.321478978 -0700
@@ -4805,7 +4805,6 @@
                                          ThunarWindow *window)
 {
   gboolean      show_full_path;
-  gchar        *title;
   gchar        *parse_name = NULL;
   const gchar  *name;
 
@@ -4821,9 +4820,7 @@
     name = thunar_file_get_display_name (current_directory);
 
   /* set window title */
-  title = g_strdup_printf ("%s - %s", name, "Thunar");
-  gtk_window_set_title (GTK_WINDOW (window), title);
-  g_free (title);
+  gtk_window_set_title (GTK_WINDOW (window), name);
   g_free (parse_name);
 
   /* set window icon */
