diff -ru thunar.ORIG/src/thunar-4.16.10/thunar/thunar-window.c thunar/src/thunar-4.16.10/thunar/thunar-window.c
--- thunar.ORIG/src/thunar-4.16.10/thunar/thunar-window.c	2021-05-06 16:23:05.256867618 -0700
+++ thunar/src/thunar-4.16.10/thunar/thunar-window.c	2021-05-06 16:23:05.723528522 -0700
@@ -815,8 +815,9 @@
   /* add the location bar itself */
   gtk_container_add (GTK_CONTAINER (tool_item), window->location_bar);
 
-  /* display the toolbar */
-  gtk_widget_show_all (window->location_toolbar);
+  gtk_widget_show (tool_item);
+  gtk_widget_show (window->location_bar);
+  gtk_widget_show (window->location_toolbar);
 
   g_free (last_location_bar);
 
