diff -ru thunar.ORIG/src/Thunar-1.8.9/thunar/thunar-application.c thunar/src/Thunar-1.8.9/thunar/thunar-application.c
--- thunar.ORIG/src/Thunar-1.8.9/thunar/thunar-application.c	2019-02-12 11:51:06.180076421 -0800
+++ thunar/src/Thunar-1.8.9/thunar/thunar-application.c	2019-02-12 11:51:07.656693780 -0800
@@ -1167,16 +1167,8 @@
 {
   _thunar_return_if_fail (THUNAR_IS_APPLICATION (application));
 
-  if (application->daemon != daemonize)
-    {
-      application->daemon = daemonize;
-      g_object_notify (G_OBJECT (application), "daemon");
-
-      if (daemonize)
-        g_application_hold (G_APPLICATION (application));
-      else
-        g_application_release (G_APPLICATION (application));
-    }
+  application->daemon = FALSE;
+  g_object_notify (G_OBJECT (application), "daemon");
 }
 
 
