diff -ru thunar.ORIG/src/Thunar-1.8.4/thunar/thunar-window.c thunar/src/Thunar-1.8.4/thunar/thunar-window.c
--- thunar.ORIG/src/Thunar-1.8.4/thunar/thunar-window.c	2018-09-26 15:17:32.621627555 -0700
+++ thunar/src/Thunar-1.8.4/thunar/thunar-window.c	2018-09-26 15:19:47.469619820 -0700
@@ -1987,7 +1987,7 @@
     {
       /* allocate the new side pane widget */
       window->sidepane = g_object_new (type, NULL);
-      gtk_widget_set_size_request (window->sidepane, 0, -1);
+      gtk_widget_set_size_request (window->sidepane, 165, -1);
       thunar_component_set_ui_manager (THUNAR_COMPONENT (window->sidepane), window->ui_manager);
       exo_binding_new (G_OBJECT (window), "show-hidden", G_OBJECT (window->sidepane), "show-hidden");
       exo_binding_new (G_OBJECT (window), "current-directory", G_OBJECT (window->sidepane), "current-directory");
