diff -ru ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-emblem-chooser.c ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-emblem-chooser.c
--- ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-emblem-chooser.c	2012-05-04 13:52:16.712724004 -0700
+++ ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-emblem-chooser.c	2012-05-04 13:52:27.016057019 -0700
@@ -400,7 +400,7 @@
   GdkPixbuf   *icon;
 
   /* lookup the icon info for the emblem */
-  info = gtk_icon_theme_lookup_icon (chooser->icon_theme, emblem, 48, 0);
+  info = gtk_icon_theme_lookup_icon (chooser->icon_theme, emblem, 16, 0);
   if (G_UNLIKELY (info == NULL))
     return NULL;
 
diff -ru ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-icon-renderer.c ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-icon-renderer.c
--- ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-icon-renderer.c	2012-05-04 13:52:16.712724004 -0700
+++ ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-icon-renderer.c	2012-05-04 13:52:27.016057019 -0700
@@ -461,7 +461,7 @@
           for (lp = emblems, position = 0; lp != NULL && position < max_emblems; lp = lp->next)
             {
               /* check if we have the emblem in the icon theme */
-              emblem = thunar_icon_factory_load_icon (icon_factory, lp->data, icon_renderer->size, NULL, FALSE);
+              emblem = thunar_icon_factory_load_icon (icon_factory, lp->data, 16, NULL, FALSE);
               if (G_UNLIKELY (emblem == NULL))
                 continue;
 
@@ -469,19 +469,6 @@
               emblem_area.width = gdk_pixbuf_get_width (emblem);
               emblem_area.height = gdk_pixbuf_get_height (emblem);
 
-              /* shrink insane emblems */
-              if (G_UNLIKELY (MAX (emblem_area.width, emblem_area.height) > (gint) MIN ((2 * icon_renderer->size) / 3, 32)))
-                {
-                  /* scale down the emblem */
-                  temp = exo_gdk_pixbuf_scale_ratio (emblem, MIN ((2 * icon_renderer->size) / 3, 32));
-                  g_object_unref (G_OBJECT (emblem));
-                  emblem = temp;
-
-                  /* determine the size again */
-                  emblem_area.width = gdk_pixbuf_get_width (emblem);
-                  emblem_area.height = gdk_pixbuf_get_height (emblem);
-                }
-
               /* determine a good position for the emblem, depending on the position index */
               switch (position)
                 {
