diff -ru thunar.ORIG/src/thunar/thunar/thunar-application.c thunar/src/thunar/thunar/thunar-application.c
--- thunar.ORIG/src/thunar/thunar/thunar-application.c	2025-09-15 13:01:18.518973581 -0700
+++ thunar/src/thunar/thunar/thunar-application.c	2025-09-15 13:01:20.382496438 -0700
@@ -700,19 +700,9 @@
   gtk_css_provider_load_from_data (css_provider,
                                    /* for the location-buttons any margin looks ugly */
                                    ".location-button { margin-right: 0; }"
-                                   /* add missing top border to side pane */
-                                   ".sidebar { border-top-style: solid; }"
-                                   /* remove extra borders */
-                                   ".preview-pane { border-left-width: 0px; border-right-width: 0px; border-bottom-width: 0px; }"
-                                   ".standard-view { border-left-width: 0px; border-right-width: 0px; }"
-                                   /* make border thicker during DnD */
-                                   ".standard-view:drop(active) { border-width: 2px; }"
-                                   /* change background color of inactive split view pane */
-                                   ".split-view-inactive-pane .view { background-color: @theme_unfocused_bg_color; }"
-                                   /* for the toorbar in headerbar looks ugly */
-                                   "headerbar toolbar { background: none; }"
-                                   /* for the example box in properties dialog > highlight tab */
-                                   "#example { border-radius: 10px; }",
+                                   ".sidebar { border-width: 1px; border-style: solid; }"
+                                   ".shortcuts-pane, .tree-pane { border-right-width: 0px; }"
+                                   ".standard-view { margin-left: 5px; }",
                                    -1, NULL);
   screen = gdk_screen_get_default ();
   gtk_style_context_add_provider_for_screen (screen, GTK_STYLE_PROVIDER (css_provider), GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
diff -ru thunar.ORIG/src/thunar/thunar/thunar-tree-pane.c thunar/src/thunar/thunar/thunar-tree-pane.c
--- thunar.ORIG/src/thunar/thunar/thunar-tree-pane.c	2025-09-15 13:01:18.519037204 -0700
+++ thunar/src/thunar/thunar/thunar-tree-pane.c	2025-09-15 13:01:20.382845273 -0700
@@ -157,6 +157,8 @@
   g_signal_connect_swapped (G_OBJECT (tree_pane->view), "open-new-tab", G_CALLBACK (thunar_navigator_open_new_tab), tree_pane);
   gtk_container_add (GTK_CONTAINER (tree_pane), tree_pane->view);
   gtk_widget_show (tree_pane->view);
+
+  gtk_style_context_add_class (gtk_widget_get_style_context (GTK_WIDGET (tree_pane)), "tree-pane");
 }
 
 
diff -ru thunar.ORIG/src/thunar/thunar/thunar-window.c thunar/src/thunar/thunar/thunar-window.c
--- thunar.ORIG/src/thunar/thunar/thunar-window.c	2025-09-15 13:01:18.519037204 -0700
+++ thunar/src/thunar/thunar/thunar-window.c	2025-09-15 13:01:20.383308027 -0700
@@ -66,10 +66,6 @@
 
 
 
-#define DEFAULT_LOCATION_BAR_MARGIN 5
-
-
-
 /* Property identifiers */
 enum
 {
@@ -1070,7 +1066,7 @@
 
   /* main paned widget */
   window->paned = gtk_paned_new (GTK_ORIENTATION_HORIZONTAL);
-  gtk_container_set_border_width (GTK_CONTAINER (window->paned), 0);
+  gtk_container_set_border_width (GTK_CONTAINER (window->paned), 5);
   gtk_widget_set_hexpand (window->paned, TRUE);
   gtk_widget_set_vexpand (window->paned, TRUE);
   gtk_grid_attach (GTK_GRID (window->grid), window->paned, 0, 4, 1, 1);
@@ -2060,8 +2056,8 @@
       below_header_bar = window->location_toolbar;
 
       /* restore default location bar margins */
-      gtk_widget_set_margin_start (window->location_bar, DEFAULT_LOCATION_BAR_MARGIN);
-      gtk_widget_set_margin_end (window->location_bar, DEFAULT_LOCATION_BAR_MARGIN);
+      gtk_widget_set_margin_start (window->location_bar, 1);
+      gtk_widget_set_margin_end (window->location_bar, 2);
     }
   else
     {
@@ -6696,8 +6692,9 @@
 {
   /* allocate the new location bar widget */
   window->location_bar = thunar_location_bar_new ();
-  gtk_widget_set_margin_start (window->location_bar, DEFAULT_LOCATION_BAR_MARGIN);
-  gtk_widget_set_margin_end (window->location_bar, DEFAULT_LOCATION_BAR_MARGIN);
+  gtk_widget_set_margin_top (GTK_WIDGET (window->location_bar), 3);
+  gtk_widget_set_margin_start (GTK_WIDGET (window->location_bar), 1);
+  gtk_widget_set_margin_end (GTK_WIDGET (window->location_bar), 2);
   g_object_bind_property (G_OBJECT (window), "current-directory", G_OBJECT (window->location_bar), "current-directory", G_BINDING_SYNC_CREATE);
   g_signal_connect_swapped (G_OBJECT (window->location_bar), "change-directory", G_CALLBACK (thunar_window_set_current_directory), window);
   g_signal_connect_swapped (G_OBJECT (window->location_bar), "open-new-tab", G_CALLBACK (thunar_window_notebook_open_new_tab), window);
