diff -ru thunar.ORIG/src/thunar-4.18.11/thunar/thunar-file.c thunar/src/thunar-4.18.11/thunar/thunar-file.c
--- thunar.ORIG/src/thunar-4.18.11/thunar/thunar-file.c	2023-10-24 15:53:44.921473063 -0700
+++ thunar/src/thunar-4.18.11/thunar/thunar-file.c	2023-10-24 15:53:45.244811178 -0700
@@ -1086,9 +1086,6 @@
           display_name = g_file_info_get_display_name (file->info);
           if (G_LIKELY (display_name != NULL))
             {
-              if (strcmp (display_name, "/") == 0)
-                file->display_name = g_strdup (_("File System"));
-              else
                 file->display_name = g_strdup (display_name);
             }
         }
