diff -ru thunar.ORIG/src/Thunar-1.8.9/thunar/thunar-standard-view.c thunar/src/Thunar-1.8.9/thunar/thunar-standard-view.c
--- thunar.ORIG/src/Thunar-1.8.9/thunar/thunar-standard-view.c	2019-02-12 11:51:06.183409643 -0800
+++ thunar/src/Thunar-1.8.9/thunar/thunar-standard-view.c	2019-02-12 11:51:07.976683097 -0800
@@ -653,7 +653,7 @@
 
   /* initialize the scrolled window */
   gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (standard_view),
-                                  GTK_POLICY_AUTOMATIC,
+                                  GTK_POLICY_NEVER,
                                   GTK_POLICY_AUTOMATIC);
   gtk_scrolled_window_set_hadjustment (GTK_SCROLLED_WINDOW (standard_view), NULL);
   gtk_scrolled_window_set_vadjustment (GTK_SCROLLED_WINDOW (standard_view), NULL);
