diff -ru thunar.ORIG/src/thunar-4.18.0/thunar/thunar-enum-types.c thunar/src/thunar-4.18.0/thunar/thunar-enum-types.c
--- thunar.ORIG/src/thunar-4.18.0/thunar/thunar-enum-types.c	2022-12-15 19:42:45.971362899 -0800
+++ thunar/src/thunar-4.18.0/thunar/thunar-enum-types.c	2022-12-15 19:42:46.478052605 -0800
@@ -148,7 +148,7 @@
         { THUNAR_COLUMN_MIME_TYPE,     "THUNAR_COLUMN_MIME_TYPE",     N_ ("MIME Type"),     },
         { THUNAR_COLUMN_NAME,          "THUNAR_COLUMN_NAME",          N_ ("Name"),          },
         { THUNAR_COLUMN_OWNER,         "THUNAR_COLUMN_OWNER",         N_ ("Owner"),         },
-        { THUNAR_COLUMN_PERMISSIONS,   "THUNAR_COLUMN_PERMISSIONS",   N_ ("Permissions"),   },
+        { THUNAR_COLUMN_PERMISSIONS,   "THUNAR_COLUMN_PERMISSIONS",   N_ ("Mode"),   },
         { THUNAR_COLUMN_SIZE,          "THUNAR_COLUMN_SIZE",          N_ ("Size"),          },
         { THUNAR_COLUMN_SIZE_IN_BYTES, "THUNAR_COLUMN_SIZE_IN_BYTES", N_ ("Size in Bytes"), },
         { THUNAR_COLUMN_TYPE,          "THUNAR_COLUMN_TYPE",          N_ ("Type"),          },
