diff -ru thunar.ORIG/src/thunar/thunar/thunar-application.c thunar/src/thunar/thunar/thunar-application.c
--- thunar.ORIG/src/thunar/thunar/thunar-application.c	2024-12-30 15:42:25.706759478 -0800
+++ thunar/src/thunar/thunar/thunar-application.c	2024-12-30 15:42:26.330103370 -0800
@@ -1308,16 +1308,8 @@
 {
   _thunar_return_if_fail (THUNAR_IS_APPLICATION (application));
 
-  if (application->daemon != daemonize)
-    {
-      application->daemon = daemonize;
-      g_object_notify (G_OBJECT (application), "daemon");
-
-      if (daemonize)
-        g_application_hold (G_APPLICATION (application));
-      else
-        g_application_release (G_APPLICATION (application));
-    }
+  application->daemon = FALSE;
+  g_object_notify (G_OBJECT (application), "daemon");
 }
 
 
