diff -ru abs/thunar.ORIG/src/Thunar-1.6.0/thunar/thunar-shortcuts-model.c abs/thunar/src/Thunar-1.6.0/thunar/thunar-shortcuts-model.c
--- abs/thunar.ORIG/src/Thunar-1.6.0/thunar/thunar-shortcuts-model.c	2012-12-05 19:56:09.619174692 -0800
+++ abs/thunar/src/Thunar-1.6.0/thunar/thunar-shortcuts-model.c	2012-12-05 20:31:01.597691860 -0800
@@ -884,7 +884,7 @@
   shortcut->group = THUNAR_SHORTCUT_GROUP_NETWORK_DEFAULT;
   shortcut->name = g_strdup (_("Browse Network"));
   shortcut->location = g_file_new_for_uri ("network://");
-  shortcut->gicon = g_themed_icon_new (GTK_STOCK_NETWORK);
+  shortcut->gicon = g_themed_icon_new ("network-server");
   shortcut->hidden = thunar_shortcuts_model_get_hidden (model, shortcut);
   thunar_shortcuts_model_add_shortcut (model, shortcut);
 }
@@ -916,7 +916,7 @@
       shortcut = g_slice_new0 (ThunarShortcut);
       shortcut->group = THUNAR_SHORTCUT_GROUP_PLACES_DEFAULT;
       shortcut->file = file;
-      shortcut->gicon = g_themed_icon_new (GTK_STOCK_HOME);
+      shortcut->gicon = g_themed_icon_new ("user-home");
       shortcut->sort_id = 0;
       shortcut->hidden = thunar_shortcuts_model_get_hidden (model, shortcut);
       thunar_shortcuts_model_add_shortcut (model, shortcut);
