diff -ru thunar.ORIG/src/thunar-4.16.8/thunar/thunar-application.c thunar/src/thunar-4.16.8/thunar/thunar-application.c
--- thunar.ORIG/src/thunar-4.16.8/thunar/thunar-application.c	2021-05-06 16:23:05.256867618 -0700
+++ thunar/src/thunar-4.16.8/thunar/thunar-application.c	2021-05-06 16:23:05.633529633 -0700
@@ -1153,16 +1153,8 @@
 {
   _thunar_return_if_fail (THUNAR_IS_APPLICATION (application));
 
-  if (application->daemon != daemonize)
-    {
-      application->daemon = daemonize;
-      g_object_notify (G_OBJECT (application), "daemon");
-
-      if (daemonize)
-        g_application_hold (G_APPLICATION (application));
-      else
-        g_application_release (G_APPLICATION (application));
-    }
+  application->daemon = FALSE;
+  g_object_notify (G_OBJECT (application), "daemon");
 }
 
 
