diff -ru thunar.ORIG/src/thunar/thunar/thunar-dialogs.c thunar/src/thunar/thunar/thunar-dialogs.c
--- thunar.ORIG/src/thunar/thunar/thunar-dialogs.c	2025-10-20 03:25:36.232836195 -0700
+++ thunar/src/thunar/thunar/thunar-dialogs.c	2025-10-20 03:25:36.675923398 -0700
@@ -94,7 +94,6 @@
   GdkScreen         *screen;
   GError            *error = NULL;
   gchar             *name = NULL;
-  GtkWidget         *label;
   GtkWidget         *grid;
   GtkWidget         *image;
   XfceFilenameInput *filename_input;
@@ -141,10 +140,6 @@
       gtk_widget_show (image);
     }
 
-  label = g_object_new (GTK_TYPE_LABEL, "label", _("Enter the name:"), "xalign", 0.0f, "hexpand", TRUE, NULL);
-  gtk_grid_attach (GTK_GRID (grid), label, 1, row, 1, 1);
-  gtk_widget_show (label);
-
   /* set up the widget for entering the filename */
   filename_input = g_object_new (XFCE_TYPE_FILENAME_INPUT, "original-filename", filename, NULL);
   gtk_widget_set_hexpand (GTK_WIDGET (filename_input), TRUE);
@@ -161,8 +156,6 @@
   row++;
 
   gtk_grid_attach (GTK_GRID (grid), GTK_WIDGET (filename_input), 1, row, 1, 1);
-  thunar_gtk_label_set_a11y_relation (GTK_LABEL (label),
-                                      GTK_WIDGET (xfce_filename_input_get_entry (filename_input)));
   gtk_widget_show_all (GTK_WIDGET (filename_input));
 
   /* ensure that the sensitivity of the Create button is set correctly */
@@ -229,7 +222,6 @@
   const gchar       *text;
   ThunarJob         *job = NULL;
   GtkWidget         *dialog;
-  GtkWidget         *label;
   GtkWidget         *image;
   GtkWidget         *grid;
   XfceFilenameInput *filename_input;
@@ -299,12 +291,6 @@
   cairo_surface_destroy (surface);
   gtk_widget_show (image);
 
-  label = gtk_label_new (_("Enter the new name:"));
-  gtk_label_set_xalign (GTK_LABEL (label), 0.0f);
-  gtk_widget_set_hexpand (label, TRUE);
-  gtk_grid_attach (GTK_GRID (grid), label, 1, row, 1, 1);
-  gtk_widget_show (label);
-
   /* set up the widget for entering the filename */
   filename_input = g_object_new (XFCE_TYPE_FILENAME_INPUT, "original-filename", filename, NULL);
   filename_input_entry = xfce_filename_input_get_entry (filename_input);
@@ -322,7 +308,6 @@
   row++;
 
   gtk_grid_attach (GTK_GRID (grid), GTK_WIDGET (filename_input), 1, row, 1, 1);
-  thunar_gtk_label_set_a11y_relation (GTK_LABEL (label), GTK_WIDGET (filename_input_entry));
   gtk_widget_show_all (GTK_WIDGET (filename_input));
 
   /* ensure that the sensitivity of the Create button is set correctly */
