diff -ru thunar.ORIG/src/thunar-1.8.12/thunar/thunar-application.c thunar/src/thunar-1.8.12/thunar/thunar-application.c
--- thunar.ORIG/src/thunar-1.8.12/thunar/thunar-application.c	2019-11-11 08:24:30.426793390 -0800
+++ thunar/src/thunar-1.8.12/thunar/thunar-application.c	2019-11-11 08:24:31.013479328 -0800
@@ -1148,16 +1148,8 @@
 {
   _thunar_return_if_fail (THUNAR_IS_APPLICATION (application));
 
-  if (application->daemon != daemonize)
-    {
-      application->daemon = daemonize;
-      g_object_notify (G_OBJECT (application), "daemon");
-
-      if (daemonize)
-        g_application_hold (G_APPLICATION (application));
-      else
-        g_application_release (G_APPLICATION (application));
-    }
+  application->daemon = FALSE;
+  g_object_notify (G_OBJECT (application), "daemon");
 }
 
 
