diff -ru abs/thunar.ORIG/src/Thunar-1.6.7/thunar/thunar-location-dialog.c abs/thunar/src/Thunar-1.6.7/thunar/thunar-location-dialog.c
--- abs/thunar.ORIG/src/Thunar-1.6.7/thunar/thunar-location-dialog.c	2013-05-07 00:23:01.655099745 -0700
+++ abs/thunar/src/Thunar-1.6.7/thunar/thunar-location-dialog.c	2013-05-07 00:29:49.510219156 -0700
@@ -57,11 +57,10 @@
 {
   GtkWidget *open_button;
   GtkWidget *hbox;
-  GtkWidget *label;
 
   gtk_box_set_spacing (GTK_BOX (GTK_DIALOG (location_dialog)->vbox), 2);
   gtk_container_set_border_width (GTK_CONTAINER (location_dialog), 5);
-  gtk_window_set_default_size (GTK_WINDOW (location_dialog), 350, -1);
+  gtk_window_set_default_size (GTK_WINDOW (location_dialog), 400, -1);
   gtk_window_set_title (GTK_WINDOW (location_dialog), _("Open Location"));
 
   gtk_dialog_add_button (GTK_DIALOG (location_dialog), GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL);
@@ -76,15 +75,9 @@
   gtk_box_pack_start (GTK_BOX (GTK_DIALOG (location_dialog)->vbox), hbox, FALSE, FALSE, 0);
   gtk_widget_show (hbox);
 
-  label = gtk_label_new_with_mnemonic (_("_Location:"));
-  gtk_box_pack_start (GTK_BOX (hbox), label, FALSE, FALSE, 0);
-  gtk_widget_show (label);
-
   location_dialog->entry = thunar_path_entry_new ();
   gtk_entry_set_activates_default (GTK_ENTRY (location_dialog->entry), TRUE);
   gtk_box_pack_start (GTK_BOX (hbox), location_dialog->entry, TRUE, TRUE, 0);
-  thunar_gtk_label_set_a11y_relation (GTK_LABEL (label), location_dialog->entry);
-  gtk_label_set_mnemonic_widget (GTK_LABEL (label), location_dialog->entry);
   gtk_widget_show (location_dialog->entry);
 
   /* the "Open" button is only sensitive if a valid file is entered */
diff -ru abs/thunar.ORIG/src/Thunar-1.6.7/thunar/thunar-path-entry.c abs/thunar/src/Thunar-1.6.7/thunar/thunar-path-entry.c
--- abs/thunar.ORIG/src/Thunar-1.6.7/thunar/thunar-path-entry.c	2013-05-07 00:23:01.658432393 -0700
+++ abs/thunar/src/Thunar-1.6.7/thunar/thunar-path-entry.c	2013-05-07 00:29:49.510219156 -0700
@@ -707,10 +707,6 @@
 
   /* let the GtkEntry class handle the insert */
   (*thunar_path_entry_editable_parent_iface->do_insert_text) (editable, new_text, new_text_length, position);
-
-  /* queue a completion check if this insert operation was triggered by the user */
-  if (G_LIKELY (!path_entry->in_change))
-    thunar_path_entry_queue_check_completion (path_entry);
 }
 
 
