diff -ru thunar.ORIG/src/thunar-4.18.4/thunar/thunar-notify.c thunar/src/thunar-4.18.4/thunar/thunar-notify.c
--- thunar.ORIG/src/thunar-4.18.4/thunar/thunar-notify.c	2023-01-10 23:31:26.872889245 -0800
+++ thunar/src/thunar-4.18.4/thunar/thunar-notify.c	2023-01-10 23:31:27.526252075 -0800
@@ -278,17 +278,13 @@
   if (thunar_notify_device_readonly (device))
     {
       summary = _("Unmounting device");
-      message = g_strdup_printf (_("The device \"%s\" is being unmounted by the system. "
-                                   "Please do not remove the media or disconnect the "
-                                   "drive"), name);
+      message = g_strdup_printf (_("The device \"%s\" is being unmounted"), name);
     }
   else
     {
       summary = _("Writing data to device");
       message = g_strdup_printf (_("There is data that needs to be written to the "
-                                   "device \"%s\" before it can be removed. Please "
-                                   "do not remove the media or disconnect the drive"),
-                                   name);
+                                   "device \"%s\" before it can be removed"), name);
     }
 
   thunar_notify_show (device, summary, message);
@@ -318,16 +314,13 @@
   if (thunar_notify_device_readonly (device))
     {
       summary = _("Ejecting device");
-      message = g_strdup_printf (_("The device \"%s\" is being ejected. "
-                                   "This may take some time"), name);
+      message = g_strdup_printf (_("The device \"%s\" is being ejected"), name);
     }
   else
     {
       summary = _("Writing data to device");
       message = g_strdup_printf (_("There is data that needs to be written to the "
-                                   "device \"%s\" before it can be removed. Please "
-                                   "do not remove the media or disconnect the drive"),
-                                   name);
+                                   "device \"%s\" before it can be removed"), name);
     }
 
   thunar_notify_show (device, summary, message);
