diff -ru thunar.ORIG/src/thunar-4.18.0/thunar/thunar-file.c thunar/src/thunar-4.18.0/thunar/thunar-file.c
--- thunar.ORIG/src/thunar-4.18.0/thunar/thunar-file.c	2022-12-15 19:42:45.971362899 -0800
+++ thunar/src/thunar-4.18.0/thunar/thunar-file.c	2022-12-15 19:42:46.314711844 -0800
@@ -1081,9 +1081,6 @@
           display_name = g_file_info_get_display_name (file->info);
           if (G_LIKELY (display_name != NULL))
             {
-              if (strcmp (display_name, "/") == 0)
-                file->display_name = g_strdup (_("File System"));
-              else
                 file->display_name = g_strdup (display_name);
             }
         }
