diff -ru thunar.ORIG/src/thunar-4.16.5/thunar/thunar-launcher.c thunar/src/thunar-4.16.5/thunar/thunar-launcher.c
--- thunar.ORIG/src/thunar-4.16.5/thunar/thunar-launcher.c	2020-12-26 22:33:39.644938041 -0800
+++ thunar/src/thunar-4.16.5/thunar/thunar-launcher.c	2020-12-26 22:33:40.231595073 -0800
@@ -2852,8 +2852,6 @@
 
   if (launcher->n_files_to_process == launcher->n_directories_to_process && launcher->n_directories_to_process >= 1)
     {
-      if (support_tabs)
-        thunar_launcher_append_menu_item (launcher, GTK_MENU_SHELL (menu), THUNAR_LAUNCHER_ACTION_OPEN_IN_TAB, FALSE);
       thunar_launcher_append_menu_item (launcher, GTK_MENU_SHELL (menu), THUNAR_LAUNCHER_ACTION_OPEN_IN_WINDOW, FALSE);
     }
 
diff -ru thunar.ORIG/src/thunar-4.16.5/thunar/thunar-shortcuts-view.c thunar/src/thunar-4.16.5/thunar/thunar-shortcuts-view.c
--- thunar.ORIG/src/thunar-4.16.5/thunar/thunar-shortcuts-view.c	2020-12-26 22:33:39.644938041 -0800
+++ thunar/src/thunar-4.16.5/thunar/thunar-shortcuts-view.c	2020-12-26 22:33:40.231595073 -0800
@@ -1127,12 +1127,6 @@
   gtk_image_menu_item_set_image (GTK_IMAGE_MENU_ITEM (item), image);
 G_GNUC_END_IGNORE_DEPRECATIONS
 
-  /* append the "Open in New Tab" menu action */
-  item = gtk_menu_item_new_with_mnemonic (_("Open in New Tab"));
-  g_signal_connect_swapped (G_OBJECT (item), "activate", G_CALLBACK (thunar_shortcuts_view_open_in_new_tab_clicked), view);
-  gtk_menu_shell_append (GTK_MENU_SHELL (menu), item);
-  gtk_widget_show (item);
-
   /* append the "Open in New Window" menu action */
   item = gtk_menu_item_new_with_mnemonic (_("Open in New Window"));
   g_signal_connect_swapped (G_OBJECT (item), "activate", G_CALLBACK (thunar_shortcuts_view_open_in_new_window_clicked), view);
