diff -ru thunar.ORIG/src/thunar-4.18.4/thunar/thunar-window.c thunar/src/thunar-4.18.4/thunar/thunar-window.c
--- thunar.ORIG/src/thunar-4.18.4/thunar/thunar-window.c	2023-03-23 12:05:40.124032949 -0700
+++ thunar/src/thunar-4.18.4/thunar/thunar-window.c	2023-03-23 12:05:40.907386135 -0700
@@ -2946,7 +2946,7 @@
 {
   gboolean      change_window_icon;
   GtkIconTheme *icon_theme;
-  const gchar  *icon_name = "folder";
+  const gchar  *icon_name = "thunar-window";
 
   g_object_get (window->preferences, "misc-change-window-icon", &change_window_icon, NULL);
 
