--- ../abs/thunar/src/Thunar-1.2.3/thunar/thunar-tree-model.c.ORIG	2010-11-19 07:21:48.000000000 -0800
+++ ../abs/thunar/src/Thunar-1.2.3/thunar/thunar-tree-model.c	2011-02-01 22:44:30.421938428 -0800
@@ -326,9 +326,10 @@
   /* append the root file system */
   system_paths = g_list_append (system_paths, thunar_g_file_new_for_root ());
 
+  /* gavin: no network icon in the tree pane */
   /* append the network icon if browsing the network is supported */
-  if (thunar_g_vfs_is_uri_scheme_supported ("network"))
-    system_paths = g_list_append (system_paths, g_file_new_for_uri ("network://"));
+  /* if (thunar_g_vfs_is_uri_scheme_supported ("network"))
+    system_paths = g_list_append (system_paths, g_file_new_for_uri ("network://")); */
 
   /* append the system defined nodes ('Home', 'Trash', 'File System') */
   for (lp = system_paths; lp != NULL; lp = lp->next)
