diff -ru thunar.ORIG/src/thunar-4.18.0/thunar/thunar-standard-view.c thunar/src/thunar-4.18.0/thunar/thunar-standard-view.c
--- thunar.ORIG/src/thunar-4.18.0/thunar/thunar-standard-view.c	2022-12-15 19:42:45.971362899 -0800
+++ thunar/src/thunar-4.18.0/thunar/thunar-standard-view.c	2022-12-15 19:42:46.191372903 -0800
@@ -813,7 +813,7 @@
 
   /* initialize the scrolled window */
   gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (standard_view),
-                                  GTK_POLICY_AUTOMATIC,
+                                  GTK_POLICY_NEVER,
                                   GTK_POLICY_AUTOMATIC);
   gtk_scrolled_window_set_hadjustment (GTK_SCROLLED_WINDOW (standard_view), NULL);
   gtk_scrolled_window_set_vadjustment (GTK_SCROLLED_WINDOW (standard_view), NULL);
