diff --color -ru thunar.ORIG/src/thunar/thunar/thunar-standard-view.c thunar/src/thunar/thunar/thunar-standard-view.c
--- thunar.ORIG/src/thunar/thunar/thunar-standard-view.c	2024-10-21 01:57:29.568774667 -0700
+++ thunar/src/thunar/thunar/thunar-standard-view.c	2024-10-21 01:57:47.142374244 -0700
@@ -1839,7 +1839,6 @@
                                      ThunarZoomLevel zoom_level)
 {
   ThunarStandardView *standard_view = THUNAR_STANDARD_VIEW (view);
-  gboolean newThumbnailSize = FALSE;
 
   /* check if we have a new zoom-level here */
   if (G_LIKELY (standard_view->priv->zoom_level != zoom_level))
@@ -1857,15 +1856,11 @@
             }
         }
 
-      if (thunar_zoom_level_to_thumbnail_size (zoom_level) != thunar_zoom_level_to_thumbnail_size (standard_view->priv->zoom_level))
-        newThumbnailSize = TRUE;
-
       standard_view->priv->zoom_level = zoom_level;
 
       g_object_notify_by_pspec (G_OBJECT (standard_view), standard_view_props[PROP_ZOOM_LEVEL]);
 
-      if (newThumbnailSize)
-        thunar_standard_view_reload (view, TRUE);
+      thunar_standard_view_reload (view, TRUE);
     }
 }
 
