diff -ru thunar.ORIG/src/thunar-4.16.7/thunar/thunar-details-view.c thunar/src/thunar-4.16.7/thunar/thunar-details-view.c
--- thunar.ORIG/src/thunar-4.16.7/thunar/thunar-details-view.c	2021-05-06 16:23:05.256867618 -0700
+++ thunar/src/thunar-4.16.7/thunar/thunar-details-view.c	2021-05-06 16:23:05.453531856 -0700
@@ -989,8 +989,6 @@
   ThunarDetailsView *details_view = THUNAR_DETAILS_VIEW (standard_view);
 
   _thunar_return_if_fail (THUNAR_IS_DETAILS_VIEW (details_view));
-
-  xfce_gtk_menu_item_new_from_action_entry (get_action_entry (THUNAR_DETAILS_VIEW_ACTION_CONFIGURE_COLUMNS), G_OBJECT (details_view), GTK_MENU_SHELL (menu));
 }
 
 
diff -ru thunar.ORIG/src/thunar-4.16.7/thunar/thunar-standard-view.c thunar/src/thunar-4.16.7/thunar/thunar-standard-view.c
--- thunar.ORIG/src/thunar-4.16.7/thunar/thunar-standard-view.c	2021-05-06 16:23:05.256867618 -0700
+++ thunar/src/thunar-4.16.7/thunar/thunar-standard-view.c	2021-05-06 16:23:05.453531856 -0700
@@ -3503,8 +3503,6 @@
     {
       thunar_menu_add_sections (context_menu, THUNAR_MENU_SECTION_OPEN
                                             | THUNAR_MENU_SECTION_SENDTO
-                                            | THUNAR_MENU_SECTION_CUT
-                                            | THUNAR_MENU_SECTION_COPY_PASTE
                                             | THUNAR_MENU_SECTION_TRASH_DELETE
                                             | THUNAR_MENU_SECTION_EMPTY_TRASH
                                             | THUNAR_MENU_SECTION_RESTORE
@@ -3515,13 +3513,10 @@
   else /* right click on some empty space */
     {
       thunar_menu_add_sections (context_menu, THUNAR_MENU_SECTION_CREATE_NEW_FILES
-                                            | THUNAR_MENU_SECTION_COPY_PASTE
                                             | THUNAR_MENU_SECTION_EMPTY_TRASH
                                             | THUNAR_MENU_SECTION_CUSTOM_ACTIONS);
       thunar_standard_view_append_menu_items (standard_view, GTK_MENU (context_menu), NULL);
-      xfce_gtk_menu_append_seperator (GTK_MENU_SHELL (context_menu));
-      thunar_menu_add_sections (context_menu, THUNAR_MENU_SECTION_ZOOM
-                                            | THUNAR_MENU_SECTION_PROPERTIES);
+      thunar_menu_add_sections (context_menu, THUNAR_MENU_SECTION_PROPERTIES);
     }
   thunar_menu_hide_accel_labels (context_menu);
   gtk_widget_show_all (GTK_WIDGET (context_menu));
