diff -ru ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-location-dialog.c ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-location-dialog.c
--- ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-location-dialog.c	2012-05-01 12:38:53.374916211 -0700
+++ ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-location-dialog.c	2012-05-01 12:52:14.078225105 -0700
@@ -62,7 +62,7 @@
   gtk_box_set_spacing (GTK_BOX (GTK_DIALOG (location_dialog)->vbox), 2);
   gtk_container_set_border_width (GTK_CONTAINER (location_dialog), 5);
   gtk_dialog_set_has_separator (GTK_DIALOG (location_dialog), FALSE);
-  gtk_window_set_default_size (GTK_WINDOW (location_dialog), 350, -1);
+  gtk_window_set_default_size (GTK_WINDOW (location_dialog), 400, -1);
   gtk_window_set_title (GTK_WINDOW (location_dialog), _("Open Location"));
 
   gtk_dialog_add_button (GTK_DIALOG (location_dialog), GTK_STOCK_CANCEL, GTK_RESPONSE_CANCEL);
@@ -79,7 +79,6 @@
 
   label = gtk_label_new_with_mnemonic (_("_Location:"));
   gtk_box_pack_start (GTK_BOX (hbox), label, FALSE, FALSE, 0);
-  gtk_widget_show (label);
 
   location_dialog->entry = thunar_path_entry_new ();
   gtk_entry_set_activates_default (GTK_ENTRY (location_dialog->entry), TRUE);
diff -ru ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-path-entry.c ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-path-entry.c
--- ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-path-entry.c	2012-05-01 12:38:53.374916211 -0700
+++ ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-path-entry.c	2012-05-01 12:53:08.698223439 -0700
@@ -881,10 +881,6 @@
 
   /* let the GtkEntry class handle the insert */
   (*thunar_path_entry_editable_parent_iface->do_insert_text) (editable, new_text, new_text_length, position);
-
-  /* queue a completion check if this insert operation was triggered by the user */
-  if (G_LIKELY (!path_entry->in_change))
-    thunar_path_entry_queue_check_completion (path_entry);
 }
 
 
