diff -ru thunar.ORIG/src/thunar/thunar/thunar-window.c thunar/src/thunar/thunar/thunar-window.c
--- thunar.ORIG/src/thunar/thunar/thunar-window.c	2024-12-30 15:42:25.710092867 -0800
+++ thunar/src/thunar/thunar/thunar-window.c	2024-12-30 15:42:27.073449293 -0800
@@ -5497,8 +5497,6 @@
 thunar_window_set_current_directory (ThunarWindow *window,
                                      ThunarFile   *current_directory)
 {
-  gboolean is_trash;
-  gboolean is_recent;
   GType    type;
   gchar   *type_name;
   gint     num_pages;
@@ -5574,20 +5572,7 @@
   if (window->view != NULL && window->view_type != type)
     thunar_window_replace_view (window, window->view, type);
 
-  is_trash = thunar_file_is_trash (current_directory);
-  is_recent = thunar_file_is_recent (current_directory);
-  if (is_trash)
-    gtk_widget_show (window->trash_infobar);
-  else
-    gtk_widget_hide (window->trash_infobar);
-
-  if (THUNAR_IS_DETAILS_VIEW (window->view) == FALSE)
-    return;
-
-  /* show/hide date_deleted column/sortBy in the trash directory */
-  thunar_details_view_set_date_deleted_column_visible (THUNAR_DETAILS_VIEW (window->view), is_trash);
-  thunar_details_view_set_recency_column_visible (THUNAR_DETAILS_VIEW (window->view), is_recent);
-  thunar_details_view_set_location_column_visible (THUNAR_DETAILS_VIEW (window->view), is_recent);
+  gtk_widget_hide (window->trash_infobar);
 }
 
 
