diff -ru thunar.ORIG/src/thunar-4.18.6/thunar/thunar-window.c thunar/src/thunar-4.18.6/thunar/thunar-window.c
--- thunar.ORIG/src/thunar-4.18.6/thunar/thunar-window.c	2023-03-23 12:05:40.124032949 -0700
+++ thunar/src/thunar-4.18.6/thunar/thunar-window.c	2023-03-23 12:05:40.540710176 -0700
@@ -4772,7 +4772,6 @@
                                          ThunarWindow *window)
 {
   gboolean      show_full_path;
-  gchar        *title;
   gchar        *parse_name = NULL;
   const gchar  *name;
 
@@ -4788,9 +4787,7 @@
     name = thunar_file_get_display_name (current_directory);
 
   /* set window title */
-  title = g_strdup_printf ("%s - %s", name, "Thunar");
-  gtk_window_set_title (GTK_WINDOW (window), title);
-  g_free (title);
+  gtk_window_set_title (GTK_WINDOW (window), name);
   g_free (parse_name);
 
   /* set window icon */
