diff -ru thunar.ORIG/src/thunar/thunar/thunar-window.c thunar/src/thunar/thunar/thunar-window.c
--- thunar.ORIG/src/thunar/thunar/thunar-window.c	2025-10-20 03:25:36.233107088 -0700
+++ thunar/src/thunar/thunar/thunar-window.c	2025-10-20 03:25:36.870147868 -0700
@@ -4260,7 +4260,7 @@
   window->menubar_visible = !window->menubar_visible;
 
   gtk_widget_set_visible (window->menubar, window->menubar_visible);
-  gtk_widget_set_visible (window->location_toolbar_item_menu, !window->menubar_visible);
+  gtk_widget_set_visible (window->location_toolbar_item_menu, FALSE);
 
   g_object_set (G_OBJECT (window->preferences), "last-menubar-visible", window->menubar_visible, NULL);
 
@@ -6787,7 +6787,7 @@
   gtk_widget_show_all (window->location_toolbar);
 
   /* only show the menu button when the menubar is hidden */
-  gtk_widget_set_visible (window->location_toolbar_item_menu, !window->menubar_visible);
+  gtk_widget_set_visible (window->location_toolbar_item_menu, FALSE);
 
   /* add the location bar itself after gtk_widget_show_all to not mess with the visibility of the location buttons */
   gtk_container_add (GTK_CONTAINER (tool_item), window->location_bar);
