diff -ru thunar.ORIG/src/thunar-4.16.6/thunar/thunar-location-button.c thunar/src/thunar-4.16.6/thunar/thunar-location-button.c
--- thunar.ORIG/src/thunar-4.16.6/thunar/thunar-location-button.c	2020-12-26 22:33:39.641604763 -0800
+++ thunar/src/thunar-4.16.6/thunar/thunar-location-button.c	2020-12-26 22:33:40.364926216 -0800
@@ -394,44 +394,16 @@
   icon_theme = gtk_icon_theme_get_for_screen (gtk_widget_get_screen (GTK_WIDGET (location_button)));
   /* TODO: listen for icon theme changes */
 
-  /* update and show the label widget (hide for the local root folder) */
-  if (thunar_file_is_local (file) && thunar_file_is_root (file))
-    {
-      /* hide the label would otherwise show up */
-      gtk_widget_hide (location_button->label);
-    }
-  else
-    {
-      /* set label to the file's display name and show the label */
-      gtk_label_set_text (GTK_LABEL (location_button->label), thunar_file_get_display_name (file));
+  /* set label to the file's display name and show the label */
+  gtk_label_set_text (GTK_LABEL (location_button->label), thunar_file_get_display_name (file));
 
-      /* set the label's size request in such a way that a bold label will not change the button size */
-      if (gtk_widget_get_mapped (GTK_WIDGET (location_button)))
-        thunar_location_button_apply_label_size (location_button);
+  /* set the label's size request in such a way that a bold label will not change the button size */
+  if (gtk_widget_get_mapped (GTK_WIDGET (location_button)))
+    thunar_location_button_apply_label_size (location_button);
 
-      gtk_widget_show (location_button->label);
-    }
+  gtk_widget_show (location_button->label);
 
-  /* the image is only visible for certain special paths */
-  if (thunar_file_is_home (file) || thunar_file_is_desktop (file) || thunar_file_is_root (file))
-    {
-      icon_name = g_strdup_printf ("%s-symbolic", thunar_file_get_icon_name (file,
-                                                                             location_button->file_icon_state,
-                                                                             icon_theme));
-
-      /* update the icon for the image */
-      gtk_image_set_from_icon_name (GTK_IMAGE (location_button->image), icon_name, GTK_ICON_SIZE_BUTTON);
-
-      /* show the image widget */
-      gtk_widget_show (location_button->image);
-
-      g_free (icon_name);
-    }
-  else
-    {
-      /* hide the image widget */
-      gtk_widget_hide (location_button->image);
-    }
+  gtk_widget_hide (location_button->image);
 
   /* setup the DnD icon for the button */
   dnd_icon_name = thunar_file_get_custom_icon (file);
