diff -ru thunar.ORIG/src/thunar-4.16.10/thunar/thunar-util.c thunar/src/thunar-4.16.10/thunar/thunar-util.c
--- thunar.ORIG/src/thunar-4.16.10/thunar/thunar-util.c	2021-05-06 16:23:05.256867618 -0700
+++ thunar/src/thunar-4.16.10/thunar/thunar-util.c	2021-05-06 16:23:05.406865765 -0700
@@ -483,7 +483,10 @@
             return g_strdup ("");
 
           /* use custom date formatting */
-          return exo_strdup_strftime (date_custom_style, &tfile);
+          char buf[80];
+          strftime (buf, sizeof (buf), date_custom_style, &tfile);
+
+          return g_strdup (buf);
         }
     }
 
