diff -ru abs/thunar.ORIG/src/Thunar-1.6.2/thunar/thunar-launcher.c abs/thunar/src/Thunar-1.6.2/thunar/thunar-launcher.c
--- abs/thunar.ORIG/src/Thunar-1.6.2/thunar/thunar-launcher.c	2012-12-29 11:55:27.581478345 -0800
+++ abs/thunar/src/Thunar-1.6.2/thunar/thunar-launcher.c	2012-12-29 11:59:57.473686084 -0800
@@ -934,7 +934,7 @@
       else if (G_LIKELY (applications != NULL))
         {
           /* turn the "Open" action into "Open With DEFAULT" */
-          label = g_strdup_printf (_("_Open With \"%s\""), g_app_info_get_name (applications->data));
+          label = g_strdup_printf (_("_Open With %s"), g_app_info_get_name (applications->data));
           tooltip = g_strdup_printf (ngettext ("Use \"%s\" to open the selected file",
                                                "Use \"%s\" to open the selected files",
                                                n_selected_files), g_app_info_get_name (applications->data));
@@ -1020,7 +1020,7 @@
 
               /* generate a unique label, unique id and tooltip for the application's action */
               name = g_strdup_printf ("thunar-launcher-addon-application%d-%p", n, launcher);
-              label = g_strdup_printf (_("Open With \"%s\""), g_app_info_get_name (lp->data));
+              label = g_strdup_printf (_("Open With %s"), g_app_info_get_name (lp->data));
               tooltip = g_strdup_printf (ngettext ("Use \"%s\" to open the selected file",
                                                    "Use \"%s\" to open the selected files",
                                                    n_selected_files), g_app_info_get_name (lp->data));
Only in abs/thunar/src/Thunar-1.6.2/thunar: thunar-launcher.c.orig
