diff -ru thunar.ORIG/src/thunar-4.16.1/thunar/thunar-dialogs.c thunar/src/thunar-4.16.1/thunar/thunar-dialogs.c
--- thunar.ORIG/src/thunar-4.16.1/thunar/thunar-dialogs.c	2020-12-26 22:33:39.641604763 -0800
+++ thunar/src/thunar-4.16.1/thunar/thunar-dialogs.c	2020-12-26 22:33:39.878267542 -0800
@@ -79,7 +79,6 @@
   GdkScreen         *screen;
   GError            *error = NULL;
   gchar             *name = NULL;
-  GtkWidget         *label;
   GtkWidget         *grid;
   GtkWidget         *image;
   XfceFilenameInput *filename_input;
@@ -117,17 +116,14 @@
   /* setup the image */
   if (G_LIKELY (icon != NULL))
     {
-      image = g_object_new (GTK_TYPE_IMAGE, "xpad", 6, "ypad", 6, NULL);
-      gtk_image_set_from_gicon (GTK_IMAGE (image), icon, GTK_ICON_SIZE_DIALOG);
+      image = g_object_new (GTK_TYPE_IMAGE, NULL);
+      gtk_widget_set_margin_bottom (GTK_WIDGET(image), 12);
+      gtk_image_set_from_gicon (GTK_IMAGE (image), icon, GTK_ICON_SIZE_LARGE_TOOLBAR);
       gtk_grid_attach (GTK_GRID (grid), image, 0, 0, 1, 2);
       g_object_unref (icon);
       gtk_widget_show (image);
     }
 
-  label = g_object_new (GTK_TYPE_LABEL, "label", _("Enter the name:"), "xalign", 0.0f, "hexpand", TRUE, NULL);
-  gtk_grid_attach (GTK_GRID (grid), label, 1, 0, 1, 1);
-  gtk_widget_show (label);
-
   /* set up the widget for entering the filename */
   filename_input = g_object_new (XFCE_TYPE_FILENAME_INPUT, "original-filename", filename, NULL);
   gtk_widget_set_hexpand (GTK_WIDGET (filename_input), TRUE);
@@ -141,8 +137,6 @@
                             gtk_dialog_get_widget_for_response (GTK_DIALOG (dialog), GTK_RESPONSE_OK));
 
   gtk_grid_attach (GTK_GRID (grid), GTK_WIDGET (filename_input), 1, 1, 1, 1);
-  thunar_gtk_label_set_a11y_relation (GTK_LABEL (label),
-                                      GTK_WIDGET (xfce_filename_input_get_entry (filename_input)));
   gtk_widget_show_all ( GTK_WIDGET (filename_input));
 
   /* ensure that the sensitivity of the Create button is set correctly */
@@ -201,7 +195,6 @@
   const gchar       *text;
   ThunarJob         *job = NULL;
   GtkWidget         *dialog;
-  GtkWidget         *label;
   GtkWidget         *image;
   GtkWidget         *grid;
   GtkWindow         *window;
@@ -249,24 +242,15 @@
 
   icon_theme = gtk_icon_theme_get_for_screen (gtk_widget_get_screen (dialog));
   icon_factory = thunar_icon_factory_get_for_icon_theme (icon_theme);
-  icon = thunar_icon_factory_load_file_icon (icon_factory, file, THUNAR_FILE_ICON_STATE_DEFAULT, 48);
+  icon = thunar_icon_factory_load_file_icon (icon_factory, file, THUNAR_FILE_ICON_STATE_DEFAULT, 24);
   g_object_unref (G_OBJECT (icon_factory));
 
   image = gtk_image_new_from_pixbuf (icon);
-  gtk_widget_set_margin_start (GTK_WIDGET(image), 6);
-  gtk_widget_set_margin_end (GTK_WIDGET(image), 6);
-  gtk_widget_set_margin_top (GTK_WIDGET(image), 6);
-  gtk_widget_set_margin_bottom (GTK_WIDGET(image), 6);
+  gtk_widget_set_margin_bottom (GTK_WIDGET(image), 12);
   gtk_grid_attach (GTK_GRID (grid), image, 0, 0, 1, 2);
   g_object_unref (G_OBJECT (icon));
   gtk_widget_show (image);
 
-  label = gtk_label_new (_("Enter the new name:"));
-  gtk_label_set_xalign (GTK_LABEL (label), 0.0f);
-  gtk_widget_set_hexpand (label, TRUE);
-  gtk_grid_attach (GTK_GRID (grid), label, 1, 0, 1, 1);
-  gtk_widget_show (label);
-
   /* set up the widget for entering the filename */
   filename_input = g_object_new (XFCE_TYPE_FILENAME_INPUT, "original-filename", filename, NULL);
   gtk_widget_set_hexpand (GTK_WIDGET (filename_input), TRUE);
@@ -280,8 +264,6 @@
                             gtk_dialog_get_widget_for_response (GTK_DIALOG (dialog), GTK_RESPONSE_OK));
 
   gtk_grid_attach (GTK_GRID (grid), GTK_WIDGET (filename_input), 1, 1, 1, 1);
-  thunar_gtk_label_set_a11y_relation (GTK_LABEL (label),
-                                      GTK_WIDGET (xfce_filename_input_get_entry (filename_input)));
   gtk_widget_show_all ( GTK_WIDGET (filename_input));
 
   /* ensure that the sensitivity of the Create button is set correctly */
@@ -848,7 +830,7 @@
   gtk_widget_show (label);
   g_free (text);
 
-  icon = thunar_icon_factory_load_file_icon (icon_factory, dst_file, THUNAR_FILE_ICON_STATE_DEFAULT, 48);
+  icon = thunar_icon_factory_load_file_icon (icon_factory, dst_file, THUNAR_FILE_ICON_STATE_DEFAULT, 24);
   image = gtk_image_new_from_pixbuf (icon);
   gtk_widget_set_margin_start (GTK_WIDGET(image), 6);
   gtk_widget_set_margin_end (GTK_WIDGET(image), 6);
@@ -887,7 +869,7 @@
   gtk_widget_show (label);
   g_free (text);
 
-  icon = thunar_icon_factory_load_file_icon (icon_factory, src_file, THUNAR_FILE_ICON_STATE_DEFAULT, 48);
+  icon = thunar_icon_factory_load_file_icon (icon_factory, src_file, THUNAR_FILE_ICON_STATE_DEFAULT, 24);
   image = gtk_image_new_from_pixbuf (icon);
   gtk_widget_set_margin_start (GTK_WIDGET(image), 6);
   gtk_widget_set_margin_end (GTK_WIDGET(image), 6);
