diff -ru thunar.ORIG/src/thunar-4.18.2/thunar/thunar-enum-types.c thunar/src/thunar-4.18.2/thunar/thunar-enum-types.c
--- thunar.ORIG/src/thunar-4.18.2/thunar/thunar-enum-types.c	2023-01-10 23:31:26.872889245 -0800
+++ thunar/src/thunar-4.18.2/thunar/thunar-enum-types.c	2023-01-10 23:31:27.486250269 -0800
@@ -224,7 +224,9 @@
         { THUNAR_ICON_SIZE_160,  "THUNAR_ICON_SIZE_160",       "160px",  },
         { THUNAR_ICON_SIZE_192,  "THUNAR_ICON_SIZE_192",       "192px",  },
         { THUNAR_ICON_SIZE_256,  "THUNAR_ICON_SIZE_256",       "256px",  },
+        { THUNAR_ICON_SIZE_384,  "THUNAR_ICON_SIZE_384",       "384px",  },
         { THUNAR_ICON_SIZE_512,  "THUNAR_ICON_SIZE_512",       "512px",  },
+        { THUNAR_ICON_SIZE_768,  "THUNAR_ICON_SIZE_768",       "768px",  },
         { THUNAR_ICON_SIZE_1024, "THUNAR_ICON_SIZE_1024",      "1024px", },
         /* Support of old type-strings for two thunar stable releases. Old strings will be transformed to new ones on write*/
         { THUNAR_ICON_SIZE_16,   "THUNAR_ICON_SIZE_SMALLEST",  "16px",   },
@@ -313,7 +315,9 @@
         { THUNAR_ZOOM_LEVEL_250_PERCENT,  "THUNAR_ZOOM_LEVEL_250_PERCENT",   "250%", },
         { THUNAR_ZOOM_LEVEL_300_PERCENT,  "THUNAR_ZOOM_LEVEL_300_PERCENT",   "300%", },
         { THUNAR_ZOOM_LEVEL_400_PERCENT,  "THUNAR_ZOOM_LEVEL_400_PERCENT",   "400%", },
+        { THUNAR_ZOOM_LEVEL_600_PERCENT,  "THUNAR_ZOOM_LEVEL_600_PERCENT",   "600%", },
         { THUNAR_ZOOM_LEVEL_800_PERCENT,  "THUNAR_ZOOM_LEVEL_800_PERCENT",   "800%", },
+        { THUNAR_ZOOM_LEVEL_1200_PERCENT, "THUNAR_ZOOM_LEVEL_1200_PERCENT",  "1200%",},
         { THUNAR_ZOOM_LEVEL_1600_PERCENT, "THUNAR_ZOOM_LEVEL_1600_PERCENT",  "1600%",},
         /* Support of old type-strings for two thunar stable releases. Old strings will be transformed to new ones on write*/
         { THUNAR_ZOOM_LEVEL_25_PERCENT,   "THUNAR_ZOOM_LEVEL_SMALLEST",      "25%",  },
@@ -423,7 +427,9 @@
       {
         { THUNAR_THUMBNAIL_SIZE_NORMAL,     "THUNAR_THUMBNAIL_SIZE_NORMAL",   "normal",   },
         { THUNAR_THUMBNAIL_SIZE_LARGE,      "THUNAR_THUMBNAIL_SIZE_LARGE",    "large",    },
+        { THUNAR_THUMBNAIL_SIZE_LARGE_PLUS, "THUNAR_THUMBNAIL_SIZE_LARGE_PLUS", "large-plus", },
         { THUNAR_THUMBNAIL_SIZE_X_LARGE,    "THUNAR_THUMBNAIL_SIZE_X_LARGE",  "x-large",  },
+        { THUNAR_THUMBNAIL_SIZE_X_LARGE_PLUS, "THUNAR_THUMBNAIL_SIZE_X_LARGE_PLUS", "x-large-plus", },
         { THUNAR_THUMBNAIL_SIZE_XX_LARGE,   "THUNAR_THUMBNAIL_SIZE_XX_LARGE", "xx-large", },
         { 0,                             NULL,                           NULL,     },
       };
@@ -491,7 +497,9 @@
     case THUNAR_ZOOM_LEVEL_250_PERCENT:  return THUNAR_ICON_SIZE_160;
     case THUNAR_ZOOM_LEVEL_300_PERCENT:  return THUNAR_ICON_SIZE_192;
     case THUNAR_ZOOM_LEVEL_400_PERCENT:  return THUNAR_ICON_SIZE_256;
+    case THUNAR_ZOOM_LEVEL_600_PERCENT:  return THUNAR_ICON_SIZE_384;
     case THUNAR_ZOOM_LEVEL_800_PERCENT:  return THUNAR_ICON_SIZE_512;
+    case THUNAR_ZOOM_LEVEL_1200_PERCENT: return THUNAR_ICON_SIZE_768;
     case THUNAR_ZOOM_LEVEL_1600_PERCENT: return THUNAR_ICON_SIZE_1024;
     default:                             return THUNAR_ICON_SIZE_64; // default = 100 %zoom
     }
@@ -505,9 +513,15 @@
   if (icon_size >= THUNAR_ICON_SIZE_1024)
     return THUNAR_THUMBNAIL_SIZE_XX_LARGE;
 
+  if (icon_size >= THUNAR_ICON_SIZE_768)
+    return THUNAR_THUMBNAIL_SIZE_X_LARGE_PLUS;
+
   if (icon_size >= THUNAR_ICON_SIZE_512)
     return THUNAR_THUMBNAIL_SIZE_X_LARGE;
 
+  if (icon_size >= THUNAR_ICON_SIZE_384)
+    return THUNAR_THUMBNAIL_SIZE_LARGE_PLUS;
+
   if (icon_size >= THUNAR_ICON_SIZE_128)
     return THUNAR_THUMBNAIL_SIZE_LARGE;
 
diff -ru thunar.ORIG/src/thunar-4.18.2/thunar/thunar-enum-types.h thunar/src/thunar-4.18.2/thunar/thunar-enum-types.h
--- thunar.ORIG/src/thunar-4.18.2/thunar/thunar-enum-types.h	2023-01-10 23:31:26.872889245 -0800
+++ thunar/src/thunar-4.18.2/thunar/thunar-enum-types.h	2023-01-10 23:31:27.486250269 -0800
@@ -160,7 +160,9 @@
   THUNAR_ICON_SIZE_160   =  160,
   THUNAR_ICON_SIZE_192   =  192,
   THUNAR_ICON_SIZE_256   =  256,
+  THUNAR_ICON_SIZE_384   =  384,
   THUNAR_ICON_SIZE_512   =  512,
+  THUNAR_ICON_SIZE_768   =  768,
   THUNAR_ICON_SIZE_1024  = 1024,
 } ThunarIconSize;
 
@@ -191,14 +193,18 @@
  * ThunarThumbnailSize:
  * @THUNAR_THUMBNAIL_NORMAL      : max 128px x 128px
  * @THUNAR_THUMBNAIL_LARGE       : max 256px x 256px
+ * @THUNAR_THUMBNAIL_LARGE_PLUS  : max 384px x 384px
  * @THUNAR_THUMBNAIL_X_LARGE     : max 512px x 512px
+ * @THUNAR_THUMBNAIL_X_LARGE_PLUS: max 768px x 768px
  * @THUNAR_THUMBNAIL_XX_LARGE    : max 1024px x 1024px
  **/
 typedef enum
 {
   THUNAR_THUMBNAIL_SIZE_NORMAL,
   THUNAR_THUMBNAIL_SIZE_LARGE,
+  THUNAR_THUMBNAIL_SIZE_LARGE_PLUS,
   THUNAR_THUMBNAIL_SIZE_X_LARGE,
+  THUNAR_THUMBNAIL_SIZE_X_LARGE_PLUS,
   THUNAR_THUMBNAIL_SIZE_XX_LARGE,
 
   THUNAR_THUMBNAIL_SIZE_DEFAULT = -1
@@ -286,7 +292,9 @@
   THUNAR_ZOOM_LEVEL_250_PERCENT,
   THUNAR_ZOOM_LEVEL_300_PERCENT,
   THUNAR_ZOOM_LEVEL_400_PERCENT,
+  THUNAR_ZOOM_LEVEL_600_PERCENT,
   THUNAR_ZOOM_LEVEL_800_PERCENT,
+  THUNAR_ZOOM_LEVEL_1200_PERCENT,
   THUNAR_ZOOM_LEVEL_1600_PERCENT,
 
   /*< private >*/
diff -ru thunar.ORIG/src/thunar-4.18.2/thunar/thunar-icon-view.c thunar/src/thunar-4.18.2/thunar/thunar-icon-view.c
--- thunar.ORIG/src/thunar-4.18.2/thunar/thunar-icon-view.c	2023-01-10 23:31:26.872889245 -0800
+++ thunar/src/thunar-4.18.2/thunar/thunar-icon-view.c	2023-01-10 23:31:27.486250269 -0800
@@ -201,8 +201,12 @@
       wrap_width = 112;
       break;
 
+    case THUNAR_ZOOM_LEVEL_200_PERCENT:
+      wrap_width = 192;
+      break;
+
     default:
-      wrap_width = 128;
+      wrap_width = 384;
       break;
     }
 
