diff -ru ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-shortcuts-model.c ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-shortcuts-model.c
--- ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-shortcuts-model.c	2012-05-01 12:38:53.378249539 -0700
+++ ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-shortcuts-model.c	2012-05-01 12:43:23.334907966 -0700
@@ -259,10 +259,6 @@
   /* append the root file system */
   system_paths = g_list_append (system_paths, thunar_g_file_new_for_root ());
 
-  /* append the network icon if browsing the network is supported */
-  if (thunar_g_vfs_is_uri_scheme_supported ("network"))
-    system_paths = g_list_append (system_paths, g_file_new_for_uri ("network://"));
-
   /* will be used to append the shortcuts to the list */
   path = gtk_tree_path_new_from_indices (0, -1);
 
diff -ru ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-tree-model.c ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-tree-model.c
--- ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-tree-model.c	2012-05-01 12:38:53.378249539 -0700
+++ ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-tree-model.c	2012-05-01 12:46:15.061569394 -0700
@@ -326,10 +326,6 @@
   /* append the root file system */
   system_paths = g_list_append (system_paths, thunar_g_file_new_for_root ());
 
-  /* append the network icon if browsing the network is supported */
-  if (thunar_g_vfs_is_uri_scheme_supported ("network"))
-    system_paths = g_list_append (system_paths, g_file_new_for_uri ("network://"));
-
   /* append the system defined nodes ('Home', 'Trash', 'File System') */
   for (lp = system_paths; lp != NULL; lp = lp->next)
     {
diff -ru ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-window.c ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-window.c
--- ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-window.c	2012-05-01 12:38:53.378249539 -0700
+++ ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-window.c	2012-05-01 12:44:35.411572435 -0700
@@ -1435,28 +1435,6 @@
                            gtk_action_get_name (GTK_ACTION (action)),
                            GTK_UI_MANAGER_MENUITEM, FALSE);
   }
-
-  /* setup the "open-network" action */
-  if (thunar_g_vfs_is_uri_scheme_supported ("network"))
-    {
-      /* create the network action */
-      action = gtk_action_new ("open-network", _("Network"), _("Browse the network"), 
-                               GTK_STOCK_NETWORK);
-      g_signal_connect (action, "activate", G_CALLBACK (thunar_window_action_open_network), window);
-
-      /* add the network action to the action group */
-      gtk_action_group_add_action (window->action_group, action);
-
-      /* add the action to the UI manager */
-      gtk_ui_manager_add_ui (window->ui_manager,
-                             window->go_items_actions_merge_id,
-                             "/main-menu/go-menu/placeholder-go-items-actions",
-                             gtk_action_get_name (GTK_ACTION (action)),
-                             gtk_action_get_name (GTK_ACTION (action)),
-                             GTK_UI_MANAGER_MENUITEM, FALSE);
-
-      g_object_unref (action);
-    }
 }
 
 
