diff -ru ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-window.c ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-window.c
--- ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-window.c	2012-05-01 12:38:53.378249539 -0700
+++ ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-window.c	2012-05-01 13:10:02.911525822 -0700
@@ -823,7 +823,7 @@
     }
 
   window->paned = gtk_hpaned_new ();
-  gtk_container_set_border_width (GTK_CONTAINER (window->paned), 0);
+  gtk_container_set_border_width (GTK_CONTAINER (window->paned), 5);
   gtk_table_attach (GTK_TABLE (window->table), window->paned, 0, 1, 4, 5, GTK_EXPAND | GTK_FILL, GTK_EXPAND | GTK_FILL, 0, 0);
   gtk_widget_show (window->paned);
 
