diff -ru abs/thunar.ORIG/src/Thunar-1.6.7/thunar/thunar-application.c abs/thunar/src/Thunar-1.6.7/thunar/thunar-application.c
--- abs/thunar.ORIG/src/Thunar-1.6.7/thunar/thunar-application.c	2013-05-07 00:23:01.658432393 -0700
+++ abs/thunar/src/Thunar-1.6.7/thunar/thunar-application.c	2013-05-07 00:29:49.590215696 -0700
@@ -814,11 +814,8 @@
 {
   _thunar_return_if_fail (THUNAR_IS_APPLICATION (application));
 
-  if (application->daemon != daemonize)
-    {
-      application->daemon = daemonize;
-      g_object_notify (G_OBJECT (application), "daemon");
-    }
+  application->daemon = FALSE;
+  g_object_notify (G_OBJECT (application), "daemon");
 }
 
 
