diff -ru thunar.ORIG/src/Thunar-1.8.7/thunar/thunar-enum-types.c thunar/src/Thunar-1.8.7/thunar/thunar-enum-types.c
--- thunar.ORIG/src/Thunar-1.8.7/thunar/thunar-enum-types.c	2019-02-12 11:51:06.180076421 -0800
+++ thunar/src/Thunar-1.8.7/thunar/thunar-enum-types.c	2019-02-12 11:51:09.106645381 -0800
@@ -104,7 +104,7 @@
         { THUNAR_COLUMN_MIME_TYPE,     "THUNAR_COLUMN_MIME_TYPE",     N_ ("MIME Type"),     },
         { THUNAR_COLUMN_NAME,          "THUNAR_COLUMN_NAME",          N_ ("Name"),          },
         { THUNAR_COLUMN_OWNER,         "THUNAR_COLUMN_OWNER",         N_ ("Owner"),         },
-        { THUNAR_COLUMN_PERMISSIONS,   "THUNAR_COLUMN_PERMISSIONS",   N_ ("Permissions"),   },
+        { THUNAR_COLUMN_PERMISSIONS,   "THUNAR_COLUMN_PERMISSIONS",   N_ ("Mode"),   },
         { THUNAR_COLUMN_SIZE,          "THUNAR_COLUMN_SIZE",          N_ ("Size"),          },
         { THUNAR_COLUMN_SIZE_IN_BYTES, "THUNAR_COLUMN_SIZE_IN_BYTES", N_ ("Size in Bytes"), },
         { THUNAR_COLUMN_TYPE,          "THUNAR_COLUMN_TYPE",          N_ ("Type"),          },
