diff -ru thunar.ORIG/src/thunar-4.18.0/thunar/thunar-application.c thunar/src/thunar-4.18.0/thunar/thunar-application.c
--- thunar.ORIG/src/thunar-4.18.0/thunar/thunar-application.c	2022-12-15 19:42:45.971362899 -0800
+++ thunar/src/thunar-4.18.0/thunar/thunar-application.c	2022-12-15 19:42:46.148037599 -0800
@@ -1265,16 +1265,8 @@
 {
   _thunar_return_if_fail (THUNAR_IS_APPLICATION (application));
 
-  if (application->daemon != daemonize)
-    {
-      application->daemon = daemonize;
-      g_object_notify (G_OBJECT (application), "daemon");
-
-      if (daemonize)
-        g_application_hold (G_APPLICATION (application));
-      else
-        g_application_release (G_APPLICATION (application));
-    }
+  application->daemon = FALSE;
+  g_object_notify (G_OBJECT (application), "daemon");
 }
 
 
