diff -ru abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-notify.c abs/thunar/src/Thunar-1.4.0/thunar/thunar-notify.c
--- abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-notify.c	2012-07-18 19:26:16.348035022 -0700
+++ abs/thunar/src/Thunar-1.4.0/thunar/thunar-notify.c	2012-07-18 19:29:06.189261318 -0700
@@ -129,8 +129,7 @@
     {
       summary = _("Writing data to device");
       message = g_strdup_printf (_("There is data that needs to be written to the "
-                                   "device \"%s\" before it can be removed. Please "
-                                   "do not remove the media or disconnect the drive"),
+                                   "device \"%s\" before it can be removed."),
                                    name);
     }
 
@@ -247,8 +246,7 @@
     {
       summary = _("Writing data to device");
       message = g_strdup_printf (_("There is data that needs to be written to the "
-                                   "device \"%s\" before it can be removed. Please "
-                                   "do not remove the media or disconnect the drive"),
+                                   "device \"%s\" before it can be removed."),
                                    name);
     }
 
