--- ../abs/thunar/src/Thunar-1.2.3/thunar/thunar-progress-dialog.c.ORIG	2011-09-24 23:05:00.575948554 -0700
+++ ../abs/thunar/src/Thunar-1.2.3/thunar/thunar-progress-dialog.c	2011-09-24 23:06:06.735946535 -0700
@@ -150,7 +150,7 @@
   _thunar_return_if_fail (THUNAR_IS_PROGRESS_DIALOG (dialog));
 
   /* show the status icon */
-  if (dialog->status_icon == NULL)
+  if (0) /* gavin: no status icon for file operations */
     {
       dialog->status_icon = gtk_status_icon_new_from_icon_name ("stock_folder-copy");
       thunar_progress_dialog_update_status_icon (dialog);
