diff -ru thunar.ORIG/src/thunar/thunar/thunar-location-button.c thunar/src/thunar/thunar/thunar-location-button.c
--- thunar.ORIG/src/thunar/thunar/thunar-location-button.c	2024-12-30 15:42:25.710092867 -0800
+++ thunar/src/thunar/thunar/thunar-location-button.c	2024-12-30 15:42:27.190117936 -0800
@@ -426,11 +426,9 @@
 thunar_location_button_file_changed (ThunarLocationButton *location_button,
                                      ThunarFile           *file)
 {
-  ThunarPreferences *preferences;
   GtkIconTheme      *icon_theme;
-  gchar             *icon_name;
   const gchar       *dnd_icon_name;
-  gboolean           use_symbolic_icons;
+  const gchar       *diplay_name;
 
   _thunar_return_if_fail (THUNAR_IS_LOCATION_BUTTON (location_button));
   _thunar_return_if_fail (location_button->file == file);
@@ -440,54 +438,21 @@
   icon_theme = gtk_icon_theme_get_for_screen (gtk_widget_get_screen (GTK_WIDGET (location_button)));
   /* TODO: listen for icon theme changes */
 
-  /* update and show the label widget (hide for the local root folder) */
-  if (thunar_file_is_local (file) && thunar_file_is_root (file))
-    {
-      /* hide the label would otherwise show up */
-      gtk_widget_hide (location_button->label);
-    }
-  else
-    {
-      const gchar *diplay_name;
-
-      /* set label to the file's display name and show the label */
-      diplay_name = thunar_file_get_display_name (file);
-      if (diplay_name != NULL)
-        gtk_label_set_text (GTK_LABEL (location_button->label), diplay_name);
-
-      /* reset the size request before applying the new label size so that the button can shrink */
-      gtk_widget_set_size_request (location_button->label, -1, -1);
-
-      /* set the label's size request in such a way that a bold label will not change the button size */
-      if (gtk_widget_get_mapped (GTK_WIDGET (location_button)))
-        thunar_location_button_apply_label_size (location_button);
-
-      gtk_widget_show (location_button->label);
-    }
-
-  /* the image is only visible for certain special paths */
-  if (thunar_file_is_home (file) || thunar_file_is_desktop (file) || thunar_file_is_root (file))
-    {
-      preferences = thunar_preferences_get ();
-      g_object_get (G_OBJECT (preferences), "misc-symbolic-icons-in-toolbar", &use_symbolic_icons, NULL);
-      g_object_unref (G_OBJECT (preferences));
-
-      icon_name = g_strdup_printf ("%s%s", thunar_file_get_icon_name (file, location_button->file_icon_state, icon_theme),
-                                   use_symbolic_icons ? "-symbolic" : "");
-
-      /* update the icon for the image */
-      gtk_image_set_from_icon_name (GTK_IMAGE (location_button->image), icon_name, GTK_ICON_SIZE_BUTTON);
-
-      /* show the image widget */
-      gtk_widget_show (location_button->image);
-
-      g_free (icon_name);
-    }
-  else
-    {
-      /* hide the image widget */
-      gtk_widget_hide (location_button->image);
-    }
+  /* set label to the file's display name and show the label */
+  diplay_name = thunar_file_get_display_name (file);
+  if (diplay_name != NULL)
+    gtk_label_set_text (GTK_LABEL (location_button->label), diplay_name);
+
+  /* reset the size request before applying the new label size so that the button can shrink */
+  gtk_widget_set_size_request (location_button->label, -1, -1);
+
+  /* set the label's size request in such a way that a bold label will not change the button size */
+  if (gtk_widget_get_mapped (GTK_WIDGET (location_button)))
+    thunar_location_button_apply_label_size (location_button);
+
+  gtk_widget_show (location_button->label);
+
+  gtk_widget_hide (location_button->image);
 
   /* setup the DnD icon for the button */
   dnd_icon_name = thunar_file_get_custom_icon (file);
