diff -ru thunar.ORIG/src/thunar-4.16.5/thunar/thunar-launcher.c thunar/src/thunar-4.16.5/thunar/thunar-launcher.c
--- thunar.ORIG/src/thunar-4.16.5/thunar/thunar-launcher.c	2021-01-13 09:20:01.795139780 -0800
+++ thunar/src/thunar-4.16.5/thunar/thunar-launcher.c	2021-01-13 09:25:46.663049264 -0800
@@ -253,7 +253,7 @@
     { THUNAR_LAUNCHER_ACTION_RENAME,           "<Actions>/ThunarStandardView/rename",              "F2",                XFCE_GTK_MENU_ITEM,       N_ ("_Rename..."),                      NULL,                                                                                            NULL,                   G_CALLBACK (thunar_launcher_action_rename),              },
     { THUNAR_LAUNCHER_ACTION_EMPTY_TRASH,      "<Actions>/ThunarWindow/empty-trash",               "",                  XFCE_GTK_IMAGE_MENU_ITEM, N_ ("_Empty Trash"),                    N_ ("Delete all files and folders in the Trash"),                                                NULL,                   G_CALLBACK (thunar_launcher_action_empty_trash),         },
     { THUNAR_LAUNCHER_ACTION_CREATE_FOLDER,    "<Actions>/ThunarStandardView/create-folder",       "<Primary><shift>N", XFCE_GTK_IMAGE_MENU_ITEM, N_ ("Create _Folder..."),               N_ ("Create an empty folder within the current folder"),                                         "folder-new",           G_CALLBACK (thunar_launcher_action_create_folder),       },
-    { THUNAR_LAUNCHER_ACTION_CREATE_DOCUMENT,  "<Actions>/ThunarStandardView/create-document",     "",                  XFCE_GTK_IMAGE_MENU_ITEM, N_ ("Create _Document"),                N_ ("Create a new document from a template"),                                                    "document-new",         G_CALLBACK (NULL),                                       },
+    { THUNAR_LAUNCHER_ACTION_CREATE_EMPTY_FILE,"<Actions>/ThunarStandardView/create-empty-file",   "<Primary><shift>F", XFCE_GTK_IMAGE_MENU_ITEM, N_ ("Create _Empty File..."),           N_ ("Create an empty file within the current folder"),                                           "gtk-new",              G_CALLBACK (thunar_launcher_action_create_document),     },
 
     { THUNAR_LAUNCHER_ACTION_RESTORE,          "<Actions>/ThunarLauncher/restore",                 "",                  XFCE_GTK_MENU_ITEM,       N_ ("_Restore"),                        NULL,                                                                                            NULL,                   G_CALLBACK (thunar_launcher_action_restore),             },
     { THUNAR_LAUNCHER_ACTION_MOVE_TO_TRASH,    "<Actions>/ThunarLauncher/move-to-trash",           "",                  XFCE_GTK_IMAGE_MENU_ITEM, N_ ("Mo_ve to Trash"),                  NULL,                                                                                            "user-trash",           G_CALLBACK (thunar_launcher_action_move_to_trash),       },
@@ -1451,6 +1451,7 @@
         return NULL;
 
       case THUNAR_LAUNCHER_ACTION_CREATE_FOLDER:
+      case THUNAR_LAUNCHER_ACTION_CREATE_EMPTY_FILE:
         if (THUNAR_IS_TREE_VIEW (launcher->widget) && launcher->files_are_selected && launcher->single_directory_to_process)
           parent = launcher->single_folder;
         else
diff -ru thunar.ORIG/src/thunar-4.16.5/thunar/thunar-launcher.h thunar/src/thunar-4.16.5/thunar/thunar-launcher.h
--- thunar.ORIG/src/thunar-4.16.5/thunar/thunar-launcher.h	2021-01-13 09:20:01.795139780 -0800
+++ thunar/src/thunar-4.16.5/thunar/thunar-launcher.h	2021-01-13 09:23:46.694941848 -0800
@@ -52,6 +52,7 @@
   THUNAR_LAUNCHER_ACTION_RENAME,
   THUNAR_LAUNCHER_ACTION_EMPTY_TRASH,
   THUNAR_LAUNCHER_ACTION_CREATE_FOLDER,
+  THUNAR_LAUNCHER_ACTION_CREATE_EMPTY_FILE,
   THUNAR_LAUNCHER_ACTION_CREATE_DOCUMENT,
   THUNAR_LAUNCHER_ACTION_RESTORE,
   THUNAR_LAUNCHER_ACTION_MOVE_TO_TRASH,
diff -ru thunar.ORIG/src/thunar-4.16.5/thunar/thunar-menu.c thunar/src/thunar-4.16.5/thunar/thunar-menu.c
--- thunar.ORIG/src/thunar-4.16.5/thunar/thunar-menu.c	2021-01-13 09:20:01.795139780 -0800
+++ thunar/src/thunar-4.16.5/thunar/thunar-menu.c	2021-01-13 09:25:11.510271000 -0800
@@ -257,6 +257,7 @@
     {
       item_added = FALSE;
       item_added |= (thunar_launcher_append_menu_item (menu->launcher, GTK_MENU_SHELL (menu), THUNAR_LAUNCHER_ACTION_CREATE_FOLDER, force) != NULL);
+      item_added |= (thunar_launcher_append_menu_item (menu->launcher, GTK_MENU_SHELL (menu), THUNAR_LAUNCHER_ACTION_CREATE_EMPTY_FILE, force) != NULL);
 
       /* No document creation for tree-view */
       if (menu->type != THUNAR_MENU_TYPE_CONTEXT_TREE_VIEW)
