diff -ru thunar.ORIG/src/thunar/thunar/thunar-file.c thunar/src/thunar/thunar/thunar-file.c
--- thunar.ORIG/src/thunar/thunar/thunar-file.c	2025-10-20 03:25:36.232836195 -0700
+++ thunar/src/thunar/thunar/thunar-file.c	2025-10-20 03:25:36.965678679 -0700
@@ -1090,9 +1090,6 @@
           display_name = g_file_info_get_display_name (file->info);
           if (G_LIKELY (display_name != NULL))
             {
-              if (strcmp (display_name, "/") == 0)
-                file->display_name = g_strdup (_("File System"));
-              else
                 file->display_name = g_strdup (display_name);
             }
         }
