diff -ru thunar.ORIG/src/thunar-4.18.4/thunar/thunar-dialogs.c thunar/src/thunar-4.18.4/thunar/thunar-dialogs.c
--- thunar.ORIG/src/thunar-4.18.4/thunar/thunar-dialogs.c	2023-01-10 23:31:26.872889245 -0800
+++ thunar/src/thunar-4.18.4/thunar/thunar-dialogs.c	2023-01-10 23:31:26.886223180 -0800
@@ -92,7 +92,6 @@
   GdkScreen         *screen;
   GError            *error = NULL;
   gchar             *name = NULL;
-  GtkWidget         *label;
   GtkWidget         *grid;
   GtkWidget         *image;
   XfceFilenameInput *filename_input;
@@ -131,18 +130,15 @@
   /* setup the image */
   if (G_LIKELY (icon != NULL))
     {
-      image = g_object_new (GTK_TYPE_IMAGE, "xpad", 6, "ypad", 6, NULL);
-      gtk_image_set_from_gicon (GTK_IMAGE (image), icon, GTK_ICON_SIZE_DIALOG);
+      image = g_object_new (GTK_TYPE_IMAGE, NULL);
+      gtk_widget_set_margin_bottom (GTK_WIDGET(image), 12);
+      gtk_image_set_from_gicon (GTK_IMAGE (image), icon, GTK_ICON_SIZE_LARGE_TOOLBAR);
       gtk_grid_attach (GTK_GRID (grid), image, 0, row, 1, 2);
       gtk_widget_set_valign (GTK_WIDGET (image), GTK_ALIGN_START);
       g_object_unref (icon);
       gtk_widget_show (image);
     }
 
-  label = g_object_new (GTK_TYPE_LABEL, "label", _("Enter the name:"), "xalign", 0.0f, "hexpand", TRUE, NULL);
-  gtk_grid_attach (GTK_GRID (grid), label, 1, row, 1, 1);
-  gtk_widget_show (label);
-
   /* set up the widget for entering the filename */
   filename_input = g_object_new (XFCE_TYPE_FILENAME_INPUT, "original-filename", filename, NULL);
   gtk_widget_set_hexpand (GTK_WIDGET (filename_input), TRUE);
@@ -159,8 +155,6 @@
   row++;
 
   gtk_grid_attach (GTK_GRID (grid), GTK_WIDGET (filename_input), 1, row, 1, 1);
-  thunar_gtk_label_set_a11y_relation (GTK_LABEL (label),
-                                      GTK_WIDGET (xfce_filename_input_get_entry (filename_input)));
   gtk_widget_show_all ( GTK_WIDGET (filename_input));
 
   /* ensure that the sensitivity of the Create button is set correctly */
@@ -226,7 +220,6 @@
   const gchar       *text;
   ThunarJob         *job = NULL;
   GtkWidget         *dialog;
-  GtkWidget         *label;
   GtkWidget         *image;
   GtkWidget         *grid;
   GtkWindow         *window;
@@ -275,25 +268,16 @@
 
   icon_theme = gtk_icon_theme_get_for_screen (gtk_widget_get_screen (dialog));
   icon_factory = thunar_icon_factory_get_for_icon_theme (icon_theme);
-  icon = thunar_icon_factory_load_file_icon (icon_factory, file, THUNAR_FILE_ICON_STATE_DEFAULT, 48);
+  icon = thunar_icon_factory_load_file_icon (icon_factory, file, THUNAR_FILE_ICON_STATE_DEFAULT, 24);
   g_object_unref (G_OBJECT (icon_factory));
 
   image = gtk_image_new_from_pixbuf (icon);
-  gtk_widget_set_margin_start (GTK_WIDGET (image), 6);
-  gtk_widget_set_margin_end (GTK_WIDGET (image), 6);
-  gtk_widget_set_margin_top (GTK_WIDGET (image), 6);
-  gtk_widget_set_margin_bottom (GTK_WIDGET (image), 6);
+  gtk_widget_set_margin_bottom (GTK_WIDGET (image), 12);
   gtk_widget_set_valign (GTK_WIDGET (image), GTK_ALIGN_START);
   gtk_grid_attach (GTK_GRID (grid), image, 0, row, 1, 2);
   g_object_unref (G_OBJECT (icon));
   gtk_widget_show (image);
 
-  label = gtk_label_new (_("Enter the new name:"));
-  gtk_label_set_xalign (GTK_LABEL (label), 0.0f);
-  gtk_widget_set_hexpand (label, TRUE);
-  gtk_grid_attach (GTK_GRID (grid), label, 1, row, 1, 1);
-  gtk_widget_show (label);
-
   /* set up the widget for entering the filename */
   filename_input = g_object_new (XFCE_TYPE_FILENAME_INPUT, "original-filename", filename, NULL);
   gtk_widget_set_hexpand (GTK_WIDGET (filename_input), TRUE);
@@ -310,8 +294,6 @@
   row++;
 
   gtk_grid_attach (GTK_GRID (grid), GTK_WIDGET (filename_input), 1, row, 1, 1);
-  thunar_gtk_label_set_a11y_relation (GTK_LABEL (label),
-                                      GTK_WIDGET (xfce_filename_input_get_entry (filename_input)));
   gtk_widget_show_all ( GTK_WIDGET (filename_input));
 
   /* ensure that the sensitivity of the Create button is set correctly */
@@ -895,7 +877,7 @@
   /* next row */
   row++;
 
-  icon = thunar_icon_factory_load_file_icon (icon_factory, dst_file, THUNAR_FILE_ICON_STATE_DEFAULT, 48);
+  icon = thunar_icon_factory_load_file_icon (icon_factory, dst_file, THUNAR_FILE_ICON_STATE_DEFAULT, 24);
   image = gtk_image_new_from_pixbuf (icon);
   gtk_widget_set_margin_start (GTK_WIDGET(image), 6);
   gtk_widget_set_margin_end (GTK_WIDGET(image), 6);
@@ -940,7 +922,7 @@
   /* next row */
   row++;
 
-  icon = thunar_icon_factory_load_file_icon (icon_factory, src_file, THUNAR_FILE_ICON_STATE_DEFAULT, 48);
+  icon = thunar_icon_factory_load_file_icon (icon_factory, src_file, THUNAR_FILE_ICON_STATE_DEFAULT, 24);
   image = gtk_image_new_from_pixbuf (icon);
   gtk_widget_set_margin_start (GTK_WIDGET(image), 6);
   gtk_widget_set_margin_end (GTK_WIDGET(image), 6);
