diff -ru thunar.ORIG/src/thunar-4.18.0/thunar/thunar-window.c thunar/src/thunar-4.18.0/thunar/thunar-window.c
--- thunar.ORIG/src/thunar-4.18.0/thunar/thunar-window.c	2022-12-15 19:42:45.971362899 -0800
+++ thunar/src/thunar-4.18.0/thunar/thunar-window.c	2022-12-15 19:42:46.271376540 -0800
@@ -3879,7 +3879,7 @@
   window->menubar_visible = !window->menubar_visible;
 
   gtk_widget_set_visible (window->menubar, window->menubar_visible);
-  gtk_widget_set_visible (window->location_toolbar_item_view_menubar, !window->menubar_visible);
+  gtk_widget_set_visible (window->location_toolbar_item_view_menubar, FALSE);
 
   g_object_set (G_OBJECT (window->preferences), "last-menubar-visible", window->menubar_visible, NULL);
 
@@ -6093,7 +6093,7 @@
   gtk_widget_show_all (window->location_toolbar);
 
   /* view_menubar is only visible when the menu is hidden */
-  gtk_widget_set_visible (window->location_toolbar_item_view_menubar, !window->menubar_visible);
+  gtk_widget_set_visible (window->location_toolbar_item_view_menubar, FALSE);
 
   /* add the location bar itself after gtk_widget_show_all to not mess with the visibility of the location buttons */
   gtk_container_add (GTK_CONTAINER (tool_item), window->location_bar);
