diff -ru thunar.ORIG/src/thunar-4.16.7/thunar/thunar-emblem-chooser.c thunar/src/thunar-4.16.7/thunar/thunar-emblem-chooser.c
--- thunar.ORIG/src/thunar-4.16.7/thunar/thunar-emblem-chooser.c	2021-05-06 16:23:05.256867618 -0700
+++ thunar/src/thunar-4.16.7/thunar/thunar-emblem-chooser.c	2021-05-06 16:23:05.270200786 -0700
@@ -454,7 +454,7 @@
   GdkPixbuf   *icon;
 
   /* lookup the icon info for the emblem */
-  info = gtk_icon_theme_lookup_icon (chooser->icon_theme, emblem, 48, 0);
+  info = gtk_icon_theme_lookup_icon (chooser->icon_theme, emblem, 16, 0);
   if (G_UNLIKELY (info == NULL))
     return NULL;
 
diff -ru thunar.ORIG/src/thunar-4.16.7/thunar/thunar-icon-renderer.c thunar/src/thunar-4.16.7/thunar/thunar-icon-renderer.c
--- thunar.ORIG/src/thunar-4.16.7/thunar/thunar-icon-renderer.c	2021-05-06 16:23:05.256867618 -0700
+++ thunar/src/thunar-4.16.7/thunar/thunar-icon-renderer.c	2021-05-06 16:23:05.270200786 -0700
@@ -505,7 +505,7 @@
           for (lp = emblems, position = 0; lp != NULL && position < max_emblems; lp = lp->next)
             {
               /* calculate the emblem size */
-              emblem_size = MIN ((2 * icon_renderer->size) / 3, 32);
+              emblem_size = 16;
 
               /* check if we have the emblem in the icon theme */
               emblem = thunar_icon_factory_load_icon (icon_factory, lp->data, emblem_size, FALSE);
@@ -516,19 +516,6 @@
               emblem_area.width = gdk_pixbuf_get_width (emblem);
               emblem_area.height = gdk_pixbuf_get_height (emblem);
 
-              /* shrink insane emblems */
-              if (G_UNLIKELY (MAX (emblem_area.width, emblem_area.height) > emblem_size))
-                {
-                  /* scale down the emblem */
-                  temp = exo_gdk_pixbuf_scale_ratio (emblem, emblem_size);
-                  g_object_unref (G_OBJECT (emblem));
-                  emblem = temp;
-
-                  /* determine the size again */
-                  emblem_area.width = gdk_pixbuf_get_width (emblem);
-                  emblem_area.height = gdk_pixbuf_get_height (emblem);
-                }
-
               /* determine a good position for the emblem, depending on the position index */
               switch (position)
                 {
