diff -ru thunar.ORIG/src/thunar/thunar/thunar-file.c thunar/src/thunar/thunar/thunar-file.c
--- thunar.ORIG/src/thunar/thunar/thunar-file.c	2024-12-30 15:42:25.706759478 -0800
+++ thunar/src/thunar/thunar/thunar-file.c	2024-12-30 15:42:26.703443026 -0800
@@ -1089,9 +1089,6 @@
           display_name = g_file_info_get_display_name (file->info);
           if (G_LIKELY (display_name != NULL))
             {
-              if (strcmp (display_name, "/") == 0)
-                file->display_name = g_strdup (_("File System"));
-              else
                 file->display_name = g_strdup (display_name);
             }
         }
