diff -ru abs/thunar.ORIG/src/Thunar-1.6.2/thunar/thunar-application.c abs/thunar/src/Thunar-1.6.2/thunar/thunar-application.c
--- abs/thunar.ORIG/src/Thunar-1.6.2/thunar/thunar-application.c	2012-12-29 11:55:27.581478345 -0800
+++ abs/thunar/src/Thunar-1.6.2/thunar/thunar-application.c	2012-12-29 11:57:20.890471323 -0800
@@ -814,11 +814,8 @@
 {
   _thunar_return_if_fail (THUNAR_IS_APPLICATION (application));
 
-  if (application->daemon != daemonize)
-    {
-      application->daemon = daemonize;
-      g_object_notify (G_OBJECT (application), "daemon");
-    }
+  application->daemon = FALSE;
+  g_object_notify (G_OBJECT (application), "daemon");
 }
 
 
