diff -ru thunar.ORIG/src/thunar-4.16.5/thunar/thunar-util.c thunar/src/thunar-4.16.5/thunar/thunar-util.c
--- thunar.ORIG/src/thunar-4.16.5/thunar/thunar-util.c	2020-12-26 22:33:39.641604763 -0800
+++ thunar/src/thunar-4.16.5/thunar/thunar-util.c	2020-12-26 22:33:39.791602298 -0800
@@ -483,7 +483,10 @@
             return g_strdup ("");
 
           /* use custom date formatting */
-          return exo_strdup_strftime (date_custom_style, &tfile);
+          char buf[80];
+          strftime (buf, sizeof (buf), date_custom_style, &tfile);
+
+          return g_strdup (buf);
         }
     }
 
