diff -ru thunar.ORIG/src/thunar-1.8.11/thunar/thunar-history.c thunar/src/thunar-1.8.11/thunar/thunar-history.c
--- thunar.ORIG/src/thunar-1.8.11/thunar/thunar-history.c	2019-11-11 08:24:30.423459947 -0800
+++ thunar/src/thunar-1.8.11/thunar/thunar-history.c	2019-11-11 08:24:30.900142272 -0800
@@ -167,6 +167,7 @@
   g_signal_connect (G_OBJECT (history->action_back), "show-menu", G_CALLBACK (thunar_history_show_menu), history);
 G_GNUC_BEGIN_IGNORE_DEPRECATIONS
   gtk_action_set_sensitive (history->action_back, FALSE);
+  gtk_action_set_visible (history->action_back, FALSE);
 G_GNUC_END_IGNORE_DEPRECATIONS
 
   /* create the "forward" action */
@@ -175,6 +176,7 @@
   g_signal_connect (G_OBJECT (history->action_forward), "show-menu", G_CALLBACK (thunar_history_show_menu), history);
 G_GNUC_BEGIN_IGNORE_DEPRECATIONS
   gtk_action_set_sensitive (history->action_forward, FALSE);
+  gtk_action_set_visible (history->action_forward, FALSE);
 G_GNUC_END_IGNORE_DEPRECATIONS
 }
 
diff -ru thunar.ORIG/src/thunar-1.8.11/thunar/thunar-window-ui.xml thunar/src/thunar-1.8.11/thunar/thunar-window-ui.xml
--- thunar.ORIG/src/thunar-1.8.11/thunar/thunar-window-ui.xml	2019-11-11 08:24:30.426793390 -0800
+++ thunar/src/thunar-1.8.11/thunar/thunar-window-ui.xml	2019-11-11 08:24:30.900142272 -0800
@@ -126,9 +126,6 @@
   </popup>
 
   <toolbar name="location-toolbar">
-    <placeholder name="placeholder-history-actions" />
-    <toolitem action="open-parent" />
-    <toolitem action="open-home" />
   </toolbar>
 </ui>
 
