diff -ru ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-launcher.c ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-launcher.c
--- ../abs/thunar.ORIG/src/Thunar-1.4.0/thunar/thunar-launcher.c	2012-05-01 12:38:53.378249539 -0700
+++ ../abs/thunar/src/Thunar-1.4.0/thunar/thunar-launcher.c	2012-05-01 12:57:39.684881837 -0700
@@ -867,7 +867,7 @@
       else if (G_LIKELY (applications != NULL))
         {
           /* turn the "Open" action into "Open With DEFAULT" */
-          label = g_strdup_printf (_("_Open With \"%s\""), g_app_info_get_name (applications->data));
+          label = g_strdup_printf (_("_Open With %s"), g_app_info_get_name (applications->data));
           tooltip = g_strdup_printf (ngettext ("Use \"%s\" to open the selected file",
                                                "Use \"%s\" to open the selected files",
                                                n_selected_files), g_app_info_get_name (applications->data));
@@ -953,7 +953,7 @@
 
               /* generate a unique label, unique id and tooltip for the application's action */
               name = g_strdup_printf ("thunar-launcher-addon-application%d-%p", n, launcher);
-              label = g_strdup_printf (_("Open With \"%s\""), g_app_info_get_name (lp->data));
+              label = g_strdup_printf (_("Open With %s"), g_app_info_get_name (lp->data));
               tooltip = g_strdup_printf (ngettext ("Use \"%s\" to open the selected file",
                                                    "Use \"%s\" to open the selected files",
                                                    n_selected_files), g_app_info_get_name (lp->data));
