diff -ru thunar.ORIG/src/thunar-1.8.14/thunar/thunar-standard-view.c thunar/src/thunar-1.8.14/thunar/thunar-standard-view.c
--- thunar.ORIG/src/thunar-1.8.14/thunar/thunar-standard-view.c	2019-11-11 08:24:30.426793390 -0800
+++ thunar/src/thunar-1.8.14/thunar/thunar-standard-view.c	2019-11-11 08:24:31.126816382 -0800
@@ -649,7 +649,7 @@
 
   /* initialize the scrolled window */
   gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (standard_view),
-                                  GTK_POLICY_AUTOMATIC,
+                                  GTK_POLICY_NEVER,
                                   GTK_POLICY_AUTOMATIC);
   gtk_scrolled_window_set_hadjustment (GTK_SCROLLED_WINDOW (standard_view), NULL);
   gtk_scrolled_window_set_vadjustment (GTK_SCROLLED_WINDOW (standard_view), NULL);
