diff -ru thunar.ORIG/src/thunar/thunar/thunar-window.c thunar/src/thunar/thunar/thunar-window.c
--- thunar.ORIG/src/thunar/thunar/thunar-window.c	2025-09-15 13:01:18.519037204 -0700
+++ thunar/src/thunar/thunar/thunar-window.c	2025-09-15 13:01:18.937032936 -0700
@@ -3340,7 +3340,7 @@
         window->sidepane = g_object_new (THUNAR_TYPE_SHORTCUTS_PANE, NULL);
       else
         window->sidepane = g_object_new (THUNAR_TYPE_TREE_PANE, NULL);
-      gtk_widget_set_size_request (window->sidepane, 0, -1);
+      gtk_widget_set_size_request (window->sidepane, 165, -1);
       g_object_bind_property (G_OBJECT (window), "current-directory", G_OBJECT (window->sidepane), "current-directory", G_BINDING_SYNC_CREATE);
       g_signal_connect_swapped (G_OBJECT (window->sidepane), "change-directory", G_CALLBACK (thunar_window_set_current_directory), window);
       g_signal_connect_swapped (G_OBJECT (window->sidepane), "open-new-tab", G_CALLBACK (thunar_window_notebook_open_new_tab), window);
