diff -ru thunar.ORIG/src/Thunar-1.8.9/thunar/thunar-application.c thunar/src/Thunar-1.8.9/thunar/thunar-application.c
--- thunar.ORIG/src/Thunar-1.8.9/thunar/thunar-application.c	2019-02-12 11:51:06.180076421 -0800
+++ thunar/src/Thunar-1.8.9/thunar/thunar-application.c	2019-02-12 11:51:06.523398289 -0800
@@ -607,13 +607,11 @@
   gtk_css_provider_load_from_data (css_provider,
     /* for the pathbar-buttons any margin looks ugly*/
     ".path-bar-button { margin-right: 0; }"
-    /* remove extra border between side pane and view */
-    ".shortcuts-pane { border-right-width: 0px; }"
+    ".sidebar { border-width: 1px; border-style: solid; }"
+    ".shortcuts-pane, .tree-pane { border-right-width: 0px; }"
+    ".sidebar ~ grid .standard-view { margin-left: 5px; }"
     /* add missing top border to side pane */
-    ".shortcuts-pane { border-top-style: solid; }"
-    /* make border thicker during DnD */
-    ".standard-view { border-left-width: 0px; }"
-    ".standard-view:drop(active) { border-width: 2px; }", -1, NULL);
+    ".shortcuts-pane { border-top-style: solid; }", -1, NULL);
   screen = gdk_screen_get_default ();
   gtk_style_context_add_provider_for_screen (screen, GTK_STYLE_PROVIDER (css_provider), GTK_STYLE_PROVIDER_PRIORITY_APPLICATION);
   g_object_unref (css_provider);
diff -ru thunar.ORIG/src/Thunar-1.8.9/thunar/thunar-tree-pane.c thunar/src/Thunar-1.8.9/thunar/thunar-tree-pane.c
--- thunar.ORIG/src/Thunar-1.8.9/thunar/thunar-tree-pane.c	2019-02-12 11:51:06.190076087 -0800
+++ thunar/src/Thunar-1.8.9/thunar/thunar-tree-pane.c	2019-02-12 11:51:06.523398289 -0800
@@ -151,6 +151,9 @@
   g_signal_connect_swapped (G_OBJECT (tree_pane->view), "open-new-tab", G_CALLBACK (thunar_navigator_open_new_tab), tree_pane);
   gtk_container_add (GTK_CONTAINER (tree_pane), tree_pane->view);
   gtk_widget_show (tree_pane->view);
+
+  /* add widget to css class */
+  gtk_style_context_add_class (gtk_widget_get_style_context (GTK_WIDGET (tree_pane)), "tree-pane");
 }
 
 
diff -ru thunar.ORIG/src/Thunar-1.8.9/thunar/thunar-window.c thunar/src/Thunar-1.8.9/thunar/thunar-window.c
--- thunar.ORIG/src/Thunar-1.8.9/thunar/thunar-window.c	2019-02-12 11:51:06.186742865 -0800
+++ thunar/src/Thunar-1.8.9/thunar/thunar-window.c	2019-02-12 11:51:06.523398289 -0800
@@ -874,7 +874,7 @@
     }
 
   window->paned = gtk_paned_new (GTK_ORIENTATION_HORIZONTAL);
-  gtk_container_set_border_width (GTK_CONTAINER (window->paned), 0);
+  gtk_container_set_border_width (GTK_CONTAINER (window->paned), 5);
   gtk_widget_set_hexpand (window->paned, TRUE);
   gtk_widget_set_vexpand (window->paned, TRUE);
   gtk_grid_attach (GTK_GRID (window->grid), window->paned, 0, 4, 1, 1);
@@ -908,6 +908,9 @@
 
   /* allocate the new location bar widget */
   window->location_bar = thunar_location_bar_new ();
+  gtk_widget_set_margin_top (GTK_WIDGET (window->location_bar), 2);
+  gtk_widget_set_margin_start (GTK_WIDGET (window->location_bar), 1);
+  gtk_widget_set_margin_end (GTK_WIDGET (window->location_bar), 2);
   g_object_bind_property (G_OBJECT (window), "current-directory", G_OBJECT (window->location_bar), "current-directory", G_BINDING_SYNC_CREATE);
   g_signal_connect_swapped (G_OBJECT (window->location_bar), "change-directory", G_CALLBACK (thunar_window_set_current_directory), window);
   g_signal_connect_swapped (G_OBJECT (window->location_bar), "open-new-tab", G_CALLBACK (thunar_window_notebook_insert), window);
