--- ../abs/thunar/src/Thunar-1.2.3/thunar/thunar-path-entry.c.ORIG	2010-11-13 07:37:13.000000000 -0800
+++ ../abs/thunar/src/Thunar-1.2.3/thunar/thunar-path-entry.c	2011-02-01 22:43:31.626055760 -0800
@@ -883,8 +883,8 @@
   (*thunar_path_entry_editable_parent_iface->do_insert_text) (editable, new_text, new_text_length, position);
 
   /* queue a completion check if this insert operation was triggered by the user */
-  if (G_LIKELY (!path_entry->in_change))
-    thunar_path_entry_queue_check_completion (path_entry);
+  /* if (G_LIKELY (!path_entry->in_change))
+    thunar_path_entry_queue_check_completion (path_entry); */ /* gavin: only complete with tab press */
 }
 
 
