diff -ru thunar.ORIG/src/thunar-4.16.6/thunar/thunar-progress-dialog.c thunar/src/thunar-4.16.6/thunar/thunar-progress-dialog.c
--- thunar.ORIG/src/thunar-4.16.6/thunar/thunar-progress-dialog.c	2020-12-26 22:33:39.641604763 -0800
+++ thunar/src/thunar-4.16.6/thunar/thunar-progress-dialog.c	2020-12-26 22:33:40.188262451 -0800
@@ -152,19 +152,6 @@
 thunar_progress_dialog_shown (ThunarProgressDialog *dialog)
 {
   _thunar_return_if_fail (THUNAR_IS_PROGRESS_DIALOG (dialog));
-
-  /* show the status icon */
-  if (dialog->status_icon == NULL)
-    {
-G_GNUC_BEGIN_IGNORE_DEPRECATIONS
-      dialog->status_icon = gtk_status_icon_new_from_icon_name ("edit-copy");
-G_GNUC_END_IGNORE_DEPRECATIONS
-
-      thunar_progress_dialog_update_status_icon (dialog);
-      g_signal_connect_swapped (dialog->status_icon, "button-press-event",
-                                G_CALLBACK (thunar_progress_dialog_toggled),
-                                GTK_WIDGET (dialog));
-    }
 }
 
 
