diff -ru thunar.ORIG/src/thunar-1.8.11/thunar/thunar-shortcuts-model.c thunar/src/thunar-1.8.11/thunar/thunar-shortcuts-model.c
--- thunar.ORIG/src/thunar-1.8.11/thunar/thunar-shortcuts-model.c	2019-11-11 08:24:30.430126833 -0800
+++ thunar/src/thunar-1.8.11/thunar/thunar-shortcuts-model.c	2019-11-11 08:24:31.640166566 -0800
@@ -969,7 +969,7 @@
   shortcut->group = THUNAR_SHORTCUT_GROUP_NETWORK_DEFAULT;
   shortcut->name = g_strdup (_("Browse Network"));
   shortcut->location = g_file_new_for_uri ("network://");
-  shortcut->gicon = g_themed_icon_new ("network-workgroup");
+  shortcut->gicon = g_themed_icon_new ("network-server");
   shortcut->hidden = thunar_shortcuts_model_get_hidden (model, shortcut);
   thunar_shortcuts_model_add_shortcut (model, shortcut);
 }
@@ -1001,7 +1001,7 @@
       shortcut = g_slice_new0 (ThunarShortcut);
       shortcut->group = THUNAR_SHORTCUT_GROUP_PLACES_DEFAULT;
       shortcut->file = file;
-      shortcut->gicon = g_themed_icon_new ("go-home");
+      shortcut->gicon = g_themed_icon_new ("user-home");
       shortcut->sort_id = 0;
       shortcut->hidden = thunar_shortcuts_model_get_hidden (model, shortcut);
       thunar_shortcuts_model_add_shortcut (model, shortcut);
