--- ../abs/thunar/src/Thunar-1.2.3/thunar/thunar-window.c.ORIG	2011-09-22 20:56:30.838958398 -0700
+++ ../abs/thunar/src/Thunar-1.2.3/thunar/thunar-window.c	2011-09-22 20:57:57.588955750 -0700
@@ -1436,7 +1436,7 @@
   }
 
   /* setup the "open-network" action */
-  if (thunar_g_vfs_is_uri_scheme_supported ("network"))
+  if (0) /* gavin: no network icon */
     {
       /* create the network action */
       action = gtk_action_new ("open-network", _("Network"), _("Browse the network"), 
@@ -2417,7 +2417,7 @@
   g_free (title);
 
   /* set window icon */
-  icon = thunar_icon_factory_load_file_icon (window->icon_factory, current_directory, THUNAR_FILE_ICON_STATE_DEFAULT, 48);
+  icon = thunar_icon_factory_load_file_icon (window->icon_factory, current_directory, THUNAR_FILE_ICON_STATE_DEFAULT, 16); /* gavin: 16px icons in taskbar */
   if (G_LIKELY (icon != NULL))
     {
       gtk_window_set_icon (GTK_WINDOW (window), icon);
