diff -ru thunar.ORIG/src/thunar/thunar/thunar-standard-view.c thunar/src/thunar/thunar/thunar-standard-view.c
--- thunar.ORIG/src/thunar/thunar/thunar-standard-view.c	2024-12-30 15:42:25.710092867 -0800
+++ thunar/src/thunar/thunar/thunar-standard-view.c	2024-12-30 15:42:26.453438791 -0800
@@ -938,7 +938,7 @@
 
   /* initialize the scrolled window */
   gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (standard_view),
-                                  GTK_POLICY_AUTOMATIC,
+                                  GTK_POLICY_NEVER,
                                   GTK_POLICY_AUTOMATIC);
   gtk_scrolled_window_set_hadjustment (GTK_SCROLLED_WINDOW (standard_view), NULL);
   gtk_scrolled_window_set_vadjustment (GTK_SCROLLED_WINDOW (standard_view), NULL);
