diff -ru thunar.ORIG/src/Thunar-1.8.9/thunar/thunar-util.c thunar/src/Thunar-1.8.9/thunar/thunar-util.c
--- thunar.ORIG/src/Thunar-1.8.9/thunar/thunar-util.c	2019-03-20 15:05:20.552210261 -0700
+++ thunar/src/Thunar-1.8.9/thunar/thunar-util.c	2019-03-20 15:23:40.057300290 -0700
@@ -463,7 +463,10 @@
             return g_strdup ("");
 
           /* use custom date formatting */
-          return exo_strdup_strftime (date_custom_style, tfile);
+          char buf[80];
+          strftime (buf, sizeof (buf), date_custom_style, tfile);
+
+          return g_strdup (buf);
         }
     }
 
