--- ../abs/thunar/src/Thunar-1.2.3/thunar/thunar-launcher.c.ORIG	2010-10-24 14:04:46.000000000 -0700
+++ ../abs/thunar/src/Thunar-1.2.3/thunar/thunar-launcher.c	2011-02-01 22:42:05.806930171 -0800
@@ -859,7 +859,7 @@
       else if (G_LIKELY (applications != NULL))
         {
           /* turn the "Open" action into "Open With DEFAULT" */
-          label = g_strdup_printf (_("_Open With \"%s\""), g_app_info_get_name (applications->data));
+          label = g_strdup_printf (_("_Open With %s"), g_app_info_get_name (applications->data)); /* gavin: no quotes around open with app */
           tooltip = g_strdup_printf (ngettext ("Use \"%s\" to open the selected file",
                                                "Use \"%s\" to open the selected files",
                                                n_selected_files), g_app_info_get_name (applications->data));
@@ -945,7 +945,7 @@
 
               /* generate a unique label, unique id and tooltip for the application's action */
               name = g_strdup_printf ("thunar-launcher-addon-application%d-%p", n, launcher);
-              label = g_strdup_printf (_("Open With \"%s\""), g_app_info_get_name (lp->data));
+              label = g_strdup_printf (_("Open With %s"), g_app_info_get_name (lp->data)); /* gavin: no quotes around open with app */
               tooltip = g_strdup_printf (ngettext ("Use \"%s\" to open the selected file",
                                                    "Use \"%s\" to open the selected files",
                                                    n_selected_files), g_app_info_get_name (lp->data));
