diff -ru thunar.ORIG/src/thunar-4.18.7/thunar/thunar-window.c thunar/src/thunar-4.18.7/thunar/thunar-window.c
--- thunar.ORIG/src/thunar-4.18.7/thunar/thunar-window.c	2023-09-01 16:30:12.429245583 -0700
+++ thunar/src/thunar-4.18.7/thunar/thunar-window.c	2023-09-01 16:30:12.919252885 -0700
@@ -5163,9 +5163,6 @@
 thunar_window_set_current_directory (ThunarWindow *window,
                                      ThunarFile   *current_directory)
 {
-  gboolean is_trash;
-  gboolean is_recent;
-
   _thunar_return_if_fail (THUNAR_IS_WINDOW (window));
   _thunar_return_if_fail (current_directory == NULL || THUNAR_IS_FILE (current_directory));
 
@@ -5248,18 +5245,7 @@
   if (current_directory == NULL)
     return;
 
-  is_trash = thunar_file_is_trash (current_directory);
-  is_recent = thunar_file_is_recent (current_directory);
-  if (is_trash)
-    gtk_widget_show (window->trash_infobar);
-  else
-    gtk_widget_hide (window->trash_infobar);
-
-  if (THUNAR_IS_DETAILS_VIEW (window->view) == FALSE)
-    return;
-  thunar_details_view_set_date_deleted_column_visible (THUNAR_DETAILS_VIEW (window->view), is_trash);
-  thunar_details_view_set_recency_column_visible (THUNAR_DETAILS_VIEW (window->view), is_recent);
-  thunar_details_view_set_location_column_visible (THUNAR_DETAILS_VIEW (window->view), is_recent);
+  gtk_widget_hide (window->trash_infobar);
 }
 
 
