diff -ru tumbler.ORIG/src/tumbler-4.18.1/plugins/xdg-cache/xdg-cache-cache.c tumbler/src/tumbler-4.18.1/plugins/xdg-cache/xdg-cache-cache.c
--- tumbler.ORIG/src/tumbler-4.18.1/plugins/xdg-cache/xdg-cache-cache.c	2022-12-15 19:42:43.371244674 -0800
+++ tumbler/src/tumbler-4.18.1/plugins/xdg-cache/xdg-cache-cache.c	2022-12-15 19:42:43.401246038 -0800
@@ -140,9 +140,15 @@
   flavor = tumbler_thumbnail_flavor_new_large ();
   cache->flavors = g_list_prepend (cache->flavors, flavor);
 
+  flavor = tumbler_thumbnail_flavor_new("large-plus", 384, 384);
+  cache->flavors = g_list_prepend (cache->flavors, flavor);
+
   flavor = tumbler_thumbnail_flavor_new_x_large ();
   cache->flavors = g_list_prepend (cache->flavors, flavor);
 
+  flavor = tumbler_thumbnail_flavor_new("x-large-plus", 768, 768);
+  cache->flavors = g_list_prepend (cache->flavors, flavor);
+
   flavor = tumbler_thumbnail_flavor_new_xx_large ();
   cache->flavors = g_list_prepend (cache->flavors, flavor);
 }
