diff -ru tumbler.ORIG/src/tumbler-4.18.0/plugins/jpeg-thumbnailer/jpeg-thumbnailer.c tumbler/src/tumbler-4.18.0/plugins/jpeg-thumbnailer/jpeg-thumbnailer.c
--- tumbler.ORIG/src/tumbler-4.18.0/plugins/jpeg-thumbnailer/jpeg-thumbnailer.c	2022-12-15 19:42:43.371244674 -0800
+++ tumbler/src/tumbler-4.18.0/plugins/jpeg-thumbnailer/jpeg-thumbnailer.c	2022-12-15 19:42:43.387912098 -0800
@@ -711,6 +711,8 @@
                           gint          height,
                           guint        *exif_orientation)
 {
+  return NULL;
+
   GdkPixbuf *pixbuf = NULL;
   guint marker_len;
   guint marker;
