diff --color -ru tumbler.ORIG/src/tumbler-4.16.0/plugins/xdg-cache/xdg-cache-cache.c tumbler/src/tumbler-4.16.0/plugins/xdg-cache/xdg-cache-cache.c
--- tumbler.ORIG/src/tumbler-4.16.0/plugins/xdg-cache/xdg-cache-cache.c	2022-01-13 13:39:40.166764861 -0800
+++ tumbler/src/tumbler-4.16.0/plugins/xdg-cache/xdg-cache-cache.c	2022-01-13 13:50:28.954553392 -0800
@@ -531,6 +531,7 @@
 {
   XDGCacheCache *xdg_cache = XDG_CACHE_CACHE (cache);
   const gchar   *cachedir;
+  const gchar   *datadir;
   const gchar   *dirname;
   gboolean       is_thumbnail = FALSE;
   GList         *iter;
@@ -544,6 +545,8 @@
   for (iter = xdg_cache->flavors; !is_thumbnail && iter != NULL; iter = iter->next)
     {
       cachedir = g_get_user_cache_dir ();
+      datadir = g_get_user_data_dir ();
+
       dirname = tumbler_thumbnail_flavor_get_name (iter->data);
       path = g_build_filename (cachedir, "thumbnails", dirname, NULL);
 
@@ -552,6 +555,12 @@
 
       if (g_file_has_prefix (file, flavor_dir))
         is_thumbnail = TRUE;
+
+      path = g_build_filename (datadir, "thumbnails", dirname, NULL);
+      flavor_dir = g_file_new_for_path (path);
+
+      if (g_file_has_prefix (file, flavor_dir))
+        is_thumbnail = TRUE;
 
       g_object_unref (file);
       g_object_unref (flavor_dir);
