diff -ru gtk3.ORIG/src/gtk/gtk/gtkplacessidebar.c gtk3/src/gtk/gtk/gtkplacessidebar.c
--- gtk3.ORIG/src/gtk/gtk/gtkplacessidebar.c	2019-03-20 16:42:12.051047637 -0700
+++ gtk3/src/gtk/gtk/gtkplacessidebar.c	2019-03-20 16:44:00.258716717 -0700
@@ -265,25 +265,25 @@
 };
 
 /* Names for themed icons */
-#define ICON_NAME_HOME     "user-home-symbolic"
-#define ICON_NAME_DESKTOP  "user-desktop-symbolic"
-#define ICON_NAME_FILESYSTEM     "drive-harddisk-symbolic"
-#define ICON_NAME_EJECT    "media-eject-symbolic"
-#define ICON_NAME_NETWORK  "network-workgroup-symbolic"
-#define ICON_NAME_NETWORK_SERVER "network-server-symbolic"
-#define ICON_NAME_FOLDER_NETWORK "folder-remote-symbolic"
-#define ICON_NAME_OTHER_LOCATIONS "list-add-symbolic"
+#define ICON_NAME_HOME     "user-home"
+#define ICON_NAME_DESKTOP  "user-desktop"
+#define ICON_NAME_FILESYSTEM     "drive-harddisk"
+#define ICON_NAME_EJECT    "media-eject"
+#define ICON_NAME_NETWORK  "network-workgroup"
+#define ICON_NAME_NETWORK_SERVER "network-server"
+#define ICON_NAME_FOLDER_NETWORK "folder-remote"
+#define ICON_NAME_OTHER_LOCATIONS "list-add"
 
-#define ICON_NAME_FOLDER                "folder-symbolic"
-#define ICON_NAME_FOLDER_DESKTOP  "user-desktop-symbolic"
-#define ICON_NAME_FOLDER_DOCUMENTS      "folder-documents-symbolic"
-#define ICON_NAME_FOLDER_DOWNLOAD       "folder-download-symbolic"
-#define ICON_NAME_FOLDER_MUSIC    "folder-music-symbolic"
-#define ICON_NAME_FOLDER_PICTURES       "folder-pictures-symbolic"
-#define ICON_NAME_FOLDER_PUBLIC_SHARE   "folder-publicshare-symbolic"
-#define ICON_NAME_FOLDER_TEMPLATES      "folder-templates-symbolic"
-#define ICON_NAME_FOLDER_VIDEOS   "folder-videos-symbolic"
-#define ICON_NAME_FOLDER_SAVED_SEARCH   "folder-saved-search-symbolic"
+#define ICON_NAME_FOLDER                "folder"
+#define ICON_NAME_FOLDER_DESKTOP  "user-desktop"
+#define ICON_NAME_FOLDER_DOCUMENTS      "folder-documents"
+#define ICON_NAME_FOLDER_DOWNLOAD       "folder-download"
+#define ICON_NAME_FOLDER_MUSIC    "folder-music"
+#define ICON_NAME_FOLDER_PICTURES       "folder-pictures"
+#define ICON_NAME_FOLDER_PUBLIC_SHARE   "folder-publicshare"
+#define ICON_NAME_FOLDER_TEMPLATES      "folder-templates"
+#define ICON_NAME_FOLDER_VIDEOS   "folder-videos"
+#define ICON_NAME_FOLDER_SAVED_SEARCH   "folder-saved-search"
 
 static guint places_sidebar_signals [LAST_SIGNAL] = { 0 };
 static GParamSpec *properties[NUM_PROPERTIES] = { NULL, };
