diff -ru gtk3.ORIG/src/gtk/gtk/gtkwindow.c gtk3/src/gtk/gtk/gtkwindow.c
--- gtk3.ORIG/src/gtk/gtk/gtkwindow.c	2024-07-11 18:45:20.372376493 -0700
+++ gtk3/src/gtk/gtk/gtkwindow.c	2024-07-11 18:45:22.859082813 -0700
@@ -8053,9 +8053,6 @@
   GtkWindow *window = GTK_WINDOW (widget);
   GtkWindowPrivate *priv = window->priv;
 
-  if (event->changed_mask & GDK_WINDOW_STATE_FOCUSED)
-    ensure_state_flag_backdrop (widget);
-
   if (event->changed_mask & GDK_WINDOW_STATE_FULLSCREEN)
     {
       priv->fullscreen =
