diff -ru gtk3.ORIG/src/gtk/gtk/gtkwindow.c gtk3/src/gtk/gtk/gtkwindow.c
--- gtk3.ORIG/src/gtk/gtk/gtkwindow.c	2018-06-13 01:21:58.205498574 -0700
+++ gtk3/src/gtk/gtk/gtkwindow.c	2018-06-13 01:21:58.475501335 -0700
@@ -7955,9 +7955,6 @@
   GtkWindow *window = GTK_WINDOW (widget);
   GtkWindowPrivate *priv = window->priv;
 
-  if (event->changed_mask & GDK_WINDOW_STATE_FOCUSED)
-    ensure_state_flag_backdrop (widget);
-
   if (event->changed_mask & GDK_WINDOW_STATE_FULLSCREEN)
     {
       priv->fullscreen =
