diff -ru gtk3.ORIG/src/gtk/gtk/gtkpathbar.c gtk3/src/gtk/gtk/gtkpathbar.c
--- gtk3.ORIG/src/gtk/gtk/gtkpathbar.c	2024-07-11 18:45:20.365709720 -0700
+++ gtk3/src/gtk/gtk/gtkpathbar.c	2024-07-11 18:45:24.255771752 -0700
@@ -1566,20 +1566,7 @@
   switch (button_data->type)
     {
     case ROOT_BUTTON:
-      button_data->image = gtk_image_new ();
-      child = button_data->image;
-      button_data->label = NULL;
       atk_object_set_name (atk_obj, _("File System Root"));
-      break;
-    case HOME_BUTTON:
-    case DESKTOP_BUTTON:
-      button_data->image = gtk_image_new ();
-      button_data->label = gtk_label_new (NULL);
-      child = gtk_box_new (GTK_ORIENTATION_HORIZONTAL, 0);
-      gtk_box_pack_start (GTK_BOX (child), button_data->image, FALSE, FALSE, 0);
-      gtk_box_pack_start (GTK_BOX (child), button_data->label, FALSE, FALSE, 0);
-      break;
-    case NORMAL_BUTTON:
     default:
       button_data->label = gtk_label_new (NULL);
       child = button_data->label;
