--- ../abs/xfce4-session/src/xfce4-session-4.8.2/xfce4-session/shutdown.c.ORIG	2011-02-14 20:55:12.269904354 -0800
+++ ../abs/xfce4-session/src/xfce4-session-4.8.2/xfce4-session/shutdown.c	2011-02-14 20:58:45.458117142 -0800
@@ -402,7 +402,8 @@
   gtk_widget_show (vbox2);
   gtk_container_add (GTK_CONTAINER (logout_button), vbox2);
 
-  icon = gtk_icon_theme_load_icon (icon_theme,
+  /* gavin: no icon on logout button */
+  /* icon = gtk_icon_theme_load_icon (icon_theme,
                                    "system-log-out",
                                    32,
                                    0,
@@ -417,7 +418,7 @@
   image = gtk_image_new_from_pixbuf (icon);
   gtk_widget_show (image);
   gtk_box_pack_start (GTK_BOX (vbox2), image, FALSE, FALSE, 0);
-  g_object_unref (icon);
+  g_object_unref (icon); */
 
   label = gtk_label_new (_("Log Out"));
   gtk_widget_show (label);
@@ -436,7 +437,8 @@
   gtk_widget_show (vbox2);
   gtk_container_add (GTK_CONTAINER (reboot_button), vbox2);
 
-  icon = gtk_icon_theme_load_icon (icon_theme,
+  /* gavin: no icon on reboot button */
+  /* icon = gtk_icon_theme_load_icon (icon_theme,
                                    "system-reboot",
                                    32,
                                    0,
@@ -452,7 +454,7 @@
   image = gtk_image_new_from_pixbuf (icon);
   gtk_widget_show (image);
   gtk_box_pack_start (GTK_BOX (vbox2), image, FALSE, FALSE, 0);
-  g_object_unref (icon);
+  g_object_unref (icon); */
 
   label = gtk_label_new (_("Restart"));
   gtk_widget_show (label);
@@ -478,7 +480,8 @@
   gtk_widget_show (vbox2);
   gtk_container_add (GTK_CONTAINER (halt_button), vbox2);
 
-  icon = gtk_icon_theme_load_icon (icon_theme,
+  /* gavin: no icon on shutdown button */
+  /* icon = gtk_icon_theme_load_icon (icon_theme,
                                    "system-shutdown",
                                    32,
                                    0,
@@ -494,7 +497,7 @@
   image = gtk_image_new_from_pixbuf (icon);
   gtk_widget_show (image);
   gtk_box_pack_start (GTK_BOX (vbox2), image, FALSE, FALSE, 0);
-  g_object_unref (icon);
+  g_object_unref (icon); */
 
   label = gtk_label_new (_("Shut Down"));
   gtk_widget_show (label);
@@ -540,7 +543,8 @@
       gtk_widget_show (vbox2);
       gtk_container_add (GTK_CONTAINER (suspend_button), vbox2);
 
-      icon = gtk_icon_theme_load_icon (icon_theme,
+      /* gavin: no icon on suspend button */
+      /* icon = gtk_icon_theme_load_icon (icon_theme,
                                        "system-suspend",
                                        32,
                                        0,
@@ -556,7 +560,7 @@
       image = gtk_image_new_from_pixbuf (icon);
       gtk_widget_show (image);
       gtk_box_pack_start (GTK_BOX (vbox2), image, FALSE, FALSE, 0);
-      g_object_unref (icon);
+      g_object_unref (icon); */
 
       label = gtk_label_new (_("Suspend"));
       gtk_widget_show (label);
@@ -578,7 +582,8 @@
       gtk_widget_show (vbox2);
       gtk_container_add (GTK_CONTAINER (hibernate_button), vbox2);
 
-      icon = gtk_icon_theme_load_icon (icon_theme,
+      /* gavin: no icon on hibernate button */
+      /* icon = gtk_icon_theme_load_icon (icon_theme,
                                        "system-hibernate",
                                        32,
                                        0,
@@ -594,7 +599,7 @@
       image = gtk_image_new_from_pixbuf (icon);
       gtk_widget_show (image);
       gtk_box_pack_start (GTK_BOX (vbox2), image, FALSE, FALSE, 0);
-      g_object_unref (icon);
+      g_object_unref (icon); */
 
       label = gtk_label_new (_("Hibernate"));
       gtk_widget_show (label);
@@ -608,7 +613,7 @@
           _("_Save session for future logins"));
       gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(checkbox), saveonexit);
       gtk_box_pack_start(GTK_BOX(vbox), checkbox, FALSE, TRUE, BORDER);
-      gtk_widget_show(checkbox);
+      /* gtk_widget_show(checkbox); */ /* gavin: don't show save-session checkbox */
     }
   else
     {
