--- ../abs/xfce4-session/src/xfce4-session-4.8.2/xfce4-session/shutdown.c.ORIG	2011-12-10 15:30:12.142326877 -0800
+++ ../abs/xfce4-session/src/xfce4-session-4.8.2/xfce4-session/shutdown.c	2011-12-10 15:30:24.855659821 -0800
@@ -61,7 +61,6 @@
 #include <xfce4-session/shutdown.h>
 #include <xfce4-session/xfsm-compat-gnome.h>
 #include <xfce4-session/xfsm-compat-kde.h>
-#include <xfce4-session/xfsm-fadeout.h>
 #include <xfce4-session/xfsm-global.h>
 #include <xfce4-session/xfsm-legacy.h>
 #include <xfce4-session/xfsm-shutdown-helper.h>
@@ -169,7 +168,6 @@
 {
   gboolean accessibility;
   GtkIconTheme *icon_theme;
-  XfsmFadeout *fadeout = NULL;
   GdkScreen *screen;
   GtkWidget *dialog;
   GtkWidget *label;
@@ -327,10 +325,6 @@
       g_object_unref (G_OBJECT (screenshot_gc));
 #endif
 
-      /* display fadeout */
-      fadeout = xfsm_fadeout_new (gtk_widget_get_display (hidden));
-      gdk_flush ();
-
       /* create confirm dialog */
       dialog = g_object_new (GTK_TYPE_DIALOG,
                              "type", GTK_WINDOW_POPUP,
@@ -402,7 +396,8 @@
   gtk_widget_show (vbox2);
   gtk_container_add (GTK_CONTAINER (logout_button), vbox2);
 
-  icon = gtk_icon_theme_load_icon (icon_theme,
+  /* gavin: no icon on logout button */
+  /* icon = gtk_icon_theme_load_icon (icon_theme,
                                    "system-log-out",
                                    32,
                                    0,
@@ -417,7 +412,7 @@
   image = gtk_image_new_from_pixbuf (icon);
   gtk_widget_show (image);
   gtk_box_pack_start (GTK_BOX (vbox2), image, FALSE, FALSE, 0);
-  g_object_unref (icon);
+  g_object_unref (icon); */
 
   label = gtk_label_new (_("Log Out"));
   gtk_widget_show (label);
@@ -436,7 +431,8 @@
   gtk_widget_show (vbox2);
   gtk_container_add (GTK_CONTAINER (reboot_button), vbox2);
 
-  icon = gtk_icon_theme_load_icon (icon_theme,
+  /* gavin: no icon on reboot button */
+  /* icon = gtk_icon_theme_load_icon (icon_theme,
                                    "system-reboot",
                                    32,
                                    0,
@@ -452,7 +448,7 @@
   image = gtk_image_new_from_pixbuf (icon);
   gtk_widget_show (image);
   gtk_box_pack_start (GTK_BOX (vbox2), image, FALSE, FALSE, 0);
-  g_object_unref (icon);
+  g_object_unref (icon); */
 
   label = gtk_label_new (_("Restart"));
   gtk_widget_show (label);
@@ -478,7 +474,8 @@
   gtk_widget_show (vbox2);
   gtk_container_add (GTK_CONTAINER (halt_button), vbox2);
 
-  icon = gtk_icon_theme_load_icon (icon_theme,
+  /* gavin: no icon on shutdown button */
+  /* icon = gtk_icon_theme_load_icon (icon_theme,
                                    "system-shutdown",
                                    32,
                                    0,
@@ -494,7 +491,7 @@
   image = gtk_image_new_from_pixbuf (icon);
   gtk_widget_show (image);
   gtk_box_pack_start (GTK_BOX (vbox2), image, FALSE, FALSE, 0);
-  g_object_unref (icon);
+  g_object_unref (icon); */
 
   label = gtk_label_new (_("Shut Down"));
   gtk_widget_show (label);
@@ -540,7 +537,8 @@
       gtk_widget_show (vbox2);
       gtk_container_add (GTK_CONTAINER (suspend_button), vbox2);
 
-      icon = gtk_icon_theme_load_icon (icon_theme,
+      /* gavin: no icon on suspend button */
+      /* icon = gtk_icon_theme_load_icon (icon_theme,
                                        "system-suspend",
                                        32,
                                        0,
@@ -556,7 +554,7 @@
       image = gtk_image_new_from_pixbuf (icon);
       gtk_widget_show (image);
       gtk_box_pack_start (GTK_BOX (vbox2), image, FALSE, FALSE, 0);
-      g_object_unref (icon);
+      g_object_unref (icon); */
 
       label = gtk_label_new (_("Suspend"));
       gtk_widget_show (label);
@@ -578,7 +576,8 @@
       gtk_widget_show (vbox2);
       gtk_container_add (GTK_CONTAINER (hibernate_button), vbox2);
 
-      icon = gtk_icon_theme_load_icon (icon_theme,
+      /* gavin: no icon on hibernate button */
+      /* icon = gtk_icon_theme_load_icon (icon_theme,
                                        "system-hibernate",
                                        32,
                                        0,
@@ -594,7 +593,7 @@
       image = gtk_image_new_from_pixbuf (icon);
       gtk_widget_show (image);
       gtk_box_pack_start (GTK_BOX (vbox2), image, FALSE, FALSE, 0);
-      g_object_unref (icon);
+      g_object_unref (icon); */
 
       label = gtk_label_new (_("Hibernate"));
       gtk_widget_show (label);
@@ -608,7 +607,7 @@
           _("_Save session for future logins"));
       gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(checkbox), saveonexit);
       gtk_box_pack_start(GTK_BOX(vbox), checkbox, FALSE, TRUE, BORDER);
-      gtk_widget_show(checkbox);
+      /* gtk_widget_show(checkbox); */ /* gavin: don't show save-session checkbox */
     }
   else
     {
@@ -744,8 +743,6 @@
   /* Release Keyboard/Mouse pointer grab */
   if (!accessibility)
     {
-      xfsm_fadeout_destroy (fadeout);
-
       gdk_pointer_ungrab (GDK_CURRENT_TIME);
       gdk_keyboard_ungrab (GDK_CURRENT_TIME);
       gdk_flush ();
