--- ../abs/xfce4-session/src/xfce4-session-4.8.2/xfce4-session/xfsm-fadeout.c.ORIG	2011-12-10 15:01:42.915712369 -0800
+++ ../abs/xfce4-session/src/xfce4-session-4.8.2/xfce4-session/xfsm-fadeout.c	2011-12-10 15:01:59.492378531 -0800
@@ -111,8 +111,11 @@
       fadeout->screens = g_list_append (fadeout->screens, screen);
     }
 
+  /* gavin: no fadeout screen */
+  /*
   for (lp = fadeout->screens; lp != NULL; lp = lp->next)
     gdk_window_show (((FoScreen *) lp->data)->window);
+  */
 
   gdk_cursor_unref (cursor);
 
