diff -ru ../abs/xfce4-session.ORIG/src/xfce4-session-4.10.0/xfce4-session/xfsm-fadeout.c ../abs/xfce4-session/src/xfce4-session-4.10.0/xfce4-session/xfsm-fadeout.c
--- ../abs/xfce4-session.ORIG/src/xfce4-session-4.10.0/xfce4-session/xfsm-fadeout.c	2012-07-25 23:47:05.646074602 -0700
+++ ../abs/xfce4-session/src/xfce4-session-4.10.0/xfce4-session/xfsm-fadeout.c	2012-07-25 23:47:16.213086312 -0700
@@ -51,7 +51,6 @@
   GdkPixmap     *backbuf;
   GdkScreen     *gdk_screen;
   GdkWindow     *window;
-  GdkColor       black = { 0, };
 
   fadeout = g_slice_new0 (XfsmFadeout);
 
@@ -80,9 +79,8 @@
       if (gdk_screen_is_composited (gdk_screen)
           && gdk_screen_get_rgba_colormap (gdk_screen) != NULL)
         {
-          /* transparent black window */
-          gdk_window_set_background (window, &black);
-          gdk_window_set_opacity (window, 0.50);
+          /* transparent window */
+          gdk_window_set_opacity (window, 0.0);
         }
       else
         {
@@ -97,9 +95,8 @@
           cairo_paint (cr);
           g_object_unref (G_OBJECT (root_pixbuf));
 
-          /* draw black layer */
-          gdk_cairo_set_source_color (cr, &black);
-          cairo_paint_with_alpha (cr, 0.50);
+          /* draw transparent layer */
+          cairo_paint_with_alpha (cr, 0.0);
           cairo_destroy (cr);
 
           gdk_window_set_back_pixmap (window, backbuf, FALSE);
diff -ru ../abs/xfce4-session.ORIG/src/xfce4-session-4.10.0/xfce4-session/xfsm-logout-dialog.c ../abs/xfce4-session/src/xfce4-session-4.10.0/xfce4-session/xfsm-logout-dialog.c
--- ../abs/xfce4-session.ORIG/src/xfce4-session-4.10.0/xfce4-session/xfsm-logout-dialog.c	2012-07-25 23:47:05.646074602 -0700
+++ ../abs/xfce4-session/src/xfce4-session-4.10.0/xfce4-session/xfsm-logout-dialog.c	2012-07-25 23:48:26.585382281 -0700
@@ -61,7 +61,7 @@
 
 
 
-#define BORDER   6
+#define BORDER   4
 #define SHOTSIZE 64
 
 
@@ -96,9 +96,6 @@
   /* set when a button is clicked */
   XfsmShutdownType  type_clicked;
 
-  /* save session checkbox */
-  GtkWidget        *save_session;
-
   /* mode widgets */
   GtkWidget        *box[N_MODES];
 
@@ -137,9 +134,7 @@
 static void
 xfsm_logout_dialog_init (XfsmLogoutDialog *dialog)
 {
-  const gchar   *username;
   GtkWidget     *label;
-  gchar         *label_str;
   PangoAttrList *attrs;
   GtkWidget     *vbox;
   GtkWidget     *button_vbox;
@@ -157,14 +152,12 @@
   XfconfChannel *channel;
   GtkWidget     *entry;
   GtkWidget     *image;
-  GtkWidget     *separator;
   gboolean       upower_not_found = FALSE;
 
   dialog->type_clicked = XFSM_SHUTDOWN_LOGOUT;
   dialog->shutdown = xfsm_shutdown_get ();
 
   gtk_dialog_set_default_response (GTK_DIALOG (dialog), GTK_RESPONSE_CANCEL);
-  gtk_dialog_set_has_separator (GTK_DIALOG (dialog), FALSE);
   gtk_window_set_position (GTK_WINDOW (dialog), GTK_WIN_POS_CENTER);
   gtk_window_set_resizable (GTK_WINDOW (dialog), FALSE);
 
@@ -178,30 +171,6 @@
   gtk_container_set_border_width (GTK_CONTAINER (main_vbox), BORDER);
   gtk_widget_show (main_vbox);
 
-  /* label showing the users' name */
-  username = g_get_real_name ();
-  if (username == NULL
-      || *username == '\0'
-      || strcmp ("Unknown", username) == 0)
-    username = g_get_user_name ();
-
-  label_str = g_strdup_printf (_("Log out %s"), username);
-  label = gtk_label_new (label_str);
-  gtk_label_set_justify (GTK_LABEL (label), GTK_JUSTIFY_CENTER);
-  gtk_box_pack_start (GTK_BOX (main_vbox), label, FALSE, TRUE, 0);
-  gtk_widget_show (label);
-  g_free (label_str);
-
-  attrs = pango_attr_list_new ();
-  pango_attr_list_insert (attrs, pango_attr_scale_new (PANGO_SCALE_LARGE));
-  pango_attr_list_insert (attrs, pango_attr_weight_new (PANGO_WEIGHT_BOLD));
-  gtk_label_set_attributes (GTK_LABEL (label), attrs);
-  pango_attr_list_unref (attrs);
-
-  separator = gtk_hseparator_new ();
-  gtk_box_pack_start (GTK_BOX (main_vbox), separator, FALSE, TRUE, 0);
-  gtk_widget_show (separator);
-
   /**
    * Start mode MODE_LOGOUT_BUTTONS
    **/
@@ -359,18 +328,6 @@
         }
     }
 
-  /**
-   * Save session
-   **/
-  if (xfsm_shutdown_can_save_session (dialog->shutdown)
-      && !xfconf_channel_get_bool (channel, "/general/AutoSave", FALSE))
-    {
-      dialog->save_session = gtk_check_button_new_with_mnemonic (_("_Save session for future logins"));
-      gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (dialog->save_session), save_session);
-      gtk_box_pack_start (GTK_BOX (vbox), dialog->save_session, FALSE, TRUE, BORDER);
-      gtk_widget_show (dialog->save_session);
-    }
-
   attrs = pango_attr_list_new ();
   pango_attr_list_insert (attrs, pango_attr_weight_new (PANGO_WEIGHT_BOLD));
 
@@ -518,27 +475,6 @@
   gtk_container_add (GTK_CONTAINER (button), vbox);
   gtk_widget_show (vbox);
 
-  if (G_UNLIKELY (icon_size == 0))
-    {
-      if (gtk_icon_size_lookup (GTK_ICON_SIZE_DND, &w, &h))
-        icon_size = MAX (w, h);
-      else
-        icon_size = 32;
-    }
-
-  icon_theme = gtk_icon_theme_get_for_screen (gtk_window_get_screen (GTK_WINDOW (dialog)));
-  pixbuf = gtk_icon_theme_load_icon (icon_theme, icon_name, icon_size, 0, NULL);
-  if (G_UNLIKELY (pixbuf == NULL))
-    {
-      pixbuf = gtk_icon_theme_load_icon (icon_theme, icon_name_fallback,
-                                         icon_size, GTK_ICON_LOOKUP_GENERIC_FALLBACK,
-                                         NULL);
-    }
-
-  image = gtk_image_new_from_pixbuf (pixbuf);
-  gtk_box_pack_start (GTK_BOX (vbox), image, FALSE, FALSE, 0);
-  gtk_widget_show (image);
-
   label = gtk_label_new_with_mnemonic (title);
   gtk_box_pack_start (GTK_BOX (vbox), label, FALSE, FALSE, 0);
   gtk_widget_show (label);
@@ -882,8 +818,6 @@
           /* store autosave state */
           if (autosave)
             *return_save_session = TRUE;
-          else if (xfsm_dialog->save_session != NULL)
-            *return_save_session = gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (xfsm_dialog->save_session));
           else
             *return_save_session = FALSE;
 
