diff -ru aur/notification-daemon-gtk2.ORIG/src/notification-daemon-0.5.0/src/daemon/stack.h aur/notification-daemon-gtk2/src/notification-daemon-0.5.0/src/daemon/stack.h
--- aur/notification-daemon-gtk2.ORIG/src/notification-daemon-0.5.0/src/daemon/stack.h	2016-12-22 01:07:10.761572206 -0800
+++ aur/notification-daemon-gtk2/src/notification-daemon-0.5.0/src/daemon/stack.h	2016-12-22 01:09:17.640069270 -0800
@@ -33,7 +33,7 @@
         NOTIFY_STACK_LOCATION_TOP_RIGHT,
         NOTIFY_STACK_LOCATION_BOTTOM_LEFT,
         NOTIFY_STACK_LOCATION_BOTTOM_RIGHT,
-        NOTIFY_STACK_LOCATION_DEFAULT = NOTIFY_STACK_LOCATION_BOTTOM_RIGHT
+        NOTIFY_STACK_LOCATION_DEFAULT = NOTIFY_STACK_LOCATION_TOP_RIGHT
 } NotifyStackLocation;
 
 typedef struct _NotifyStack NotifyStack;
