diff --color -ru firefox.ORIG/PKGBUILD firefox/PKGBUILD
--- firefox.ORIG/PKGBUILD	2024-11-19 10:17:05.589088242 -0800
+++ firefox/PKGBUILD	2024-11-19 10:22:53.984894037 -0800
@@ -106,6 +106,8 @@
   mkdir mozbuild
   cd firefox-$pkgver
 
+  patch -p0 < "$srcdir"/../firefox_use-monotonic-time.diff
+
   echo -n "$_google_api_key" >google-api-key
 
   cat >../mozconfig <<END
@@ -278,4 +280,8 @@
 END
 }
 
+source+=('firefox_use-monotonic-time.diff')
+sha256sums+=('00d82e0f1d820ebf64861f0ca837f476bc923e12456e1d5140ef3248c914a555')
+b2sums+=('9cc228b75a1e8d485bbc51144b176682c7bb78cdf62b36ab2e228e52eece2fdbb16b22561c5bf3b005bbb6d7b17c179d0738d842d529375d16626b420e33a693')
+
 # vim:set sw=2 sts=-1 et:
