diff --color -ru gtk-nocsd-git.ORIG/src/gtk-nocsd-git/Source/GTK-NoCSD.c gtk-nocsd-git/src/gtk-nocsd-git/Source/GTK-NoCSD.c
--- gtk-nocsd-git.ORIG/src/gtk-nocsd-git/Source/GTK-NoCSD.c	2026-09-14 08:03:25.741972452 -0700
+++ gtk-nocsd-git/src/gtk-nocsd-git/Source/GTK-NoCSD.c	2026-09-14 08:03:28.762474912 -0700
@@ -2349,7 +2349,7 @@
 
 	// The CSS that is globally loaded
 	const char *CSS = "window { border-radius: 0; box-shadow: none; }"
-		"headerbar { border-radius: 0; min-height: 0pt; padding: 0pt; }"
+		"headerbar { border-radius: 0; min-height: 0pt; }"
 		"headerbar * { margin: 0pt; }"
 		"decoration { margin: 0px; }"
 		"headerbar viewswitcher * { padding: 0pt; min-height: 0pt; }";
