diff -ru rhythmbox.ORIG/src/rhythmbox/widgets/rb-header.c rhythmbox/src/rhythmbox/widgets/rb-header.c
--- rhythmbox.ORIG/src/rhythmbox/widgets/rb-header.c	2024-11-22 14:42:56.679174837 -0800
+++ rhythmbox/src/rhythmbox/widgets/rb-header.c	2024-11-22 14:42:56.779175754 -0800
@@ -553,7 +553,7 @@
 		if (header->priv->art_key == NULL ||
 		    rhythmdb_entry_matches_ext_db_key (header->priv->db, entry, header->priv->art_key) == FALSE ||
 		    rb_ext_db_key_is_null_match (key, header->priv->art_key)) {
-			rb_fading_image_start (RB_FADING_IMAGE (header->priv->image), 2000);
+			rb_fading_image_start (RB_FADING_IMAGE (header->priv->image), 0);
 			rb_ext_db_request (header->priv->art_store,
 					   key,
 					   (RBExtDBRequestCallback) art_cb,
@@ -571,7 +571,7 @@
 					  G_CALLBACK (playback_status_changed_cb),
 					  header);
 	} else {
-		rb_fading_image_start (RB_FADING_IMAGE (header->priv->image), 2000);
+		rb_fading_image_start (RB_FADING_IMAGE (header->priv->image), 0);
 		header->priv->duration = 0;
 	}
 
