diff -ru rhythmbox.ORIG/src/rhythmbox/widgets/rb-entry-view.c rhythmbox/src/rhythmbox/widgets/rb-entry-view.c
--- rhythmbox.ORIG/src/rhythmbox/widgets/rb-entry-view.c	2026-08-03 17:01:29.026305695 -0700
+++ rhythmbox/src/rhythmbox/widgets/rb-entry-view.c	2026-08-03 17:01:29.685746451 -0700
@@ -2691,13 +2691,7 @@
 void
 rb_entry_view_set_status (RBEntryView *view, const char *status, gboolean busy)
 {
-	if (status == NULL) {
-		gtk_widget_hide (view->priv->status);
-	} else {
-		nautilus_floating_bar_set_primary_label (NAUTILUS_FLOATING_BAR (view->priv->status), status);
-		nautilus_floating_bar_set_show_spinner (NAUTILUS_FLOATING_BAR (view->priv->status), busy);
-		gtk_widget_show (view->priv->status);
-	}
+	gtk_widget_hide (view->priv->status);
 }
 
 /* This should really be standard. */
