--- pre/rhythmbox/src/rhythmbox-0.13.3/plugins/audioscrobbler/rb-audioscrobbler-profile-page.c.ORIG	2011-02-03 23:38:45.303417424 -0800
+++ pre/rhythmbox/src/rhythmbox-0.13.3/plugins/audioscrobbler/rb-audioscrobbler-profile-page.c	2011-02-03 23:40:02.291239428 -0800
@@ -1163,7 +1163,7 @@
 		}
 
 		/* if the list of stations is still empty then add some defaults */
-		if (page->priv->radio_sources == NULL) {
+		if (0) { /* gavin: don't add these default radio stations */
 			char *url;
 			char *name;
 
