--- ../pre/rhythmbox/src/rhythmbox-0.13.3/widgets/rb-header.c.ORIG	2011-02-03 22:57:53.344179071 -0800
+++ ../pre/rhythmbox/src/rhythmbox-0.13.3/widgets/rb-header.c	2011-02-03 22:58:38.214819132 -0800
@@ -120,9 +120,10 @@
 	PROP_SLIDER_DRAGGING
 };
 
+/* gavin: "Now Playing" formatting */
 #define TITLE_FORMAT  "<big><b>%s</b></big>"
-#define ALBUM_FORMAT  "<i>%s</i>"
-#define ARTIST_FORMAT "<i>%s</i>"
+#define ALBUM_FORMAT  "<b><i>%s</i></b>"
+#define ARTIST_FORMAT "<b>%s</b>"
 #define STREAM_FORMAT "(%s)"
 
 #define SCROLL_UP_SEEK_OFFSET	5
