diff -ru rhythmbox.ORIG/src/rhythmbox/shell/rb-shell.c rhythmbox/src/rhythmbox/shell/rb-shell.c
--- rhythmbox.ORIG/src/rhythmbox/shell/rb-shell.c	2026-08-03 17:01:29.022305720 -0700
+++ rhythmbox/src/rhythmbox/shell/rb-shell.c	2026-08-03 17:01:29.134168687 -0700
@@ -700,6 +700,7 @@
 
 	rb_shell_append_display_page (shell, RB_DISPLAY_PAGE (shell->priv->library_source), page_group);
 	rb_shell_append_display_page (shell, RB_DISPLAY_PAGE (shell->priv->podcast_source), page_group);
+	g_object_set (shell->priv->podcast_source, "visibility", FALSE, NULL);
 	rb_shell_append_display_page (shell, RB_DISPLAY_PAGE (shell->priv->missing_files_source), page_group);
 	rb_shell_append_display_page (shell, RB_DISPLAY_PAGE (shell->priv->import_errors_source), page_group);
 
