diff -ru batterymon-clone.ORIG/src/batterymon-clone/batterymon batterymon-clone/src/batterymon-clone/batterymon
--- batterymon-clone.ORIG/src/batterymon-clone/batterymon	2016-12-23 12:56:43.406320012 -0800
+++ batterymon-clone/src/batterymon-clone/batterymon	2016-12-23 12:57:17.465738696 -0800
@@ -411,15 +411,20 @@
     def __init__(self, theme):
         self.theme = theme
 
-    def notify(self, title, message, icon):
+    def notify(self, title, message, iconn):
         if pynotify:
-            n = pynotify.Notification(title, message)
-            iconf = self.theme.get_icon(icon)
+            n = pynotify.Notification(title, message) if title else None
+            iconf = self.theme.get_icon(iconn)
             logger.debug("DEBUG Notification icon " +iconf)
             icon = gtk.gdk.pixbuf_new_from_file_at_size(iconf, 46, 46)
-            n.set_icon_from_pixbuf(icon)
-            n.show()
-            logger.debug("Notification shown")
+
+            if n is not None:
+                if iconn is "empty":
+                    n.set_urgency(pynotify.URGENCY_CRITICAL)
+
+                n.set_icon_from_pixbuf(icon)
+                n.show()
+                logger.debug("Notification shown")
 
 
 def parse_args():
@@ -574,9 +579,9 @@
     def ac_property_modified(self, present):
         # TODO: use the appropriate icon, instead of just the "full" versions
         if present:
-            self.n.notify(_("On AC"), _("You are currently running on AC"),"charging_full")
+            self.n.notify(None, None, "charging_full")
         else:
-            self.n.notify(_("On Battery"), _("AC adapter unplugged, running on battery"),"full")
+            self.n.notify(None, None, "full")
 
     def battery_property_modified(self, battery):
         # TODO: use the appropriate icon, instead of just the "full" versions
