diff -ru batterymon-clone.ORIG/src/batterymon-clone/batterymon batterymon-clone/src/batterymon-clone/batterymon
--- batterymon-clone.ORIG/src/batterymon-clone/batterymon	2018-01-08 14:54:54.056605289 -0800
+++ batterymon-clone/src/batterymon-clone/batterymon	2018-01-08 15:37:47.249589425 -0800
@@ -266,9 +266,9 @@
             return True
 
     def validate(self, theme):
-        all_icons = ["1", "2", "3", "4", "5", "empty", "full",
+        all_icons = ["1", "2", "3", "4", "5", "6", "7", "empty", "full",
         "charging_1","charging_2","charging_3","charging_4","charging_5",
-        "charging_full","charging_empty"]
+        "charging_6","charging_7", "charging_full", "charging_empty"]
 
         for icon in all_icons:
             if not self.file_exists(self.get_icon(icon)):
@@ -346,11 +346,13 @@
 
         levels_and_icons = [
             (96, 'full'),
-            (80, '5'),
-            (64, '4'),
-            (48, '3'),
-            (32, '2'),
-            (16, '1'),
+            (84, '7'),
+            (72, '6'),
+            (60, '5'),
+            (48, '4'),
+            (36, '3'),
+            (24, '2'),
+            (12, '1'),
             ( 0, 'empty'),
         ]
 
