diff -ru xfce4-screenshooter.ORIG/src/xfce4-screenshooter/lib/screenshooter-capture-x11.c xfce4-screenshooter/src/xfce4-screenshooter/lib/screenshooter-capture-x11.c
--- xfce4-screenshooter.ORIG/src/xfce4-screenshooter/lib/screenshooter-capture-x11.c	2024-11-03 11:18:30.290333545 -0800
+++ xfce4-screenshooter/src/xfce4-screenshooter/lib/screenshooter-capture-x11.c	2024-11-03 11:18:30.297000169 -0800
@@ -879,7 +879,6 @@
                        event->x + x_offset,
                        event->y + y_offset);
 
-      gtk_label_set_text (GTK_LABEL (rbdata->size_label), coords);
       g_free (coords);
 
       region = cairo_region_create_rectangle (&old_rect);
