diff -ru abs/gtk2.ORIG/src/gtk+/gtk/gtkfilechooserdefault.c abs/gtk2/src/gtk+/gtk/gtkfilechooserdefault.c
--- abs/gtk2.ORIG/src/gtk+/gtk/gtkfilechooserdefault.c	2016-08-10 11:24:34.748465893 -0700
+++ abs/gtk2/src/gtk+/gtk/gtkfilechooserdefault.c	2016-08-10 11:24:35.461780190 -0700
@@ -6017,7 +6017,7 @@
   switch (impl->startup_mode)
     {
     case STARTUP_MODE_RECENT:
-      recent_shortcut_handler (impl);
+      home_folder_handler (impl);
       break;
 
     case STARTUP_MODE_CWD:
