diff --color -ru mirage-python3.ORIG/src/mirage-0.11.1/resources/gtk/menus.ui mirage-python3/src/mirage-0.11.1/resources/gtk/menus.ui
--- mirage-python3.ORIG/src/mirage-0.11.1/resources/gtk/menus.ui	2022-07-01 18:26:53.659567826 -0700
+++ mirage-python3/src/mirage-0.11.1/resources/gtk/menus.ui	2022-07-01 18:46:35.741048615 -0700
@@ -5,32 +5,27 @@
       <attribute name="label" translatable="yes">_File</attribute>
       <section>
         <item>
-          <attribute name="icon">document-open</attribute>
           <attribute name="label" translatable="yes">_Open Image...</attribute>
           <attribute name="action">app.open-file</attribute>
           <attribute name="accel">&lt;Control&gt;o</attribute>
         </item>
         <item>
-          <attribute name="icon">folder</attribute>
           <attribute name="label" translatable="yes">Open _Folder...</attribute>
           <attribute name="action">app.open-folder</attribute>
           <attribute name="accel">&lt;Control&gt;f</attribute>
         </item>
         <item>
-          <attribute name="icon">network-workgroup</attribute>
           <attribute name="label" translatable="yes">Open _Remote image...</attribute>
           <attribute name="action">app.open-file-remote</attribute>
         </item>
       </section>
       <section>
         <item>
-          <attribute name="icon">document-save</attribute>
           <attribute name="label" translatable="yes">_Save Image</attribute>
           <attribute name="action">app.save-image</attribute>
           <attribute name="accel">&lt;Control&gt;s</attribute>
         </item>
         <item>
-          <attribute name="icon">document-save-as</attribute>
           <attribute name="label" translatable="yes">Save Image _As...</attribute>
           <attribute name="action">app.save-image-as</attribute>
           <attribute name="accel">&lt;Shift&gt;&lt;Control&gt;s</attribute>
@@ -38,14 +33,12 @@
       </section>
       <section>
         <item>
-          <attribute name="icon">camera-photo</attribute>
           <attribute name="label" translatable="yes">_Take Screenshot...</attribute>
           <attribute name="action">app.screenshot</attribute>
         </item>
       </section>
       <section>
         <item>
-          <attribute name="icon">document-properties</attribute>
           <attribute name="label" translatable="yes">_Properties...</attribute>
           <attribute name="action">app.show-properties</attribute>
         </item>
@@ -53,7 +46,6 @@
       <section id="recent-files-section"></section>
       <section>
         <item>
-          <attribute name="icon">application-exit</attribute>
           <attribute name="label" translatable="yes">_Quit</attribute>
           <attribute name="action">app.exit-app</attribute>
           <attribute name="accel">&lt;Control&gt;q</attribute>
@@ -64,25 +56,21 @@
       <attribute name="label" translatable="yes">_Edit</attribute>
       <section>
         <item>
-          <attribute name="icon">object-rotate-left</attribute>
           <attribute name="label" translatable="yes">Rotate _Left</attribute>
           <attribute name="action">app.rotate-left</attribute>
           <attribute name="accel">&lt;Control&gt;Left</attribute>
         </item>
         <item>
-          <attribute name="icon">object-rotate-right</attribute>
           <attribute name="label" translatable="yes">Rotate _Right</attribute>
           <attribute name="action">app.rotate-right</attribute>
           <attribute name="accel">&lt;Control&gt;Right</attribute>
         </item>
         <item>
-          <attribute name="icon">object-flip-vertical</attribute>
           <attribute name="label" translatable="yes">Flip _Vertically</attribute>
           <attribute name="action">app.flip-image-vert</attribute>
           <attribute name="accel">&lt;Control&gt;v</attribute>
         </item>
         <item>
-          <attribute name="icon">object-flip-horizontal</attribute>
           <attribute name="label" translatable="yes">Flip _Horizontally</attribute>
           <attribute name="action">app.flip-image-horiz</attribute>
           <attribute name="accel">&lt;Control&gt;h</attribute>
@@ -109,7 +97,6 @@
           <attribute name="accel">F2</attribute>
         </item>
         <item>
-          <attribute name="icon">edit-delete</attribute>
           <attribute name="label" translatable="yes">_Delete...</attribute>
           <attribute name="action">app.delete-image</attribute>
           <attribute name="accel">Delete</attribute>
@@ -128,7 +115,6 @@
           </section>
         </submenu>
         <item>
-          <attribute name="icon">preferences-system</attribute>
           <attribute name="label" translatable="yes">_Preferences...</attribute>
           <attribute name="action">app.show-prefs</attribute>
           <attribute name="accel">&lt;Control&gt;p</attribute>
@@ -139,25 +125,21 @@
       <attribute name="label" translatable="yes">_View</attribute>
       <section>
         <item>
-          <attribute name="icon">zoom-out</attribute>
           <attribute name="label" translatable="yes">Zoom _Out</attribute>
           <attribute name="action">app.zoom-out</attribute>
           <attribute name="accel">&lt;Control&gt;Down</attribute>
         </item>
         <item>
-          <attribute name="icon">zoom-in</attribute>
           <attribute name="label" translatable="yes">Zoom _In</attribute>
           <attribute name="action">app.zoom-in</attribute>
           <attribute name="accel">&lt;Control&gt;Up</attribute>
         </item>
         <item>
-          <attribute name="icon">zoom-original</attribute>
           <attribute name="label" translatable="yes">_1:1</attribute>
           <attribute name="action">app.zoom-1-to-1</attribute>
           <attribute name="accel">&lt;Control&gt;1</attribute>
         </item>
         <item>
-          <attribute name="icon">zoom-fit-best</attribute>
           <attribute name="label" translatable="yes">Zoom To _Fit</attribute>
           <attribute name="action">app.zoom-to-fit-window</attribute>
           <attribute name="accel">&lt;Control&gt;0</attribute>
@@ -179,14 +161,12 @@
       </section>
       <section>
         <item>
-          <attribute name="icon">view-fullscreen</attribute>
           <attribute name="label" translatable="yes">_Full Screen</attribute>
           <attribute name="action">app.enter-fullscreen</attribute>
           <attribute name="accel">F11</attribute>
           <attribute name="hidden-when">action-missing</attribute>
         </item>
         <item>
-          <attribute name="icon">view-restore</attribute>
           <attribute name="label" translatable="yes">E_xit Full Screen</attribute>
           <attribute name="action">app.leave-fullscreen</attribute>
           <attribute name="accel">F11</attribute>
@@ -198,19 +178,16 @@
       <attribute name="label" translatable="yes">_Go</attribute>
       <section>
         <item>
-          <attribute name="icon">go-next</attribute>
           <attribute name="label" translatable="yes">_Next Image</attribute>
           <attribute name="action">app.goto-next-image</attribute>
           <attribute name="accel">Right</attribute>
         </item>
         <item>
-          <attribute name="icon">go-previous</attribute>
           <attribute name="label" translatable="yes">_Previous Image</attribute>
           <attribute name="action">app.goto-prev-image</attribute>
           <attribute name="accel">Left</attribute>
         </item>
         <item>
-          <attribute name="icon">media-playlist-shuffle</attribute>
           <attribute name="label" translatable="yes">_Random Image</attribute>
           <attribute name="action">app.goto-random-image</attribute>
           <attribute name="accel">R</attribute>
@@ -218,13 +195,11 @@
       </section>
       <section>
         <item>
-          <attribute name="icon">go-first</attribute>
           <attribute name="label" translatable="yes">_First Image</attribute>
           <attribute name="action">app.goto-first-image</attribute>
           <attribute name="accel">Home</attribute>
         </item>
         <item>
-          <attribute name="icon">go-last</attribute>
           <attribute name="label" translatable="yes">_Last Image</attribute>
           <attribute name="action">app.goto-last-image</attribute>
           <attribute name="accel">End</attribute>
@@ -232,14 +207,12 @@
       </section>
       <section>
         <item>
-          <attribute name="icon">media-playback-start</attribute>
           <attribute name="label" translatable="yes">_Start Slideshow</attribute>
           <attribute name="action">app.start-slideshow</attribute>
           <attribute name="accel">F5</attribute>
           <attribute name="hidden-when">action-missing</attribute>
         </item>
         <item>
-          <attribute name="icon">media-playback-stop</attribute>
           <attribute name="label" translatable="yes">_Stop Slideshow</attribute>
           <attribute name="action">app.stop-slideshow</attribute>
           <attribute name="accel">F5</attribute>
@@ -251,7 +224,6 @@
       <attribute name="label" translatable="yes">_Help</attribute>
       <section>
         <item>
-          <attribute name="icon">help-about</attribute>
           <attribute name="label" translatable="yes">_About</attribute>
           <attribute name="action">app.show-about</attribute>
         </item>
@@ -262,47 +234,39 @@
   <menu id="popup-menu">
     <section>
       <item>
-        <attribute name="icon">go-next</attribute>
         <attribute name="label" translatable="yes">Next Image</attribute>
         <attribute name="action">app.goto-next-image</attribute>
       </item>
       <item>
-        <attribute name="icon">go-previous</attribute>
         <attribute name="label" translatable="yes">Previous Image</attribute>
         <attribute name="action">app.goto-prev-image</attribute>
       </item>
     </section>
     <section>
       <item>
-        <attribute name="icon">zoom-out</attribute>
         <attribute name="label" translatable="yes">Zoom Out</attribute>
         <attribute name="action">app.zoom-out</attribute>
       </item>
       <item>
-        <attribute name="icon">zoom-in</attribute>
         <attribute name="label" translatable="yes">Zoom In</attribute>
         <attribute name="action">app.zoom-in</attribute>
       </item>
       <item>
-        <attribute name="icon">zoom-original</attribute>
         <attribute name="label" translatable="yes">1:1</attribute>
         <attribute name="action">app.zoom-1-to-1</attribute>
       </item>
       <item>
-        <attribute name="icon">zoom-fit-best</attribute>
         <attribute name="label" translatable="yes">Zoom to Fit</attribute>
         <attribute name="action">app.zoom-to-fit-window</attribute>
       </item>
     </section>
     <section>
       <item>
-        <attribute name="icon">media-playback-start</attribute>
         <attribute name="label" translatable="yes">Start Slideshow</attribute>
         <attribute name="action">app.start-slideshow</attribute>
         <attribute name="hidden-when">action-missing</attribute>
       </item>
       <item>
-        <attribute name="icon">media-playback-start</attribute>
         <attribute name="label" translatable="yes">Stop Slideshow</attribute>
         <attribute name="action">app.stop-slideshow</attribute>
         <attribute name="hidden-when">action-missing</attribute>
@@ -310,13 +274,11 @@
     </section>
     <section>
       <item>
-        <attribute name="icon">view-fullscreen</attribute>
         <attribute name="label" translatable="yes">Full Screen</attribute>
         <attribute name="action">app.enter-fullscreen</attribute>
         <attribute name="hidden-when">action-missing</attribute>
       </item>
       <item>
-        <attribute name="icon">view-restore</attribute>
         <attribute name="label" translatable="yes">Exit Full Screen</attribute>
         <attribute name="action">app.leave-fullscreen</attribute>
         <attribute name="hidden-when">action-missing</attribute>
