diff --color -ru mirage-python3.ORIG/src/mirage-0.11.1/mirage/__init__.py mirage-python3/src/mirage-0.11.1/mirage/__init__.py
--- mirage-python3.ORIG/src/mirage-0.11.1/mirage/__init__.py	2022-07-01 20:32:11.909600526 -0700
+++ mirage-python3/src/mirage-0.11.1/mirage/__init__.py	2022-07-01 20:32:19.209924956 -0700
@@ -1546,7 +1546,12 @@
             return True
         self.stop_now = True
         self.closing_app = True
-        self.save_settings()
+
+        try:
+            self.save_settings()
+        except:
+            print(traceback.format_exc(), file=sys.stderr)
+
         sys.exit(0)
 
     def put_zoom_image_to_window(self, currimg_preloaded):
