diff -ru roxterm.ORIG/src/roxterm-3.10.1/src/roxterm.c roxterm/src/roxterm-3.10.1/src/roxterm.c
--- roxterm.ORIG/src/roxterm-3.10.1/src/roxterm.c	2020-07-28 11:56:32.795141854 -0700
+++ roxterm/src/roxterm-3.10.1/src/roxterm.c	2020-07-28 11:56:32.811809150 -0700
@@ -4256,8 +4256,10 @@
         }
         g_free(modpath);
     }
-    resources_access_icon();
-    gtk_window_set_default_icon_name("roxterm");
+
+    char *logo_filename = g_build_filename(ICON_DIR, "roxterm.png", NULL);
+    gtk_window_set_default_icon_from_file(logo_filename, NULL);
+    g_free(logo_filename);
 
     optsdbus_listen_for_opt_signals(roxterm_opt_signal_handler);
     optsdbus_listen_for_stuff_changed_signals(roxterm_stuff_changed_handler);
