diff -ru roxterm.ORIG/src/roxterm-3.7.5/src/roxterm.c roxterm/src/roxterm-3.7.5/src/roxterm.c
--- roxterm.ORIG/src/roxterm-3.7.5/src/roxterm.c	2019-07-29 10:50:33.508577932 -0700
+++ roxterm/src/roxterm-3.7.5/src/roxterm.c	2019-07-29 11:04:57.610643882 -0700
@@ -1133,7 +1133,7 @@
     double spacing = (double) options_lookup_int_with_default(roxterm->profile,
             "vspacing", 0) / 100.0;
 
-    vte_terminal_set_cell_height_scale(vte, CLAMP(spacing, 0.0, 1.0) + 1.0);
+    vte_terminal_set_cell_height_scale(vte, CLAMP(spacing, -0.5, 1.0) + 1.0);
 }
 
 static void
@@ -1142,7 +1142,7 @@
     double spacing = (double) options_lookup_int_with_default(roxterm->profile,
             "hspacing", 0) / 100.0;
 
-    vte_terminal_set_cell_width_scale(vte, CLAMP(spacing, 0.0, 1.0) + 1.0);
+    vte_terminal_set_cell_width_scale(vte, CLAMP(spacing, -0.5, 1.0) + 1.0);
 }
 
 static void
diff -ru roxterm.ORIG/src/roxterm-3.7.5/src/roxterm-config.ui roxterm/src/roxterm-3.7.5/src/roxterm-config.ui
--- roxterm.ORIG/src/roxterm-3.7.5/src/roxterm-config.ui	2019-07-29 10:50:33.508577932 -0700
+++ roxterm/src/roxterm-3.7.5/src/roxterm-config.ui	2019-07-29 11:06:56.626848368 -0700
@@ -1296,6 +1296,7 @@
     <property name="page_increment">10</property>
   </object>
   <object class="GtkAdjustment" id="hspacing_adjustment">
+    <property name="lower">-50</property>
     <property name="upper">100</property>
     <property name="step_increment">1</property>
     <property name="page_increment">10</property>
@@ -1319,6 +1320,7 @@
     <property name="page_increment">10</property>
   </object>
   <object class="GtkAdjustment" id="vspacing_adjustment">
+    <property name="lower">-50</property>
     <property name="upper">100</property>
     <property name="step_increment">1</property>
     <property name="page_increment">10</property>
