diff -ru abs/pidgin-otr.ORIG/src/pidgin-otr-4.0.1/gtk-dialog.c abs/pidgin-otr/src/pidgin-otr-4.0.1/gtk-dialog.c
--- abs/pidgin-otr.ORIG/src/pidgin-otr-4.0.1/gtk-dialog.c	2015-06-16 12:38:42.312699310 -0700
+++ abs/pidgin-otr/src/pidgin-otr-4.0.1/gtk-dialog.c	2015-06-16 12:42:49.842732460 -0700
@@ -2677,8 +2677,6 @@
 		    purple_normalize(account,
 			    purple_conversation_get_name(currentConv)));
 
-	    otr_add_buddy_top_menu(gtkconv, convctx, active_conv, username,
-		    accountname, &pos);
 	    g_free(username);
 
 	} else if (num_contexts == 2 &&
@@ -2703,10 +2701,6 @@
 		convctx->context = currentContext;
 	    }
 
-	    otr_add_buddy_top_menu(gtkconv, convctx, active_conv,
-		    currentContext->username, currentContext->accountname,
-		    &pos);
-
 	} else {
 	    /* Multi-instances */
 	    *is_multi_instance = TRUE;
