diff -ru abs/vte.ORIG/src/vte-0.28.2/src/vte.c abs/vte/src/vte-0.28.2/src/vte.c
--- abs/vte.ORIG/src/vte-0.28.2/src/vte.c	2014-01-26 11:18:34.010986123 -0800
+++ abs/vte/src/vte-0.28.2/src/vte.c	2014-01-26 11:18:48.840985664 -0800
@@ -11398,7 +11398,7 @@
 	}
 
 	adj = terminal->adjustment;
-	v = MAX (1., ceil (gtk_adjustment_get_page_increment (adj) / 10.));
+	v = 2.0; /* MAX (1., ceil (gtk_adjustment_get_page_increment (adj) / 10.)); */ /* gavin: always scroll by 2 */
 	switch (event->direction) {
 	case GDK_SCROLL_UP:
 		v = -v;
