diff --color -ru 1password.ORIG/PKGBUILD 1password/PKGBUILD
--- 1password.ORIG/PKGBUILD	2022-01-26 13:15:11.113472001 -0800
+++ 1password/PKGBUILD	2022-01-26 13:18:14.118098841 -0800
@@ -43,6 +43,19 @@
     install -dm0755 "${pkgdir}"/opt
     mv "1password-${_tarver}.x64" "${pkgdir}/opt/1Password"
 
+    _ASAR_PATH="${pkgdir}"/opt/1Password/resources/app.asar
+    _ASAR_TMP=$(mktemp -dp "${srcdir}")
+
+    asar extract "${_ASAR_PATH}" "${_ASAR_TMP}"
+
+    cp "${srcdir}"/icons/* "${_ASAR_TMP}"/images
+
+    js-beautify "${_ASAR_TMP}"/main.js > "${_ASAR_TMP}"/main.full.js
+    patch -d "${_ASAR_TMP}" -p0 < "${srcdir}"/enable-context-menu.diff
+    terser "${_ASAR_TMP}"/main.full.js > "${_ASAR_TMP}"/main.js
+    rm "${_ASAR_TMP}"/main.full.js
+    asar pack "${_ASAR_TMP}" "${_ASAR_PATH}"
+
     # Symlink /usr/bin executable to opt
     install -dm0755 "${pkgdir}"/usr/bin
     ln -s /opt/1Password/1password "${pkgdir}"/usr/bin/1password
