diff --color -ru 1password.ORIG/PKGBUILD 1password/PKGBUILD
--- 1password.ORIG/PKGBUILD	2022-01-26 08:16:34.349970796 -0800
+++ 1password/PKGBUILD	2022-01-26 08:21:22.014573190 -0800
@@ -43,6 +43,13 @@
     install -dm0755 "${pkgdir}"/opt
     mv "1password-${_tarver}.x64" "${pkgdir}/opt/1Password"
 
+    _ASAR_TMP=$(mktemp -dp "${srcdir}")
+    asar extract "${pkgdir}"/opt/1Password/resources/app.asar "${_ASAR_TMP}"
+    cp "${srcdir}"/tray_{un,}locked.png "${_ASAR_TMP}"/images
+    js-beautify -r "${_ASAR_TMP}"/main.js
+    patch -d "${_ASAR_TMP}" -p0 < "${srcdir}"/enable-context-menu.diff
+    asar pack "${_ASAR_TMP}" "${pkgdir}"/opt/1Password/resources/app.asar
+
     # Symlink /usr/bin executable to opt
     install -dm0755 "${pkgdir}"/usr/bin
     ln -s /opt/1Password/1password "${pkgdir}"/usr/bin/1password
