--- main.js.ORIG	2022-01-26 10:48:58.939268711 -0800
+++ main.js	2022-01-26 10:49:19.386808170 -0800
@@ -1736,7 +1736,7 @@
                                 click: () => {
                                     s.app.quit()
                                 }
-                            }), this.contextMenu = s.Menu.buildFromTemplate(e), "menu" === this.userSettings.get("trayAction") && this.tray.setContextMenu(this.contextMenu)
+                            }), this.contextMenu = s.Menu.buildFromTemplate(e), this.tray.setContextMenu(this.contextMenu)
                         }, this.setupGlobalShortcuts = () => {
                             const e = () => {
                                     this.windowManager.toggleQuickAccess()
