--- main.js.ORIG	2022-03-15 08:54:32.125692637 -0700
+++ main.js	2022-03-15 08:56:03.869675244 -0700
@@ -12028,7 +12028,7 @@
                                 click: () => {
                                     s.app.quit()
                                 }
-                            }), this.contextMenu = s.Menu.buildFromTemplate(e), "menu" === this.userSettings.get("trayAction") ? this.tray.setContextMenu(this.contextMenu) : this.tray.setContextMenu(null)
+                            }), this.contextMenu = s.Menu.buildFromTemplate(e), this.tray.setContextMenu(this.contextMenu)
                         }, this.setupGlobalShortcuts = () => {
                             const e = () => {
                                     this.windowManager.toggleQuickAccess()
