--- main.full.js.ORIG	2022-11-01 16:00:00.426710440 -0700
+++ main.full.js	2022-11-01 16:01:25.359565072 -0700
@@ -51877,7 +51877,7 @@
                             click: () => {
                                 i.app.quit()
                             }
-                        }), this.contextMenu = i.Menu.buildFromTemplate(e), "menu" === this.userSettings.get("trayAction") ? this.tray.setContextMenu(this.contextMenu) : this.tray.setContextMenu(null)
+                        }), this.contextMenu = i.Menu.buildFromTemplate(e), this.tray.setContextMenu(this.contextMenu)
                     };
                     getTrayImagePath = e => r.join(__dirname, `images/tray_${e===a.OpAppLockState.Locked?"locked":"unlocked"}.png`);
                     showBackgroundSshPrompt = () => {
