--- main.full.js.ORIG	2022-12-06 21:52:13.416357999 -0800
+++ main.full.js	2022-12-06 21:53:40.693571857 -0800
@@ -2827,7 +2827,7 @@
                             click: () => {
                                 s.app.quit()
                             }
-                        }), this.contextMenu = s.Menu.buildFromTemplate(e), "menu" === this.userSettings.get("trayAction") ? this.tray.setContextMenu(this.contextMenu) : this.tray.setContextMenu(null)
+                        }), this.contextMenu = s.Menu.buildFromTemplate(e), this.tray.setContextMenu(this.contextMenu)
                     };
                     getTrayImagePath = e => n.join(__dirname, `images/tray_${e===r.TtD.Locked?"locked":"unlocked"}.png`);
                     showBackgroundSshPrompt = () => {
