--- main.js.ORIG	2022-06-09 12:24:19.048385789 -0700
+++ main.js	2022-06-09 12:25:09.730724406 -0700
@@ -46955,7 +46955,7 @@
                             click: () => {
                                 a.app.quit()
                             }
-                        }), this.contextMenu = a.Menu.buildFromTemplate(e), "menu" === this.userSettings.get("trayAction") ? this.tray.setContextMenu(this.contextMenu) : this.tray.setContextMenu(null)
+                        }), this.contextMenu = a.Menu.buildFromTemplate(e), this.tray.setContextMenu(this.contextMenu)
                     };
                     getTrayImagePath = e => r.join(__dirname, `images/tray_${"Locked"===e?"locked":"unlocked"}.png`);
                     destroy() {
