diff -ru aur/mirage.ORIG/src/mirage-0.9.5.2/mirage.py aur/mirage/src/mirage-0.9.5.2/mirage.py
--- aur/mirage.ORIG/src/mirage-0.9.5.2/mirage.py	2018-02-09 15:05:10.176129994 -0800
+++ aur/mirage/src/mirage-0.9.5.2/mirage.py	2018-02-09 15:05:10.802113556 -0800
@@ -4504,9 +4504,7 @@
 		if len(self.image_list) == 0:
 			title = "Mirage"
 		else:
-			title = "Mirage - " +_("[%(current)i of %(total)i]") % {'current': self.curr_img_in_list+1, 'total': len(self.image_list)} + ' ' + os.path.basename(self.currimg_name)
-			if self.slideshow_mode:
-				title = title + ' - ' + _('Slideshow Mode')
+			title = os.path.basename(self.currimg_name) + " - Mirage"
 		self.window.set_title(title)
 
 	def slideshow_controls_show(self):
