diff -ru aur/mirage.ORIG/src/mirage-1.0_pre1/mirage.py aur/mirage/src/mirage-1.0_pre1/mirage.py
--- aur/mirage.ORIG/src/mirage-1.0_pre1/mirage.py	2018-05-11 10:32:19.539846616 -0700
+++ aur/mirage/src/mirage-1.0_pre1/mirage.py	2018-05-11 10:34:01.126131370 -0700
@@ -492,7 +492,6 @@
 		self.refresh_recent_files_menu()
 		self.window.add_accel_group(self.UIManager.get_accel_group())
 		self.menubar = self.UIManager.get_widget('/MainMenu')
-		vbox.pack_start(self.menubar, False, False, 0)
 		self.toolbar = self.UIManager.get_widget('/MainToolbar')
 		vbox.pack_start(self.toolbar, False, False, 0)
 		self.layout = gtk.Layout()
