--- ../aur/thinkfan/src/thinkfan-0.7.3/system.c.ORIG	2011-09-04 00:39:11.743338485 -0700
+++ ../aur/thinkfan/src//thinkfan-0.7.3/system.c	2011-09-04 00:40:04.996670193 -0700
@@ -84,8 +84,13 @@
 	}
 	else {
 		if (unlikely(cur_lvl == INT_MIN)) strcpy(buf, "level disengaged\n");
-		else snprintf(buf, 10, "level %d\n", cur_lvl);
-		if (unlikely(write(ibm_fan, buf, strlen(buf)) < strlen(buf))) {
+		else {
+			if (cur_lvl == 8)
+				snprintf(buf, 18, "level full-speed\n");
+			else
+				snprintf(buf, 10, "level %d\n", cur_lvl);
+		}
+		if (unlikely(write(ibm_fan, buf, strlen(buf)) < strlen(buf))) {
 			report(LOG_ERR, LOG_ERR, MSG_ERR_FANCTRL);
 			errcnt++;
 		}
