diff -ru libwnck3.ORIG/src/libwnck/libwnck/pager.c libwnck3/src/libwnck/libwnck/pager.c
--- libwnck3.ORIG/src/libwnck/libwnck/pager.c	2023-03-23 22:46:54.874260105 -0700
+++ libwnck3/src/libwnck/libwnck/pager.c	2023-03-23 22:46:54.897593789 -0700
@@ -1634,6 +1634,8 @@
                         gint                y,
                         guint               time)
 {
+  return FALSE;
+
   WnckPager *pager;
   gint previous_workspace;
 
