diff --color -ru libwnck3.ORIG/src/libwnck/libwnck/pager.c libwnck3/src/libwnck/libwnck/pager.c
--- libwnck3.ORIG/src/libwnck/libwnck/pager.c	2024-02-26 21:42:52.600640823 -0800
+++ libwnck3/src/libwnck/libwnck/pager.c	2024-02-26 21:59:44.474954466 -0800
@@ -1334,10 +1334,10 @@
                       if (j == verti_views - 1)
                         vh = rect->height + rect->y - vy;
 
-                      if (active_i == i && active_j == j)
+                      if (active_i == i && active_j == j) {
                         rec_state = GTK_STATE_FLAG_SELECTED;
-
-                      draw_dark_rectangle (context, cr, rec_state, vx, vy, vw, vh);
+                        draw_dark_rectangle (context, cr, rec_state, vx, vy, vw + 2, vh);
+                      }
                     }
                 }
             }
