diff -ru libwnck3.ORIG/src/libwnck/libwnck/pager.c libwnck3/src/libwnck/libwnck/pager.c
--- libwnck3.ORIG/src/libwnck/libwnck/pager.c	2020-05-06 06:07:24.561748528 -0700
+++ libwnck3/src/libwnck/libwnck/pager.c	2020-05-06 06:10:59.722973757 -0700
@@ -994,43 +999,6 @@
 
   icon_w = icon_h = 0;
 
-  if (icon)
-    {
-      icon_w = gdk_pixbuf_get_width (icon);
-      icon_h = gdk_pixbuf_get_height (icon);
-
-      /* If the icon is too big, fall back to mini icon.
-       * We don't arbitrarily scale the icon, because it's
-       * just too slow on my Athlon 850.
-       */
-      if (icon_w > (winrect->width - 2) ||
-          icon_h > (winrect->height - 2))
-        {
-          icon = wnck_window_get_mini_icon (win);
-          if (icon)
-            {
-              icon_w = gdk_pixbuf_get_width (icon);
-              icon_h = gdk_pixbuf_get_height (icon);
-
-              /* Give up. */
-              if (icon_w > (winrect->width - 2) ||
-                  icon_h > (winrect->height - 2))
-                icon = NULL;
-            }
-        }
-    }
-
-  if (icon)
-    {
-      icon_x = winrect->x + (winrect->width - icon_w) / 2;
-      icon_y = winrect->y + (winrect->height - icon_h) / 2;
-
-      cairo_push_group (cr);
-      gtk_render_icon (context, cr, icon, icon_x, icon_y);
-      cairo_pop_group_to_source (cr);
-      cairo_paint_with_alpha (cr, translucency);
-    }
-
   cairo_push_group (cr);
   gtk_render_frame (context, cr, winrect->x + 0.5, winrect->y + 0.5,
                     MAX (0, winrect->width - 1), MAX (0, winrect->height - 1));
