diff -ru pam.ORIG/src/Linux-PAM-1.5.0/modules/pam_faillock/pam_faillock.c pam/src/Linux-PAM-1.5.0/modules/pam_faillock/pam_faillock.c
--- pam.ORIG/src/Linux-PAM-1.5.0/modules/pam_faillock/pam_faillock.c	2020-08-19 13:13:09.984256067 -0700
+++ pam/src/Linux-PAM-1.5.0/modules/pam_faillock/pam_faillock.c	2020-08-19 13:34:27.619201772 -0700
@@ -685,7 +685,7 @@
 	if (rv != PAM_SUCCESS)
 		goto err;
 
-	pam_fail_delay(pamh, 2000000);	/* 2 sec delay on failure */
+	pam_fail_delay(pamh, 1000000);	/* 1 sec delay on failure */
 
 	if ((rv=get_pam_user(pamh, &opts)) != PAM_SUCCESS) {
 		goto err;
diff -ru pam.ORIG/src/Linux-PAM-1.5.0/modules/pam_unix/support.c pam/src/Linux-PAM-1.5.0/modules/pam_unix/support.c
--- pam.ORIG/src/Linux-PAM-1.5.0/modules/pam_unix/support.c	2020-08-19 13:13:09.974255754 -0700
+++ pam/src/Linux-PAM-1.5.0/modules/pam_unix/support.c	2020-08-19 13:34:33.845897252 -0700
@@ -686,7 +686,7 @@
 #ifdef HAVE_PAM_FAIL_DELAY
 	if (off(UNIX_NODELAY, ctrl)) {
 		D(("setting delay"));
-		(void) pam_fail_delay(pamh, 2000000);	/* 2 sec delay for on failure */
+		(void) pam_fail_delay(pamh, 1000000);	/* 1 sec delay for on failure */
 	}
 #endif
 
