diff --color -ru cbatticon.ORIG/src/cbatticon/cbatticon.c cbatticon/src/cbatticon/cbatticon.c
--- cbatticon.ORIG/src/cbatticon/cbatticon.c	2023-02-21 01:56:02.084507179 -0800
+++ cbatticon/src/cbatticon/cbatticon.c	2023-02-21 01:58:02.433747257 -0800
@@ -781,9 +781,9 @@
         if (ac_only == FALSE) {
             ac_only = TRUE;
 
-            NOTIFY_MESSAGE (&notification, _("AC only, no battery!"), NULL, NOTIFY_EXPIRES_NEVER, NOTIFY_URGENCY_NORMAL);
+            NOTIFY_MESSAGE (&notification, _("AC only, no battery"), NULL, NOTIFY_EXPIRES_NEVER, NOTIFY_URGENCY_NORMAL);
 
-            gtk_status_icon_set_tooltip_text (tray_icon, _("AC only, no battery!"));
+            gtk_status_icon_set_tooltip_text (tray_icon, _("AC only, no battery"));
             gtk_status_icon_set_from_icon_name (tray_icon, "ac-adapter");
         }
 
@@ -925,7 +925,7 @@
 
 #ifdef WITH_NOTIFY
                         static NotifyNotification *spawn_notification = NULL;
-                        NOTIFY_MESSAGE (&spawn_notification, _("Cannot spawn low battery level command!"), configuration.command_low_level, NOTIFY_EXPIRES_NEVER, NOTIFY_URGENCY_CRITICAL);
+                        NOTIFY_MESSAGE (&spawn_notification, _("Cannot spawn low battery level command"), configuration.command_low_level, NOTIFY_EXPIRES_NEVER, NOTIFY_URGENCY_CRITICAL);
 #endif
                     }
                 }
@@ -953,7 +953,7 @@
 
 #ifdef WITH_NOTIFY
                         static NotifyNotification *spawn_notification = NULL;
-                        NOTIFY_MESSAGE (&spawn_notification, _("Cannot spawn critical battery level command!"), configuration.command_critical_level, NOTIFY_EXPIRES_NEVER, NOTIFY_URGENCY_CRITICAL);
+                        NOTIFY_MESSAGE (&spawn_notification, _("Cannot spawn critical battery level command"), configuration.command_critical_level, NOTIFY_EXPIRES_NEVER, NOTIFY_URGENCY_CRITICAL);
 #endif
                     }
                 }
@@ -975,7 +975,7 @@
 
 #ifdef WITH_NOTIFY
             static NotifyNotification *spawn_notification = NULL;
-            NOTIFY_MESSAGE (&spawn_notification, _("Cannot spawn left click command!"), configuration.command_left_click, NOTIFY_EXPIRES_DEFAULT, NOTIFY_URGENCY_CRITICAL);
+            NOTIFY_MESSAGE (&spawn_notification, _("Cannot spawn left click command"), configuration.command_left_click, NOTIFY_EXPIRES_DEFAULT, NOTIFY_URGENCY_CRITICAL);
 #endif
         }
     }
@@ -1039,15 +1039,15 @@
 
     switch (state) {
         case MISSING:
-            g_strlcpy (battery_string, _("Battery is missing!"), STR_LTH);
+            g_strlcpy (battery_string, _("Battery is missing"), STR_LTH);
             break;
 
         case UNKNOWN:
-            g_strlcpy (battery_string, _("Battery status is unknown!"), STR_LTH);
+            g_strlcpy (battery_string, _("Battery status is unknown"), STR_LTH);
             break;
 
         case CHARGED:
-            g_strlcpy (battery_string, _("Battery is charged!"), STR_LTH);
+            g_strlcpy (battery_string, _("Battery is charged"), STR_LTH);
             break;
 
         case DISCHARGING:
