diff -ru aur/physlock-git.ORIG/src/physlock-git/config.h aur/physlock-git/src/physlock-git/config.h
--- aur/physlock-git.ORIG/src/physlock-git/config.h	2014-07-23 17:07:55.194000823 -0700
+++ aur/physlock-git/src/physlock-git/config.h	2014-07-23 17:08:24.487335664 -0700
@@ -9,5 +9,5 @@
 static const char * const SYSRQ_PATH = "/proc/sys/kernel/sysrq";
 
 /* timeout (seconds) after failed authentication try:  */
-enum { AUTH_FAIL_TIMEOUT = 2 };
+enum { AUTH_FAIL_TIMEOUT = 1 };
 
