diff --color -ru compiz.ORIG/src/compiz-0.9.14.2/plugins/blur/blur.xml.in compiz/src/compiz-0.9.14.2/plugins/blur/blur.xml.in
--- compiz.ORIG/src/compiz-0.9.14.2/plugins/blur/blur.xml.in	2024-11-04 10:04:22.949540926 -0800
+++ compiz/src/compiz-0.9.14.2/plugins/blur/blur.xml.in	2024-11-04 15:40:20.759917399 -0800
@@ -73,7 +73,7 @@
 		<_long>Gaussian radius</_long>
 		<default>3</default>
 		<min>1</min>
-		<max>15</max>
+		<max>100</max>
 	    </option>
 	    <option name="gaussian_strength" type="float">
 		<_short>Gaussian Strength</_short>
diff --color -ru compiz.ORIG/src/compiz-0.9.14.2/plugins/blur/src/blur.cpp compiz/src/compiz-0.9.14.2/plugins/blur/src/blur.cpp
--- compiz.ORIG/src/compiz-0.9.14.2/plugins/blur/src/blur.cpp	2024-11-04 10:04:22.949540926 -0800
+++ compiz/src/compiz-0.9.14.2/plugins/blur/src/blur.cpp	2024-11-04 15:40:17.946537762 -0800
@@ -57,17 +57,19 @@
     int   mySize = ceil (radius / 2.0f);
     int   i, j;
 
-    ar1[0] = 1.0;
-    ar1[1] = 1.0;
+    float scalar = 0.85;
+
+    ar1[0] = scalar;
+    ar1[1] = scalar;
 
     for (i = 3; i <= size; i++)
     {
-	ar2[0] = 1;
+	ar2[0] = scalar;
 
 	for (j = 1; j < i - 1; j++)
-	    ar2[j] = (ar1[j - 1] + ar1[j]) * factor;
+	    ar2[j] = std::pow(ar1[j - 1] + ar1[j], scalar) * factor;
 
-	ar2[i - 1] = 1;
+	ar2[i - 1] = scalar;
 
 	tmp = ar1;
 	ar1 = ar2;
diff --color -ru compiz.ORIG/src/compiz-0.9.14.2/plugins/blur/src/blur.h compiz/src/compiz-0.9.14.2/plugins/blur/src/blur.h
--- compiz.ORIG/src/compiz-0.9.14.2/plugins/blur/src/blur.h	2024-11-04 10:04:22.949540926 -0800
+++ compiz/src/compiz-0.9.14.2/plugins/blur/src/blur.h	2024-11-04 15:40:19.653232517 -0800
@@ -42,7 +42,7 @@
 #include <decoration.h>
 
 
-const unsigned short BLUR_GAUSSIAN_RADIUS_MAX = 15;
+const unsigned short BLUR_GAUSSIAN_RADIUS_MAX = 100;
 
 const unsigned short BLUR_STATE_CLIENT = 0;
 const unsigned short BLUR_STATE_DECOR  = 1;
