diff --color -ru compiz.ORIG/src/compiz-0.9.14.2/plugins/blur/src/blur.cpp compiz/src/compiz-0.9.14.2/plugins/blur/src/blur.cpp
--- compiz.ORIG/src/compiz-0.9.14.2/plugins/blur/src/blur.cpp	2026-06-24 12:29:55.060587626 -0700
+++ compiz/src/compiz-0.9.14.2/plugins/blur/src/blur.cpp	2026-06-24 12:31:28.064862136 -0700
@@ -185,8 +185,8 @@
     {
 	region += CompRect (-window->output ().left,
 			    -window->output ().top,
-			    window->width () + window->output ().right,
-			    window->height () + window->output ().bottom);
+			    window->width () + window->output ().left + window->output ().right,
+			    window->height () + window->output ().top + window->output ().bottom);
 
 	region -= CompRect (0, 0, window->width (), window->height ());
 
@@ -1622,8 +1622,8 @@
 	    if (!clientThreshold)
 	    {
 		bScreen->tmpRegion -= CompRect (window->x (),
-						window->x () + window->width (),
 						window->y (),
+						window->x () + window->width (),
 						window->y () + window->height ());
 	    }
 
