diff -ru vte3.ORIG/src/vte/src/vtedraw.cc vte3/src/vte/src/vtedraw.cc
--- vte3.ORIG/src/vte/src/vtedraw.cc	2019-07-29 11:14:05.326734993 -0700
+++ vte3/src/vte/src/vtedraw.cc	2019-07-29 12:17:25.269349662 -0700
@@ -909,8 +909,8 @@
         draw->cell_width = draw->fonts[VTE_DRAW_NORMAL]->width * cell_width_scale;
         draw->char_spacing.left = (draw->cell_width - draw->fonts[VTE_DRAW_NORMAL]->width) / 2;
         draw->char_spacing.right = (draw->cell_width - draw->fonts[VTE_DRAW_NORMAL]->width + 1) / 2;
-        draw->cell_height = draw->fonts[VTE_DRAW_NORMAL]->height * cell_height_scale;
-        draw->char_spacing.top = (draw->cell_height - draw->fonts[VTE_DRAW_NORMAL]->height + 1) / 2;
+        draw->cell_height = (draw->fonts[VTE_DRAW_NORMAL]->height * cell_height_scale) - 1;
+        draw->char_spacing.top = ((draw->cell_height - draw->fonts[VTE_DRAW_NORMAL]->height + 1) / 2) - 1;
         draw->char_spacing.bottom = (draw->cell_height - draw->fonts[VTE_DRAW_NORMAL]->height) / 2;
 
         /* Drop the undercurl's cached look. Will recache on demand. */
