diff -ru vte3.ORIG/src/vte/src/vtedefines.hh vte3/src/vte/src/vtedefines.hh
--- vte3.ORIG/src/vte/src/vtedefines.hh	2019-07-29 11:01:41.131742526 -0700
+++ vte3/src/vte/src/vtedefines.hh	2019-07-29 11:01:50.381326037 -0700
@@ -97,7 +97,7 @@
 
 #define VTE_FONT_SCALE_MIN (.25)
 #define VTE_FONT_SCALE_MAX (4.)
-#define VTE_CELL_SCALE_MIN (1.)
+#define VTE_CELL_SCALE_MIN (0.5)
 #define VTE_CELL_SCALE_MAX (2.)
 
 /* Minimum time between two beeps (µs) */
