diff -ru npm.ORIG/src/cli-7.12.0/lib/cli.js npm/src/cli-7.12.0/lib/cli.js
--- npm.ORIG/src/cli-7.12.0/lib/cli.js	2021-03-29 10:53:37.703481157 -0700
+++ npm/src/cli-7.12.0/lib/cli.js	2021-03-29 10:57:17.923188665 -0700
@@ -67,7 +67,6 @@
         if (cmd) {
           const didYouMean = require('./utils/did-you-mean.js')
           const suggestions = await didYouMean(npm, npm.localPrefix, cmd)
-          npm.output(`Unknown command: "${cmd}"${suggestions}\n\nTo see a list of supported npm commands, run:\n  npm help`)
         } else
           npm.output(npm.usage)
         process.exitCode = 1
