diff -ru npm.ORIG/src/cli-7.23.0/Makefile npm/src/cli-7.23.0/Makefile
--- npm.ORIG/src/cli-7.23.0/Makefile	2021-03-11 09:32:23.014872392 -0800
+++ npm/src/cli-7.23.0/Makefile	2021-03-11 09:32:23.598196625 -0800
@@ -28,6 +28,9 @@
 
 all: docs
 
+install: all
+	node bin/npm-cli.js install -g -f ${NPMOPTS} $(shell node bin/npm-cli.js pack | tail -1)
+
 docs: mandocs htmldocs
 
 mandocs: dev-deps $(mandocs)
