diff --color -ru npm.ORIG/src/cli-8.2.0/lib/cli.js npm/src/cli-8.2.0/lib/cli.js
--- npm.ORIG/src/cli-8.2.0/lib/cli.js	2021-12-13 04:44:53.634030707 -0800
+++ npm/src/cli-8.2.0/lib/cli.js	2021-12-13 04:45:09.147995854 -0800
@@ -68,8 +68,6 @@
     if (err.code === 'EUNKNOWNCOMMAND') {
       const didYouMean = require('./utils/did-you-mean.js')
       const suggestions = await didYouMean(npm, npm.localPrefix, cmd)
-      npm.output(`Unknown command: "${cmd}"${suggestions}\n`)
-      npm.output('To see a list of supported npm commands, run:\n  npm help')
       process.exitCode = 1
       return exitHandler()
     }
