diff -ru npm.ORIG/src/cli-7.6.0/lib/utils/reify-output.js npm/src/cli-7.6.0/lib/utils/reify-output.js
--- npm.ORIG/src/cli-7.6.0/lib/utils/reify-output.js	2020-11-09 14:53:05.490494089 -0800
+++ npm/src/cli-7.6.0/lib/utils/reify-output.js	2020-11-09 14:53:05.647160363 -0800
@@ -76,7 +76,6 @@
     output(JSON.stringify(summary, 0, 2))
   } else {
     packagesChangedMessage(summary)
-    packagesFundingMessage(summary)
     printAuditReport(auditReport)
   }
 }
