diff -ru abs/pidgin.ORIG/src/pidgin-2.13.0/pidgin/gtkcellrendererexpander.c abs/pidgin/src/pidgin-2.13.0/pidgin/gtkcellrendererexpander.c
--- abs/pidgin.ORIG/src/pidgin-2.13.0/pidgin/gtkcellrendererexpander.c	2016-12-29 10:18:15.917359805 -0800
+++ abs/pidgin/src/pidgin-2.13.0/pidgin/gtkcellrendererexpander.c	2016-12-29 10:18:16.350684785 -0800
@@ -264,12 +264,6 @@
 			    cell_area->x + cell->xpad + (width / 2),
 			    cell_area->y + cell->ypad + (height / 2),
 			    cell->is_expanded ? GTK_EXPANDER_EXPANDED : GTK_EXPANDER_COLLAPSED);
-
-	/* only draw the line if the color isn't set - this prevents a bug where the hline appears only under the expander */
-	g_object_get(cellexpander, "cell-background-set", &set, NULL);
-	if (cell->is_expanded && !set)
-		gtk_paint_hline (widget->style, window, state, NULL, widget, NULL, 0,
-				 widget->allocation.width, cell_area->y + cell_area->height);
 }
 
 static gboolean pidgin_cell_renderer_expander_activate(GtkCellRenderer *r,
